<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2022-03-30" />

<title>Analysis of CF_BAL_Pilot &amp; C133_Neeland Data Combined</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-cf-cite-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of CF_BAL_Pilot &amp; C133_Neeland Data Combined</h1>
<h3 class="subtitle">Compare HCLA v1.0 versus HCLA v2.0 Annotation</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">March 30, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-cf-cite-seq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210524code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210524)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210524)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtreef45417b7c5bb3842ca44bd716cb78aab8559d9b4targetblankf45417ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/f45417b7c5bb3842ca44bd716cb78aab8559d9b4" target="_blank">f45417b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtreef45417b7c5bb3842ca44bd716cb78aab8559d9b4targetblankf45417ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/f45417b7c5bb3842ca44bd716cb78aab8559d9b4" target="_blank">f45417b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/CellRanger/
    Ignored:    data/emptyDrops/bin/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/03_COMBO.annotation_markers.Rmd
    Untracked:  analysis/03_COMBO.protein_analysis.Rmd
    Untracked:  analysis/scGateDB/
    Untracked:  code/03_C133_Neeland-dropletutils.R
    Untracked:  code/03_COMBO.proteins_geneBasis.R
    Untracked:  data/190930_A00152_0150_BHTYCMDSXX/
    Untracked:  data/Flow_data_CITEseq_2-8.csv
    Untracked:  data/GSE127465_RAW/
    Untracked:  data/GSE130148_barcodes_cell_types.txt
    Untracked:  data/GSE130148_raw_counts.RData
    Untracked:  data/Macrophage_subcluster_annotation_110322.csv
    Untracked:  data/Other_subcluster_annotation_110322.csv
    Untracked:  data/T-NK_subcluster_annotation_110322.csv
    Untracked:  data/T-NK_subcluster_proteins.csv
    Untracked:  data/T-NK_subcluster_proteins_110322.csv
    Untracked:  data/T-NK_subclusters_cytokines.csv
    Untracked:  data/T-NK_subclusters_markergenes_110322.csv
    Untracked:  data/all_azimuth_cluster_cytokines.csv
    Untracked:  data/all_azimuth_clusters_proteins.csv
    Untracked:  data/cell_type_category_rna_Granulocytes_Cell.tsv
    Untracked:  data/consolidated_markers.rds
    Untracked:  data/gb-genes-50.rds
    Untracked:  data/genome1K.phase3.SNP_AF5e4.chr1toX.hg38.vcf.gz
    Untracked:  data/macrophage_subcluster_cytokines.csv
    Untracked:  data/macrophage_subcluster_markergenes_110322.csv
    Untracked:  data/macrophage_subcluster_markergenes_160322.csv
    Untracked:  data/macrophage_subcluster_proteins.csv
    Untracked:  data/macrophage_subcluster_proteins_110322.csv
    Untracked:  data/neutrophil.tsv
    Untracked:  data/other_subcluster_proteins.csv
    Untracked:  data/other_subcluster_proteins_110322.csv
    Untracked:  data/other_subclusters_cytokines.csv
    Untracked:  data/other_subclusters_markergenes_110322.csv
    Untracked:  data/proteins_ranked.rds
    Untracked:  data/sample_sheets/Hashtag_Feature_Ref_With_Clin.csv
    Untracked:  data/sample_sheets/TotalSeq_A_Human_Universal_Cocktail_Proteins of interest_29.09.21.csv
    Untracked:  data/sample_sheets/TotalSeq_A_Human_Universal_Cocktail_Proteins_of_interest_28.06.21.csv
    Untracked:  output/ADT-counts.csv
    Untracked:  output/C133_Neeland.azimuth.individual_cell_type_proportions.csv
    Untracked:  output/C133_Neeland.azimuth.phenotype_information.csv
    Untracked:  output/C133_Neeland.individual_cell_type_proportions.csv
    Untracked:  output/C133_Neeland.phenotype_information.csv
    Untracked:  output/DEGs/
    Untracked:  output/glimma-plots/js/labelled-singlets-md-plot.js
    Untracked:  output/glimma-plots/js/qc-md-plot_03_C133_Neeland.js
    Untracked:  output/glimma-plots/js/qc-md-plot_C133_Neeland.js
    Untracked:  output/glimma-plots/js/qc-md-plot_CF-BAL-Pilot.js
    Untracked:  output/glimma-plots/labelled-singlets-md-plot.html
    Untracked:  output/glimma-plots/qc-md-plot_03_C133_Neeland.html
    Untracked:  output/glimma-plots/qc-md-plot_C133_Neeland.html
    Untracked:  output/glimma-plots/qc-md-plot_CF-BAL-Pilot.html
    Untracked:  output/marker-analysis-mod/
    Untracked:  output/marker-analysis/
    Untracked:  output/metadata.csv
    Untracked:  output/protein_labels.pdf
    Untracked:  scGateDB/
    Untracked:  wflow_background.R

Unstaged changes:
    Modified:   analysis/03_COMBO.proportion_analysis.Rmd
    Modified:   data/SCEs/C133_Neeland.CellRanger.SCE.rds
    Modified:   data/SCEs/C133_Neeland.demultiplexed.SCE.rds
    Deleted:    data/SCEs/C133_Neeland.preprocessed.SCE.hto.rds
    Deleted:    data/SCEs/C133_Neeland.preprocessed.SCE_hto.rds
    Deleted:    data/SCEs/C133_Neeland.preprocessed.SCE_snp.rds
    Modified:   data/sample_sheets/TotalSeq-A_Universal_Cocktail_v1.0.csv
    Modified:   data/vireo/C133_1/GT_donors.vireo.vcf.gz
    Modified:   data/vireo/C133_1/_log.txt
    Modified:   data/vireo/C133_1/donor_ids.tsv
    Modified:   data/vireo/C133_1/fig_GT_distance_estimated.pdf
    Modified:   data/vireo/C133_1/prob_doublet.tsv.gz
    Modified:   data/vireo/C133_1/prob_singlet.tsv.gz
    Modified:   data/vireo/C133_1/summary.tsv
    Modified:   data/vireo/C133_2/GT_donors.vireo.vcf.gz
    Modified:   data/vireo/C133_2/_log.txt
    Modified:   data/vireo/C133_2/donor_ids.tsv
    Modified:   data/vireo/C133_2/fig_GT_distance_estimated.pdf
    Modified:   data/vireo/C133_2/prob_doublet.tsv.gz
    Modified:   data/vireo/C133_2/prob_singlet.tsv.gz
    Modified:   data/vireo/C133_2/summary.tsv
    Modified:   output/glimma-plots/js/qc-md-plot.js
    Modified:   output/glimma-plots/js/stripped-nuclei-md-plot.js
    Modified:   renv/.gitignore
    Modified:   renv/activate.R
    Modified:   renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03_COMBO.compare_annotations.Rmd</code>) and HTML (<code>docs/03_COMBO.compare_annotations.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f45417b7c5bb3842ca44bd716cb78aab8559d9b4/analysis/03_COMBO.compare_annotations.Rmd" target="_blank">f45417b</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish("analysis/03_COMBO.compare_annotations.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<p>Load integrated, clustered data that has been mapped to Zilionis reference.</p>
<pre class="r"><code>out &lt;- here(&quot;data/SCEs/03_COMBO.zilionis_mapped.SEU.rds&quot;)
seuInt &lt;- readRDS(out)

table(seuInt$predicted.celltype)</code></pre>
<pre><code>
               bB cells              bMonocytes bMyeloid precursor-like 
                    989                     596                      32 
           bNeutrophils               bNK cells                    bpDC 
                     12                     590                      28 
               bT cells          Ciliated cells              Club cells 
                   3867                     314                      84 
      Endothelial cells             Fibroblasts                      ND 
                    255                       1                     937 
    Smooth muscle cells                tB cells             tMast cells 
                     34                     442                      49 
               tMoMacDC            tNeutrophils               tNK cells 
                  35172                     114                     612 
                   tpDC           tPlasma cells                tT cells 
                    111                      44                    1307 </code></pre>
</div>
<div id="annotate-data-using-azimuth-human-lung-cell-reference" class="section level1">
<h1>Annotate data using Azimuth &amp; <em>Human Lung Cell Reference</em></h1>
<div id="add-azimuth-hcla-v1.0-labels" class="section level2">
<h2>Add Azimuth HCLA v1.0 labels</h2>
<p>Add Azimuth labels to data.</p>
<pre class="r"><code>seuInt &lt;- AddAzimuthResults(seuInt,
                            filename = here(&quot;data/SCEs/03_COMBO.clustered_azimuth.SEU.rds&quot;))
seuInt$predicted.annotation.l1 &lt;- fct_drop(seuInt$predicted.annotation.l1)

table(seuInt$predicted.annotation.l1)</code></pre>
<pre><code>
             Basophil/Mast                   Ciliated 
                        48                        384 
Alveolar Epithelial Type 2           Bronchial Vessel 
                        19                         87 
    Capillary Intermediate                      Basal 
                       134                        363 
                 Dendritic             CD16+ Monocyte 
                      2494                         24 
            CD14+ Monocyte              Smooth Muscle 
                       620                          1 
                     CD8 T                      CD4 T 
                      2416                       3578 
    Plasmacytoid Dendritic                          B 
                       141                       1403 
                Macrophage                       Vein 
                     32039                         37 
                    Artery                     Mucous 
                        12                         10 
                 Lymphatic   Proliferating Macrophage 
                        28                        758 
        Proliferating NK/T           Natural Killer T 
                       102                        257 
            Natural Killer                     Plasma 
                       121                         33 
            Neuroendocrine                   Ionocyte 
                         4                        412 
                    Goblet 
                        65 </code></pre>
</div>
<div id="add-azimuth-hcla-v2.0-labels" class="section level2">
<h2>Add Azimuth HCLA v2.0 labels</h2>
<p>Add Azimuth labels to data.</p>
<pre class="r"><code>seuInt &lt;- AddAzimuthResults(seuInt,
                            filename = here(&quot;data/SCEs/03_COMBO.clustered_azimuth_v2.SEU.Rds&quot;))
seuInt$predicted.ann_level_1 &lt;- fct_drop(seuInt$predicted.ann_level_1)
seuInt$predicted.ann_level_2 &lt;- fct_drop(seuInt$predicted.ann_level_2)
seuInt$predicted.ann_level_3 &lt;- fct_drop(seuInt$predicted.ann_level_3)
seuInt$predicted.ann_level_4 &lt;- fct_drop(seuInt$predicted.ann_level_4)
seuInt$predicted.ann_finest_level &lt;- fct_drop(seuInt$predicted.ann_finest_level)

table(seuInt$predicted.ann_level_1) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="right">322</td>
</tr>
<tr class="even">
<td align="left">Epithelial</td>
<td align="right">1800</td>
</tr>
<tr class="odd">
<td align="left">Immune</td>
<td align="right">43468</td>
</tr>
</tbody>
</table>
<pre class="r"><code>table(seuInt$predicted.ann_level_2) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Airway epithelium</td>
<td align="right">1488</td>
</tr>
<tr class="even">
<td align="left">Alveolar epithelium</td>
<td align="right">299</td>
</tr>
<tr class="odd">
<td align="left">Blood vessels</td>
<td align="right">325</td>
</tr>
<tr class="even">
<td align="left">Lymphatic EC</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Lymphoid</td>
<td align="right">7955</td>
</tr>
<tr class="even">
<td align="left">Myeloid</td>
<td align="right">35521</td>
</tr>
</tbody>
</table>
<pre class="r"><code>table(seuInt$predicted.ann_level_3) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AT1</td>
<td align="right">369</td>
</tr>
<tr class="even">
<td align="left">AT2</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">B cell lineage</td>
<td align="right">1492</td>
</tr>
<tr class="even">
<td align="left">Basal</td>
<td align="right">541</td>
</tr>
<tr class="odd">
<td align="left">Dendritic cells</td>
<td align="right">2081</td>
</tr>
<tr class="even">
<td align="left">EC arterial</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">EC capillary</td>
<td align="right">121</td>
</tr>
<tr class="even">
<td align="left">EC venous</td>
<td align="right">187</td>
</tr>
<tr class="odd">
<td align="left">Innate lymphoid cell NK</td>
<td align="right">120</td>
</tr>
<tr class="even">
<td align="left">Lymphatic EC mature</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">Macrophages</td>
<td align="right">33161</td>
</tr>
<tr class="even">
<td align="left">Mast cells</td>
<td align="right">51</td>
</tr>
<tr class="odd">
<td align="left">Monocytes</td>
<td align="right">204</td>
</tr>
<tr class="even">
<td align="left">Multiciliated lineage</td>
<td align="right">386</td>
</tr>
<tr class="odd">
<td align="left">Rare</td>
<td align="right">99</td>
</tr>
<tr class="even">
<td align="left">Secretory</td>
<td align="right">397</td>
</tr>
<tr class="odd">
<td align="left">T cell lineage</td>
<td align="right">6342</td>
</tr>
</tbody>
</table>
<pre class="r"><code>table(seuInt$predicted.ann_level_4) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Alveolar macrophages</td>
<td align="right">28722</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="right">1471</td>
</tr>
<tr class="odd">
<td align="left">Basal resting</td>
<td align="right">244</td>
</tr>
<tr class="even">
<td align="left">CD4 T cells</td>
<td align="right">3020</td>
</tr>
<tr class="odd">
<td align="left">CD8 T cells</td>
<td align="right">3237</td>
</tr>
<tr class="even">
<td align="left">Classical monocytes</td>
<td align="right">122</td>
</tr>
<tr class="odd">
<td align="left">Club</td>
<td align="right">258</td>
</tr>
<tr class="even">
<td align="left">DC1</td>
<td align="right">63</td>
</tr>
<tr class="odd">
<td align="left">DC2</td>
<td align="right">1789</td>
</tr>
<tr class="even">
<td align="left">Deuterosomal</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">EC general capillary</td>
<td align="right">135</td>
</tr>
<tr class="even">
<td align="left">EC venous pulmonary</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">EC venous systemic</td>
<td align="right">171</td>
</tr>
<tr class="even">
<td align="left">Goblet</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">Interstitial macrophages</td>
<td align="right">4403</td>
</tr>
<tr class="even">
<td align="left">Ionocyte</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Migratory DCs</td>
<td align="right">120</td>
</tr>
<tr class="even">
<td align="left">Multiciliated</td>
<td align="right">384</td>
</tr>
<tr class="odd">
<td align="left">NK cells</td>
<td align="right">120</td>
</tr>
<tr class="even">
<td align="left">Non-classical monocytes</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">None</td>
<td align="right">505</td>
</tr>
<tr class="even">
<td align="left">Plasma cells</td>
<td align="right">37</td>
</tr>
<tr class="odd">
<td align="left">Plasmacytoid DCs</td>
<td align="right">143</td>
</tr>
<tr class="even">
<td align="left">SMG duct</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Suprabasal</td>
<td align="right">276</td>
</tr>
<tr class="even">
<td align="left">T cells proliferating</td>
<td align="right">74</td>
</tr>
<tr class="odd">
<td align="left">Tuft</td>
<td align="right">99</td>
</tr>
</tbody>
</table>
<pre class="r"><code>table(seuInt$predicted.ann_finest_level) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Alveolar macrophages</td>
<td align="right">27390</td>
</tr>
<tr class="even">
<td align="left">Alveolar Mφ CCL3+</td>
<td align="right">573</td>
</tr>
<tr class="odd">
<td align="left">Alveolar Mφ proliferating</td>
<td align="right">686</td>
</tr>
<tr class="even">
<td align="left">AT1</td>
<td align="right">400</td>
</tr>
<tr class="odd">
<td align="left">AT2</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="right">1471</td>
</tr>
<tr class="odd">
<td align="left">Basal resting</td>
<td align="right">245</td>
</tr>
<tr class="even">
<td align="left">CD4 T cells</td>
<td align="right">3020</td>
</tr>
<tr class="odd">
<td align="left">CD8 T cells</td>
<td align="right">3237</td>
</tr>
<tr class="even">
<td align="left">Classical monocytes</td>
<td align="right">133</td>
</tr>
<tr class="odd">
<td align="left">Club (nasal)</td>
<td align="right">230</td>
</tr>
<tr class="even">
<td align="left">Club (non-nasal)</td>
<td align="right">32</td>
</tr>
<tr class="odd">
<td align="left">DC1</td>
<td align="right">63</td>
</tr>
<tr class="even">
<td align="left">DC2</td>
<td align="right">1909</td>
</tr>
<tr class="odd">
<td align="left">Deuterosomal</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">EC arterial</td>
<td align="right">27</td>
</tr>
<tr class="odd">
<td align="left">EC general capillary</td>
<td align="right">145</td>
</tr>
<tr class="even">
<td align="left">EC venous pulmonary</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">EC venous systemic</td>
<td align="right">171</td>
</tr>
<tr class="even">
<td align="left">Goblet (bronchial)</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Goblet (nasal)</td>
<td align="right">116</td>
</tr>
<tr class="even">
<td align="left">Interstitial Mφ perivascular</td>
<td align="right">267</td>
</tr>
<tr class="odd">
<td align="left">Ionocyte</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Lymphatic EC mature</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">Mast cells</td>
<td align="right">51</td>
</tr>
<tr class="even">
<td align="left">Migratory DCs</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">Monocyte-derived Mφ</td>
<td align="right">4054</td>
</tr>
<tr class="even">
<td align="left">Multiciliated (nasal)</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">Multiciliated (non-nasal)</td>
<td align="right">383</td>
</tr>
<tr class="even">
<td align="left">NK cells</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">Non-classical monocytes</td>
<td align="right">86</td>
</tr>
<tr class="even">
<td align="left">Plasma cells</td>
<td align="right">37</td>
</tr>
<tr class="odd">
<td align="left">Plasmacytoid DCs</td>
<td align="right">143</td>
</tr>
<tr class="even">
<td align="left">SMG duct</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Suprabasal</td>
<td align="right">276</td>
</tr>
<tr class="even">
<td align="left">T cells proliferating</td>
<td align="right">74</td>
</tr>
<tr class="odd">
<td align="left">Tuft</td>
<td align="right">100</td>
</tr>
</tbody>
</table>
</div>
<div id="visualise-reference-mapping" class="section level2">
<h2>Visualise reference mapping</h2>
<pre class="r"><code>azimuth_results &lt;- readRDS(here(&quot;data/SCEs/03_COMBO.clustered_azimuth.SEU.rds&quot;))
seuInt[[&quot;umap.proj.v1&quot;]] &lt;- azimuth_results$umap

DimPlot(seuInt, reduction = &#39;umap.proj.v1&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.annotation.l1&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuInt, reduction = &#39;umap.proj&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.ann_level_1&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuInt, reduction = &#39;umap.proj&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.ann_level_2&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuInt, reduction = &#39;umap.proj&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.ann_level_3&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuInt, reduction = &#39;umap.proj&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.ann_level_4&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuInt, reduction = &#39;umap.proj&#39;, 
        label = TRUE, repel = TRUE, label.size = 2.5, 
        group.by = &quot;predicted.ann_finest_level&quot;) + NoLegend()</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-6-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>labels &lt;- colnames(seuInt@meta.data)[grepl(&quot;(?!.*score$)predicted.ann&quot;,
                                           colnames(seuInt@meta.data), 
                                           perl = TRUE)]
p &lt;- vector(&quot;list&quot;,length(labels))

for(label in labels){
  seuInt@meta.data %&gt;%
    ggplot(aes(x = !!sym(label), fill = !!sym(label))) +
    geom_bar() +
    geom_text(aes(label = ..count..), stat = &quot;count&quot;,
              vjust = -0.5, colour = &quot;black&quot;, size = 2) +
    scale_y_log10() +
    theme(axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()) +
    NoLegend() +
    labs(y = &quot;No. Cells (log scale)&quot;) -&gt; p1
  
  seuInt@meta.data %&gt;%
    dplyr::select(!!sym(label), experiment) %&gt;%
    group_by(!!sym(label), experiment) %&gt;%
    summarise(num = n()) %&gt;%
    mutate(prop = num / sum(num)) %&gt;%
  ggplot(aes(x = !!sym(label), y = prop * 100, 
             fill = experiment)) + 
    geom_bar(stat = &quot;identity&quot;) +
    theme(axis.text.x = element_text(angle = 90, 
                                     vjust = 0.5, 
                                     hjust = 1)) +
    labs(y = &quot;% Cells&quot;, fill = &quot;Experiment&quot;) -&gt; p2
  
  p1 / p2 -&gt; p[[label]]
}

p</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL

[[5]]
NULL

[[6]]
NULL

[[7]]
NULL

$predicted.annotation.l1</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_level_1</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_level_2</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_level_3</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_level_4</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-5.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_level_5</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-6.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
$predicted.ann_finest_level</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-7-7.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>check &lt;- levels(seuInt$predicted.annotation.l1)
p &lt;- vector(&quot;list&quot;, length(check))

for(i in 1:length(p)){
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    ggplot(aes(x = predicted.ann_level_3.score,
               colour = predicted.ann_level_3)) +
    geom_density() +
    ggtitle(paste0(&quot;Cluster &quot;, check[i])) +
    theme(legend.key.size = unit(4, &quot;pt&quot;)) -&gt; p1
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    ggplot(aes(x = predicted.ann_level_3,
               fill = predicted.ann_level_3)) +
    geom_bar() +
    geom_text(aes(label = ..count..), stat = &quot;count&quot;, 
              vjust = -0.5, colour = &quot;black&quot;, size = 2) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    NoLegend() -&gt; p2  
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    ggplot(aes(x = predicted.ann_finest_level.score,
               colour = predicted.ann_finest_level)) +
    geom_density() +
    ggtitle(paste0(&quot;Cluster &quot;, check[i])) +
    theme(legend.key.size = unit(4, &quot;pt&quot;)) -&gt; p3
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    
    ggplot(aes(x = predicted.ann_finest_level,
               fill = predicted.ann_finest_level)) +
    geom_bar() +
    geom_text(aes(label = ..count..), stat = &quot;count&quot;, 
              vjust = -0.5, colour = &quot;black&quot;, size = 2) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    NoLegend() -&gt; p4 
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    ggplot(aes(x = predicted.ann_level_4.score,
               colour = predicted.ann_level_4)) +
    geom_density() +
    ggtitle(paste0(&quot;Cluster &quot;, check[i])) +
    theme(legend.key.size = unit(4, &quot;pt&quot;)) -&gt; p5
  
  seuInt@meta.data %&gt;%
    dplyr::filter(predicted.annotation.l1 == check[i]) %&gt;%
    
    ggplot(aes(x = predicted.ann_level_4,
               fill = predicted.ann_level_4)) +
    geom_bar() +
    geom_text(aes(label = ..count..), stat = &quot;count&quot;, 
              vjust = -0.5, colour = &quot;black&quot;, size = 2) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    NoLegend() -&gt; p6  
  
  ((p1 | p2) / (p5 | p6) / (p3 | p4)) + 
    plot_annotation(title = check[i]) &amp; 
    theme(text = element_text(size = 8)) -&gt; p[[i]]
}

p</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-4.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-5.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-6.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-7.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-8.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[9]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-9.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[10]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-10.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[11]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-11.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[12]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-12.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[13]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-13.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[14]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-14.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[15]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-15.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[16]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-16.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[17]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-17.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[18]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-18.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[19]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-19.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[20]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-20.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[21]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-21.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[22]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-22.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[23]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-23.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[24]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-24.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[25]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-25.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[26]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-26.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[27]]</code></pre>
<p><img src="figure/03_COMBO.compare_annotations.Rmd/unnamed-chunk-8-27.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.0 (2021-05-18)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-03-30
 pandoc   2.14.0.3 @ /usr/lib/rstudio-server/bin/pandoc/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package                * version   date (UTC) lib source
 P abind                    1.4-5     2016-07-21 [?] CRAN (R 4.1.0)
 P AnnotationDbi          * 1.56.2    2021-11-09 [?] Bioconductor
 P AnnotationFilter       * 1.18.0    2021-10-26 [?] Bioconductor
 P AnnotationHub          * 3.2.0     2021-10-26 [?] Bioconductor
 P assertthat               0.2.1     2019-03-21 [?] CRAN (R 4.1.0)
 P backports                1.4.1     2021-12-13 [?] CRAN (R 4.1.0)
 P beachmat                 2.10.0    2021-10-26 [?] Bioconductor
 P beeswarm                 0.4.0     2021-06-01 [?] CRAN (R 4.1.0)
 P Biobase                * 2.54.0    2021-10-26 [?] Bioconductor
 P BiocFileCache          * 2.2.0     2021-10-26 [?] Bioconductor
 P BiocGenerics           * 0.40.0    2021-10-26 [?] Bioconductor
 P BiocIO                   1.4.0     2021-10-26 [?] Bioconductor
 P BiocManager              1.30.16   2021-06-15 [?] CRAN (R 4.1.0)
 P BiocNeighbors            1.12.0    2021-10-26 [?] Bioconductor
 P BiocParallel           * 1.28.3    2021-12-09 [?] Bioconductor
 P BiocSingular             1.10.0    2021-10-26 [?] Bioconductor
 P BiocVersion              3.14.0    2021-05-19 [?] Bioconductor
 P biomaRt                  2.50.1    2021-11-21 [?] Bioconductor
 P Biostrings               2.62.0    2021-10-26 [?] Bioconductor
 P bit                      4.0.4     2020-08-04 [?] CRAN (R 4.1.0)
 P bit64                    4.0.5     2020-08-30 [?] CRAN (R 4.0.2)
 P bitops                   1.0-7     2021-04-24 [?] CRAN (R 4.0.2)
 P blob                     1.2.2     2021-07-23 [?] CRAN (R 4.1.0)
 P bluster                  1.4.0     2021-10-26 [?] Bioconductor
 P broom                    0.7.11    2022-01-03 [?] CRAN (R 4.1.0)
 P bslib                    0.3.1     2021-10-06 [?] CRAN (R 4.1.0)
 P cachem                   1.0.6     2021-08-19 [?] CRAN (R 4.1.0)
 P callr                    3.7.0     2021-04-20 [?] CRAN (R 4.1.0)
 P cellranger               1.1.0     2016-07-27 [?] CRAN (R 4.1.0)
 P cli                      3.1.0     2021-10-27 [?] CRAN (R 4.1.0)
 P cluster                  2.1.2     2021-04-17 [?] CRAN (R 4.1.0)
 P clustree               * 0.4.4     2021-11-08 [?] CRAN (R 4.1.0)
 P codetools                0.2-18    2020-11-04 [?] CRAN (R 4.1.0)
 P colorspace               2.0-2     2021-06-24 [?] CRAN (R 4.0.2)
 P cowplot                  1.1.1     2020-12-30 [?] CRAN (R 4.0.2)
 P crayon                   1.4.2     2021-10-29 [?] CRAN (R 4.1.0)
 P curl                     4.3.2     2021-06-23 [?] CRAN (R 4.1.0)
 P data.table             * 1.14.2    2021-09-27 [?] CRAN (R 4.1.0)
 P DBI                      1.1.2     2021-12-20 [?] CRAN (R 4.1.0)
 P dbplyr                 * 2.1.1     2021-04-06 [?] CRAN (R 4.1.0)
 P DelayedArray             0.20.0    2021-10-26 [?] Bioconductor
 P DelayedMatrixStats       1.16.0    2021-10-26 [?] Bioconductor
 P deldir                   1.0-6     2021-10-23 [?] CRAN (R 4.1.0)
 P digest                   0.6.29    2021-12-01 [?] CRAN (R 4.1.0)
 P dplyr                  * 1.0.7     2021-06-18 [?] CRAN (R 4.1.0)
 P dqrng                    0.3.0     2021-05-01 [?] CRAN (R 4.1.0)
 P DropletUtils           * 1.14.1    2021-11-08 [?] Bioconductor
 P edgeR                  * 3.36.0    2021-10-26 [?] Bioconductor
 P ellipsis                 0.3.2     2021-04-29 [?] CRAN (R 4.0.2)
 P ensembldb              * 2.18.2    2021-11-08 [?] Bioconductor
 P evaluate                 0.14      2019-05-28 [?] CRAN (R 4.0.2)
 P fansi                    1.0.0     2022-01-10 [?] CRAN (R 4.1.0)
 P farver                   2.1.0     2021-02-28 [?] CRAN (R 4.0.2)
 P fastmap                  1.1.0     2021-01-25 [?] CRAN (R 4.1.0)
 P filelock                 1.0.2     2018-10-05 [?] CRAN (R 4.1.0)
 P fitdistrplus             1.1-6     2021-09-28 [?] CRAN (R 4.1.0)
 P forcats                * 0.5.1     2021-01-27 [?] CRAN (R 4.1.0)
 P fs                       1.5.2     2021-12-08 [?] CRAN (R 4.1.0)
 P future                   1.23.0    2021-10-31 [?] CRAN (R 4.1.0)
 P future.apply             1.8.1     2021-08-10 [?] CRAN (R 4.1.0)
 P generics                 0.1.1     2021-10-25 [?] CRAN (R 4.1.0)
   GenomeInfoDb           * 1.30.1    2022-01-30 [1] Bioconductor
 P GenomeInfoDbData         1.2.7     2021-12-21 [?] Bioconductor
 P GenomicAlignments        1.30.0    2021-10-26 [?] Bioconductor
 P GenomicFeatures        * 1.46.3    2021-12-30 [?] Bioconductor
 P GenomicRanges          * 1.46.1    2021-11-18 [?] Bioconductor
 P getPass                  0.2-2     2017-07-21 [?] CRAN (R 4.0.2)
 P ggalluvial             * 0.12.3    2020-12-05 [?] CRAN (R 4.1.0)
 P ggbeeswarm               0.6.0     2017-08-07 [?] CRAN (R 4.1.0)
 P ggforce                  0.3.3     2021-03-05 [?] CRAN (R 4.1.0)
 P ggplot2                * 3.3.5     2021-06-25 [?] CRAN (R 4.0.2)
 P ggraph                 * 2.0.5     2021-02-23 [?] CRAN (R 4.1.0)
 P ggrepel                  0.9.1     2021-01-15 [?] CRAN (R 4.1.0)
 P ggridges               * 0.5.3     2021-01-08 [?] CRAN (R 4.1.0)
 P git2r                    0.29.0    2021-11-22 [?] CRAN (R 4.1.0)
 P glmGamPoi              * 1.6.0     2021-10-26 [?] Bioconductor
 P globals                  0.14.0    2020-11-22 [?] CRAN (R 4.0.2)
 P glue                   * 1.6.0     2021-12-17 [?] CRAN (R 4.1.0)
 P goftest                  1.2-3     2021-10-07 [?] CRAN (R 4.1.0)
 P graphlayouts             0.8.0     2022-01-03 [?] CRAN (R 4.1.0)
 P gridExtra                2.3       2017-09-09 [?] CRAN (R 4.1.0)
 P gtable                   0.3.0     2019-03-25 [?] CRAN (R 4.1.0)
 P haven                    2.4.3     2021-08-04 [?] CRAN (R 4.1.0)
 P HDF5Array                1.22.1    2021-11-14 [?] Bioconductor
 P here                   * 1.0.1     2020-12-13 [?] CRAN (R 4.0.2)
 P highr                    0.9       2021-04-16 [?] CRAN (R 4.1.0)
 P hms                      1.1.1     2021-09-26 [?] CRAN (R 4.1.0)
 P htmltools                0.5.2     2021-08-25 [?] CRAN (R 4.1.0)
 P htmlwidgets              1.5.4     2021-09-08 [?] CRAN (R 4.1.0)
 P httpuv                   1.6.5     2022-01-05 [?] CRAN (R 4.1.0)
 P httr                     1.4.2     2020-07-20 [?] CRAN (R 4.1.0)
 P ica                      1.0-2     2018-05-24 [?] CRAN (R 4.1.0)
 P igraph                   1.2.11    2022-01-04 [?] CRAN (R 4.1.0)
 P interactiveDisplayBase   1.32.0    2021-10-26 [?] Bioconductor
 P IRanges                * 2.28.0    2021-10-26 [?] Bioconductor
 P irlba                    2.3.5     2021-12-06 [?] CRAN (R 4.1.0)
 P janitor                * 2.1.0     2021-01-05 [?] CRAN (R 4.0.2)
 P jquerylib                0.1.4     2021-04-26 [?] CRAN (R 4.1.0)
 P jsonlite                 1.7.2     2020-12-09 [?] CRAN (R 4.0.2)
 P KEGGREST                 1.34.0    2021-10-26 [?] Bioconductor
 P KernSmooth               2.23-20   2021-05-03 [?] CRAN (R 4.1.0)
 P knitr                    1.37      2021-12-16 [?] CRAN (R 4.1.0)
 P labeling                 0.4.2     2020-10-20 [?] CRAN (R 4.0.2)
 P later                    1.3.0     2021-08-18 [?] CRAN (R 4.1.0)
 P lattice                  0.20-45   2021-09-22 [?] CRAN (R 4.1.0)
 P lazyeval                 0.2.2     2019-03-15 [?] CRAN (R 4.1.0)
 P leiden                   0.3.9     2021-07-27 [?] CRAN (R 4.1.0)
 P lifecycle                1.0.1     2021-09-24 [?] CRAN (R 4.1.0)
 P limma                  * 3.50.0    2021-10-26 [?] Bioconductor
 P listenv                  0.8.0     2019-12-05 [?] CRAN (R 4.1.0)
 P lmtest                   0.9-39    2021-11-07 [?] CRAN (R 4.1.0)
 P locfit                   1.5-9.4   2020-03-25 [?] CRAN (R 4.1.0)
 P lubridate                1.8.0     2021-10-07 [?] CRAN (R 4.1.0)
 P magrittr                 2.0.1     2020-11-17 [?] CRAN (R 4.0.2)
 P MASS                     7.3-53.1  2021-02-12 [?] CRAN (R 4.0.2)
 P Matrix                 * 1.4-0     2021-12-08 [?] CRAN (R 4.1.0)
 P MatrixGenerics         * 1.6.0     2021-10-26 [?] Bioconductor
 P matrixStats            * 0.61.0    2021-09-17 [?] CRAN (R 4.1.0)
 P memoise                  2.0.1     2021-11-26 [?] CRAN (R 4.1.0)
 P metapod                  1.2.0     2021-10-26 [?] Bioconductor
 P mgcv                     1.8-38    2021-10-06 [?] CRAN (R 4.1.0)
 P mime                     0.12      2021-09-28 [?] CRAN (R 4.1.0)
 P miniUI                   0.1.1.1   2018-05-18 [?] CRAN (R 4.1.0)
 P modelr                   0.1.8     2020-05-19 [?] CRAN (R 4.0.2)
 P munsell                  0.5.0     2018-06-12 [?] CRAN (R 4.1.0)
 P nlme                     3.1-153   2021-09-07 [?] CRAN (R 4.1.0)
 P org.Hs.eg.db           * 3.14.0    2021-12-21 [?] Bioconductor
 P parallelly               1.30.0    2021-12-17 [?] CRAN (R 4.1.0)
 P patchwork              * 1.1.1     2020-12-17 [?] CRAN (R 4.0.2)
 P pbapply                  1.5-0     2021-09-16 [?] CRAN (R 4.1.0)
 P pillar                   1.6.4     2021-10-18 [?] CRAN (R 4.1.0)
 P pkgconfig                2.0.3     2019-09-22 [?] CRAN (R 4.1.0)
 P plotly                   4.10.0    2021-10-09 [?] CRAN (R 4.1.0)
 P plyr                     1.8.6     2020-03-03 [?] CRAN (R 4.0.2)
 P png                      0.1-7     2013-12-03 [?] CRAN (R 4.1.0)
 P polyclip                 1.10-0    2019-03-14 [?] CRAN (R 4.1.0)
 P prettyunits              1.1.1     2020-01-24 [?] CRAN (R 4.0.2)
 P processx                 3.5.2     2021-04-30 [?] CRAN (R 4.1.0)
 P progress                 1.2.2     2019-05-16 [?] CRAN (R 4.1.0)
 P promises                 1.2.0.1   2021-02-11 [?] CRAN (R 4.0.2)
 P ProtGenerics             1.26.0    2021-10-26 [?] Bioconductor
 P ps                       1.6.0     2021-02-28 [?] CRAN (R 4.1.0)
 P purrr                  * 0.3.4     2020-04-17 [?] CRAN (R 4.0.2)
 P R.methodsS3              1.8.1     2020-08-26 [?] CRAN (R 4.0.2)
 P R.oo                     1.24.0    2020-08-26 [?] CRAN (R 4.0.2)
 P R.utils                  2.11.0    2021-09-26 [?] CRAN (R 4.1.0)
 P R6                       2.5.1     2021-08-19 [?] CRAN (R 4.1.0)
 P RANN                     2.6.1     2019-01-08 [?] CRAN (R 4.1.0)
 P rappdirs                 0.3.3     2021-01-31 [?] CRAN (R 4.0.2)
 P RColorBrewer             1.1-2     2014-12-07 [?] CRAN (R 4.0.2)
 P Rcpp                     1.0.7     2021-07-07 [?] CRAN (R 4.1.0)
 P RcppAnnoy                0.0.19    2021-07-30 [?] CRAN (R 4.1.0)
   RCurl                    1.98-1.6  2022-02-08 [1] CRAN (R 4.1.0)
 P readr                  * 2.1.1     2021-11-30 [?] CRAN (R 4.1.0)
 P readxl                   1.3.1     2019-03-13 [?] CRAN (R 4.1.0)
 P renv                     0.15.0-14 2022-01-10 [?] Github (rstudio/renv@a3b90eb)
 P reprex                   2.0.1     2021-08-05 [?] CRAN (R 4.1.0)
 P reshape2               * 1.4.4     2020-04-09 [?] CRAN (R 4.1.0)
 P restfulr                 0.0.13    2017-08-06 [?] CRAN (R 4.1.0)
 P reticulate               1.22      2021-09-17 [?] CRAN (R 4.1.0)
 P rhdf5                    2.38.0    2021-10-26 [?] Bioconductor
 P rhdf5filters             1.6.0     2021-10-26 [?] Bioconductor
 P Rhdf5lib                 1.16.0    2021-10-26 [?] Bioconductor
 P rjson                    0.2.21    2022-01-09 [?] CRAN (R 4.1.0)
 P rlang                    0.4.12    2021-10-18 [?] CRAN (R 4.1.0)
 P rmarkdown                2.11      2021-09-14 [?] CRAN (R 4.1.0)
 P ROCR                     1.0-11    2020-05-02 [?] CRAN (R 4.1.0)
 P rpart                    4.1-15    2019-04-12 [?] CRAN (R 4.1.0)
 P rprojroot                2.0.2     2020-11-15 [?] CRAN (R 4.0.2)
 P Rsamtools                2.10.0    2021-10-26 [?] Bioconductor
 P RSQLite                  2.2.9     2021-12-06 [?] CRAN (R 4.1.0)
 P rstudioapi               0.13      2020-11-12 [?] CRAN (R 4.0.2)
 P rsvd                     1.0.5     2021-04-16 [?] CRAN (R 4.1.0)
 P rtracklayer              1.54.0    2021-10-26 [?] Bioconductor
 P Rtsne                    0.15      2018-11-10 [?] CRAN (R 4.1.0)
 P rvest                    1.0.2     2021-10-16 [?] CRAN (R 4.1.0)
 P S4Vectors              * 0.32.3    2021-11-21 [?] Bioconductor
 P sass                     0.4.0     2021-05-12 [?] CRAN (R 4.1.0)
 P ScaledMatrix             1.2.0     2021-10-26 [?] Bioconductor
 P scales                 * 1.1.1     2020-05-11 [?] CRAN (R 4.0.2)
 P scater                 * 1.22.0    2021-10-26 [?] Bioconductor
 P scattermore              0.7       2020-11-24 [?] CRAN (R 4.1.0)
   scGate                 * 1.0.1     2022-02-15 [1] Github (carmonalab/scGate@4cd1db6)
 P scran                  * 1.22.1    2021-11-14 [?] Bioconductor
 P sctransform              0.3.2     2020-12-16 [?] CRAN (R 4.1.0)
 P scuttle                * 1.4.0     2021-10-26 [?] Bioconductor
 P sessioninfo              1.2.2     2021-12-06 [?] CRAN (R 4.1.0)
 P Seurat                 * 4.0.6     2021-12-16 [?] CRAN (R 4.1.0)
 P SeuratObject           * 4.0.4     2021-11-23 [?] CRAN (R 4.1.0)
 P shiny                    1.7.1     2021-10-02 [?] CRAN (R 4.1.0)
 P SingleCellExperiment   * 1.16.0    2021-10-26 [?] Bioconductor
 P snakecase                0.11.0    2019-05-25 [?] CRAN (R 4.0.2)
 P sparseMatrixStats        1.6.0     2021-10-26 [?] Bioconductor
 P spatstat.core            2.3-2     2021-11-26 [?] CRAN (R 4.1.0)
 P spatstat.data            2.1-2     2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.geom            2.3-1     2021-12-10 [?] CRAN (R 4.1.0)
 P spatstat.sparse          2.1-0     2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.utils           2.3-0     2021-12-12 [?] CRAN (R 4.1.0)
 P statmod                  1.4.36    2021-05-10 [?] CRAN (R 4.1.0)
 P stringi                  1.7.6     2021-11-29 [?] CRAN (R 4.1.0)
 P stringr                * 1.4.0     2019-02-10 [?] CRAN (R 4.0.2)
 P SummarizedExperiment   * 1.24.0    2021-10-26 [?] Bioconductor
 P survival                 3.2-13    2021-08-24 [?] CRAN (R 4.1.0)
 P tensor                   1.5       2012-05-05 [?] CRAN (R 4.1.0)
 P tibble                 * 3.1.6     2021-11-07 [?] CRAN (R 4.1.0)
 P tidygraph                1.2.0     2020-05-12 [?] CRAN (R 4.0.2)
 P tidyr                  * 1.1.4     2021-09-27 [?] CRAN (R 4.1.0)
 P tidyselect               1.1.1     2021-04-30 [?] CRAN (R 4.1.0)
 P tidyverse              * 1.3.1     2021-04-15 [?] CRAN (R 4.1.0)
 P tweenr                   1.0.2     2021-03-23 [?] CRAN (R 4.1.0)
 P tzdb                     0.2.0     2021-10-27 [?] CRAN (R 4.1.0)
   UCell                  * 1.99.1    2022-02-15 [1] Github (carmonalab/UCell@77fad0f)
 P utf8                     1.2.2     2021-07-24 [?] CRAN (R 4.1.0)
 P uwot                     0.1.11    2021-12-02 [?] CRAN (R 4.1.0)
 P vctrs                    0.3.8     2021-04-29 [?] CRAN (R 4.0.2)
 P vipor                    0.4.5     2017-03-22 [?] CRAN (R 4.1.0)
 P viridis                  0.6.2     2021-10-13 [?] CRAN (R 4.1.0)
 P viridisLite              0.4.0     2021-04-13 [?] CRAN (R 4.0.2)
 P whisker                  0.4       2019-08-28 [?] CRAN (R 4.0.2)
 P withr                    2.4.3     2021-11-30 [?] CRAN (R 4.1.0)
 P workflowr              * 1.7.0     2021-12-21 [?] CRAN (R 4.1.0)
 P xfun                     0.29      2021-12-14 [?] CRAN (R 4.1.0)
 P XML                      3.99-0.8  2021-09-17 [?] CRAN (R 4.1.0)
 P xml2                     1.3.3     2021-11-30 [?] CRAN (R 4.1.0)
 P xtable                   1.8-4     2019-04-21 [?] CRAN (R 4.1.0)
 P XVector                  0.34.0    2021-10-26 [?] Bioconductor
 P yaml                     2.2.1     2020-02-01 [?] CRAN (R 4.0.2)
 P zlibbioc                 1.40.0    2021-10-26 [?] Bioconductor
 P zoo                      1.8-9     2021-03-09 [?] CRAN (R 4.1.0)

 [1] /oshlack_lab/jovana.maksimovic/projects/MCRI/melanie.neeland/paed-cf-cite-seq/renv/library/R-4.1/x86_64-pc-linux-gnu
 [2] /config/binaries/R/4.1.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</details>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.1.0/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.1.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] scGate_1.0.1                reshape2_1.4.4             
 [3] ggridges_0.5.3              UCell_1.99.1               
 [5] data.table_1.14.2           ggalluvial_0.12.3          
 [7] scales_1.1.1                janitor_2.1.0              
 [9] BiocParallel_1.28.3         glmGamPoi_1.6.0            
[11] clustree_0.4.4              ggraph_2.0.5               
[13] org.Hs.eg.db_3.14.0         ensembldb_2.18.2           
[15] AnnotationFilter_1.18.0     GenomicFeatures_1.46.3     
[17] AnnotationDbi_1.56.2        AnnotationHub_3.2.0        
[19] BiocFileCache_2.2.0         dbplyr_2.1.1               
[21] patchwork_1.1.1             edgeR_3.36.0               
[23] limma_3.50.0                Matrix_1.4-0               
[25] scater_1.22.0               scran_1.22.1               
[27] scuttle_1.4.0               DropletUtils_1.14.1        
[29] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0
[31] Biobase_2.54.0              GenomicRanges_1.46.1       
[33] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[35] S4Vectors_0.32.3            BiocGenerics_0.40.0        
[37] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[39] glue_1.6.0                  here_1.0.1                 
[41] forcats_0.5.1               stringr_1.4.0              
[43] dplyr_1.0.7                 purrr_0.3.4                
[45] readr_2.1.1                 tidyr_1.1.4                
[47] tibble_3.1.6                ggplot2_3.3.5              
[49] tidyverse_1.3.1             SeuratObject_4.0.4         
[51] Seurat_4.0.6                workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                rtracklayer_1.54.0           
  [3] scattermore_0.7               R.methodsS3_1.8.1            
  [5] bit64_4.0.5                   knitr_1.37                   
  [7] irlba_2.3.5                   DelayedArray_0.20.0          
  [9] R.utils_2.11.0                rpart_4.1-15                 
 [11] KEGGREST_1.34.0               RCurl_1.98-1.6               
 [13] generics_0.1.1                ScaledMatrix_1.2.0           
 [15] callr_3.7.0                   cowplot_1.1.1                
 [17] RSQLite_2.2.9                 RANN_2.6.1                   
 [19] future_1.23.0                 bit_4.0.4                    
 [21] tzdb_0.2.0                    spatstat.data_2.1-2          
 [23] xml2_1.3.3                    lubridate_1.8.0              
 [25] httpuv_1.6.5                  assertthat_0.2.1             
 [27] viridis_0.6.2                 xfun_0.29                    
 [29] hms_1.1.1                     jquerylib_0.1.4              
 [31] evaluate_0.14                 promises_1.2.0.1             
 [33] restfulr_0.0.13               progress_1.2.2               
 [35] fansi_1.0.0                   readxl_1.3.1                 
 [37] igraph_1.2.11                 DBI_1.1.2                    
 [39] htmlwidgets_1.5.4             spatstat.geom_2.3-1          
 [41] ellipsis_0.3.2                backports_1.4.1              
 [43] biomaRt_2.50.1                deldir_1.0-6                 
 [45] sparseMatrixStats_1.6.0       vctrs_0.3.8                  
 [47] ROCR_1.0-11                   abind_1.4-5                  
 [49] cachem_1.0.6                  withr_2.4.3                  
 [51] ggforce_0.3.3                 sctransform_0.3.2            
 [53] GenomicAlignments_1.30.0      prettyunits_1.1.1            
 [55] goftest_1.2-3                 cluster_2.1.2                
 [57] lazyeval_0.2.2                crayon_1.4.2                 
 [59] labeling_0.4.2                pkgconfig_2.0.3              
 [61] tweenr_1.0.2                  ProtGenerics_1.26.0          
 [63] nlme_3.1-153                  vipor_0.4.5                  
 [65] rlang_0.4.12                  globals_0.14.0               
 [67] lifecycle_1.0.1               miniUI_0.1.1.1               
 [69] filelock_1.0.2                modelr_0.1.8                 
 [71] rsvd_1.0.5                    cellranger_1.1.0             
 [73] rprojroot_2.0.2               polyclip_1.10-0              
 [75] lmtest_0.9-39                 Rhdf5lib_1.16.0              
 [77] zoo_1.8-9                     reprex_2.0.1                 
 [79] beeswarm_0.4.0                whisker_0.4                  
 [81] processx_3.5.2                rjson_0.2.21                 
 [83] png_0.1-7                     viridisLite_0.4.0            
 [85] bitops_1.0-7                  getPass_0.2-2                
 [87] R.oo_1.24.0                   KernSmooth_2.23-20           
 [89] rhdf5filters_1.6.0            Biostrings_2.62.0            
 [91] blob_1.2.2                    DelayedMatrixStats_1.16.0    
 [93] parallelly_1.30.0             beachmat_2.10.0              
 [95] memoise_2.0.1                 magrittr_2.0.1               
 [97] plyr_1.8.6                    ica_1.0-2                    
 [99] zlibbioc_1.40.0               compiler_4.1.0               
[101] BiocIO_1.4.0                  dqrng_0.3.0                  
[103] RColorBrewer_1.1-2            fitdistrplus_1.1-6           
[105] snakecase_0.11.0              Rsamtools_2.10.0             
[107] cli_3.1.0                     XVector_0.34.0               
[109] listenv_0.8.0                 pbapply_1.5-0                
[111] ps_1.6.0                      MASS_7.3-53.1                
[113] mgcv_1.8-38                   tidyselect_1.1.1             
[115] stringi_1.7.6                 highr_0.9                    
[117] yaml_2.2.1                    BiocSingular_1.10.0          
[119] locfit_1.5-9.4                ggrepel_0.9.1                
[121] grid_4.1.0                    sass_0.4.0                   
[123] tools_4.1.0                   future.apply_1.8.1           
[125] parallel_4.1.0                rstudioapi_0.13              
[127] bluster_1.4.0                 git2r_0.29.0                 
[129] metapod_1.2.0                 gridExtra_2.3                
[131] farver_2.1.0                  Rtsne_0.15                   
[133] digest_0.6.29                 BiocManager_1.30.16          
[135] shiny_1.7.1                   Rcpp_1.0.7                   
[137] broom_0.7.11                  BiocVersion_3.14.0           
[139] later_1.3.0                   RcppAnnoy_0.0.19             
[141] httr_1.4.2                    colorspace_2.0-2             
[143] XML_3.99-0.8                  rvest_1.0.2                  
[145] fs_1.5.2                      tensor_1.5                   
[147] reticulate_1.22               splines_4.1.0                
[149] uwot_0.1.11                   statmod_1.4.36               
[151] spatstat.utils_2.3-0          graphlayouts_0.8.0           
[153] renv_0.15.0-14                sessioninfo_1.2.2            
[155] plotly_4.10.0                 xtable_1.8-4                 
[157] jsonlite_1.7.2                tidygraph_1.2.0              
[159] R6_2.5.1                      pillar_1.6.4                 
[161] htmltools_0.5.2               mime_0.12                    
[163] fastmap_1.1.0                 BiocNeighbors_1.12.0         
[165] interactiveDisplayBase_1.32.0 codetools_0.2-18             
[167] utf8_1.2.2                    lattice_0.20-45              
[169] bslib_0.3.1                   spatstat.sparse_2.1-0        
[171] curl_4.3.2                    ggbeeswarm_0.6.0             
[173] leiden_0.3.9                  survival_3.2-13              
[175] rmarkdown_2.11                munsell_0.5.0                
[177] rhdf5_2.38.0                  GenomeInfoDbData_1.2.7       
[179] HDF5Array_1.22.1              haven_2.4.3                  
[181] gtable_0.3.0                  spatstat.core_2.3-2          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
