<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2022-05-26" />

<title>Analysis of DGE Between Groups, Within Cell Labels in C133_Neeland Data</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-cf-cite-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of DGE Between Groups, Within Cell Labels in C133_Neeland Data</h1>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">May 26, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-cf-cite-seq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210524code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210524)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210524)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree8708a60068488f532dbe530d9873a81dbb67e774targetblank8708a60a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/8708a60068488f532dbe530d9873a81dbb67e774" target="_blank">8708a60</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree8708a60068488f532dbe530d9873a81dbb67e774targetblank8708a60a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/8708a60068488f532dbe530d9873a81dbb67e774" target="_blank">8708a60</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/CellRanger/
    Ignored:    data/emptyDrops/bin/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/03_COMBO.annotation_markers.Rmd
    Untracked:  analysis/03_COMBO.protein_analysis.Rmd
    Untracked:  analysis/04_COMBO.compare_annotations.Rmd
    Untracked:  analysis/obsolete/
    Untracked:  analysis/scGateDB/
    Untracked:  code/03_C133_Neeland-dropletutils.R
    Untracked:  code/03_COMBO.proteins_geneBasis.R
    Untracked:  code/deidentify.R
    Untracked:  data/190930_A00152_0150_BHTYCMDSXX/
    Untracked:  data/CITE-seq_pilot_proportions_final.csv
    Untracked:  data/Flow_data_CITEseq_2-8.csv
    Untracked:  data/GSE127465_RAW/
    Untracked:  data/GSE130148_barcodes_cell_types.txt
    Untracked:  data/GSE130148_raw_counts.RData
    Untracked:  data/Macrophage_subcluster_annotation_110322.csv
    Untracked:  data/Other_subcluster_annotation_110322.csv
    Untracked:  data/Proteins_T-NK_22.04.22.csv
    Untracked:  data/Proteins_broad_22.04.22.csv
    Untracked:  data/Proteins_macs_22.04.22.csv
    Untracked:  data/Proteins_other_22.04.22.csv
    Untracked:  data/T-NK_subcluster_annotation_110322.csv
    Untracked:  data/T-NK_subcluster_proteins.csv
    Untracked:  data/T-NK_subcluster_proteins_110322.csv
    Untracked:  data/T-NK_subclusters_cytokines.csv
    Untracked:  data/T-NK_subclusters_markergenes_110322.csv
    Untracked:  data/TNK_subcluster_annotation_09.04.22.csv
    Untracked:  data/all_azimuth_cluster_cytokines.csv
    Untracked:  data/all_azimuth_clusters_proteins.csv
    Untracked:  data/cell_type_category_rna_Granulocytes_Cell.tsv
    Untracked:  data/consolidated_markers.rds
    Untracked:  data/gb-genes-50.rds
    Untracked:  data/genome1K.phase3.SNP_AF5e4.chr1toX.hg38.vcf.gz
    Untracked:  data/macrophage_subcluster_annotation_09.04.22.csv
    Untracked:  data/macrophage_subcluster_cytokines.csv
    Untracked:  data/macrophage_subcluster_markergenes_110322.csv
    Untracked:  data/macrophage_subcluster_markergenes_160322.csv
    Untracked:  data/macrophage_subcluster_proteins.csv
    Untracked:  data/macrophage_subcluster_proteins_110322.csv
    Untracked:  data/nCoV.all.cell.annotation.meta.txt
    Untracked:  data/nCoV.rds
    Untracked:  data/neutrophil.tsv
    Untracked:  data/other_subcluster_proteins.csv
    Untracked:  data/other_subcluster_proteins_110322.csv
    Untracked:  data/other_subclusters_cytokines.csv
    Untracked:  data/other_subclusters_markergenes_110322.csv
    Untracked:  data/others_subcluster_annotation_09.04.22.csv
    Untracked:  data/proteins_ranked.rds
    Untracked:  data/sample_sheets/Hashtag_Feature_Ref_With_Clin.csv
    Untracked:  data/sample_sheets/Sample_information.csv
    Untracked:  data/sample_sheets/TotalSeq_A_Human_Universal_Cocktail_Proteins of interest_29.09.21.csv
    Untracked:  data/sample_sheets/TotalSeq_A_Human_Universal_Cocktail_Proteins_of_interest_28.06.21.csv
    Untracked:  output/ADT-counts.csv
    Untracked:  output/C133_Neeland.azimuth.individual_cell_type_proportions.csv
    Untracked:  output/C133_Neeland.azimuth.phenotype_information.csv
    Untracked:  output/C133_Neeland.individual_cell_type_proportions.csv
    Untracked:  output/C133_Neeland.phenotype_information.csv
    Untracked:  output/DEGs-New/
    Untracked:  output/DEGs/
    Untracked:  output/glimma-plots/js/labelled-singlets-md-plot.js
    Untracked:  output/glimma-plots/js/qc-md-plot_03_C133_Neeland.js
    Untracked:  output/glimma-plots/js/qc-md-plot_C133_Neeland.js
    Untracked:  output/glimma-plots/js/qc-md-plot_CF-BAL-Pilot.js
    Untracked:  output/glimma-plots/labelled-singlets-md-plot.html
    Untracked:  output/glimma-plots/qc-md-plot_03_C133_Neeland.html
    Untracked:  output/glimma-plots/qc-md-plot_C133_Neeland.html
    Untracked:  output/glimma-plots/qc-md-plot_CF-BAL-Pilot.html
    Untracked:  output/marker-analysis-mod/
    Untracked:  output/marker-analysis/
    Untracked:  output/metadata.csv
    Untracked:  output/protein_labels.pdf
    Untracked:  scGateDB/
    Untracked:  wflow_background.R

Unstaged changes:
    Modified:   analysis/03_COMBO.cluster_macrophages.Rmd
    Modified:   analysis/03_COMBO.compare_annotations.Rmd
    Modified:   analysis/03_COMBO.proportion_analysis.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/utility.R
    Modified:   data/SCEs/C133_Neeland.CellRanger.SCE.rds
    Modified:   data/SCEs/C133_Neeland.demultiplexed.SCE.rds
    Modified:   data/sample_sheets/TotalSeq-A_Universal_Cocktail_v1.0.csv
    Modified:   data/vireo/C133_1/GT_donors.vireo.vcf.gz
    Modified:   data/vireo/C133_1/_log.txt
    Modified:   data/vireo/C133_1/donor_ids.tsv
    Modified:   data/vireo/C133_1/fig_GT_distance_estimated.pdf
    Modified:   data/vireo/C133_1/prob_doublet.tsv.gz
    Modified:   data/vireo/C133_1/prob_singlet.tsv.gz
    Modified:   data/vireo/C133_1/summary.tsv
    Modified:   data/vireo/C133_2/GT_donors.vireo.vcf.gz
    Modified:   data/vireo/C133_2/_log.txt
    Modified:   data/vireo/C133_2/donor_ids.tsv
    Modified:   data/vireo/C133_2/fig_GT_distance_estimated.pdf
    Modified:   data/vireo/C133_2/prob_doublet.tsv.gz
    Modified:   data/vireo/C133_2/prob_singlet.tsv.gz
    Modified:   data/vireo/C133_2/summary.tsv
    Modified:   output/glimma-plots/js/qc-md-plot.js
    Modified:   output/glimma-plots/js/stripped-nuclei-md-plot.js
    Modified:   renv.lock
    Modified:   renv/.gitignore
    Modified:   renv/activate.R
    Modified:   renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/04_COMBO.expression_analysis.Rmd</code>) and HTML (<code>docs/04_COMBO.expression_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/8708a60068488f532dbe530d9873a81dbb67e774/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">8708a60</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-26
</td>
<td>
wflow_publish(c("analysis/04_COMBO.expression_analysis.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/04_COMBO.expression_analysis.html" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f0d21d0191712aff1ec0552ffab590af7f7f794c/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">f0d21d0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
<td>
wflow_publish(c("analysis/04_COMBO.expression_analysis.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9bb5b6fcac1a249a12145f29696ff4a33cfc8abd/docs/04_COMBO.expression_analysis.html" target="_blank">9bb5b6f</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/58f2eeadbf8d449ededa92dd212929b4539ccf56/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">58f2eea</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
<td>
wflow_publish(c("analysis/04_COMBO.expression_analysis.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/4f769d452aa920442673e5c102cd46a3cafe88ba/docs/04_COMBO.expression_analysis.html" target="_blank">4f769d4</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/535a3b2a2b22f788e63c2eed6e0806dfb0cb5cee/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">535a3b2</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
<td>
wflow_publish(c("analysis/04_COMBO.expression_analysis.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/04_COMBO.expression_analysis.html" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/23ba2d207646c114553b4fc83a60070bfdd994e5/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">23ba2d2</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
<td>
wflow_publish("analysis/04_COMBO.expression_analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9a6422117a9d227a8ac3ff661b08e5b39e2ced51/docs/04_COMBO.expression_analysis.html" target="_blank">9a64221</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/82df4fb24302c183a10833a871eb04ca64f96758/analysis/04_COMBO.expression_analysis.Rmd" target="_blank">82df4fb</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
<td>
wflow_publish("analysis/04_COMBO.expression_analysis.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>library(Seurat)
library(tidyverse)
library(here)
library(glue)
library(DropletUtils)
library(scran)
library(scater)
library(scuttle)
library(Matrix)
library(edgeR)
library(patchwork)
library(AnnotationHub)
library(ensembldb)
library(EGSEA)
library(org.Hs.eg.db)
library(glmGamPoi)
library(BiocParallel)
library(janitor)
library(scales)
library(speckle)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(missMethyl)
source(here(&quot;code/utility.R&quot;))
source(here(&quot;code/helper_functions.R&quot;))

set.seed(42)
options(scipen=0)
options(future.globals.maxSize = 6500 * 1024^2)</code></pre>
<div id="load-data" class="section level2">
<h2>Load Data</h2>
<pre class="r"><code>seu1 &lt;- readRDS(here(&quot;data/SCEs/04_COMBO.clean_macrophages_diet.SEU.rds&quot;))
seu2 &lt;- readRDS(here(&quot;data/SCEs/04_COMBO.clean_tcells_diet.SEU.rds&quot;))
seu3 &lt;- readRDS(here(&quot;data/SCEs/04_COMBO.clean_others_diet.SEU.rds&quot;))
seu &lt;- merge(seu1, y = c(seu2, seu3))</code></pre>
<pre><code>            used   (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells  21758060 1162.1   34390851  1836.7   21952564  1172.4
Vcells 699211520 5334.6 1697073267 12947.7 1414102097 10788.8</code></pre>
</div>
<div id="load-clinical-information" class="section level2">
<h2>Load clinical information</h2>
<p>Import clinical characteristics and patient information and associate with <code>genetic_donor</code> IDs.</p>
<pre class="r"><code>info &lt;- read.csv(file = here(&quot;data/sample_sheets/Sample_information.csv&quot;))
tab &lt;- table(seu$HTO, seu$donor)

apply(tab, 2, function(x){
  names(which(x == max(x)))
}) %&gt;% data.frame %&gt;%
  dplyr::rename(&quot;HTO&quot; = &quot;.&quot;) %&gt;%
  rownames_to_column(var = &quot;donor&quot;) %&gt;%
  inner_join(info, by = c(&quot;HTO&quot; = &quot;Sample&quot;)) %&gt;%
  mutate(Batch = factor(Batch)) -&gt; info

info %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="left">HTO</th>
<th align="left">Participant</th>
<th align="left">Sex</th>
<th align="right">Age</th>
<th align="left">Disease</th>
<th align="left">Batch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DM</td>
<td align="left">DM</td>
<td align="left">B1_CF</td>
<td align="left">M</td>
<td align="right">2.99</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">donor_A</td>
<td align="left">Human_HTO_8</td>
<td align="left">L2_CF</td>
<td align="left">M</td>
<td align="right">5.95</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_B</td>
<td align="left">Human_HTO_1</td>
<td align="left">E2_CF</td>
<td align="left">F</td>
<td align="right">5.99</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_C</td>
<td align="left">Human_HTO_4</td>
<td align="left">H2_CF</td>
<td align="left">F</td>
<td align="right">5.89</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_D</td>
<td align="left">Human_HTO_6</td>
<td align="left">J2_CF</td>
<td align="left">M</td>
<td align="right">5.05</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_E</td>
<td align="left">Human_HTO_3</td>
<td align="left">G2_CF</td>
<td align="left">F</td>
<td align="right">4.91</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_F</td>
<td align="left">Human_HTO_5</td>
<td align="left">I2_CF</td>
<td align="left">F</td>
<td align="right">5.93</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_G</td>
<td align="left">Human_HTO_2</td>
<td align="left">F2_CF</td>
<td align="left">F</td>
<td align="right">6.02</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_H</td>
<td align="left">Human_HTO_7</td>
<td align="left">K2_CF</td>
<td align="left">M</td>
<td align="right">4.92</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">JC</td>
<td align="left">JC</td>
<td align="left">C1_CF</td>
<td align="left">M</td>
<td align="right">2.99</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">JLG</td>
<td align="left">JLG</td>
<td align="left">A1_Ctrl</td>
<td align="left">M</td>
<td align="right">3.00</td>
<td align="left">Ctrl</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">LC</td>
<td align="left">LC</td>
<td align="left">D1_CF</td>
<td align="left">M</td>
<td align="right">3.03</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="differences-in-gene-expression" class="section level1">
<h1>Differences in gene expression</h1>
<div id="test-for-dge-within-cell-types" class="section level2">
<h2>Test for DGE within cell types</h2>
<pre class="r"><code>seu@meta.data %&gt;%
  left_join(info, by = c(&quot;donor&quot; = &quot;donor&quot;)) -&gt; mdata

# Create single cell experiment object
sce &lt;- SingleCellExperiment(assays = list(counts = seu[[&quot;RNA&quot;]]@counts), 
                            colData = mdata)</code></pre>
<div id="add-gene-annotation" class="section level3">
<h3>Add gene annotation</h3>
<pre class="r"><code>columns(org.Hs.eg.db)</code></pre>
<pre><code> [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
 [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;    
[11] &quot;GENETYPE&quot;     &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;         
[16] &quot;OMIM&quot;         &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;        
[21] &quot;PMID&quot;         &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      
[26] &quot;UNIPROT&quot;     </code></pre>
<pre class="r"><code>ann &lt;- AnnotationDbi:::select(org.Hs.eg.db,
                              keys = rownames(sce),
                              columns=c(&quot;SYMBOL&quot;,
                                        &quot;ENTREZID&quot;,
                                        &quot;ENSEMBL&quot;,
                                        &quot;GENENAME&quot;,
                                        &quot;CHR&quot;),
                              keytype = &quot;SYMBOL&quot;)
m &lt;- match(rownames(sce), ann$SYMBOL)
ann &lt;- ann[m,]
table(ann$SYMBOL == rownames(sce))</code></pre>
<pre><code>
 TRUE 
15578 </code></pre>
<pre class="r"><code>rowData(sce) &lt;- DataFrame(ann)</code></pre>
<p>Heatmap showing the abundance of cells with each combination of donor (row) and cell type label (column). The color scale represents the log2-count for each combination.</p>
<pre class="r"><code>library(tidyHeatmap)
table(sce$donor, sce$Broad) %&gt;% data.frame %&gt;%
  dplyr::rename(donor = Var1, `Cell Label` = Var2) %&gt;%
  inner_join(info) %&gt;%
  mutate(`Log2 No. Cells` = log2(Freq + 1),
         Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.row = Participant, 
          .column = `Cell Label`,
          .value = `Log2 No. Cells`,
          scale = &quot;row&quot;,
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2), 
                                             viridis::magma(5))) %&gt;%
  add_tile(Batch) %&gt;%
  add_tile(Sex) %&gt;%
  add_tile(Age) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9bb5b6fcac1a249a12145f29696ff4a33cfc8abd/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">9bb5b6f</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/4f769d452aa920442673e5c102cd46a3cafe88ba/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">4f769d4</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="create-pseudobulk-samples" class="section level1">
<h1>Create pseudobulk samples</h1>
<p>Using ‘label’ and ‘sample’ as our two factors; each column of the output corresponds to one unique combination of these two factors.</p>
<pre class="r"><code>summed &lt;- aggregateAcrossCells(sce, 
                               id = colData(sce)[, c(&quot;Broad&quot;, &quot;donor&quot;)])
summed</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 15578 137 
metadata(0):
assays(1): counts
rownames(15578): FAM87B LINC00115 ... PRMT2 MAFIP
rowData names(5): SYMBOL ENTREZID ENSEMBL GENENAME CHR
colnames: NULL
colData names(68): orig.ident nCount_RNA ... donor ncells
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):</code></pre>
<pre class="r"><code>y.pb &lt;- DGEList(counts(summed), 
                samples = colData(summed))
y.pb</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
          Sample1 Sample2 Sample3 Sample4 Sample5 Sample6 Sample7 Sample8
FAM87B          0       0       0       0       0       0       0       0
LINC00115      20       0       0       0       2       0       1       0
FAM41C         33       3       4       0       0       2       0       0
SAMD11          0       0       0       0       0       0       0       0
NOC2L         280      25      47       2       6       3       1       2
          Sample9 Sample10 Sample11 Sample12 Sample13 Sample14 Sample15
FAM87B          0        0        0        0        0        0        0
LINC00115       1        2        4        1       12        2        0
FAM41C          0        1       12        2       12        1        1
SAMD11          0        1        0        1        1        0        0
NOC2L           2       49      122       30      109       42        3
          Sample16 Sample17 Sample18 Sample19 Sample20 Sample21 Sample22
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        0        4        1        9
FAM41C           1        1        1        0        2        1        2
SAMD11           0        0        0        0        0        0        0
NOC2L           19        7        8       11       57        2       63
          Sample23 Sample24 Sample25 Sample26 Sample27 Sample28 Sample29
FAM87B           0        0        0        0        0        0        0
LINC00115        5        3        1        0        0        0        2
FAM41C           4        5        6        0        0        1        0
SAMD11           0        0        0        0        0        0        1
NOC2L           92       60       72       14        2        8        8
          Sample30 Sample31 Sample32 Sample33 Sample34 Sample35 Sample36
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        1        0        2        4        7
FAM41C           0        0        1        1        1        4        4
SAMD11           0        0        0        0        0        0        0
NOC2L            3        3       27        3       19      111      101
          Sample37 Sample38 Sample39 Sample40 Sample41 Sample42 Sample43
FAM87B           1        0        0        0        0        0        0
LINC00115        6        3        0        0        0        0        0
FAM41C           4        0        0        3        2        2        0
SAMD11           0        0        1        0        0        0        0
NOC2L           34       27       12       10       11       18        5
          Sample44 Sample45 Sample46 Sample47 Sample48 Sample49 Sample50
FAM87B           0        0        1        0        0        0        0
LINC00115        2        1       15       37       13        2        0
FAM41C           1        0        3       13        2        0        0
SAMD11           1        0        2        1        0        0        0
NOC2L           22       14       93      201      103        1        2
          Sample51 Sample52 Sample53 Sample54 Sample55 Sample56 Sample57
FAM87B           0        0        0        0        0        0        0
LINC00115        1        1        0        0        0        0        0
FAM41C          21        3        7        0        0        1        0
SAMD11           1        0        0        0        0        0        0
NOC2L           55       19       31        1        0        1        1
          Sample58 Sample59 Sample60 Sample61 Sample62 Sample63 Sample64
FAM87B           0        0        0        0        0        0        0
LINC00115        0        3        3        3        0        0        0
FAM41C           1        0       12       12        0        1        0
SAMD11           0        1        0        1        0        0        0
NOC2L            3       25       45       79        3       23        0
          Sample65 Sample66 Sample67 Sample68 Sample69 Sample70 Sample71
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        1        0        0        0
FAM41C           0        0        0        1        0        0        0
SAMD11           0        0        0        0        0        0        0
NOC2L            1        0        0        6        0        2        5
          Sample72 Sample73 Sample74 Sample75 Sample76 Sample77 Sample78
FAM87B           0       12        0        3        2        2        7
LINC00115        0       78        7        5        7       19       42
FAM41C           0      239       10       30       66       38      113
SAMD11           0        1        0        1        2        1        4
NOC2L            5     1385      210      358      672      693     1197
          Sample79 Sample80 Sample81 Sample82 Sample83 Sample84 Sample85
FAM87B           2        3        3       19       24       18        0
LINC00115       26       18       29      127      138      231        0
FAM41C          68       50       77      236      309      278        2
SAMD11           5        2        3        2        2        5        0
NOC2L         1193      641      972     3197     5091     5766        2
          Sample86 Sample87 Sample88 Sample89 Sample90 Sample91 Sample92
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        0        0        0        0
FAM41C           1        0        0        0        0        0        2
SAMD11           0        0        0        0        0        0        0
NOC2L            3        0        0        1        1        5        1
          Sample93 Sample94 Sample95 Sample96 Sample97 Sample98 Sample99
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        2        1        0        0
FAM41C           0        0        0        0        0        1        0
SAMD11           0        0        0        0        0        0        0
NOC2L            0        4        5        8        5        3        7
          Sample100 Sample101 Sample102 Sample103 Sample104 Sample105 Sample106
FAM87B            0         1         0         0         0         0         0
LINC00115         2        13         5         0         0         1         0
FAM41C            1        13         0         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L            31        73        36         8         3         1         0
          Sample107 Sample108 Sample109 Sample110 Sample111 Sample112 Sample113
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         0         0         0         0         0
FAM41C            0         0         0         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             0         0         0         0         0         5         5
          Sample114 Sample115 Sample116 Sample117 Sample118 Sample119 Sample120
FAM87B            0         0         0         0         0         0         0
LINC00115         0         3         1         0         0         0         0
FAM41C            0         1         1         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             2        29         2         1         4         0         1
          Sample121 Sample122 Sample123 Sample124 Sample125 Sample126 Sample127
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         0         3         3         0         1
FAM41C            0         0         0         0         2         3         0
SAMD11            0         0         0         1         1         0         0
NOC2L             2         9         0        17        19        23         7
          Sample128 Sample129 Sample130 Sample131 Sample132 Sample133 Sample134
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         0         0         0         0         0
FAM41C            0         0         1         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             4         3         3         0         0         1         3
          Sample135 Sample136 Sample137
FAM87B            0         0         0
LINC00115         0         1         0
FAM41C            1         0         0
SAMD11            0         0         0
NOC2L            25        16         6
15573 more rows ...

$samples
        group lib.size norm.factors orig.ident nCount_RNA nFeature_RNA Barcode
Sample1     1  2896339            1        A-1         NA           NA    &lt;NA&gt;
Sample2     1   287510            1       &lt;NA&gt;         NA           NA    &lt;NA&gt;
Sample3     1   504697            1       &lt;NA&gt;         NA           NA    &lt;NA&gt;
Sample4     1    27511            1       &lt;NA&gt;         NA           NA    &lt;NA&gt;
Sample5     1    80548            1       &lt;NA&gt;         NA           NA    &lt;NA&gt;
        capture nuclear_fraction cell_status       HTO.x   donor experiment sum
Sample1      DM               NA        cell          DM      DM          1  NA
Sample2    &lt;NA&gt;               NA        cell Human_HTO_8 donor_A          2  NA
Sample3    &lt;NA&gt;               NA        &lt;NA&gt; Human_HTO_1 donor_B          2  NA
Sample4    &lt;NA&gt;               NA        cell Human_HTO_4 donor_C          2  NA
Sample5    &lt;NA&gt;               NA        cell        &lt;NA&gt; donor_D          2  NA
        detected subsets_Mito_sum subsets_Mito_detected subsets_Mito_percent
Sample1       NA               NA                    NA                   NA
Sample2       NA               NA                    NA                   NA
Sample3       NA               NA                    NA                   NA
Sample4       NA               NA                    NA                   NA
Sample5       NA               NA                    NA                   NA
        subsets_Ribo_sum subsets_Ribo_detected subsets_Ribo_percent total
Sample1               NA                    NA                   NA    NA
Sample2               NA                    NA                   NA    NA
Sample3               NA                    NA                   NA    NA
Sample4               NA                    NA                   NA    NA
Sample5               NA                    NA                   NA    NA
        zero_percent predicted.annotation.l1 predicted.annotation.l1.score
Sample1           NA                    &lt;NA&gt;                            NA
Sample2           NA                    &lt;NA&gt;                            NA
Sample3           NA                    &lt;NA&gt;                            NA
Sample4           NA                    &lt;NA&gt;                            NA
Sample5           NA                    &lt;NA&gt;                            NA
        mapping.score nCount_SCT nFeature_SCT integrated_snn_res.0.1
Sample1            NA         NA           NA                   &lt;NA&gt;
Sample2            NA         NA           NA                   &lt;NA&gt;
Sample3            NA         NA           NA                   &lt;NA&gt;
Sample4            NA         NA           NA                      2
Sample5            NA         NA           NA                   &lt;NA&gt;
        integrated_snn_res.0.2 integrated_snn_res.0.3 integrated_snn_res.0.4
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample4                      0                      1                      1
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
        integrated_snn_res.0.5 integrated_snn_res.0.6 integrated_snn_res.0.7
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample4                      1                      0                      0
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
        integrated_snn_res.0.8 integrated_snn_res.0.9 integrated_snn_res.1
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample4                      0                      0                    0
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
        seurat_clusters predicted.celltype.score predicted.celltype
Sample1            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample2            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample3            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample4               0                       NA               &lt;NA&gt;
Sample5            &lt;NA&gt;                       NA               &lt;NA&gt;
        predicted.ann_level_1 predicted.ann_level_2 predicted.ann_level_3
Sample1                  &lt;NA&gt;                  &lt;NA&gt;                  &lt;NA&gt;
Sample2                Immune                  &lt;NA&gt;                  &lt;NA&gt;
Sample3                  &lt;NA&gt;                  &lt;NA&gt;                  &lt;NA&gt;
Sample4                Immune                  &lt;NA&gt;                  &lt;NA&gt;
Sample5                Immune              Lymphoid        B cell lineage
        predicted.ann_level_4 predicted.ann_level_5 predicted.ann_finest_level
Sample1                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample2                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample3                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample4                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample5                  &lt;NA&gt;                  None                       &lt;NA&gt;
        predicted.ann_level_1.score predicted.ann_level_2.score
Sample1                          NA                          NA
Sample2                          NA                          NA
Sample3                          NA                          NA
Sample4                          NA                          NA
Sample5                          NA                          NA
        predicted.ann_level_3.score predicted.ann_level_4.score
Sample1                          NA                          NA
Sample2                          NA                          NA
Sample3                          NA                          NA
Sample4                          NA                          NA
Sample5                          NA                          NA
        predicted.ann_level_5.score predicted.ann_finest_level.score nCount_ADT
Sample1                          NA                               NA          0
Sample2                          NA                               NA         NA
Sample3                          NA                               NA         NA
Sample4                          NA                               NA         NA
Sample5                          NA                               NA         NA
        nFeature_ADT Annotation   Broad nCount_ADT.dsb nFeature_ADT.dsb
Sample1            0       &lt;NA&gt; B cells              0                0
Sample2           NA       &lt;NA&gt; B cells             NA               NA
Sample3           NA       &lt;NA&gt; B cells             NA               NA
Sample4           NA    B cells B cells             NA               NA
Sample5           NA       &lt;NA&gt; B cells             NA               NA
        nCount_ADT.raw nFeature_ADT.raw merged.annotation       HTO.y
Sample1              0                0              &lt;NA&gt;          DM
Sample2             NA               NA              &lt;NA&gt; Human_HTO_8
Sample3             NA               NA              &lt;NA&gt; Human_HTO_1
Sample4             NA               NA              &lt;NA&gt; Human_HTO_4
Sample5             NA               NA              &lt;NA&gt; Human_HTO_6
        Participant Sex  Age Disease Batch Broad.1 donor.1 ncells
Sample1       B1_CF   M 2.99      CF     1 B cells      DM    542
Sample2       L2_CF   M 5.95      CF     2 B cells donor_A    122
Sample3       E2_CF   F 5.99      CF     2 B cells donor_B    133
Sample4       H2_CF   F 5.89      CF     2 B cells donor_C      9
Sample5       J2_CF   M 5.05      CF     2 B cells donor_D     25
132 more rows ...</code></pre>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(cpm(y.pb[, y.pb$samples$Broad == &quot;Macrophages&quot;], log=TRUE), 
                 gene.selection = &quot;common&quot;,
                 plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(rownames_to_column(y.pb$samples, var = &quot;sample&quot;)) %&gt;%
    mutate(Batch = as.factor(Batch))-&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            shape = Batch,
                            color = Disease))+
    geom_point(size = 3) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Disease&quot;,
         shape = &quot;Batch&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))  +
    scale_colour_brewer(palette = &quot;Set1&quot;) 
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/4f769d452aa920442673e5c102cd46a3cafe88ba/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">4f769d4</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9a6422117a9d227a8ac3ff661b08e5b39e2ced51/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">9a64221</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-expression-analysis-within-cell-types" class="section level1">
<h1>Differential expression analysis within cell types</h1>
<p>This analysis was adapted from Dr. Belinda Phipson’s <a href="https://bphipson.github.io/Human_Development_snRNAseq/">analysis</a> of single nuclei RNAseq data from human heart samples.</p>
<pre class="r"><code>y.pb$samples %&gt;%
  mutate(cell_label = fct_recode(Broad,
                                 Macr = &quot;Macrophages&quot;,
                                 B = &quot;B cells&quot;,
                                 CD4 = &quot;CD4 T&quot;,
                                 CD8 = &quot;CD8 T&quot;,
                                 GDT = &quot;Gamma delta T&quot;,
                                 Mast = &quot;Mast cells&quot;,
                                 NK = &quot;NK cells&quot;,
                                 NKT = &quot;NK T cells&quot;
                                 )) -&gt; y.pb$samples

newgrp &lt;- paste(y.pb$sample$cell_label, y.pb$samples$Disease, sep=&quot;.&quot;)
newgrp &lt;- factor(newgrp)

design &lt;- model.matrix(~ 0 + newgrp)
colnames(design) &lt;- levels(newgrp)

keep &lt;- filterByExpr(y.pb, design = design)
y &lt;- y.pb[keep,]
summary(keep)</code></pre>
<pre><code>   Mode   FALSE    TRUE 
logical    5135   10443 </code></pre>
<pre class="r"><code>v &lt;- voom(y, design, plot = TRUE, 
          normalize.method = &quot;cyclicloess&quot;)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9a6422117a9d227a8ac3ff661b08e5b39e2ced51/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">9a64221</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fit &lt;- lmFit(v, design)

y &lt;- calcNormFactors(y) # add normalisation factors for downstream visualization</code></pre>
<pre class="r"><code>y$samples %&gt;% dplyr::select(Broad,
                            Disease,
                            ncells) %&gt;%
  group_by(Broad, Disease) %&gt;%
  summarise(agg = sum(ncells),
            ind = n()) %&gt;%
  dplyr::rename(cell = Broad,
                grp = Disease) -&gt; dat

ggplot(dat, aes(x = cell,  y = ind, fill = grp)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  labs(y = &quot;No. individuals&quot;, x = &quot;Cell type&quot;, fill = &quot;Clinical group&quot;) +
  geom_text(size = 3, position = position_stack(vjust = 0.5),
            aes(label = agg))</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="overall-deg-summary" class="section level3">
<h3>Overall DEG summary</h3>
<pre class="r"><code>cont &lt;- makeContrasts(Macr.CvCF = Macr.Ctrl - Macr.CF,
                           B.CvCF = B.Ctrl - B.CF,
                           CD4.CvCF = CD4.Ctrl - CD4.CF,
                           CD8.CvCF = CD8.Ctrl - CD8.CF,
                           Dendritic.CvCF = Dendritic.Ctrl - Dendritic.CF,
                           Epithelial.CvCF = Epithelial.Ctrl - Epithelial.CF,
                           GDT.CvCF = GDT.Ctrl - GDT.CF,
                           Mast.CvCF = Mast.Ctrl - Mast.CF,
                           Monocytes.CvCF = Monocytes.Ctrl - Monocytes.CF,
                           Neutrophils.CvCF = Neutrophils.Ctrl - Neutrophils.CF,
                           NK.CvCF = NK.Ctrl - NK.CF,
                           NKT.CvCF = NKT.Ctrl - NKT.CF,
                           levels = design)
  
cont.fit &lt;- contrasts.fit(fit, contrasts = cont)
fit2 &lt;- eBayes(cont.fit, robust = TRUE)

summary(decideTests(fit2))</code></pre>
<pre><code>       Macr.CvCF B.CvCF CD4.CvCF CD8.CvCF Dendritic.CvCF Epithelial.CvCF
Down          68      9        1        3              0             216
NotSig     10328  10376    10440    10430          10421           10199
Up            47     58        2       10             22              28
       GDT.CvCF Mast.CvCF Monocytes.CvCF Neutrophils.CvCF NK.CvCF NKT.CvCF
Down         46        10              0               75      16       22
NotSig    10397     10431          10418            10361   10423    10417
Up            0         2             25                7       4        4</code></pre>
</div>
<div id="treat-deg-summary" class="section level3">
<h3>TREAT DEG summary</h3>
<p>This tests for DEGs relative to a logFC threshold of 0.5</p>
<pre class="r"><code>fit.treat &lt;- treat(fit2, lfc = 0.5)
dt &lt;- decideTests(fit.treat)
summary(dt)</code></pre>
<pre><code>       Macr.CvCF B.CvCF CD4.CvCF CD8.CvCF Dendritic.CvCF Epithelial.CvCF
Down          35      3        1        0              0             143
NotSig     10385  10411    10441    10440          10434           10284
Up            23     29        1        3              9              16
       GDT.CvCF Mast.CvCF Monocytes.CvCF Neutrophils.CvCF NK.CvCF NKT.CvCF
Down          7         0              0               20       3        5
NotSig    10436     10443          10435            10422   10439    10438
Up            0         0              8                1       1        0</code></pre>
<pre class="r"><code>pal &lt;- paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;, 2)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;)

summary(dt) %&gt;% 
  data.frame %&gt;%
  dplyr::rename(Direction = Var1, Cell = Var2) %&gt;%
  dplyr::filter(Direction != &quot;NotSig&quot;) %&gt;%
  mutate(Cell = gsub(&quot;.CvCF&quot;, &quot;&quot;, Cell)) %&gt;%
  ggplot(aes(x = Cell, y = Freq, fill = Direction)) +
  geom_bar(position = &quot;dodge&quot;, stat = &quot;identity&quot;) +
  geom_text(aes(label = Freq), stat = &quot;identity&quot;,
            position = position_dodge(width = .9),
            vjust = -0.5, colour = &quot;black&quot;, size = 2) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90,
                                   vjust = 0.5,
                                   hjust = 1)) +
  labs(y = &quot;No. DGE (FDR &lt; 0.05)&quot;) +
  scale_fill_manual(values = pal)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/4f769d452aa920442673e5c102cd46a3cafe88ba/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">4f769d4</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-13
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9a6422117a9d227a8ac3ff661b08e5b39e2ced51/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">9a64221</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="macrophages" class="section level2">
<h2>Macrophages</h2>
<div id="mean-difference-plot-of-degs" class="section level3">
<h3>Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,1],
                  status = as.factor(as.vector(dt[,1]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/a720c4c89a6bb3a82116a8acc971a1b069eb61fa/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">a720c4c</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9a6422117a9d227a8ac3ff661b08e5b39e2ced51/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">9a64221</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-04-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 1, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,1] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;Macr.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 20) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) -&gt; f5a
f5a</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs" class="section level3">
<h3>Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs-New&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 1, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">UTS2</td>
<td align="right">6.014453</td>
<td align="right">2.879215</td>
<td align="right">9.345958</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">RBMS3</td>
<td align="right">-5.781275</td>
<td align="right">2.946525</td>
<td align="right">-7.450658</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000366</td>
</tr>
<tr class="odd">
<td align="left">CEACAM6</td>
<td align="right">3.730140</td>
<td align="right">2.655014</td>
<td align="right">7.108379</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000001440</td>
</tr>
<tr class="even">
<td align="left">UGT2A1</td>
<td align="right">2.872195</td>
<td align="right">2.489493</td>
<td align="right">6.511095</td>
<td align="right">0.0000000008</td>
<td align="right">0.0000022055</td>
</tr>
<tr class="odd">
<td align="left">TEKT4</td>
<td align="right">3.538849</td>
<td align="right">2.999890</td>
<td align="right">5.877896</td>
<td align="right">0.0000000181</td>
<td align="right">0.0000352276</td>
</tr>
<tr class="even">
<td align="left">IGHA2</td>
<td align="right">4.593751</td>
<td align="right">2.886652</td>
<td align="right">5.840646</td>
<td align="right">0.0000000216</td>
<td align="right">0.0000352276</td>
</tr>
<tr class="odd">
<td align="left">SLC5A8</td>
<td align="right">2.962505</td>
<td align="right">2.514349</td>
<td align="right">5.821409</td>
<td align="right">0.0000000236</td>
<td align="right">0.0000352276</td>
</tr>
<tr class="even">
<td align="left">TSPAN8</td>
<td align="right">3.913386</td>
<td align="right">2.705416</td>
<td align="right">5.685265</td>
<td align="right">0.0000000446</td>
<td align="right">0.0000581884</td>
</tr>
<tr class="odd">
<td align="left">GCNT4</td>
<td align="right">3.325604</td>
<td align="right">2.662362</td>
<td align="right">5.516708</td>
<td align="right">0.0000000968</td>
<td align="right">0.0001122665</td>
</tr>
<tr class="even">
<td align="left">CCL15</td>
<td align="right">-4.392473</td>
<td align="right">3.207891</td>
<td align="right">-5.429925</td>
<td align="right">0.0000001437</td>
<td align="right">0.0001413012</td>
</tr>
<tr class="odd">
<td align="left">KDR</td>
<td align="right">-2.883208</td>
<td align="right">2.423815</td>
<td align="right">-5.421822</td>
<td align="right">0.0000001488</td>
<td align="right">0.0001413012</td>
</tr>
<tr class="even">
<td align="left">IGHG4</td>
<td align="right">5.552617</td>
<td align="right">2.833704</td>
<td align="right">5.341080</td>
<td align="right">0.0000002155</td>
<td align="right">0.0001875373</td>
</tr>
<tr class="odd">
<td align="left">CYP2F1</td>
<td align="right">3.477048</td>
<td align="right">2.707227</td>
<td align="right">5.293709</td>
<td align="right">0.0000002645</td>
<td align="right">0.0002124986</td>
</tr>
<tr class="even">
<td align="left">CXCL6</td>
<td align="right">3.129576</td>
<td align="right">2.645153</td>
<td align="right">4.804541</td>
<td align="right">0.0000022053</td>
<td align="right">0.0016449657</td>
</tr>
<tr class="odd">
<td align="left">HOXA9</td>
<td align="right">-4.470227</td>
<td align="right">2.746921</td>
<td align="right">-4.773114</td>
<td align="right">0.0000025273</td>
<td align="right">0.0017595029</td>
</tr>
<tr class="even">
<td align="left">SPRR3</td>
<td align="right">3.092153</td>
<td align="right">2.515952</td>
<td align="right">4.729487</td>
<td align="right">0.0000030198</td>
<td align="right">0.0018442855</td>
</tr>
<tr class="odd">
<td align="left">CCDC148</td>
<td align="right">-3.323340</td>
<td align="right">2.634682</td>
<td align="right">-4.722598</td>
<td align="right">0.0000031085</td>
<td align="right">0.0018442855</td>
</tr>
<tr class="even">
<td align="left">FRMPD2</td>
<td align="right">3.409853</td>
<td align="right">2.830529</td>
<td align="right">4.717255</td>
<td align="right">0.0000031789</td>
<td align="right">0.0018442855</td>
</tr>
<tr class="odd">
<td align="left">GALNT15</td>
<td align="right">-3.174378</td>
<td align="right">2.328373</td>
<td align="right">-4.634117</td>
<td align="right">0.0000044832</td>
<td align="right">0.0024640947</td>
</tr>
<tr class="even">
<td align="left">EHD2</td>
<td align="right">-2.571949</td>
<td align="right">2.371571</td>
<td align="right">-4.573562</td>
<td align="right">0.0000057440</td>
<td align="right">0.0029992156</td>
</tr>
<tr class="odd">
<td align="left">NNMT</td>
<td align="right">-2.944302</td>
<td align="right">2.731819</td>
<td align="right">-4.549927</td>
<td align="right">0.0000063257</td>
<td align="right">0.0031457018</td>
</tr>
<tr class="even">
<td align="left">PTPRG</td>
<td align="right">-2.424154</td>
<td align="right">2.422100</td>
<td align="right">-4.480499</td>
<td align="right">0.0000083769</td>
<td align="right">0.0039763609</td>
</tr>
<tr class="odd">
<td align="left">SPARCL1</td>
<td align="right">-2.714841</td>
<td align="right">2.505531</td>
<td align="right">-4.413063</td>
<td align="right">0.0000109781</td>
<td align="right">0.0049845571</td>
</tr>
<tr class="even">
<td align="left">FBN1</td>
<td align="right">-2.901561</td>
<td align="right">3.231493</td>
<td align="right">-4.400225</td>
<td align="right">0.0000115562</td>
<td align="right">0.0050284045</td>
</tr>
<tr class="odd">
<td align="left">ADGRB2</td>
<td align="right">-2.783403</td>
<td align="right">2.448059</td>
<td align="right">-4.320931</td>
<td align="right">0.0000158162</td>
<td align="right">0.0066067267</td>
</tr>
<tr class="even">
<td align="left">PTP4A1</td>
<td align="right">7.644277</td>
<td align="right">4.136643</td>
<td align="right">4.318173</td>
<td align="right">0.0000173216</td>
<td align="right">0.0069483747</td>
</tr>
<tr class="odd">
<td align="left">CHST13</td>
<td align="right">-3.540142</td>
<td align="right">3.055822</td>
<td align="right">-4.289013</td>
<td align="right">0.0000179648</td>
<td align="right">0.0069483747</td>
</tr>
<tr class="even">
<td align="left">ITGB1-DT</td>
<td align="right">-3.688922</td>
<td align="right">2.655775</td>
<td align="right">-4.273111</td>
<td align="right">0.0000191354</td>
<td align="right">0.0071368111</td>
</tr>
<tr class="odd">
<td align="left">ELF3-AS1</td>
<td align="right">-4.357530</td>
<td align="right">3.046490</td>
<td align="right">-4.244882</td>
<td align="right">0.0000215016</td>
<td align="right">0.0077427995</td>
</tr>
<tr class="even">
<td align="left">AGAP6</td>
<td align="right">5.336960</td>
<td align="right">3.229518</td>
<td align="right">4.231835</td>
<td align="right">0.0000230022</td>
<td align="right">0.0080070755</td>
</tr>
<tr class="odd">
<td align="left">TMPRSS11B</td>
<td align="right">2.156397</td>
<td align="right">2.365282</td>
<td align="right">4.186152</td>
<td align="right">0.0000267277</td>
<td align="right">0.0090037972</td>
</tr>
<tr class="even">
<td align="left">SCNN1B</td>
<td align="right">2.467106</td>
<td align="right">2.532342</td>
<td align="right">4.131453</td>
<td align="right">0.0000329723</td>
<td align="right">0.0104767475</td>
</tr>
<tr class="odd">
<td align="left">PDGFD</td>
<td align="right">-3.268289</td>
<td align="right">3.761486</td>
<td align="right">-4.130833</td>
<td align="right">0.0000331066</td>
<td align="right">0.0104767475</td>
</tr>
<tr class="even">
<td align="left">ZNF667</td>
<td align="right">2.804237</td>
<td align="right">3.392160</td>
<td align="right">4.114742</td>
<td align="right">0.0000351557</td>
<td align="right">0.0104942303</td>
</tr>
<tr class="odd">
<td align="left">IGHM</td>
<td align="right">4.135638</td>
<td align="right">4.468494</td>
<td align="right">4.113940</td>
<td align="right">0.0000355746</td>
<td align="right">0.0104942303</td>
</tr>
<tr class="even">
<td align="left">ZSWIM8-AS1</td>
<td align="right">-5.115876</td>
<td align="right">2.599761</td>
<td align="right">-4.105613</td>
<td align="right">0.0000373615</td>
<td align="right">0.0104942303</td>
</tr>
<tr class="odd">
<td align="left">TSPOAP1</td>
<td align="right">3.334592</td>
<td align="right">4.110543</td>
<td align="right">4.097453</td>
<td align="right">0.0000376143</td>
<td align="right">0.0104942303</td>
</tr>
<tr class="even">
<td align="left">SPP1</td>
<td align="right">-4.816301</td>
<td align="right">4.004563</td>
<td align="right">-4.098309</td>
<td align="right">0.0000381864</td>
<td align="right">0.0104942303</td>
</tr>
<tr class="odd">
<td align="left">FAM110D</td>
<td align="right">-2.389966</td>
<td align="right">2.390521</td>
<td align="right">-4.074147</td>
<td align="right">0.0000409995</td>
<td align="right">0.0109783914</td>
</tr>
<tr class="even">
<td align="left">SLC18A2</td>
<td align="right">-3.747931</td>
<td align="right">2.672485</td>
<td align="right">-4.041568</td>
<td align="right">0.0000466345</td>
<td align="right">0.0121751053</td>
</tr>
<tr class="odd">
<td align="left">MUCL1</td>
<td align="right">-5.073879</td>
<td align="right">2.693868</td>
<td align="right">-4.038693</td>
<td align="right">0.0000482068</td>
<td align="right">0.0122786186</td>
</tr>
<tr class="even">
<td align="left">COL4A1</td>
<td align="right">-2.539340</td>
<td align="right">2.289163</td>
<td align="right">-4.016666</td>
<td align="right">0.0000509178</td>
<td align="right">0.0126603583</td>
</tr>
<tr class="odd">
<td align="left">PRRX2</td>
<td align="right">-2.950747</td>
<td align="right">2.699328</td>
<td align="right">-3.963491</td>
<td align="right">0.0000621540</td>
<td align="right">0.0150947592</td>
</tr>
<tr class="even">
<td align="left">CACNA2D1</td>
<td align="right">-2.500340</td>
<td align="right">2.327259</td>
<td align="right">-3.900187</td>
<td align="right">0.0000784638</td>
<td align="right">0.0183134598</td>
</tr>
<tr class="odd">
<td align="left">CLDN5</td>
<td align="right">-2.247932</td>
<td align="right">2.398618</td>
<td align="right">-3.898575</td>
<td align="right">0.0000789147</td>
<td align="right">0.0183134598</td>
</tr>
<tr class="even">
<td align="left">GBP3</td>
<td align="right">-2.618082</td>
<td align="right">5.745894</td>
<td align="right">-3.878259</td>
<td align="right">0.0000850531</td>
<td align="right">0.0193088968</td>
</tr>
<tr class="odd">
<td align="left">IL33</td>
<td align="right">-2.782730</td>
<td align="right">2.532656</td>
<td align="right">-3.861792</td>
<td align="right">0.0000903694</td>
<td align="right">0.0200793114</td>
</tr>
<tr class="even">
<td align="left">SCT</td>
<td align="right">-3.740283</td>
<td align="right">2.849375</td>
<td align="right">-3.803371</td>
<td align="right">0.0001126836</td>
<td align="right">0.0245157186</td>
</tr>
<tr class="odd">
<td align="left">SPATA9</td>
<td align="right">-2.593571</td>
<td align="right">2.525008</td>
<td align="right">-3.789219</td>
<td align="right">0.0001175395</td>
<td align="right">0.0250503082</td>
</tr>
<tr class="even">
<td align="left">NTS</td>
<td align="right">-2.783062</td>
<td align="right">2.374360</td>
<td align="right">-3.761874</td>
<td align="right">0.0001297624</td>
<td align="right">0.0271021751</td>
</tr>
<tr class="odd">
<td align="left">PPP1R14C</td>
<td align="right">2.240785</td>
<td align="right">3.116639</td>
<td align="right">3.753862</td>
<td align="right">0.0001333948</td>
<td align="right">0.0273030549</td>
</tr>
<tr class="even">
<td align="left">KDF1</td>
<td align="right">-2.838331</td>
<td align="right">2.573102</td>
<td align="right">-3.748940</td>
<td align="right">0.0001359532</td>
<td align="right">0.0273030549</td>
</tr>
<tr class="odd">
<td align="left">CCL7</td>
<td align="right">-2.655975</td>
<td align="right">2.728236</td>
<td align="right">-3.702615</td>
<td align="right">0.0001602502</td>
<td align="right">0.0315753432</td>
</tr>
<tr class="even">
<td align="left">CNGA3</td>
<td align="right">-2.532619</td>
<td align="right">2.320057</td>
<td align="right">-3.642681</td>
<td align="right">0.0001979191</td>
<td align="right">0.0382753492</td>
</tr>
<tr class="odd">
<td align="left">SRXN1</td>
<td align="right">-4.614894</td>
<td align="right">3.455578</td>
<td align="right">-3.637989</td>
<td align="right">0.0002081396</td>
<td align="right">0.0395200376</td>
</tr>
<tr class="even">
<td align="left">CTTNBP2</td>
<td align="right">2.039734</td>
<td align="right">2.747408</td>
<td align="right">3.578827</td>
<td align="right">0.0002470888</td>
<td align="right">0.0460776429</td>
</tr>
<tr class="odd">
<td align="left">NR2F1</td>
<td align="right">-2.072447</td>
<td align="right">2.428419</td>
<td align="right">-3.566768</td>
<td align="right">0.0002576205</td>
<td align="right">0.0471987886</td>
</tr>
<tr class="even">
<td align="left">BICDL2</td>
<td align="right">2.171079</td>
<td align="right">2.732991</td>
<td align="right">3.556225</td>
<td align="right">0.0002671906</td>
<td align="right">0.0481081287</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/Macr-DEGs-{colnames(cont)[1]}.csv&quot;))</code></pre>
<pre class="r"><code>## calculate gene lengths
ex &lt;- exonsBy(TxDb.Hsapiens.UCSC.hg38.knownGene, &quot;gene&quot;)
## reduce to longest exon extents
ex &lt;- reduce(ex)
## compute exon lengths, by gene
exlen &lt;- relist(width(unlist(ex)), ex)
## add them up
exlens &lt;- sapply(exlen, sum)

## create full annotation
ann &lt;- rowData(summed) %&gt;% 
  data.frame %&gt;%
  right_join(data.frame(SYMBOL = rownames(y))) %&gt;%
  right_join(exlens %&gt;% 
               data.frame %&gt;% 
               rownames_to_column(var = &quot;ENTREZID&quot;) %&gt;% 
               dplyr::rename(Length = &quot;.&quot;))

gsAnnots &lt;- buildIdx(entrezIDs = ann$ENTREZID, 
                     species = &quot;human&quot;,
                     msigdb.gsets = c(&quot;c2&quot;,&quot;c5&quot;))</code></pre>
<pre><code>[1] &quot;Loading MSigDB Gene Sets ... &quot;
[1] &quot;Loaded gene sets for the collection c2 ...&quot;
[1] &quot;Indexed the collection c2 ...&quot;
[1] &quot;Created annotation for the collection c2 ...&quot;
[1] &quot;Loaded gene sets for the collection c5 ...&quot;
[1] &quot;Indexed the collection c5 ...&quot;
[1] &quot;Created annotation for the collection c5 ...&quot;
[1] &quot;Building KEGG pathways annotation object ... &quot;</code></pre>
<pre class="r"><code>react &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
                universe = ann$ENTREZID, 
                collection = gsAnnots$c2@original[grep(&quot;REACTOME&quot;, 
                                                       names(gsAnnots$c2@original))],
                gene.length = ann$Length)</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">3</td>
<td align="right">0.0002134</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">4</td>
<td align="right">0.0004622</td>
<td align="right">0.1146752</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0005104</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">5</td>
<td align="right">0.0013274</td>
<td align="right">0.2236600</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0016910</td>
<td align="right">0.2279411</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">4</td>
<td align="right">0.0032207</td>
<td align="right">0.3560318</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">3</td>
<td align="right">0.0036977</td>
<td align="right">0.3560318</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">2</td>
<td align="right">0.0052907</td>
<td align="right">0.4457399</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">3</td>
<td align="right">0.0065906</td>
<td align="right">0.4935657</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0103041</td>
<td align="right">0.6944954</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0193157</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0216091</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_DOPAMINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">0.0239236</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_SIGNALING_BY_GPCR</td>
<td align="right">902</td>
<td align="right">5</td>
<td align="right">0.0281039</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0.0341306</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NA_CL_DEPENDENT_NEUROTRANSMITTER_TRANSPORTERS</td>
<td align="right">17</td>
<td align="right">1</td>
<td align="right">0.0342693</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS</td>
<td align="right">19</td>
<td align="right">1</td>
<td align="right">0.0389305</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">1</td>
<td align="right">0.0407147</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GLUCURONIDATION</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">0.0413113</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_G_ALPHA_Q_SIGNALLING_EVENTS</td>
<td align="right">180</td>
<td align="right">2</td>
<td align="right">0.0510888</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="66%" />
<col width="5%" />
<col width="3%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_RECEPTOR_BINDING</td>
<td align="right">1452</td>
<td align="right">13</td>
<td align="right">0.0000038</td>
<td align="right">0.0235902</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_RESPONSE_TO_EXTERNAL_STIMULUS</td>
<td align="right">296</td>
<td align="right">6</td>
<td align="right">0.0000244</td>
<td align="right">0.0730257</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_SPACE</td>
<td align="right">1343</td>
<td align="right">11</td>
<td align="right">0.0000509</td>
<td align="right">0.0730257</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOKINE_ACTIVITY</td>
<td align="right">219</td>
<td align="right">5</td>
<td align="right">0.0000527</td>
<td align="right">0.0730257</td>
</tr>
<tr class="odd">
<td align="left">GO_CHEMOKINE_ACTIVITY</td>
<td align="right">48</td>
<td align="right">3</td>
<td align="right">0.0000676</td>
<td align="right">0.0730257</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_BODY_FLUID_LEVELS</td>
<td align="right">505</td>
<td align="right">7</td>
<td align="right">0.0000711</td>
<td align="right">0.0730257</td>
</tr>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_CHEMOTAXIS</td>
<td align="right">120</td>
<td align="right">4</td>
<td align="right">0.0000960</td>
<td align="right">0.0845219</td>
</tr>
<tr class="even">
<td align="left">GO_CHEMOKINE_RECEPTOR_BINDING</td>
<td align="right">57</td>
<td align="right">3</td>
<td align="right">0.0001225</td>
<td align="right">0.0881698</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CELLULAR_COMPONENT_MOVEMENT</td>
<td align="right">771</td>
<td align="right">8</td>
<td align="right">0.0001409</td>
<td align="right">0.0881698</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_ENDOGENOUS_STIMULUS</td>
<td align="right">1447</td>
<td align="right">11</td>
<td align="right">0.0001430</td>
<td align="right">0.0881698</td>
</tr>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_LOCOMOTION</td>
<td align="right">420</td>
<td align="right">6</td>
<td align="right">0.0001809</td>
<td align="right">0.1014150</td>
</tr>
<tr class="even">
<td align="left">GO_O_GLYCAN_PROCESSING</td>
<td align="right">57</td>
<td align="right">3</td>
<td align="right">0.0002314</td>
<td align="right">0.1123754</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_LEUKOCYTE_MIGRATION</td>
<td align="right">149</td>
<td align="right">4</td>
<td align="right">0.0002369</td>
<td align="right">0.1123754</td>
</tr>
<tr class="even">
<td align="left">GO_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY</td>
<td align="right">72</td>
<td align="right">3</td>
<td align="right">0.0003432</td>
<td align="right">0.1511393</td>
</tr>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_ERK1_AND_ERK2_CASCADE</td>
<td align="right">170</td>
<td align="right">4</td>
<td align="right">0.0003767</td>
<td align="right">0.1548466</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_OXYGEN_CONTAINING_COMPOUND</td>
<td align="right">1379</td>
<td align="right">10</td>
<td align="right">0.0004433</td>
<td align="right">0.1646782</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CHEMOTAXIS</td>
<td align="right">180</td>
<td align="right">4</td>
<td align="right">0.0004540</td>
<td align="right">0.1646782</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_RESPONSE_TO_EXTERNAL_STIMULUS</td>
<td align="right">926</td>
<td align="right">8</td>
<td align="right">0.0005168</td>
<td align="right">0.1770294</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_STEROID_HORMONE</td>
<td align="right">496</td>
<td align="right">6</td>
<td align="right">0.0005623</td>
<td align="right">0.1824727</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSPORT_VESICLE</td>
<td align="right">338</td>
<td align="right">5</td>
<td align="right">0.0006817</td>
<td align="right">0.2101604</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-Macr-{colnames(cont)[1]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-Macr-{colnames(cont)[1]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;Macrophages&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;))
  
dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.column = Gene,
          .row = Participant,
          .value = `Log2 CPM`,
          .scale = &quot;column&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 9),
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2),
                                               viridis::magma(5))) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;bottom&quot;,  
                       annotation_legend_side = &quot;bottom&quot;) -&gt; f5b</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-22-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.row = Gene,
          .column = Participant,
          .value = `Log2 CPM`,
          .scale = &quot;row&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 9),
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2),
                                               viridis::magma(5))) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;right&quot;,  
                       annotation_legend_side = &quot;right&quot;) -&gt; f5b</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, nrow(top))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="b-cells" class="section level2">
<h2>B cells</h2>
<div id="mean-difference-plot-of-degs-1" class="section level3">
<h3>Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,2],
                  status = as.factor(as.vector(dt[,2]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 2, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,2] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;B.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-25-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs-1" class="section level3">
<h3>Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs-New&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 2, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ALPL</td>
<td align="right">4.206994</td>
<td align="right">2.602078</td>
<td align="right">6.769809</td>
<td align="right">0.0000000002</td>
<td align="right">0.0000024213</td>
</tr>
<tr class="even">
<td align="left">PKHD1L1</td>
<td align="right">4.285125</td>
<td align="right">2.452903</td>
<td align="right">6.315525</td>
<td align="right">0.0000000022</td>
<td align="right">0.0000115505</td>
</tr>
<tr class="odd">
<td align="left">IGHG4</td>
<td align="right">6.531059</td>
<td align="right">2.833704</td>
<td align="right">6.218258</td>
<td align="right">0.0000000036</td>
<td align="right">0.0000124317</td>
</tr>
<tr class="even">
<td align="left">UTS2</td>
<td align="right">4.532193</td>
<td align="right">2.879215</td>
<td align="right">5.558057</td>
<td align="right">0.0000000802</td>
<td align="right">0.0002093474</td>
</tr>
<tr class="odd">
<td align="left">IGHD</td>
<td align="right">4.422130</td>
<td align="right">3.071104</td>
<td align="right">5.224020</td>
<td align="right">0.0000003611</td>
<td align="right">0.0007541399</td>
</tr>
<tr class="even">
<td align="left">CRY1</td>
<td align="right">-4.647383</td>
<td align="right">4.592678</td>
<td align="right">-5.106808</td>
<td align="right">0.0000006052</td>
<td align="right">0.0010533829</td>
</tr>
<tr class="odd">
<td align="left">ACY3</td>
<td align="right">4.284676</td>
<td align="right">2.866002</td>
<td align="right">4.981166</td>
<td align="right">0.0000010422</td>
<td align="right">0.0015548700</td>
</tr>
<tr class="even">
<td align="left">SCNN1B</td>
<td align="right">3.226429</td>
<td align="right">2.532342</td>
<td align="right">4.908866</td>
<td align="right">0.0000014175</td>
<td align="right">0.0018503999</td>
</tr>
<tr class="odd">
<td align="left">TNFRSF17</td>
<td align="right">3.748614</td>
<td align="right">2.783356</td>
<td align="right">4.810503</td>
<td align="right">0.0000021526</td>
<td align="right">0.0024977667</td>
</tr>
<tr class="even">
<td align="left">IGHM</td>
<td align="right">4.293534</td>
<td align="right">4.468494</td>
<td align="right">4.739099</td>
<td align="right">0.0000029112</td>
<td align="right">0.0030402025</td>
</tr>
<tr class="odd">
<td align="left">TTC24</td>
<td align="right">4.135692</td>
<td align="right">2.802868</td>
<td align="right">4.691118</td>
<td align="right">0.0000035523</td>
<td align="right">0.0033724588</td>
</tr>
<tr class="even">
<td align="left">NXPH4</td>
<td align="right">3.422732</td>
<td align="right">2.691628</td>
<td align="right">4.562985</td>
<td align="right">0.0000060025</td>
<td align="right">0.0052236891</td>
</tr>
<tr class="odd">
<td align="left">FCRL4</td>
<td align="right">4.270051</td>
<td align="right">2.817520</td>
<td align="right">4.511847</td>
<td align="right">0.0000074209</td>
<td align="right">0.0059612662</td>
</tr>
<tr class="even">
<td align="left">UGT8</td>
<td align="right">2.997405</td>
<td align="right">2.723567</td>
<td align="right">4.400098</td>
<td align="right">0.0000115635</td>
<td align="right">0.0086255131</td>
</tr>
<tr class="odd">
<td align="left">IL17A</td>
<td align="right">4.138479</td>
<td align="right">2.702547</td>
<td align="right">4.301046</td>
<td align="right">0.0000172102</td>
<td align="right">0.0119817108</td>
</tr>
<tr class="even">
<td align="left">JCHAIN</td>
<td align="right">4.177312</td>
<td align="right">3.984491</td>
<td align="right">4.283830</td>
<td align="right">0.0000184222</td>
<td align="right">0.0120239116</td>
</tr>
<tr class="odd">
<td align="left">IGLC7</td>
<td align="right">4.008947</td>
<td align="right">2.374041</td>
<td align="right">4.251674</td>
<td align="right">0.0000208596</td>
<td align="right">0.0128139146</td>
</tr>
<tr class="even">
<td align="left">FCRLA</td>
<td align="right">3.483586</td>
<td align="right">2.882056</td>
<td align="right">4.215908</td>
<td align="right">0.0000238831</td>
<td align="right">0.0133130202</td>
</tr>
<tr class="odd">
<td align="left">LINC02362</td>
<td align="right">3.345315</td>
<td align="right">3.047518</td>
<td align="right">4.212105</td>
<td align="right">0.0000242217</td>
<td align="right">0.0133130202</td>
</tr>
<tr class="even">
<td align="left">MCM10</td>
<td align="right">3.764448</td>
<td align="right">3.119826</td>
<td align="right">4.159844</td>
<td align="right">0.0000297138</td>
<td align="right">0.0155150663</td>
</tr>
<tr class="odd">
<td align="left">RHEX</td>
<td align="right">3.855230</td>
<td align="right">2.857810</td>
<td align="right">4.084174</td>
<td align="right">0.0000397238</td>
<td align="right">0.0192185824</td>
</tr>
<tr class="even">
<td align="left">FNIP2</td>
<td align="right">-4.724302</td>
<td align="right">4.879492</td>
<td align="right">-4.077093</td>
<td align="right">0.0000413403</td>
<td align="right">0.0192185824</td>
</tr>
<tr class="odd">
<td align="left">HTR3A</td>
<td align="right">3.659887</td>
<td align="right">2.444086</td>
<td align="right">4.066961</td>
<td align="right">0.0000423276</td>
<td align="right">0.0192185824</td>
</tr>
<tr class="even">
<td align="left">RASSF6</td>
<td align="right">3.270889</td>
<td align="right">2.901013</td>
<td align="right">4.012993</td>
<td align="right">0.0000517370</td>
<td align="right">0.0225120680</td>
</tr>
<tr class="odd">
<td align="left">BACE1</td>
<td align="right">-3.490219</td>
<td align="right">4.116146</td>
<td align="right">-3.980078</td>
<td align="right">0.0000586113</td>
<td align="right">0.0244831155</td>
</tr>
<tr class="even">
<td align="left">FBLN2</td>
<td align="right">3.393693</td>
<td align="right">2.712464</td>
<td align="right">3.965974</td>
<td align="right">0.0000617302</td>
<td align="right">0.0247941734</td>
</tr>
<tr class="odd">
<td align="left">EME1</td>
<td align="right">3.121446</td>
<td align="right">2.850905</td>
<td align="right">3.953190</td>
<td align="right">0.0000646263</td>
<td align="right">0.0249960301</td>
</tr>
<tr class="even">
<td align="left">GRAMD1C</td>
<td align="right">3.040849</td>
<td align="right">2.905974</td>
<td align="right">3.940740</td>
<td align="right">0.0000676551</td>
<td align="right">0.0252329428</td>
</tr>
<tr class="odd">
<td align="left">TUBB2B</td>
<td align="right">4.136561</td>
<td align="right">2.784378</td>
<td align="right">3.919957</td>
<td align="right">0.0000739045</td>
<td align="right">0.0266132689</td>
</tr>
<tr class="even">
<td align="left">E2F1</td>
<td align="right">3.305462</td>
<td align="right">3.657748</td>
<td align="right">3.803479</td>
<td align="right">0.0001120618</td>
<td align="right">0.0390087220</td>
</tr>
<tr class="odd">
<td align="left">AICDA</td>
<td align="right">3.229271</td>
<td align="right">2.993678</td>
<td align="right">3.731951</td>
<td align="right">0.0001448638</td>
<td align="right">0.0488004231</td>
</tr>
<tr class="even">
<td align="left">FOXM1</td>
<td align="right">3.480782</td>
<td align="right">3.405731</td>
<td align="right">3.717773</td>
<td align="right">0.0001528245</td>
<td align="right">0.0498733357</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/B-DEGs-{colnames(cont)[2]}.csv&quot;))</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">3</td>
<td align="right">0.0002134</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">4</td>
<td align="right">0.0004622</td>
<td align="right">0.1146752</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0005104</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">5</td>
<td align="right">0.0013274</td>
<td align="right">0.2236600</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0016910</td>
<td align="right">0.2279411</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">4</td>
<td align="right">0.0032207</td>
<td align="right">0.3560318</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">3</td>
<td align="right">0.0036977</td>
<td align="right">0.3560318</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">2</td>
<td align="right">0.0052907</td>
<td align="right">0.4457399</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">3</td>
<td align="right">0.0065906</td>
<td align="right">0.4935657</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0103041</td>
<td align="right">0.6944954</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0193157</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0216091</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_DOPAMINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">0.0239236</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_SIGNALING_BY_GPCR</td>
<td align="right">902</td>
<td align="right">5</td>
<td align="right">0.0281039</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0.0341306</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NA_CL_DEPENDENT_NEUROTRANSMITTER_TRANSPORTERS</td>
<td align="right">17</td>
<td align="right">1</td>
<td align="right">0.0342693</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS</td>
<td align="right">19</td>
<td align="right">1</td>
<td align="right">0.0389305</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">1</td>
<td align="right">0.0407147</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GLUCURONIDATION</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">0.0413113</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_G_ALPHA_Q_SIGNALLING_EVENTS</td>
<td align="right">180</td>
<td align="right">2</td>
<td align="right">0.0510888</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="70%" />
<col width="5%" />
<col width="3%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_OXYGEN_CONTAINING_COMPOUND</td>
<td align="right">1379</td>
<td align="right">8</td>
<td align="right">0.0000537</td>
<td align="right">0.1462178</td>
</tr>
<tr class="even">
<td align="left">GO_CELLULAR_RESPONSE_TO_DNA_DAMAGE_STIMULUS</td>
<td align="right">720</td>
<td align="right">6</td>
<td align="right">0.0000698</td>
<td align="right">0.1462178</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_ENDOGENOUS_STIMULUS</td>
<td align="right">1447</td>
<td align="right">8</td>
<td align="right">0.0000711</td>
<td align="right">0.1462178</td>
</tr>
<tr class="even">
<td align="left">GO_ADAPTIVE_IMMUNE_RESPONSE</td>
<td align="right">259</td>
<td align="right">4</td>
<td align="right">0.0001194</td>
<td align="right">0.1840215</td>
</tr>
<tr class="odd">
<td align="left">GO_IMMUNE_RESPONSE</td>
<td align="right">1066</td>
<td align="right">6</td>
<td align="right">0.0004848</td>
<td align="right">0.5978466</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_FIBROBLAST_PROLIFERATION</td>
<td align="right">53</td>
<td align="right">2</td>
<td align="right">0.0014178</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_BIOTIC_STIMULUS</td>
<td align="right">865</td>
<td align="right">5</td>
<td align="right">0.0014365</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_QUATERNARY_AMMONIUM_GROUP_BINDING</td>
<td align="right">50</td>
<td align="right">2</td>
<td align="right">0.0015447</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_HORMONE</td>
<td align="right">891</td>
<td align="right">5</td>
<td align="right">0.0019270</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_LIPID</td>
<td align="right">887</td>
<td align="right">5</td>
<td align="right">0.0021576</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_ORGANIC_CYCLIC_COMPOUND</td>
<td align="right">914</td>
<td align="right">5</td>
<td align="right">0.0021997</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY_IN_RESPONSE_TO_DNA_DAMAGE</td>
<td align="right">71</td>
<td align="right">2</td>
<td align="right">0.0024293</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_CYCLE_PHASE_TRANSITION</td>
<td align="right">254</td>
<td align="right">3</td>
<td align="right">0.0025050</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_NEGATIVE_REGULATION_OF_HORMONE_SECRETION</td>
<td align="right">74</td>
<td align="right">2</td>
<td align="right">0.0025333</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_AMMONIUM_ION_BINDING</td>
<td align="right">71</td>
<td align="right">2</td>
<td align="right">0.0031316</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_FIBROBLAST_PROLIFERATION</td>
<td align="right">81</td>
<td align="right">2</td>
<td align="right">0.0032603</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_SINGLE_STRANDED_DNA_BINDING</td>
<td align="right">91</td>
<td align="right">2</td>
<td align="right">0.0037410</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_OXYGEN_LEVELS</td>
<td align="right">311</td>
<td align="right">3</td>
<td align="right">0.0039067</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">GO_CELLULAR_RESPONSE_TO_STRESS</td>
<td align="right">1561</td>
<td align="right">6</td>
<td align="right">0.0043364</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_SIGNAL_TRANSDUCTION_IN_RESPONSE_TO_DNA_DAMAGE</td>
<td align="right">96</td>
<td align="right">2</td>
<td align="right">0.0044409</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-B-{colnames(cont)[2]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-B-{colnames(cont)[2]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;B cells&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;))
  
dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.column = Gene,
          .row = Participant,
          .value = `Log2 CPM`,
          .scale = &quot;column&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 9),
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2),
                                               viridis::magma(5))) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;bottom&quot;,  
                       annotation_legend_side = &quot;bottom&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, nrow(top))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-29-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="epithelial" class="section level2">
<h2>Epithelial</h2>
<div id="mean-difference-plot-of-degs-2" class="section level3">
<h3>Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,6],
                  status = as.factor(as.vector(dt[,6]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-30-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 6, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,6] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;Epithelial.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-31-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs-2" class="section level3">
<h3>Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs-New&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 6, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TMPRSS11B</td>
<td align="right">5.685095</td>
<td align="right">2.365282</td>
<td align="right">11.272511</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">SPRR3</td>
<td align="right">6.838732</td>
<td align="right">2.515952</td>
<td align="right">10.745088</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">CYGB</td>
<td align="right">-6.107909</td>
<td align="right">2.996615</td>
<td align="right">-8.287764</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000003</td>
</tr>
<tr class="even">
<td align="left">SPRR1B</td>
<td align="right">3.974893</td>
<td align="right">2.306376</td>
<td align="right">7.939061</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000014</td>
</tr>
<tr class="odd">
<td align="left">STC1</td>
<td align="right">-4.627466</td>
<td align="right">2.325665</td>
<td align="right">-6.965649</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000001795</td>
</tr>
<tr class="even">
<td align="left">KRT78</td>
<td align="right">4.316143</td>
<td align="right">2.416028</td>
<td align="right">6.316542</td>
<td align="right">0.0000000022</td>
<td align="right">0.0000038311</td>
</tr>
<tr class="odd">
<td align="left">CALB2</td>
<td align="right">-4.155494</td>
<td align="right">2.287687</td>
<td align="right">-6.259667</td>
<td align="right">0.0000000029</td>
<td align="right">0.0000043332</td>
</tr>
<tr class="even">
<td align="left">AARD</td>
<td align="right">-3.927574</td>
<td align="right">2.358606</td>
<td align="right">-5.820400</td>
<td align="right">0.0000000237</td>
<td align="right">0.0000309767</td>
</tr>
<tr class="odd">
<td align="left">UGT2A1</td>
<td align="right">3.106154</td>
<td align="right">2.489493</td>
<td align="right">5.676695</td>
<td align="right">0.0000000464</td>
<td align="right">0.0000538053</td>
</tr>
<tr class="even">
<td align="left">EMCN</td>
<td align="right">-3.752675</td>
<td align="right">2.276701</td>
<td align="right">-5.611802</td>
<td align="right">0.0000000626</td>
<td align="right">0.0000653576</td>
</tr>
<tr class="odd">
<td align="left">RASAL2-AS1</td>
<td align="right">-3.736390</td>
<td align="right">2.827875</td>
<td align="right">-5.534629</td>
<td align="right">0.0000000892</td>
<td align="right">0.0000846526</td>
</tr>
<tr class="even">
<td align="left">ECM1</td>
<td align="right">4.216926</td>
<td align="right">3.127860</td>
<td align="right">5.441700</td>
<td align="right">0.0000001362</td>
<td align="right">0.0001184901</td>
</tr>
<tr class="odd">
<td align="left">COL4A1</td>
<td align="right">-3.961332</td>
<td align="right">2.289163</td>
<td align="right">-5.409839</td>
<td align="right">0.0000001572</td>
<td align="right">0.0001202212</td>
</tr>
<tr class="even">
<td align="left">C1QTNF1</td>
<td align="right">-3.964747</td>
<td align="right">2.364877</td>
<td align="right">-5.394145</td>
<td align="right">0.0000001687</td>
<td align="right">0.0001202212</td>
</tr>
<tr class="odd">
<td align="left">PCDHGA11</td>
<td align="right">-3.834614</td>
<td align="right">2.297335</td>
<td align="right">-5.388962</td>
<td align="right">0.0000001727</td>
<td align="right">0.0001202212</td>
</tr>
<tr class="even">
<td align="left">SORCS1</td>
<td align="right">-4.183727</td>
<td align="right">2.345207</td>
<td align="right">-5.347146</td>
<td align="right">0.0000002085</td>
<td align="right">0.0001360753</td>
</tr>
<tr class="odd">
<td align="left">FAM110D</td>
<td align="right">-3.839156</td>
<td align="right">2.390521</td>
<td align="right">-5.321103</td>
<td align="right">0.0000002341</td>
<td align="right">0.0001438363</td>
</tr>
<tr class="even">
<td align="left">SYN2</td>
<td align="right">-3.888879</td>
<td align="right">2.591041</td>
<td align="right">-5.260843</td>
<td align="right">0.0000003063</td>
<td align="right">0.0001777112</td>
</tr>
<tr class="odd">
<td align="left">EDN1</td>
<td align="right">-4.214557</td>
<td align="right">2.793081</td>
<td align="right">-5.245413</td>
<td align="right">0.0000003282</td>
<td align="right">0.0001804046</td>
</tr>
<tr class="even">
<td align="left">TMEM233</td>
<td align="right">-3.807829</td>
<td align="right">2.415178</td>
<td align="right">-5.200082</td>
<td align="right">0.0000004008</td>
<td align="right">0.0002092923</td>
</tr>
<tr class="odd">
<td align="left">SOX18</td>
<td align="right">-3.754164</td>
<td align="right">2.269423</td>
<td align="right">-5.092902</td>
<td align="right">0.0000006414</td>
<td align="right">0.0003189431</td>
</tr>
<tr class="even">
<td align="left">LINC01506</td>
<td align="right">-4.063175</td>
<td align="right">3.188107</td>
<td align="right">-4.916958</td>
<td align="right">0.0000013715</td>
<td align="right">0.0006510448</td>
</tr>
<tr class="odd">
<td align="left">COL15A1</td>
<td align="right">-3.868994</td>
<td align="right">2.935201</td>
<td align="right">-4.894937</td>
<td align="right">0.0000015057</td>
<td align="right">0.0006836699</td>
</tr>
<tr class="even">
<td align="left">SLC5A8</td>
<td align="right">2.931636</td>
<td align="right">2.514349</td>
<td align="right">4.816588</td>
<td align="right">0.0000020959</td>
<td align="right">0.0008821412</td>
</tr>
<tr class="odd">
<td align="left">C7</td>
<td align="right">-3.553940</td>
<td align="right">2.258996</td>
<td align="right">-4.814945</td>
<td align="right">0.0000021118</td>
<td align="right">0.0008821412</td>
</tr>
<tr class="even">
<td align="left">MAMSTR</td>
<td align="right">-4.153512</td>
<td align="right">2.938959</td>
<td align="right">-4.722068</td>
<td align="right">0.0000031234</td>
<td align="right">0.0012545127</td>
</tr>
<tr class="odd">
<td align="left">DAAM2</td>
<td align="right">-3.662179</td>
<td align="right">2.422726</td>
<td align="right">-4.593289</td>
<td align="right">0.0000053070</td>
<td align="right">0.0020526335</td>
</tr>
<tr class="even">
<td align="left">SMAD9</td>
<td align="right">-3.901011</td>
<td align="right">2.386163</td>
<td align="right">-4.574899</td>
<td align="right">0.0000057273</td>
<td align="right">0.0021129317</td>
</tr>
<tr class="odd">
<td align="left">MMRN1</td>
<td align="right">-3.983684</td>
<td align="right">2.399571</td>
<td align="right">-4.569082</td>
<td align="right">0.0000058676</td>
<td align="right">0.0021129317</td>
</tr>
<tr class="even">
<td align="left">ADGRL4</td>
<td align="right">-3.557942</td>
<td align="right">2.237482</td>
<td align="right">-4.495463</td>
<td align="right">0.0000078981</td>
<td align="right">0.0027493363</td>
</tr>
<tr class="odd">
<td align="left">MEOX1</td>
<td align="right">-3.658330</td>
<td align="right">2.449449</td>
<td align="right">-4.477504</td>
<td align="right">0.0000084950</td>
<td align="right">0.0028262946</td>
</tr>
<tr class="even">
<td align="left">ITGA2B</td>
<td align="right">-3.673284</td>
<td align="right">2.359509</td>
<td align="right">-4.472480</td>
<td align="right">0.0000086693</td>
<td align="right">0.0028262946</td>
</tr>
<tr class="odd">
<td align="left">CXCL5</td>
<td align="right">-4.643649</td>
<td align="right">3.449096</td>
<td align="right">-4.466950</td>
<td align="right">0.0000089311</td>
<td align="right">0.0028262946</td>
</tr>
<tr class="even">
<td align="left">PCDH12</td>
<td align="right">-3.652757</td>
<td align="right">2.481848</td>
<td align="right">-4.439932</td>
<td align="right">0.0000098798</td>
<td align="right">0.0028808884</td>
</tr>
<tr class="odd">
<td align="left">HRCT1</td>
<td align="right">-3.807094</td>
<td align="right">2.360685</td>
<td align="right">-4.431872</td>
<td align="right">0.0000102120</td>
<td align="right">0.0028808884</td>
</tr>
<tr class="even">
<td align="left">HOXA10</td>
<td align="right">-3.849126</td>
<td align="right">2.834597</td>
<td align="right">-4.428055</td>
<td align="right">0.0000103718</td>
<td align="right">0.0028808884</td>
</tr>
<tr class="odd">
<td align="left">CLEC14A</td>
<td align="right">-3.785092</td>
<td align="right">2.479179</td>
<td align="right">-4.426036</td>
<td align="right">0.0000104522</td>
<td align="right">0.0028808884</td>
</tr>
<tr class="even">
<td align="left">RAMP3</td>
<td align="right">-3.555644</td>
<td align="right">2.249839</td>
<td align="right">-4.425033</td>
<td align="right">0.0000104830</td>
<td align="right">0.0028808884</td>
</tr>
<tr class="odd">
<td align="left">TNFRSF10C</td>
<td align="right">-3.918104</td>
<td align="right">2.834243</td>
<td align="right">-4.411290</td>
<td align="right">0.0000110959</td>
<td align="right">0.0029711289</td>
</tr>
<tr class="even">
<td align="left">NNMT</td>
<td align="right">-4.020390</td>
<td align="right">2.731819</td>
<td align="right">-4.392562</td>
<td align="right">0.0000119655</td>
<td align="right">0.0031239014</td>
</tr>
<tr class="odd">
<td align="left">ARHGEF15</td>
<td align="right">-3.546705</td>
<td align="right">2.297894</td>
<td align="right">-4.378196</td>
<td align="right">0.0000126345</td>
<td align="right">0.0032181029</td>
</tr>
<tr class="even">
<td align="left">ACKR1</td>
<td align="right">-3.545700</td>
<td align="right">2.303291</td>
<td align="right">-4.369902</td>
<td align="right">0.0000130575</td>
<td align="right">0.0032362285</td>
</tr>
<tr class="odd">
<td align="left">RASSF1-AS1</td>
<td align="right">-3.766496</td>
<td align="right">3.106623</td>
<td align="right">-4.364470</td>
<td align="right">0.0000133569</td>
<td align="right">0.0032362285</td>
</tr>
<tr class="even">
<td align="left">IL12RB1</td>
<td align="right">-4.410316</td>
<td align="right">5.036335</td>
<td align="right">-4.360267</td>
<td align="right">0.0000136557</td>
<td align="right">0.0032362285</td>
</tr>
<tr class="odd">
<td align="left">MSMB</td>
<td align="right">4.652040</td>
<td align="right">3.145096</td>
<td align="right">4.355723</td>
<td align="right">0.0000139453</td>
<td align="right">0.0032362285</td>
</tr>
<tr class="even">
<td align="left">FLRT2</td>
<td align="right">-3.506479</td>
<td align="right">2.546502</td>
<td align="right">-4.319012</td>
<td align="right">0.0000159645</td>
<td align="right">0.0036242891</td>
</tr>
<tr class="odd">
<td align="left">AATK</td>
<td align="right">-4.387474</td>
<td align="right">3.121683</td>
<td align="right">-4.296245</td>
<td align="right">0.0000175856</td>
<td align="right">0.0039073640</td>
</tr>
<tr class="even">
<td align="left">CD300LG</td>
<td align="right">-3.535748</td>
<td align="right">2.356847</td>
<td align="right">-4.285320</td>
<td align="right">0.0000182263</td>
<td align="right">0.0039132062</td>
</tr>
<tr class="odd">
<td align="left">CAV1</td>
<td align="right">-4.641160</td>
<td align="right">2.779679</td>
<td align="right">-4.286119</td>
<td align="right">0.0000183613</td>
<td align="right">0.0039132062</td>
</tr>
<tr class="even">
<td align="left">RDH16</td>
<td align="right">-3.911386</td>
<td align="right">2.528413</td>
<td align="right">-4.274241</td>
<td align="right">0.0000190814</td>
<td align="right">0.0039853413</td>
</tr>
<tr class="odd">
<td align="left">CRHBP</td>
<td align="right">-3.505092</td>
<td align="right">2.556584</td>
<td align="right">-4.263962</td>
<td align="right">0.0000198117</td>
<td align="right">0.0040567377</td>
</tr>
<tr class="even">
<td align="left">CEBPE</td>
<td align="right">-3.586584</td>
<td align="right">3.123644</td>
<td align="right">-4.244316</td>
<td align="right">0.0000213989</td>
<td align="right">0.0042974802</td>
</tr>
<tr class="odd">
<td align="left">GALNT15</td>
<td align="right">-3.541034</td>
<td align="right">2.328373</td>
<td align="right">-4.209913</td>
<td align="right">0.0000244524</td>
<td align="right">0.0048180358</td>
</tr>
<tr class="even">
<td align="left">MS4A2</td>
<td align="right">-3.945582</td>
<td align="right">2.723631</td>
<td align="right">-4.176194</td>
<td align="right">0.0000279454</td>
<td align="right">0.0053181794</td>
</tr>
<tr class="odd">
<td align="left">CXCL12</td>
<td align="right">-3.549056</td>
<td align="right">2.285241</td>
<td align="right">-4.174779</td>
<td align="right">0.0000280092</td>
<td align="right">0.0053181794</td>
</tr>
<tr class="even">
<td align="left">GRIK1</td>
<td align="right">-3.651797</td>
<td align="right">2.482757</td>
<td align="right">-4.165249</td>
<td align="right">0.0000290765</td>
<td align="right">0.0053374273</td>
</tr>
<tr class="odd">
<td align="left">LINC02361</td>
<td align="right">-4.014716</td>
<td align="right">3.628236</td>
<td align="right">-4.165589</td>
<td align="right">0.0000291328</td>
<td align="right">0.0053374273</td>
</tr>
<tr class="even">
<td align="left">TNXB</td>
<td align="right">-3.996811</td>
<td align="right">2.529075</td>
<td align="right">-4.128781</td>
<td align="right">0.0000335538</td>
<td align="right">0.0060414111</td>
</tr>
<tr class="odd">
<td align="left">CDK15</td>
<td align="right">-3.550446</td>
<td align="right">2.277764</td>
<td align="right">-4.109073</td>
<td align="right">0.0000360337</td>
<td align="right">0.0063779662</td>
</tr>
<tr class="even">
<td align="left">CCM2L</td>
<td align="right">-3.639220</td>
<td align="right">2.572611</td>
<td align="right">-4.099325</td>
<td align="right">0.0000374229</td>
<td align="right">0.0065134559</td>
</tr>
<tr class="odd">
<td align="left">ACOD1</td>
<td align="right">-3.499084</td>
<td align="right">2.600866</td>
<td align="right">-4.045507</td>
<td align="right">0.0000458465</td>
<td align="right">0.0077731330</td>
</tr>
<tr class="even">
<td align="left">PI3</td>
<td align="right">2.975514</td>
<td align="right">2.567189</td>
<td align="right">4.043051</td>
<td align="right">0.0000461490</td>
<td align="right">0.0077731330</td>
</tr>
<tr class="odd">
<td align="left">FSCN1</td>
<td align="right">-4.598580</td>
<td align="right">3.964831</td>
<td align="right">-4.028155</td>
<td align="right">0.0000496735</td>
<td align="right">0.0081742336</td>
</tr>
<tr class="even">
<td align="left">TSPAN18</td>
<td align="right">-3.902358</td>
<td align="right">2.669838</td>
<td align="right">-4.023038</td>
<td align="right">0.0000500958</td>
<td align="right">0.0081742336</td>
</tr>
<tr class="odd">
<td align="left">ST8SIA6</td>
<td align="right">-3.959533</td>
<td align="right">2.533319</td>
<td align="right">-3.992422</td>
<td align="right">0.0000562474</td>
<td align="right">0.0090368000</td>
</tr>
<tr class="even">
<td align="left">F2RL3</td>
<td align="right">-3.526670</td>
<td align="right">2.412088</td>
<td align="right">-3.986160</td>
<td align="right">0.0000573117</td>
<td align="right">0.0090682706</td>
</tr>
<tr class="odd">
<td align="left">CNGA3</td>
<td align="right">-3.276116</td>
<td align="right">2.320057</td>
<td align="right">-3.965165</td>
<td align="right">0.0000618661</td>
<td align="right">0.0096427959</td>
</tr>
<tr class="even">
<td align="left">DNASE1L3</td>
<td align="right">-3.947237</td>
<td align="right">3.449278</td>
<td align="right">-3.934326</td>
<td align="right">0.0000698594</td>
<td align="right">0.0107285471</td>
</tr>
<tr class="odd">
<td align="left">PREX2</td>
<td align="right">-4.272977</td>
<td align="right">2.429268</td>
<td align="right">-3.919788</td>
<td align="right">0.0000741276</td>
<td align="right">0.0112190570</td>
</tr>
<tr class="even">
<td align="left">AGRP</td>
<td align="right">-3.923052</td>
<td align="right">3.794366</td>
<td align="right">-3.913150</td>
<td align="right">0.0000755413</td>
<td align="right">0.0112696816</td>
</tr>
<tr class="odd">
<td align="left">CEACAM21</td>
<td align="right">-4.632465</td>
<td align="right">4.713851</td>
<td align="right">-3.909789</td>
<td align="right">0.0000774962</td>
<td align="right">0.0113984894</td>
</tr>
<tr class="even">
<td align="left">COL13A1</td>
<td align="right">-3.663444</td>
<td align="right">2.420121</td>
<td align="right">-3.899020</td>
<td align="right">0.0000793219</td>
<td align="right">0.0115049754</td>
</tr>
<tr class="odd">
<td align="left">DLC1</td>
<td align="right">-3.573522</td>
<td align="right">3.290645</td>
<td align="right">-3.893375</td>
<td align="right">0.0000809083</td>
<td align="right">0.0115743181</td>
</tr>
<tr class="even">
<td align="left">CYP2F1</td>
<td align="right">3.039234</td>
<td align="right">2.707227</td>
<td align="right">3.886313</td>
<td align="right">0.0000826935</td>
<td align="right">0.0116698365</td>
</tr>
<tr class="odd">
<td align="left">LINC01781</td>
<td align="right">-3.726123</td>
<td align="right">2.413186</td>
<td align="right">-3.882959</td>
<td align="right">0.0000842212</td>
<td align="right">0.0117269594</td>
</tr>
<tr class="even">
<td align="left">CAPN11</td>
<td align="right">-3.473456</td>
<td align="right">2.872691</td>
<td align="right">-3.867972</td>
<td align="right">0.0000887324</td>
<td align="right">0.0121925259</td>
</tr>
<tr class="odd">
<td align="left">RBP5</td>
<td align="right">-3.526954</td>
<td align="right">2.410341</td>
<td align="right">-3.862277</td>
<td align="right">0.0000906538</td>
<td align="right">0.0122947791</td>
</tr>
<tr class="even">
<td align="left">GCSAML</td>
<td align="right">-3.510875</td>
<td align="right">2.514500</td>
<td align="right">-3.858703</td>
<td align="right">0.0000918325</td>
<td align="right">0.0122949527</td>
</tr>
<tr class="odd">
<td align="left">ZNF366</td>
<td align="right">-3.710247</td>
<td align="right">3.179310</td>
<td align="right">-3.845761</td>
<td align="right">0.0000965144</td>
<td align="right">0.0126779384</td>
</tr>
<tr class="even">
<td align="left">ARF4-AS1</td>
<td align="right">-3.543649</td>
<td align="right">3.321610</td>
<td align="right">-3.842499</td>
<td align="right">0.0000974720</td>
<td align="right">0.0126779384</td>
</tr>
<tr class="odd">
<td align="left">MGP</td>
<td align="right">-3.059595</td>
<td align="right">2.635787</td>
<td align="right">-3.838150</td>
<td align="right">0.0000986251</td>
<td align="right">0.0126779384</td>
</tr>
<tr class="even">
<td align="left">HOXB6</td>
<td align="right">-4.062862</td>
<td align="right">3.483866</td>
<td align="right">-3.838476</td>
<td align="right">0.0000996670</td>
<td align="right">0.0126779384</td>
</tr>
<tr class="odd">
<td align="left">ROBO4</td>
<td align="right">-3.528069</td>
<td align="right">2.403506</td>
<td align="right">-3.833349</td>
<td align="right">0.0001007631</td>
<td align="right">0.0126779384</td>
</tr>
<tr class="even">
<td align="left">SHE</td>
<td align="right">-3.516798</td>
<td align="right">2.472800</td>
<td align="right">-3.829909</td>
<td align="right">0.0001020211</td>
<td align="right">0.0126834101</td>
</tr>
<tr class="odd">
<td align="left">ANKDD1A</td>
<td align="right">-4.182382</td>
<td align="right">3.862890</td>
<td align="right">-3.825819</td>
<td align="right">0.0001046238</td>
<td align="right">0.0128475092</td>
</tr>
<tr class="even">
<td align="left">C11orf96</td>
<td align="right">-4.379102</td>
<td align="right">2.808789</td>
<td align="right">-3.823872</td>
<td align="right">0.0001058016</td>
<td align="right">0.0128475092</td>
</tr>
<tr class="odd">
<td align="left">PTPRB</td>
<td align="right">-3.657526</td>
<td align="right">2.449388</td>
<td align="right">-3.807023</td>
<td align="right">0.0001110698</td>
<td align="right">0.0133322110</td>
</tr>
<tr class="even">
<td align="left">LINC02207</td>
<td align="right">-3.495164</td>
<td align="right">2.635211</td>
<td align="right">-3.789327</td>
<td align="right">0.0001181849</td>
<td align="right">0.0139863950</td>
</tr>
<tr class="odd">
<td align="left">GNRH1</td>
<td align="right">-3.706924</td>
<td align="right">3.143592</td>
<td align="right">-3.787737</td>
<td align="right">0.0001191984</td>
<td align="right">0.0139863950</td>
</tr>
<tr class="even">
<td align="left">BIRC7</td>
<td align="right">-3.483411</td>
<td align="right">2.747231</td>
<td align="right">-3.776492</td>
<td align="right">0.0001237776</td>
<td align="right">0.0143623226</td>
</tr>
<tr class="odd">
<td align="left">ZNF385D</td>
<td align="right">-3.951564</td>
<td align="right">2.581844</td>
<td align="right">-3.762592</td>
<td align="right">0.0001310713</td>
<td align="right">0.0149195156</td>
</tr>
<tr class="even">
<td align="left">PRR19</td>
<td align="right">-3.678185</td>
<td align="right">2.748086</td>
<td align="right">-3.759939</td>
<td align="right">0.0001317325</td>
<td align="right">0.0149195156</td>
</tr>
<tr class="odd">
<td align="left">MUC5AC</td>
<td align="right">3.180133</td>
<td align="right">2.759285</td>
<td align="right">3.755958</td>
<td align="right">0.0001328656</td>
<td align="right">0.0149195156</td>
</tr>
<tr class="even">
<td align="left">TSKS</td>
<td align="right">-3.740501</td>
<td align="right">3.291968</td>
<td align="right">-3.754306</td>
<td align="right">0.0001345610</td>
<td align="right">0.0149491501</td>
</tr>
<tr class="odd">
<td align="left">SEMA3G</td>
<td align="right">-3.521860</td>
<td align="right">2.441634</td>
<td align="right">-3.747292</td>
<td align="right">0.0001375642</td>
<td align="right">0.0150250242</td>
</tr>
<tr class="even">
<td align="left">DCHS1</td>
<td align="right">-3.774925</td>
<td align="right">2.899018</td>
<td align="right">-3.747204</td>
<td align="right">0.0001381215</td>
<td align="right">0.0150250242</td>
</tr>
<tr class="odd">
<td align="left">CHRNA6</td>
<td align="right">-3.506115</td>
<td align="right">2.549147</td>
<td align="right">-3.735760</td>
<td align="right">0.0001433462</td>
<td align="right">0.0154326233</td>
</tr>
<tr class="even">
<td align="left">IL21</td>
<td align="right">-3.535557</td>
<td align="right">2.357877</td>
<td align="right">-3.716675</td>
<td align="right">0.0001535386</td>
<td align="right">0.0163612643</td>
</tr>
<tr class="odd">
<td align="left">ENPP1</td>
<td align="right">-4.297789</td>
<td align="right">3.137731</td>
<td align="right">-3.706737</td>
<td align="right">0.0001614423</td>
<td align="right">0.0168480584</td>
</tr>
<tr class="even">
<td align="left">ADGRA2</td>
<td align="right">-3.472340</td>
<td align="right">2.890346</td>
<td align="right">-3.700256</td>
<td align="right">0.0001626588</td>
<td align="right">0.0168480584</td>
</tr>
<tr class="odd">
<td align="left">PLEKHD1</td>
<td align="right">-3.651934</td>
<td align="right">2.560862</td>
<td align="right">-3.700498</td>
<td align="right">0.0001629468</td>
<td align="right">0.0168480584</td>
</tr>
<tr class="even">
<td align="left">NID1</td>
<td align="right">-3.587729</td>
<td align="right">3.108249</td>
<td align="right">-3.696636</td>
<td align="right">0.0001650398</td>
<td align="right">0.0168971669</td>
</tr>
<tr class="odd">
<td align="left">SPTB</td>
<td align="right">-3.528867</td>
<td align="right">2.398616</td>
<td align="right">-3.690982</td>
<td align="right">0.0001682486</td>
<td align="right">0.0169546729</td>
</tr>
<tr class="even">
<td align="left">MCM10</td>
<td align="right">-4.358188</td>
<td align="right">3.119826</td>
<td align="right">-3.693502</td>
<td align="right">0.0001695377</td>
<td align="right">0.0169546729</td>
</tr>
<tr class="odd">
<td align="left">EGOT</td>
<td align="right">-3.613223</td>
<td align="right">2.785755</td>
<td align="right">-3.687647</td>
<td align="right">0.0001704721</td>
<td align="right">0.0169546729</td>
</tr>
<tr class="even">
<td align="left">GATA2</td>
<td align="right">-3.805725</td>
<td align="right">2.612392</td>
<td align="right">-3.679882</td>
<td align="right">0.0001758257</td>
<td align="right">0.0173221536</td>
</tr>
<tr class="odd">
<td align="left">ST6GALNAC3</td>
<td align="right">-3.478027</td>
<td align="right">2.810929</td>
<td align="right">-3.668457</td>
<td align="right">0.0001821175</td>
<td align="right">0.0177743317</td>
</tr>
<tr class="even">
<td align="left">NEFH</td>
<td align="right">-3.543142</td>
<td align="right">3.590392</td>
<td align="right">-3.658151</td>
<td align="right">0.0001890642</td>
<td align="right">0.0182789131</td>
</tr>
<tr class="odd">
<td align="left">MAL</td>
<td align="right">4.034231</td>
<td align="right">3.898695</td>
<td align="right">3.658216</td>
<td align="right">0.0001907882</td>
<td align="right">0.0182789131</td>
</tr>
<tr class="even">
<td align="left">ZBTB20-AS5</td>
<td align="right">-4.002895</td>
<td align="right">2.632129</td>
<td align="right">-3.650425</td>
<td align="right">0.0001959973</td>
<td align="right">0.0186072743</td>
</tr>
<tr class="odd">
<td align="left">TPSD1</td>
<td align="right">-3.674752</td>
<td align="right">2.352035</td>
<td align="right">-3.643099</td>
<td align="right">0.0001998270</td>
<td align="right">0.0187999410</td>
</tr>
<tr class="even">
<td align="left">KIR3DL1</td>
<td align="right">-3.535047</td>
<td align="right">2.360754</td>
<td align="right">-3.626746</td>
<td align="right">0.0002111780</td>
<td align="right">0.0196904630</td>
</tr>
<tr class="odd">
<td align="left">FAM111B</td>
<td align="right">-4.940253</td>
<td align="right">3.676186</td>
<td align="right">-3.621621</td>
<td align="right">0.0002228704</td>
<td align="right">0.0205147722</td>
</tr>
<tr class="even">
<td align="left">PDE2A</td>
<td align="right">-3.787802</td>
<td align="right">2.763451</td>
<td align="right">-3.611354</td>
<td align="right">0.0002239475</td>
<td align="right">0.0205147722</td>
</tr>
<tr class="odd">
<td align="left">PVALB</td>
<td align="right">-3.588839</td>
<td align="right">3.093704</td>
<td align="right">-3.593616</td>
<td align="right">0.0002374097</td>
<td align="right">0.0215588663</td>
</tr>
<tr class="even">
<td align="left">CYSLTR2</td>
<td align="right">-3.809979</td>
<td align="right">3.658070</td>
<td align="right">-3.578682</td>
<td align="right">0.0002512269</td>
<td align="right">0.0226169171</td>
</tr>
<tr class="odd">
<td align="left">HAND2-AS1</td>
<td align="right">-3.510934</td>
<td align="right">2.514064</td>
<td align="right">-3.571063</td>
<td align="right">0.0002564920</td>
<td align="right">0.0228935523</td>
</tr>
<tr class="even">
<td align="left">UPK1B</td>
<td align="right">2.525298</td>
<td align="right">2.802679</td>
<td align="right">3.561826</td>
<td align="right">0.0002622674</td>
<td align="right">0.0232106633</td>
</tr>
<tr class="odd">
<td align="left">SIGLEC8</td>
<td align="right">-3.545322</td>
<td align="right">2.305324</td>
<td align="right">-3.557396</td>
<td align="right">0.0002691380</td>
<td align="right">0.0236185521</td>
</tr>
<tr class="even">
<td align="left">PDE1B</td>
<td align="right">-3.667423</td>
<td align="right">4.031748</td>
<td align="right">-3.542779</td>
<td align="right">0.0002838129</td>
<td align="right">0.0246988138</td>
</tr>
<tr class="odd">
<td align="left">MND1</td>
<td align="right">-4.758089</td>
<td align="right">3.441373</td>
<td align="right">-3.531155</td>
<td align="right">0.0003047352</td>
<td align="right">0.0261548973</td>
</tr>
<tr class="even">
<td align="left">NKD1</td>
<td align="right">-3.852260</td>
<td align="right">3.130586</td>
<td align="right">-3.522663</td>
<td align="right">0.0003055537</td>
<td align="right">0.0261548973</td>
</tr>
<tr class="odd">
<td align="left">ADCY4</td>
<td align="right">-3.577660</td>
<td align="right">3.327916</td>
<td align="right">-3.503009</td>
<td align="right">0.0003249978</td>
<td align="right">0.0275931049</td>
</tr>
<tr class="even">
<td align="left">GPBAR1</td>
<td align="right">-3.681946</td>
<td align="right">3.449228</td>
<td align="right">-3.491492</td>
<td align="right">0.0003388793</td>
<td align="right">0.0284329514</td>
</tr>
<tr class="odd">
<td align="left">JPH4</td>
<td align="right">-3.454344</td>
<td align="right">3.214587</td>
<td align="right">-3.488873</td>
<td align="right">0.0003403350</td>
<td align="right">0.0284329514</td>
</tr>
<tr class="even">
<td align="left">ITGB1-DT</td>
<td align="right">-4.022441</td>
<td align="right">2.655775</td>
<td align="right">-3.479708</td>
<td align="right">0.0003559385</td>
<td align="right">0.0295005222</td>
</tr>
<tr class="odd">
<td align="left">ZNF521</td>
<td align="right">-3.705054</td>
<td align="right">2.546408</td>
<td align="right">-3.474609</td>
<td align="right">0.0003592779</td>
<td align="right">0.0295428250</td>
</tr>
<tr class="even">
<td align="left">NTS</td>
<td align="right">-3.108372</td>
<td align="right">2.374360</td>
<td align="right">-3.461896</td>
<td align="right">0.0003710739</td>
<td align="right">0.0302744101</td>
</tr>
<tr class="odd">
<td align="left">CYYR1</td>
<td align="right">-3.637181</td>
<td align="right">2.672111</td>
<td align="right">-3.460371</td>
<td align="right">0.0003766152</td>
<td align="right">0.0304883163</td>
</tr>
<tr class="even">
<td align="left">LINC00996</td>
<td align="right">-3.855255</td>
<td align="right">4.198344</td>
<td align="right">-3.458339</td>
<td align="right">0.0003812773</td>
<td align="right">0.0305691051</td>
</tr>
<tr class="odd">
<td align="left">MIR9-3HG</td>
<td align="right">-3.730496</td>
<td align="right">2.646681</td>
<td align="right">-3.455736</td>
<td align="right">0.0003834677</td>
<td align="right">0.0305691051</td>
</tr>
<tr class="even">
<td align="left">GPR141</td>
<td align="right">-3.589663</td>
<td align="right">3.083238</td>
<td align="right">-3.441917</td>
<td align="right">0.0004006301</td>
<td align="right">0.0316144211</td>
</tr>
<tr class="odd">
<td align="left">PRICKLE1</td>
<td align="right">-3.975401</td>
<td align="right">2.889243</td>
<td align="right">-3.443359</td>
<td align="right">0.0004026351</td>
<td align="right">0.0316144211</td>
</tr>
<tr class="even">
<td align="left">SCNN1B</td>
<td align="right">2.515764</td>
<td align="right">2.532342</td>
<td align="right">3.426844</td>
<td align="right">0.0004155343</td>
<td align="right">0.0323837641</td>
</tr>
<tr class="odd">
<td align="left">KRT1</td>
<td align="right">-3.538412</td>
<td align="right">2.342490</td>
<td align="right">-3.402135</td>
<td align="right">0.0004578973</td>
<td align="right">0.0354209036</td>
</tr>
<tr class="even">
<td align="left">PRG4</td>
<td align="right">-3.538360</td>
<td align="right">2.342770</td>
<td align="right">-3.398484</td>
<td align="right">0.0004635699</td>
<td align="right">0.0354658951</td>
</tr>
<tr class="odd">
<td align="left">KRT17</td>
<td align="right">2.898120</td>
<td align="right">2.590091</td>
<td align="right">3.394074</td>
<td align="right">0.0004652712</td>
<td align="right">0.0354658951</td>
</tr>
<tr class="even">
<td align="left">TRBV3-1</td>
<td align="right">-3.536217</td>
<td align="right">2.354318</td>
<td align="right">-3.393061</td>
<td align="right">0.0004720969</td>
<td align="right">0.0357254198</td>
</tr>
<tr class="odd">
<td align="left">LINC00299</td>
<td align="right">-4.082393</td>
<td align="right">3.137234</td>
<td align="right">-3.383992</td>
<td align="right">0.0004942826</td>
<td align="right">0.0371351997</td>
</tr>
<tr class="even">
<td align="left">CDC45</td>
<td align="right">-4.019790</td>
<td align="right">3.169967</td>
<td align="right">-3.371883</td>
<td align="right">0.0005138909</td>
<td align="right">0.0380288034</td>
</tr>
<tr class="odd">
<td align="left">CCL7</td>
<td align="right">-3.485094</td>
<td align="right">2.728236</td>
<td align="right">-3.366801</td>
<td align="right">0.0005150194</td>
<td align="right">0.0380288034</td>
</tr>
<tr class="even">
<td align="left">SLC4A1</td>
<td align="right">-3.518466</td>
<td align="right">2.462521</td>
<td align="right">-3.363943</td>
<td align="right">0.0005203865</td>
<td align="right">0.0380288034</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS7</td>
<td align="right">-3.623448</td>
<td align="right">2.689493</td>
<td align="right">-3.364356</td>
<td align="right">0.0005210362</td>
<td align="right">0.0380288034</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="right">-4.210317</td>
<td align="right">3.128116</td>
<td align="right">-3.367768</td>
<td align="right">0.0005243845</td>
<td align="right">0.0380288034</td>
</tr>
<tr class="odd">
<td align="left">PSMA8</td>
<td align="right">-3.480728</td>
<td align="right">2.778808</td>
<td align="right">-3.357327</td>
<td align="right">0.0005315578</td>
<td align="right">0.0381989551</td>
</tr>
<tr class="even">
<td align="left">S100A14</td>
<td align="right">2.420026</td>
<td align="right">2.773758</td>
<td align="right">3.351534</td>
<td align="right">0.0005340465</td>
<td align="right">0.0381989551</td>
</tr>
<tr class="odd">
<td align="left">ELN</td>
<td align="right">-2.622152</td>
<td align="right">2.367627</td>
<td align="right">-3.348830</td>
<td align="right">0.0005395215</td>
<td align="right">0.0383280471</td>
</tr>
<tr class="even">
<td align="left">LINC00243</td>
<td align="right">-3.802577</td>
<td align="right">2.832910</td>
<td align="right">-3.350468</td>
<td align="right">0.0005485912</td>
<td align="right">0.0387090428</td>
</tr>
<tr class="odd">
<td align="left">OLFM1</td>
<td align="right">-3.986549</td>
<td align="right">2.724932</td>
<td align="right">-3.347527</td>
<td align="right">0.0005571919</td>
<td align="right">0.0390310164</td>
</tr>
<tr class="even">
<td align="left">CLEC4F</td>
<td align="right">-3.593324</td>
<td align="right">3.031412</td>
<td align="right">-3.342233</td>
<td align="right">0.0005606294</td>
<td align="right">0.0390310164</td>
</tr>
<tr class="odd">
<td align="left">CADPS</td>
<td align="right">-3.519831</td>
<td align="right">2.454118</td>
<td align="right">-3.335738</td>
<td align="right">0.0005718212</td>
<td align="right">0.0395465465</td>
</tr>
<tr class="even">
<td align="left">CACNA2D4</td>
<td align="right">-3.841095</td>
<td align="right">4.272392</td>
<td align="right">-3.334062</td>
<td align="right">0.0005802326</td>
<td align="right">0.0396369591</td>
</tr>
<tr class="odd">
<td align="left">TMEM255B</td>
<td align="right">-3.675576</td>
<td align="right">3.685307</td>
<td align="right">-3.332352</td>
<td align="right">0.0005807196</td>
<td align="right">0.0396369591</td>
</tr>
<tr class="even">
<td align="left">MAPK12</td>
<td align="right">-3.961185</td>
<td align="right">2.958276</td>
<td align="right">-3.330221</td>
<td align="right">0.0005899768</td>
<td align="right">0.0400073211</td>
</tr>
<tr class="odd">
<td align="left">TNFSF11</td>
<td align="right">-3.790650</td>
<td align="right">2.853475</td>
<td align="right">-3.320250</td>
<td align="right">0.0006066731</td>
<td align="right">0.0408741141</td>
</tr>
<tr class="even">
<td align="left">CALY</td>
<td align="right">-3.509505</td>
<td align="right">2.524477</td>
<td align="right">-3.310363</td>
<td align="right">0.0006219713</td>
<td align="right">0.0416361956</td>
</tr>
<tr class="odd">
<td align="left">MEFV</td>
<td align="right">-3.675724</td>
<td align="right">3.494053</td>
<td align="right">-3.301291</td>
<td align="right">0.0006439212</td>
<td align="right">0.0428310120</td>
</tr>
<tr class="even">
<td align="left">KCNK13</td>
<td align="right">-3.621667</td>
<td align="right">3.278639</td>
<td align="right">-3.263893</td>
<td align="right">0.0007274225</td>
<td align="right">0.0480305441</td>
</tr>
<tr class="odd">
<td align="left">SUCNR1</td>
<td align="right">-4.042335</td>
<td align="right">4.134055</td>
<td align="right">-3.266530</td>
<td align="right">0.0007312895</td>
<td align="right">0.0480305441</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/Epithelial-DEGs-{colnames(cont)[6]}.csv&quot;))</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">3</td>
<td align="right">0.0002134</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">4</td>
<td align="right">0.0004622</td>
<td align="right">0.1146752</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0005104</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">5</td>
<td align="right">0.0013274</td>
<td align="right">0.2236600</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0016910</td>
<td align="right">0.2279411</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">4</td>
<td align="right">0.0032207</td>
<td align="right">0.3560318</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">3</td>
<td align="right">0.0036977</td>
<td align="right">0.3560318</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">2</td>
<td align="right">0.0052907</td>
<td align="right">0.4457399</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">3</td>
<td align="right">0.0065906</td>
<td align="right">0.4935657</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0103041</td>
<td align="right">0.6944954</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0193157</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0216091</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_DOPAMINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">0.0239236</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_SIGNALING_BY_GPCR</td>
<td align="right">902</td>
<td align="right">5</td>
<td align="right">0.0281039</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0.0341306</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NA_CL_DEPENDENT_NEUROTRANSMITTER_TRANSPORTERS</td>
<td align="right">17</td>
<td align="right">1</td>
<td align="right">0.0342693</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS</td>
<td align="right">19</td>
<td align="right">1</td>
<td align="right">0.0389305</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">1</td>
<td align="right">0.0407147</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GLUCURONIDATION</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">0.0413113</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_G_ALPHA_Q_SIGNALLING_EVENTS</td>
<td align="right">180</td>
<td align="right">2</td>
<td align="right">0.0510888</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_INTRINSIC_COMPONENT_OF_PLASMA_MEMBRANE</td>
<td align="right">1644</td>
<td align="right">30</td>
<td align="right">0.00e+00</td>
<td align="right">0.0002402</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_CELLULAR_COMPONENT_MOVEMENT</td>
<td align="right">771</td>
<td align="right">19</td>
<td align="right">2.00e-07</td>
<td align="right">0.0005834</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_SPACE</td>
<td align="right">1343</td>
<td align="right">24</td>
<td align="right">4.00e-07</td>
<td align="right">0.0007676</td>
</tr>
<tr class="even">
<td align="left">GO_DIVALENT_INORGANIC_CATION_HOMEOSTASIS</td>
<td align="right">343</td>
<td align="right">12</td>
<td align="right">9.00e-07</td>
<td align="right">0.0012643</td>
</tr>
<tr class="odd">
<td align="left">GO_CARDIOVASCULAR_SYSTEM_DEVELOPMENT</td>
<td align="right">788</td>
<td align="right">18</td>
<td align="right">1.50e-06</td>
<td align="right">0.0012643</td>
</tr>
<tr class="even">
<td align="left">GO_CIRCULATORY_SYSTEM_DEVELOPMENT</td>
<td align="right">788</td>
<td align="right">18</td>
<td align="right">1.50e-06</td>
<td align="right">0.0012643</td>
</tr>
<tr class="odd">
<td align="left">GO_ION_HOMEOSTASIS</td>
<td align="right">575</td>
<td align="right">15</td>
<td align="right">1.60e-06</td>
<td align="right">0.0012643</td>
</tr>
<tr class="even">
<td align="left">GO_EXTRACELLULAR_MATRIX</td>
<td align="right">426</td>
<td align="right">13</td>
<td align="right">1.80e-06</td>
<td align="right">0.0012643</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEINACEOUS_EXTRACELLULAR_MATRIX</td>
<td align="right">356</td>
<td align="right">12</td>
<td align="right">1.80e-06</td>
<td align="right">0.0012643</td>
</tr>
<tr class="even">
<td align="left">GO_CELLULAR_RESPONSE_TO_CYTOKINE_STIMULUS</td>
<td align="right">606</td>
<td align="right">15</td>
<td align="right">2.60e-06</td>
<td align="right">0.0015840</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_CYTOKINE</td>
<td align="right">714</td>
<td align="right">16</td>
<td align="right">4.50e-06</td>
<td align="right">0.0024949</td>
</tr>
<tr class="even">
<td align="left">GO_RESPONSE_TO_ENDOGENOUS_STIMULUS</td>
<td align="right">1447</td>
<td align="right">24</td>
<td align="right">5.00e-06</td>
<td align="right">0.0025589</td>
</tr>
<tr class="odd">
<td align="left">GO_TISSUE_DEVELOPMENT</td>
<td align="right">1514</td>
<td align="right">24</td>
<td align="right">9.50e-06</td>
<td align="right">0.0044327</td>
</tr>
<tr class="even">
<td align="left">GO_CELLULAR_RESPONSE_TO_ORGANIC_SUBSTANCE</td>
<td align="right">1847</td>
<td align="right">27</td>
<td align="right">1.08e-05</td>
<td align="right">0.0044327</td>
</tr>
<tr class="odd">
<td align="left">GO_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY</td>
<td align="right">1158</td>
<td align="right">19</td>
<td align="right">1.14e-05</td>
<td align="right">0.0044327</td>
</tr>
<tr class="even">
<td align="left">GO_RECEPTOR_ACTIVITY</td>
<td align="right">1614</td>
<td align="right">24</td>
<td align="right">1.22e-05</td>
<td align="right">0.0044327</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_MATRIX_COMPONENT</td>
<td align="right">125</td>
<td align="right">7</td>
<td align="right">1.22e-05</td>
<td align="right">0.0044327</td>
</tr>
<tr class="even">
<td align="left">GO_ORGAN_MORPHOGENESIS</td>
<td align="right">841</td>
<td align="right">17</td>
<td align="right">1.32e-05</td>
<td align="right">0.0045014</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_NITROGEN_COMPOUND</td>
<td align="right">857</td>
<td align="right">17</td>
<td align="right">1.39e-05</td>
<td align="right">0.0045014</td>
</tr>
<tr class="even">
<td align="left">GO_CHEMICAL_HOMEOSTASIS</td>
<td align="right">871</td>
<td align="right">17</td>
<td align="right">1.64e-05</td>
<td align="right">0.0049342</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-Epithelial-{colnames(cont)[6]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-Epithelial-{colnames(cont)[6]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top)[1:50], 
           y$samples$Broad == &quot;Epithelial&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;))
  
dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.column = Gene,
          .row = Participant,
          .value = `Log2 CPM`,
          .scale = &quot;column&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 9),
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2),
                                               viridis::magma(5))) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;bottom&quot;,  
                       annotation_legend_side = &quot;bottom&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-34-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, 50)

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-35-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="neutrophils" class="section level2">
<h2>Neutrophils</h2>
<div id="mean-difference-plot-of-degs-3" class="section level3">
<h3>Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,10],
                  status = as.factor(as.vector(dt[,10]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-36-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 10, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,10] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;Neutrophils.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-37-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs-3" class="section level3">
<h3>Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs-New&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 10, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">IRF2</td>
<td align="right">-4.123220</td>
<td align="right">6.806343</td>
<td align="right">-5.115139</td>
<td align="right">5.82500e-07</td>
<td align="right">0.006082687</td>
</tr>
<tr class="even">
<td align="left">ACOD1</td>
<td align="right">4.185719</td>
<td align="right">2.600866</td>
<td align="right">4.793287</td>
<td align="right">2.31810e-06</td>
<td align="right">0.008721806</td>
</tr>
<tr class="odd">
<td align="left">FBXO11</td>
<td align="right">-3.641273</td>
<td align="right">5.994326</td>
<td align="right">-4.722531</td>
<td align="right">3.11130e-06</td>
<td align="right">0.008721806</td>
</tr>
<tr class="even">
<td align="left">TMX4</td>
<td align="right">-3.690924</td>
<td align="right">7.000616</td>
<td align="right">-4.687236</td>
<td align="right">3.60350e-06</td>
<td align="right">0.008721806</td>
</tr>
<tr class="odd">
<td align="left">PCNX4</td>
<td align="right">-3.702957</td>
<td align="right">5.826815</td>
<td align="right">-4.651615</td>
<td align="right">4.17590e-06</td>
<td align="right">0.008721806</td>
</tr>
<tr class="even">
<td align="left">UBE2G1</td>
<td align="right">-3.379038</td>
<td align="right">5.956099</td>
<td align="right">-4.498888</td>
<td align="right">7.78540e-06</td>
<td align="right">0.010140624</td>
</tr>
<tr class="odd">
<td align="left">DDRGK1</td>
<td align="right">-3.330343</td>
<td align="right">6.520868</td>
<td align="right">-4.484608</td>
<td align="right">8.24640e-06</td>
<td align="right">0.010140624</td>
</tr>
<tr class="even">
<td align="left">HNRNPUL2</td>
<td align="right">-3.914500</td>
<td align="right">6.292494</td>
<td align="right">-4.484604</td>
<td align="right">8.26560e-06</td>
<td align="right">0.010140624</td>
</tr>
<tr class="odd">
<td align="left">PRPF19</td>
<td align="right">-3.321016</td>
<td align="right">5.506008</td>
<td align="right">-4.462643</td>
<td align="right">9.00880e-06</td>
<td align="right">0.010140624</td>
</tr>
<tr class="even">
<td align="left">CDK9</td>
<td align="right">-3.604725</td>
<td align="right">5.827058</td>
<td align="right">-4.444193</td>
<td align="right">9.71050e-06</td>
<td align="right">0.010140624</td>
</tr>
<tr class="odd">
<td align="left">GOSR1</td>
<td align="right">-3.091350</td>
<td align="right">5.900311</td>
<td align="right">-4.367553</td>
<td align="right">1.31617e-05</td>
<td align="right">0.012495205</td>
</tr>
<tr class="even">
<td align="left">ST3GAL1</td>
<td align="right">-3.697461</td>
<td align="right">6.053926</td>
<td align="right">-4.313395</td>
<td align="right">1.63377e-05</td>
<td align="right">0.014217859</td>
</tr>
<tr class="odd">
<td align="left">CRIPT</td>
<td align="right">-3.228809</td>
<td align="right">5.829188</td>
<td align="right">-4.284276</td>
<td align="right">1.82788e-05</td>
<td align="right">0.014683488</td>
</tr>
<tr class="even">
<td align="left">TBC1D20</td>
<td align="right">-3.241148</td>
<td align="right">5.694579</td>
<td align="right">-4.217675</td>
<td align="right">2.36948e-05</td>
<td align="right">0.017674647</td>
</tr>
<tr class="odd">
<td align="left">CAND1</td>
<td align="right">-3.252716</td>
<td align="right">6.180806</td>
<td align="right">-4.174876</td>
<td align="right">2.79566e-05</td>
<td align="right">0.018729395</td>
</tr>
<tr class="even">
<td align="left">NRBF2</td>
<td align="right">-3.840438</td>
<td align="right">6.104421</td>
<td align="right">-4.169067</td>
<td align="right">2.86958e-05</td>
<td align="right">0.018729395</td>
</tr>
<tr class="odd">
<td align="left">GART</td>
<td align="right">-2.997201</td>
<td align="right">5.525178</td>
<td align="right">-4.062426</td>
<td align="right">4.28976e-05</td>
<td align="right">0.026351730</td>
</tr>
<tr class="even">
<td align="left">MLLT10</td>
<td align="right">-3.443529</td>
<td align="right">5.702181</td>
<td align="right">-4.045052</td>
<td align="right">4.59070e-05</td>
<td align="right">0.026633717</td>
</tr>
<tr class="odd">
<td align="left">THUMPD3</td>
<td align="right">-3.094209</td>
<td align="right">5.493968</td>
<td align="right">-3.943417</td>
<td align="right">6.70039e-05</td>
<td align="right">0.036827456</td>
</tr>
<tr class="even">
<td align="left">BAD</td>
<td align="right">-3.231151</td>
<td align="right">6.044779</td>
<td align="right">-3.863868</td>
<td align="right">8.98887e-05</td>
<td align="right">0.046935403</td>
</tr>
<tr class="odd">
<td align="left">MDM2</td>
<td align="right">-3.349842</td>
<td align="right">5.866136</td>
<td align="right">-3.837605</td>
<td align="right">9.90329e-05</td>
<td align="right">0.049247629</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/Neu-DEGs-{colnames(cont)[10]}.csv&quot;))</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">3</td>
<td align="right">0.0002134</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">4</td>
<td align="right">0.0004622</td>
<td align="right">0.1146752</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0005104</td>
<td align="right">0.1146752</td>
</tr>
<tr class="even">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">5</td>
<td align="right">0.0013274</td>
<td align="right">0.2236600</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0016910</td>
<td align="right">0.2279411</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">4</td>
<td align="right">0.0032207</td>
<td align="right">0.3560318</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">3</td>
<td align="right">0.0036977</td>
<td align="right">0.3560318</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">2</td>
<td align="right">0.0052907</td>
<td align="right">0.4457399</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">3</td>
<td align="right">0.0065906</td>
<td align="right">0.4935657</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0103041</td>
<td align="right">0.6944954</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0193157</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0216091</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_DOPAMINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">11</td>
<td align="right">1</td>
<td align="right">0.0239236</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_SIGNALING_BY_GPCR</td>
<td align="right">902</td>
<td align="right">5</td>
<td align="right">0.0281039</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0.0341306</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NA_CL_DEPENDENT_NEUROTRANSMITTER_TRANSPORTERS</td>
<td align="right">17</td>
<td align="right">1</td>
<td align="right">0.0342693</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS</td>
<td align="right">19</td>
<td align="right">1</td>
<td align="right">0.0389305</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">1</td>
<td align="right">0.0407147</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GLUCURONIDATION</td>
<td align="right">18</td>
<td align="right">1</td>
<td align="right">0.0413113</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_G_ALPHA_Q_SIGNALLING_EVENTS</td>
<td align="right">180</td>
<td align="right">2</td>
<td align="right">0.0510888</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="71%" />
<col width="5%" />
<col width="3%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_MULTI_ORGANISM_PROCESS</td>
<td align="right">157</td>
<td align="right">4</td>
<td align="right">0.0000109</td>
<td align="right">0.0672057</td>
</tr>
<tr class="even">
<td align="left">GO_PROTEIN_MODIFICATION_BY_SMALL_PROTEIN_CONJUGATION_OR_REMOVAL</td>
<td align="right">871</td>
<td align="right">6</td>
<td align="right">0.0001154</td>
<td align="right">0.2538301</td>
</tr>
<tr class="odd">
<td align="left">GO_NEGATIVE_REGULATION_OF_CELL_CYCLE_ARREST</td>
<td align="right">20</td>
<td align="right">2</td>
<td align="right">0.0001235</td>
<td align="right">0.2538301</td>
</tr>
<tr class="even">
<td align="left">GO_PROTEIN_UBIQUITINATION</td>
<td align="right">627</td>
<td align="right">5</td>
<td align="right">0.0002392</td>
<td align="right">0.3687187</td>
</tr>
<tr class="odd">
<td align="left">GO_CELLULAR_RESPONSE_TO_STRESS</td>
<td align="right">1561</td>
<td align="right">7</td>
<td align="right">0.0003417</td>
<td align="right">0.3992904</td>
</tr>
<tr class="even">
<td align="left">GO_CELLULAR_RESPONSE_TO_ALKALOID</td>
<td align="right">34</td>
<td align="right">2</td>
<td align="right">0.0004335</td>
<td align="right">0.3992904</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_K63_LINKED_UBIQUITINATION</td>
<td align="right">36</td>
<td align="right">2</td>
<td align="right">0.0005535</td>
<td align="right">0.3992904</td>
</tr>
<tr class="even">
<td align="left">GO_UBIQUITIN_LIKE_PROTEIN_TRANSFERASE_ACTIVITY</td>
<td align="right">419</td>
<td align="right">4</td>
<td align="right">0.0006334</td>
<td align="right">0.3992904</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MULTI_ORGANISM_PROCESS</td>
<td align="right">470</td>
<td align="right">4</td>
<td align="right">0.0006449</td>
<td align="right">0.3992904</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_GENE_EXPRESSION</td>
<td align="right">1730</td>
<td align="right">7</td>
<td align="right">0.0006476</td>
<td align="right">0.3992904</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_PROGESTERONE</td>
<td align="right">50</td>
<td align="right">2</td>
<td align="right">0.0007340</td>
<td align="right">0.4082925</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_SYMBIOSIS_ENCOMPASSING_MUTUALISM_THROUGH_PARASITISM</td>
<td align="right">205</td>
<td align="right">3</td>
<td align="right">0.0008339</td>
<td align="right">0.4082925</td>
</tr>
<tr class="odd">
<td align="left">GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_ACTIVITY</td>
<td align="right">199</td>
<td align="right">3</td>
<td align="right">0.0008608</td>
<td align="right">0.4082925</td>
</tr>
<tr class="even">
<td align="left">GO_TRANSFERASE_ACTIVITY_TRANSFERRING_ONE_CARBON_GROUPS</td>
<td align="right">224</td>
<td align="right">3</td>
<td align="right">0.0009499</td>
<td align="right">0.4158295</td>
</tr>
<tr class="odd">
<td align="left">GO_NEGATIVE_REGULATION_OF_CELL_CYCLE_PROCESS</td>
<td align="right">214</td>
<td align="right">3</td>
<td align="right">0.0010465</td>
<td align="right">0.4158295</td>
</tr>
<tr class="even">
<td align="left">GO_SCAFFOLD_PROTEIN_BINDING</td>
<td align="right">45</td>
<td align="right">2</td>
<td align="right">0.0010790</td>
<td align="right">0.4158295</td>
</tr>
<tr class="odd">
<td align="left">GO_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS</td>
<td align="right">233</td>
<td align="right">3</td>
<td align="right">0.0012251</td>
<td align="right">0.4443678</td>
</tr>
<tr class="even">
<td align="left">GO_CELLULAR_RESPONSE_TO_CYTOKINE_STIMULUS</td>
<td align="right">606</td>
<td align="right">4</td>
<td align="right">0.0014359</td>
<td align="right">0.4918595</td>
</tr>
<tr class="odd">
<td align="left">GO_INTRINSIC_COMPONENT_OF_GOLGI_MEMBRANE</td>
<td align="right">57</td>
<td align="right">2</td>
<td align="right">0.0015929</td>
<td align="right">0.5169315</td>
</tr>
<tr class="even">
<td align="left">GO_UBIQUITIN_LIKE_PROTEIN_LIGASE_BINDING</td>
<td align="right">264</td>
<td align="right">3</td>
<td align="right">0.0020132</td>
<td align="right">0.6088209</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-Neu-{colnames(cont)[10]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-Neu-{colnames(cont)[10]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;Neutrophils&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;))
  
dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.column = Gene,
          .row = Participant,
          .value = `Log2 CPM`,
          .scale = &quot;column&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 9),
          palette_value = circlize::colorRamp2(c(-2, -1, 0, 1, 2),
                                               viridis::magma(5))) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;bottom&quot;,  
                       annotation_legend_side = &quot;bottom&quot;) </code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-40-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, nrow(top))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-41-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="panel-figures" class="section level1">
<h1>Panel figures</h1>
<pre class="r"><code>layout = &quot;A
          A
          B
          B
          B
          B
          B&quot;

(f5a + theme(legend.position = &quot;right&quot;,
             legend.direction = &quot;vertical&quot;)) +
  wrap_elements(grid::grid.grabExpr(ComplexHeatmap::draw(f5b))) + 
  plot_layout(design = layout) +
  plot_annotation(tag_levels = &quot;A&quot;) &amp;
  theme(plot.tag = element_text(size = 14, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-42-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/9c401bfd1fdf8405e06469739bd24aae6d49d89a/docs/figure/04_COMBO.expression_analysis.Rmd/unnamed-chunk-42-1.png" target="_blank">9c401bf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.0 (2021-05-18)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-05-26
 pandoc   2.14.0.3 @ /usr/lib/rstudio-server/bin/pandoc/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package                                       * version    date (UTC) lib source
 P abind                                           1.4-5      2016-07-21 [?] CRAN (R 4.1.0)
 P annotate                                        1.72.0     2021-10-26 [?] Bioconductor
 P AnnotationDbi                                 * 1.56.2     2021-11-09 [?] Bioconductor
 P AnnotationFilter                              * 1.18.0     2021-10-26 [?] Bioconductor
 P AnnotationHub                                 * 3.2.0      2021-10-26 [?] Bioconductor
 P askpass                                         1.1        2019-01-13 [?] CRAN (R 4.0.2)
 P assertthat                                      0.2.1      2019-03-21 [?] CRAN (R 4.1.0)
 P backports                                       1.4.1      2021-12-13 [?] CRAN (R 4.1.0)
 P base64                                          2.0        2016-05-10 [?] CRAN (R 4.1.0)
 P beachmat                                        2.10.0     2021-10-26 [?] Bioconductor
 P beanplot                                        1.3.1      2022-04-09 [?] CRAN (R 4.1.0)
 P beeswarm                                        0.4.0      2021-06-01 [?] CRAN (R 4.1.0)
 P BiasedUrn                                       1.07       2015-12-28 [?] CRAN (R 4.1.0)
 P Biobase                                       * 2.54.0     2021-10-26 [?] Bioconductor
 P BiocFileCache                                 * 2.2.0      2021-10-26 [?] Bioconductor
 P BiocGenerics                                  * 0.40.0     2021-10-26 [?] Bioconductor
 P BiocIO                                          1.4.0      2021-10-26 [?] Bioconductor
 P BiocManager                                     1.30.16    2021-06-15 [?] CRAN (R 4.1.0)
 P BiocNeighbors                                   1.12.0     2021-10-26 [?] Bioconductor
 P BiocParallel                                  * 1.28.3     2021-12-09 [?] Bioconductor
 P BiocSingular                                    1.10.0     2021-10-26 [?] Bioconductor
 P BiocVersion                                     3.14.0     2021-05-19 [?] Bioconductor
 P biomaRt                                         2.50.1     2021-11-21 [?] Bioconductor
 P Biostrings                                    * 2.62.0     2021-10-26 [?] Bioconductor
 P bit                                             4.0.4      2020-08-04 [?] CRAN (R 4.1.0)
 P bit64                                           4.0.5      2020-08-30 [?] CRAN (R 4.0.2)
 P bitops                                          1.0-7      2021-04-24 [?] CRAN (R 4.0.2)
 P blob                                            1.2.2      2021-07-23 [?] CRAN (R 4.1.0)
 P bluster                                         1.4.0      2021-10-26 [?] Bioconductor
 P broom                                           0.7.11     2022-01-03 [?] CRAN (R 4.1.0)
 P bslib                                           0.3.1      2021-10-06 [?] CRAN (R 4.1.0)
 P bumphunter                                    * 1.36.0     2021-10-27 [?] Bioconductor
 P cachem                                          1.0.6      2021-08-19 [?] CRAN (R 4.1.0)
 P callr                                           3.7.0      2021-04-20 [?] CRAN (R 4.1.0)
 P caTools                                         1.18.2     2021-03-28 [?] CRAN (R 4.1.0)
 P cellranger                                      1.1.0      2016-07-27 [?] CRAN (R 4.1.0)
 P circlize                                        0.4.13     2021-06-09 [?] CRAN (R 4.1.0)
 P cli                                             3.1.0      2021-10-27 [?] CRAN (R 4.1.0)
 P clue                                            0.3-60     2021-10-11 [?] CRAN (R 4.1.0)
 P cluster                                         2.1.2      2021-04-17 [?] CRAN (R 4.1.0)
 P codetools                                       0.2-18     2020-11-04 [?] CRAN (R 4.1.0)
 P colorspace                                      2.0-2      2021-06-24 [?] CRAN (R 4.0.2)
 P ComplexHeatmap                                  2.10.0     2021-10-26 [?] Bioconductor
 P cowplot                                         1.1.1      2020-12-30 [?] CRAN (R 4.0.2)
 P crayon                                          1.4.2      2021-10-29 [?] CRAN (R 4.1.0)
 P curl                                            4.3.2      2021-06-23 [?] CRAN (R 4.1.0)
 P data.table                                      1.14.2     2021-09-27 [?] CRAN (R 4.1.0)
 P DBI                                             1.1.2      2021-12-20 [?] CRAN (R 4.1.0)
 P dbplyr                                        * 2.1.1      2021-04-06 [?] CRAN (R 4.1.0)
 P DelayedArray                                    0.20.0     2021-10-26 [?] Bioconductor
 P DelayedMatrixStats                              1.16.0     2021-10-26 [?] Bioconductor
 P deldir                                          1.0-6      2021-10-23 [?] CRAN (R 4.1.0)
 P dendextend                                      1.15.2     2021-10-28 [?] CRAN (R 4.1.0)
 P digest                                          0.6.29     2021-12-01 [?] CRAN (R 4.1.0)
 P doParallel                                      1.0.16     2020-10-16 [?] CRAN (R 4.0.2)
 P doRNG                                           1.8.2      2020-01-27 [?] CRAN (R 4.1.0)
 P dplyr                                         * 1.0.7      2021-06-18 [?] CRAN (R 4.1.0)
 P dqrng                                           0.3.0      2021-05-01 [?] CRAN (R 4.1.0)
 P DropletUtils                                  * 1.14.1     2021-11-08 [?] Bioconductor
 P DT                                              0.20       2021-11-15 [?] CRAN (R 4.1.0)
 P edgeR                                         * 3.36.0     2021-10-26 [?] Bioconductor
 P EGSEA                                         * 1.22.0     2021-10-26 [?] Bioconductor
 P EGSEAdata                                       1.22.0     2021-10-30 [?] Bioconductor
 P ellipsis                                        0.3.2      2021-04-29 [?] CRAN (R 4.0.2)
 P ensembldb                                     * 2.18.2     2021-11-08 [?] Bioconductor
 P evaluate                                        0.14       2019-05-28 [?] CRAN (R 4.0.2)
 P fansi                                           1.0.0      2022-01-10 [?] CRAN (R 4.1.0)
 P farver                                          2.1.0      2021-02-28 [?] CRAN (R 4.0.2)
 P fastmap                                         1.1.0      2021-01-25 [?] CRAN (R 4.1.0)
 P filelock                                        1.0.2      2018-10-05 [?] CRAN (R 4.1.0)
 P fitdistrplus                                    1.1-6      2021-09-28 [?] CRAN (R 4.1.0)
 P forcats                                       * 0.5.1      2021-01-27 [?] CRAN (R 4.1.0)
 P foreach                                       * 1.5.1      2020-10-15 [?] CRAN (R 4.0.2)
 P fs                                              1.5.2      2021-12-08 [?] CRAN (R 4.1.0)
 P future                                          1.23.0     2021-10-31 [?] CRAN (R 4.1.0)
 P future.apply                                    1.8.1      2021-08-10 [?] CRAN (R 4.1.0)
 P gage                                          * 2.44.0     2021-10-26 [?] Bioconductor
 P genefilter                                      1.76.0     2021-10-26 [?] Bioconductor
 P generics                                        0.1.1      2021-10-25 [?] CRAN (R 4.1.0)
   GenomeInfoDb                                  * 1.30.1     2022-01-30 [1] Bioconductor
 P GenomeInfoDbData                                1.2.7      2021-12-21 [?] Bioconductor
 P GenomicAlignments                               1.30.0     2021-10-26 [?] Bioconductor
 P GenomicFeatures                               * 1.46.3     2021-12-30 [?] Bioconductor
 P GenomicRanges                                 * 1.46.1     2021-11-18 [?] Bioconductor
 P GEOquery                                        2.62.2     2022-01-11 [?] Bioconductor
 P GetoptLong                                      1.0.5      2020-12-15 [?] CRAN (R 4.0.2)
 P getPass                                         0.2-2      2017-07-21 [?] CRAN (R 4.0.2)
 P ggbeeswarm                                      0.6.0      2017-08-07 [?] CRAN (R 4.1.0)
 P ggplot2                                       * 3.3.5      2021-06-25 [?] CRAN (R 4.0.2)
 P ggrepel                                         0.9.1      2021-01-15 [?] CRAN (R 4.1.0)
 P ggridges                                        0.5.3      2021-01-08 [?] CRAN (R 4.1.0)
 P git2r                                           0.29.0     2021-11-22 [?] CRAN (R 4.1.0)
 P glmGamPoi                                     * 1.6.0      2021-10-26 [?] Bioconductor
 P GlobalOptions                                   0.1.2      2020-06-10 [?] CRAN (R 4.1.0)
 P globals                                         0.14.0     2020-11-22 [?] CRAN (R 4.0.2)
 P globaltest                                      5.48.0     2021-10-26 [?] Bioconductor
 P glue                                          * 1.6.0      2021-12-17 [?] CRAN (R 4.1.0)
 P GO.db                                         * 3.14.0     2021-12-21 [?] Bioconductor
 P goftest                                         1.2-3      2021-10-07 [?] CRAN (R 4.1.0)
 P gplots                                          3.1.1      2020-11-28 [?] CRAN (R 4.0.2)
 P graph                                         * 1.72.0     2021-10-26 [?] Bioconductor
 P gridExtra                                       2.3        2017-09-09 [?] CRAN (R 4.1.0)
 P GSA                                             1.03.1     2019-01-31 [?] CRAN (R 4.1.0)
 P GSEABase                                        1.56.0     2021-10-26 [?] Bioconductor
 P GSVA                                            1.42.0     2021-10-26 [?] Bioconductor
 P gtable                                          0.3.0      2019-03-25 [?] CRAN (R 4.1.0)
 P gtools                                          3.9.2      2021-06-06 [?] CRAN (R 4.1.0)
 P haven                                           2.4.3      2021-08-04 [?] CRAN (R 4.1.0)
 P HDF5Array                                       1.22.1     2021-11-14 [?] Bioconductor
 P here                                          * 1.0.1      2020-12-13 [?] CRAN (R 4.0.2)
 P hgu133a.db                                      3.13.0     2022-01-24 [?] Bioconductor
 P hgu133plus2.db                                  3.13.0     2022-01-24 [?] Bioconductor
 P highr                                           0.9        2021-04-16 [?] CRAN (R 4.1.0)
 P hms                                             1.1.1      2021-09-26 [?] CRAN (R 4.1.0)
 P htmltools                                       0.5.2      2021-08-25 [?] CRAN (R 4.1.0)
 P HTMLUtils                                       0.1.7      2015-01-17 [?] CRAN (R 4.1.0)
 P htmlwidgets                                     1.5.4      2021-09-08 [?] CRAN (R 4.1.0)
 P httpuv                                          1.6.5      2022-01-05 [?] CRAN (R 4.1.0)
 P httr                                            1.4.2      2020-07-20 [?] CRAN (R 4.1.0)
 P hwriter                                         1.3.2      2014-09-10 [?] CRAN (R 4.1.0)
 P ica                                             1.0-2      2018-05-24 [?] CRAN (R 4.1.0)
 P igraph                                          1.2.11     2022-01-04 [?] CRAN (R 4.1.0)
 P IlluminaHumanMethylation450kanno.ilmn12.hg19  * 0.6.0      2022-05-10 [?] Bioconductor
 P IlluminaHumanMethylationEPICanno.ilm10b4.hg19 * 0.6.0      2022-05-10 [?] Bioconductor
 P illuminaio                                      0.36.0     2021-10-26 [?] Bioconductor
 P interactiveDisplayBase                          1.32.0     2021-10-26 [?] Bioconductor
 P IRanges                                       * 2.28.0     2021-10-26 [?] Bioconductor
 P irlba                                           2.3.5      2021-12-06 [?] CRAN (R 4.1.0)
 P iterators                                     * 1.0.13     2020-10-15 [?] CRAN (R 4.0.2)
 P janitor                                       * 2.1.0      2021-01-05 [?] CRAN (R 4.0.2)
 P jquerylib                                       0.1.4      2021-04-26 [?] CRAN (R 4.1.0)
 P jsonlite                                        1.7.2      2020-12-09 [?] CRAN (R 4.0.2)
 P KEGGdzPathwaysGEO                               1.32.0     2021-10-30 [?] Bioconductor
 P KEGGgraph                                       1.54.0     2021-10-26 [?] Bioconductor
 P KEGGREST                                        1.34.0     2021-10-26 [?] Bioconductor
 P KernSmooth                                      2.23-20    2021-05-03 [?] CRAN (R 4.1.0)
 P knitr                                           1.37       2021-12-16 [?] CRAN (R 4.1.0)
 P labeling                                        0.4.2      2020-10-20 [?] CRAN (R 4.0.2)
 P later                                           1.3.0      2021-08-18 [?] CRAN (R 4.1.0)
 P lattice                                         0.20-45    2021-09-22 [?] CRAN (R 4.1.0)
 P lazyeval                                        0.2.2      2019-03-15 [?] CRAN (R 4.1.0)
 P leiden                                          0.3.9      2021-07-27 [?] CRAN (R 4.1.0)
 P lifecycle                                       1.0.1      2021-09-24 [?] CRAN (R 4.1.0)
 P limma                                         * 3.50.0     2021-10-26 [?] Bioconductor
 P listenv                                         0.8.0      2019-12-05 [?] CRAN (R 4.1.0)
 P lmtest                                          0.9-39     2021-11-07 [?] CRAN (R 4.1.0)
 P locfit                                        * 1.5-9.4    2020-03-25 [?] CRAN (R 4.1.0)
 P lubridate                                       1.8.0      2021-10-07 [?] CRAN (R 4.1.0)
 P magrittr                                        2.0.1      2020-11-17 [?] CRAN (R 4.0.2)
 P MASS                                            7.3-53.1   2021-02-12 [?] CRAN (R 4.0.2)
 P mathjaxr                                        1.4-0      2021-03-01 [?] CRAN (R 4.1.0)
 P Matrix                                        * 1.4-0      2021-12-08 [?] CRAN (R 4.1.0)
 P MatrixGenerics                                * 1.6.0      2021-10-26 [?] Bioconductor
 P matrixStats                                   * 0.61.0     2021-09-17 [?] CRAN (R 4.1.0)
 P mclust                                          5.4.9      2021-12-17 [?] CRAN (R 4.1.0)
 P memoise                                         2.0.1      2021-11-26 [?] CRAN (R 4.1.0)
 P metap                                           1.7        2021-12-16 [?] CRAN (R 4.1.0)
 P metapod                                         1.2.0      2021-10-26 [?] Bioconductor
 P mgcv                                            1.8-38     2021-10-06 [?] CRAN (R 4.1.0)
 P mime                                            0.12       2021-09-28 [?] CRAN (R 4.1.0)
 P minfi                                         * 1.40.0     2021-10-26 [?] Bioconductor
 P miniUI                                          0.1.1.1    2018-05-18 [?] CRAN (R 4.1.0)
 P missMethyl                                    * 1.28.0     2021-10-26 [?] Bioconductor
 P mnormt                                          2.0.2      2020-09-01 [?] CRAN (R 4.0.2)
 P modelr                                          0.1.8      2020-05-19 [?] CRAN (R 4.0.2)
 P multcomp                                        1.4-18     2022-01-04 [?] CRAN (R 4.1.0)
 P multtest                                        2.50.0     2021-10-26 [?] Bioconductor
 P munsell                                         0.5.0      2018-06-12 [?] CRAN (R 4.1.0)
 P mutoss                                          0.1-12     2017-12-04 [?] CRAN (R 4.1.0)
 P mvtnorm                                         1.1-3      2021-10-08 [?] CRAN (R 4.1.0)
 P nlme                                            3.1-153    2021-09-07 [?] CRAN (R 4.1.0)
 P nor1mix                                         1.3-0      2019-06-13 [?] CRAN (R 4.1.0)
 P numDeriv                                        2016.8-1.1 2019-06-06 [?] CRAN (R 4.1.0)
 P openssl                                         1.4.6      2021-12-19 [?] CRAN (R 4.1.0)
 P org.Hs.eg.db                                  * 3.14.0     2021-12-21 [?] Bioconductor
 P org.Mm.eg.db                                    3.14.0     2022-01-24 [?] Bioconductor
 P org.Rn.eg.db                                    3.14.0     2022-01-24 [?] Bioconductor
 P PADOG                                           1.36.0     2021-10-26 [?] Bioconductor
 P paletteer                                       1.4.0      2021-07-20 [?] CRAN (R 4.1.0)
 P parallelly                                      1.30.0     2021-12-17 [?] CRAN (R 4.1.0)
 P patchwork                                     * 1.1.1      2020-12-17 [?] CRAN (R 4.0.2)
 P pathview                                      * 1.34.0     2021-10-26 [?] Bioconductor
 P pbapply                                         1.5-0      2021-09-16 [?] CRAN (R 4.1.0)
 P pillar                                          1.6.4      2021-10-18 [?] CRAN (R 4.1.0)
 P pkgconfig                                       2.0.3      2019-09-22 [?] CRAN (R 4.1.0)
 P plotly                                          4.10.0     2021-10-09 [?] CRAN (R 4.1.0)
 P plotrix                                         3.8-2      2021-09-08 [?] CRAN (R 4.1.0)
 P plyr                                            1.8.6      2020-03-03 [?] CRAN (R 4.0.2)
 P png                                             0.1-7      2013-12-03 [?] CRAN (R 4.1.0)
 P polyclip                                        1.10-0     2019-03-14 [?] CRAN (R 4.1.0)
 P preprocessCore                                  1.56.0     2021-10-26 [?] Bioconductor
 P prettyunits                                     1.1.1      2020-01-24 [?] CRAN (R 4.0.2)
 P prismatic                                       1.1.0      2021-10-17 [?] CRAN (R 4.1.0)
 P processx                                        3.5.2      2021-04-30 [?] CRAN (R 4.1.0)
 P progress                                        1.2.2      2019-05-16 [?] CRAN (R 4.1.0)
 P promises                                        1.2.0.1    2021-02-11 [?] CRAN (R 4.0.2)
 P ProtGenerics                                    1.26.0     2021-10-26 [?] Bioconductor
 P ps                                              1.6.0      2021-02-28 [?] CRAN (R 4.1.0)
 P purrr                                         * 0.3.4      2020-04-17 [?] CRAN (R 4.0.2)
 P quadprog                                        1.5-8      2019-11-20 [?] CRAN (R 4.1.0)
 P R.methodsS3                                     1.8.1      2020-08-26 [?] CRAN (R 4.0.2)
 P R.oo                                            1.24.0     2020-08-26 [?] CRAN (R 4.0.2)
 P R.utils                                         2.11.0     2021-09-26 [?] CRAN (R 4.1.0)
 P R2HTML                                          2.3.2      2016-06-23 [?] CRAN (R 4.1.0)
 P R6                                              2.5.1      2021-08-19 [?] CRAN (R 4.1.0)
 P RANN                                            2.6.1      2019-01-08 [?] CRAN (R 4.1.0)
 P rappdirs                                        0.3.3      2021-01-31 [?] CRAN (R 4.0.2)
 P rbibutils                                       2.2.7      2021-12-07 [?] CRAN (R 4.1.0)
 P RColorBrewer                                    1.1-2      2014-12-07 [?] CRAN (R 4.0.2)
 P Rcpp                                            1.0.7      2021-07-07 [?] CRAN (R 4.1.0)
 P RcppAnnoy                                       0.0.19     2021-07-30 [?] CRAN (R 4.1.0)
   RCurl                                           1.98-1.6   2022-02-08 [1] CRAN (R 4.1.0)
 P Rdpack                                          2.1.3      2021-12-08 [?] CRAN (R 4.1.0)
 P readr                                         * 2.1.1      2021-11-30 [?] CRAN (R 4.1.0)
 P readxl                                          1.3.1      2019-03-13 [?] CRAN (R 4.1.0)
 P rematch2                                        2.1.2      2020-05-01 [?] CRAN (R 4.1.0)
 P renv                                            0.15.0-14  2022-01-10 [?] Github (rstudio/renv@a3b90eb)
 P reprex                                          2.0.1      2021-08-05 [?] CRAN (R 4.1.0)
 P reshape                                         0.8.8      2018-10-23 [?] CRAN (R 4.0.2)
 P reshape2                                        1.4.4      2020-04-09 [?] CRAN (R 4.1.0)
 P restfulr                                        0.0.13     2017-08-06 [?] CRAN (R 4.1.0)
 P reticulate                                      1.22       2021-09-17 [?] CRAN (R 4.1.0)
 P Rgraphviz                                       2.38.0     2021-10-26 [?] Bioconductor
 P rhdf5                                           2.38.0     2021-10-26 [?] Bioconductor
 P rhdf5filters                                    1.6.0      2021-10-26 [?] Bioconductor
 P Rhdf5lib                                        1.16.0     2021-10-26 [?] Bioconductor
 P rjson                                           0.2.21     2022-01-09 [?] CRAN (R 4.1.0)
 P rlang                                           0.4.12     2021-10-18 [?] CRAN (R 4.1.0)
 P rmarkdown                                       2.11       2021-09-14 [?] CRAN (R 4.1.0)
 P rngtools                                        1.5.2      2021-09-20 [?] CRAN (R 4.1.0)
 P ROCR                                            1.0-11     2020-05-02 [?] CRAN (R 4.1.0)
 P rpart                                           4.1-15     2019-04-12 [?] CRAN (R 4.1.0)
 P rprojroot                                       2.0.2      2020-11-15 [?] CRAN (R 4.0.2)
 P Rsamtools                                       2.10.0     2021-10-26 [?] Bioconductor
 P RSQLite                                         2.2.9      2021-12-06 [?] CRAN (R 4.1.0)
 P rstudioapi                                      0.13       2020-11-12 [?] CRAN (R 4.0.2)
 P rsvd                                            1.0.5      2021-04-16 [?] CRAN (R 4.1.0)
 P rtracklayer                                     1.54.0     2021-10-26 [?] Bioconductor
 P Rtsne                                           0.15       2018-11-10 [?] CRAN (R 4.1.0)
 P rvest                                           1.0.2      2021-10-16 [?] CRAN (R 4.1.0)
 P S4Vectors                                     * 0.32.3     2021-11-21 [?] Bioconductor
 P safe                                            3.34.0     2021-10-26 [?] Bioconductor
 P sandwich                                        3.0-1      2021-05-18 [?] CRAN (R 4.1.0)
 P sass                                            0.4.0      2021-05-12 [?] CRAN (R 4.1.0)
 P ScaledMatrix                                    1.2.0      2021-10-26 [?] Bioconductor
 P scales                                        * 1.1.1      2020-05-11 [?] CRAN (R 4.0.2)
 P scater                                        * 1.22.0     2021-10-26 [?] Bioconductor
 P scattermore                                     0.7        2020-11-24 [?] CRAN (R 4.1.0)
 P scran                                         * 1.22.1     2021-11-14 [?] Bioconductor
 P scrime                                          1.3.5      2018-12-01 [?] CRAN (R 4.1.0)
 P sctransform                                     0.3.3      2022-01-13 [?] CRAN (R 4.1.0)
 P scuttle                                       * 1.4.0      2021-10-26 [?] Bioconductor
 P sessioninfo                                     1.2.2      2021-12-06 [?] CRAN (R 4.1.0)
 P Seurat                                        * 4.0.6      2021-12-16 [?] CRAN (R 4.1.0)
 P SeuratObject                                  * 4.0.4      2021-11-23 [?] CRAN (R 4.1.0)
 P shape                                           1.4.6      2021-05-19 [?] CRAN (R 4.1.0)
 P shiny                                           1.7.1      2021-10-02 [?] CRAN (R 4.1.0)
 P siggenes                                        1.68.0     2021-10-26 [?] Bioconductor
 P SingleCellExperiment                          * 1.16.0     2021-10-26 [?] Bioconductor
 P sn                                              2.0.1      2021-11-26 [?] CRAN (R 4.1.0)
 P snakecase                                       0.11.0     2019-05-25 [?] CRAN (R 4.0.2)
 P SparseM                                       * 1.81       2021-02-18 [?] CRAN (R 4.1.0)
 P sparseMatrixStats                               1.6.0      2021-10-26 [?] Bioconductor
 P spatstat.core                                   2.3-2      2021-11-26 [?] CRAN (R 4.1.0)
 P spatstat.data                                   2.1-2      2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.geom                                   2.3-1      2021-12-10 [?] CRAN (R 4.1.0)
 P spatstat.sparse                                 2.1-0      2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.utils                                  2.3-0      2021-12-12 [?] CRAN (R 4.1.0)
 P speckle                                       * 0.0.3      2022-03-09 [?] Github (Oshlack/speckle@fc07773)
 P statmod                                         1.4.36     2021-05-10 [?] CRAN (R 4.1.0)
 P stringi                                         1.7.6      2021-11-29 [?] CRAN (R 4.1.0)
 P stringr                                       * 1.4.0      2019-02-10 [?] CRAN (R 4.0.2)
 P SummarizedExperiment                          * 1.24.0     2021-10-26 [?] Bioconductor
 P survival                                        3.2-13     2021-08-24 [?] CRAN (R 4.1.0)
 P tensor                                          1.5        2012-05-05 [?] CRAN (R 4.1.0)
 P TFisher                                         0.2.0      2018-03-21 [?] CRAN (R 4.1.0)
 P TH.data                                         1.1-0      2021-09-27 [?] CRAN (R 4.1.0)
 P tibble                                        * 3.1.6      2021-11-07 [?] CRAN (R 4.1.0)
 P tidyHeatmap                                   * 1.7.0      2022-05-13 [?] Github (stemangiola/tidyHeatmap@241aec2)
 P tidyr                                         * 1.1.4      2021-09-27 [?] CRAN (R 4.1.0)
 P tidyselect                                      1.1.1      2021-04-30 [?] CRAN (R 4.1.0)
 P tidyverse                                     * 1.3.1      2021-04-15 [?] CRAN (R 4.1.0)
 P tmvnsim                                         1.0-2      2016-12-15 [?] CRAN (R 4.1.0)
 P topGO                                         * 2.46.0     2021-10-26 [?] Bioconductor
 P TxDb.Hsapiens.UCSC.hg38.knownGene             * 3.14.0     2022-05-10 [?] Bioconductor
 P tzdb                                            0.2.0      2021-10-27 [?] CRAN (R 4.1.0)
 P utf8                                            1.2.2      2021-07-24 [?] CRAN (R 4.1.0)
 P uwot                                            0.1.11     2021-12-02 [?] CRAN (R 4.1.0)
 P vctrs                                           0.3.8      2021-04-29 [?] CRAN (R 4.0.2)
 P vipor                                           0.4.5      2017-03-22 [?] CRAN (R 4.1.0)
 P viridis                                         0.6.2      2021-10-13 [?] CRAN (R 4.1.0)
 P viridisLite                                     0.4.0      2021-04-13 [?] CRAN (R 4.0.2)
 P vroom                                           1.5.7      2021-11-30 [?] CRAN (R 4.1.0)
 P whisker                                         0.4        2019-08-28 [?] CRAN (R 4.0.2)
 P withr                                           2.4.3      2021-11-30 [?] CRAN (R 4.1.0)
 P workflowr                                     * 1.7.0      2021-12-21 [?] CRAN (R 4.1.0)
 P xfun                                            0.29       2021-12-14 [?] CRAN (R 4.1.0)
 P XML                                             3.99-0.8   2021-09-17 [?] CRAN (R 4.1.0)
 P xml2                                            1.3.3      2021-11-30 [?] CRAN (R 4.1.0)
 P xtable                                          1.8-4      2019-04-21 [?] CRAN (R 4.1.0)
 P XVector                                       * 0.34.0     2021-10-26 [?] Bioconductor
 P yaml                                            2.2.1      2020-02-01 [?] CRAN (R 4.0.2)
 P zlibbioc                                        1.40.0     2021-10-26 [?] Bioconductor
 P zoo                                             1.8-9      2021-03-09 [?] CRAN (R 4.1.0)

 [1] /oshlack_lab/jovana.maksimovic/projects/MCRI/melanie.neeland/paed-cf-cite-seq/renv/library/R-4.1/x86_64-pc-linux-gnu
 [2] /config/binaries/R/4.1.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(fit.treat$coefficients[,1], fit.treat$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "Macrophages in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt[,1] !=0 | dt[,2] != 0 -->
<!-- text(fit.treat$coefficients[sig.genes,1], -->
<!--      fit.treat$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(fit.treat$coefficients[,3], fit.treat$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="Macrophages in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt[,1] !=0 | dt[,2] != 0 -->
<!-- text(fit.treat$coefficients[sig.genes,1], -->
<!--      fit.treat$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- ## Dendritic Cells  -->
<!-- ### Overall DEG summary -->
<!-- ```{r} -->
<!-- cont.Dend <- makeContrasts(MUvMT = Dend.MU-Dend.MT, -->
<!--                            SUvST = Dend.SU-Dend.ST, -->
<!--                            All.MvS = 0.5*(Dend.MU + Dend.MT) - 0.5*(Dend.SU + Dend.ST), -->
<!--                            UT.MvS = Dend.MU-Dend.SU, -->
<!--                            All.UvT = 0.5*(Dend.MU + Dend.SU) - 0.5*(Dend.MT + Dend.ST), -->
<!--                            levels = design) -->
<!-- fit.Dend <- contrasts.fit(fit, contrasts = cont.Dend) -->
<!-- fit.Dend <- eBayes(fit.Dend, robust = TRUE) -->
<!-- summary(decideTests(fit.Dend)) -->
<!-- ``` -->
<!-- ### TREAT DEG summary -->
<!-- This tests for DEGs relative to a logFC threshold of 0.5 -->
<!-- ```{r} -->
<!-- treat.Dend <- treat(fit.Dend, lfc = 0.5) -->
<!-- dt.Dend <- decideTests(treat.Dend) -->
<!-- summary(dt.Dend) -->
<!-- ``` -->
<!-- ### Mean-Difference plot of DEGs -->
<!-- ```{r, fig.asp=0.7} -->
<!-- par(mfrow=c(2,3)) -->
<!-- for(i in 1:ncol(cont.Dend)){ -->
<!--   plotMD(treat.Dend, coef = i, status = dt.Dend[,i], hl.col = c("blue","red")) -->
<!--   abline(h = 0, col = "grey") -->
<!-- } -->
<!-- ``` -->
<!-- ### Barplot of DEGs -->
<!-- ```{r,fig.asp=1} -->
<!-- par(mfrow=c(1,1)) -->
<!-- par(mar=c(7,4,2,2)) -->
<!-- barplot(summary(dt.Dend)[-2,], -->
<!--         beside = TRUE, -->
<!--         legend = TRUE, -->
<!--         col = c("blue","red"), -->
<!--         ylab = "Number of significant genes", -->
<!--         las = 2, -->
<!--         xlab = "") -->
<!-- title("Dendritic Cells") -->
<!-- ``` -->
<!-- ### Top DEGs -->
<!-- ```{r, results='asis', fig.asp = 1} -->
<!-- t <- vector("list", ncol(cont.Dend)) -->
<!-- names(t) <- colnames(cont.Dend) -->
<!-- for(i in 1:ncol(cont.Dend)){ -->
<!--   cat("\n", colnames(cont.Dend)[i],"\n") -->
<!--   top <- topTreat(treat.Dend, coef = i, n = 20, p.value = 0.05)   -->
<!--   topAll <- topTreat(treat.Dend, coef = i, n = Inf, p.value = 0.05) -->
<!--   write.csv(topAll, file = glue("{dirName}/Dend-DEGs-{colnames(cont.Dend)[i]}.csv")) -->
<!--   if(nrow(topAll) > 0) { -->
<!--     write.csv(topGO(goana(de = topAll$ENTREZID,  -->
<!--                           universe = treat.Dend$genes$ENTREZID,  -->
<!--                           species = "Hs"), number = 50), -->
<!--               file = here(glue("{dirName}/Dend-GO-{colnames(cont.Dend)[i]}.csv"))) -->
<!--   } -->
<!--   print(knitr::kable(top[,-c(1:3)], digits = 3)) -->
<!--   cat("\n") -->
<!--   if(nrow(top) > 1){ -->
<!--     exp <- normCounts(y.pb, log = TRUE) -->
<!--     keepCells <- y.pb$samples$cell_label == "Dend" -->
<!--     exp <- exp[rownames(exp) %in% top$SYMBOL, keepCells] -->
<!--     exp <- exp[, order(y.pb$samples$Group[keepCells])] -->
<!--     pheatmap::pheatmap(exp - rowMeans(exp), -->
<!--                        color = colorRampPalette(viridis_pal(option = "magma")(10))(100), -->
<!--                        annotation_col = data.frame(Group = y.pb$samples$Group[keepCells], -->
<!--                                                    Severity = toupper(substring(y.pb$samples$Severity[keepCells],1,1)), -->
<!--                                                    Treatment = toupper(substring(y.pb$samples$Treatment[keepCells],1,1)), -->
<!--                                                    row.names = rownames(y.pb$samples)[keepCells]), -->
<!--                        labels_col = y.pb$samples$donor[keepCells], -->
<!--                        scale = "none", -->
<!--                        cluster_cols = FALSE, -->
<!--                        cellwidth = 20, -->
<!--                        cellheight = 20, -->
<!--                        breaks = seq(-3, 3, length.out = 101)) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.Dend$coefficients[,1], treat.Dend$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "Dendritic Cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.Dend[,1] !=0 | dt.Dend[,2] != 0 -->
<!-- text(treat.Dend$coefficients[sig.genes,1], -->
<!--      treat.Dend$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.Dend)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.Dend$coefficients[,3], treat.Dend$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="Dendritic Cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.Dend[,1] !=0 | dt.Dend[,2] != 0 -->
<!-- text(treat.Dend$coefficients[sig.genes,1], -->
<!--      treat.Dend$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.Dend)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- ## CD8 T-cells  -->
<!-- ### Overall DEG summary -->
<!-- ```{r} -->
<!-- cont.CD8 <- makeContrasts(MUvMT = CD8.MU-CD8.MT, -->
<!--                            SUvST = CD8.SU-CD8.ST, -->
<!--                            All.MvS = 0.5*(CD8.MU + CD8.MT) - 0.5*(CD8.SU + CD8.ST), -->
<!--                            UT.MvS = CD8.MU-CD8.SU, -->
<!--                            All.UvT = 0.5*(CD8.MU + CD8.SU) - 0.5*(CD8.MT + CD8.ST), -->
<!--                            levels = design) -->
<!-- fit.CD8 <- contrasts.fit(fit, contrasts = cont.CD8) -->
<!-- fit.CD8 <- eBayes(fit.CD8, robust = TRUE) -->
<!-- summary(decideTests(fit.CD8)) -->
<!-- ``` -->
<!-- ### TREAT DEG summary -->
<!-- This tests for DEGs relative to a logFC threshold of 0.5 -->
<!-- ```{r} -->
<!-- treat.CD8 <- treat(fit.CD8, lfc = 0.5) -->
<!-- dt.CD8 <- decideTests(treat.CD8) -->
<!-- summary(dt.CD8) -->
<!-- ``` -->
<!-- ### Mean-Difference plot of DEGs -->
<!-- ```{r, fig.asp=0.7} -->
<!-- par(mfrow=c(2,3)) -->
<!-- for(i in 1:ncol(cont.CD8)){ -->
<!--   plotMD(treat.CD8, coef = i, status = dt.CD8[,i], hl.col = c("blue","red")) -->
<!--   abline(h = 0, col = "grey") -->
<!-- } -->
<!-- ``` -->
<!-- ### Barplot of DEGs -->
<!-- ```{r,fig.asp=1} -->
<!-- par(mfrow=c(1,1)) -->
<!-- par(mar=c(7,4,2,2)) -->
<!-- barplot(summary(dt.CD8)[-2,], -->
<!--         beside = TRUE, -->
<!--         legend = TRUE, -->
<!--         col = c("blue","red"), -->
<!--         ylab = "Number of significant genes", -->
<!--         las = 2, -->
<!--         xlab = "") -->
<!-- title("CD8 T-cells") -->
<!-- ``` -->
<!-- ### Top DEGs -->
<!-- ```{r, results='asis', fig.asp = 1} -->
<!-- t <- vector("list", ncol(cont.CD8)) -->
<!-- names(t) <- colnames(cont.CD8) -->
<!-- for(i in 1:ncol(cont.CD8)){ -->
<!--   cat("\n", colnames(cont.CD8)[i],"\n") -->
<!--   top <- topTreat(treat.CD8, coef = i, n = 20, p.value = 0.05) -->
<!--   topAll <- topTreat(treat.CD8, coef = i, n = Inf, p.value = 0.05) -->
<!--   write.csv(topAll, file = glue("{dirName}/CD8-DEGs-{colnames(cont.CD8)[i]}.csv")) -->
<!--   if(nrow(topAll) > 0) { -->
<!--     write.csv(topGO(goana(de = topAll$ENTREZID,  -->
<!--                           universe = treat.CD8$genes$ENTREZID,  -->
<!--                           species = "Hs"), number = 50), -->
<!--               file = here(glue("{dirName}/CD8-GO-{colnames(cont.CD8)[i]}.csv"))) -->
<!--   } -->
<!--   print(knitr::kable(top[,-c(1:3)], digits = 3)) -->
<!--   cat("\n") -->
<!--   if(nrow(top) > 1){ -->
<!--     exp <- normCounts(y.pb, log = TRUE) -->
<!--     keepCells <- y.pb$samples$cell_label == "CD8" -->
<!--     exp <- exp[rownames(exp) %in% top$SYMBOL, keepCells] -->
<!--     exp <- exp[, order(y.pb$samples$Group[keepCells])] -->
<!--     pheatmap::pheatmap(exp - rowMeans(exp), -->
<!--                        color = colorRampPalette(viridis_pal(option = "magma")(10))(100), -->
<!--                        annotation_col = data.frame(Group = y.pb$samples$Group[keepCells], -->
<!--                                                    Severity = toupper(substring(y.pb$samples$Severity[keepCells],1,1)), -->
<!--                                                    Treatment = toupper(substring(y.pb$samples$Treatment[keepCells],1,1)), -->
<!--                                                    row.names = rownames(y.pb$samples)[keepCells]), -->
<!--                        labels_col = y.pb$samples$donor[keepCells], -->
<!--                        scale = "none", -->
<!--                        cluster_cols = FALSE, -->
<!--                        cellwidth = 20, -->
<!--                        cellheight = 20, -->
<!--                        breaks = seq(-3, 3, length.out = 101)) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.CD8$coefficients[,1], treat.CD8$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "CD8 T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.CD8[,1] !=0 | dt.CD8[,2] != 0 -->
<!-- text(treat.CD8$coefficients[sig.genes,1], -->
<!--      treat.CD8$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.CD8)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.CD8$coefficients[,3], treat.CD8$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="CD8 T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.CD8[,1] !=0 | dt.CD8[,2] != 0 -->
<!-- text(treat.CD8$coefficients[sig.genes,1], -->
<!--      treat.CD8$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.CD8)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- ## CD4 T-cells -->
<!-- ### Overall DEG summary -->
<!-- ```{r} -->
<!-- cont.CD4 <- makeContrasts(MUvMT = CD4.MU-CD4.MT, -->
<!--                            SUvST = CD4.SU-CD4.ST, -->
<!--                            All.MvS = 0.5*(CD4.MU + CD4.MT) - 0.5*(CD4.SU + CD4.ST), -->
<!--                            UT.MvS = CD4.MU-CD4.SU, -->
<!--                            All.UvT = 0.5*(CD4.MU + CD4.SU) - 0.5*(CD4.MT + CD4.ST), -->
<!--                            levels = design) -->
<!-- fit.CD4 <- contrasts.fit(fit, contrasts = cont.CD4) -->
<!-- fit.CD4 <- eBayes(fit.CD4, robust = TRUE) -->
<!-- summary(decideTests(fit.CD4)) -->
<!-- ``` -->
<!-- ### TREAT DEG summary -->
<!-- This tests for DEGs relative to a logFC threshold of 0.5 -->
<!-- ```{r} -->
<!-- treat.CD4 <- treat(fit.CD4, lfc = 0.5) -->
<!-- dt.CD4 <- decideTests(treat.CD4) -->
<!-- summary(dt.CD4) -->
<!-- ``` -->
<!-- ### Mean-Difference plot of DEGs -->
<!-- ```{r, fig.asp=0.7} -->
<!-- par(mfrow=c(2,3)) -->
<!-- for(i in 1:ncol(cont.CD4)){ -->
<!--   plotMD(treat.CD4, coef = i, status = dt.CD4[,i], hl.col = c("blue","red")) -->
<!--   abline(h = 0, col = "grey") -->
<!-- } -->
<!-- ``` -->
<!-- ### Barplot of DEGs -->
<!-- ```{r,fig.asp=1} -->
<!-- par(mfrow=c(1,1)) -->
<!-- par(mar=c(7,4,2,2)) -->
<!-- barplot(summary(dt.CD4)[-2,], -->
<!--         beside = TRUE, -->
<!--         legend = TRUE, -->
<!--         col = c("blue","red"), -->
<!--         ylab = "Number of significant genes", -->
<!--         las = 2, -->
<!--         xlab = "") -->
<!-- title("CD4 T-cells") -->
<!-- ``` -->
<!-- ### Top DEGs -->
<!-- ```{r, results='asis', fig.asp = 1} -->
<!-- t <- vector("list", ncol(cont.CD4)) -->
<!-- names(t) <- colnames(cont.CD4) -->
<!-- for(i in 1:ncol(cont.CD4)){ -->
<!--   cat("\n", colnames(cont.CD4)[i],"\n") -->
<!--   top <- topTreat(treat.CD4, coef = i, n = 20, p.value = 0.05) -->
<!--   topAll <- topTreat(treat.CD4, coef = i, n = Inf, p.value = 0.05) -->
<!--   write.csv(topAll, file = glue("{dirName}/CD4-DEGs-{colnames(cont.CD4)[i]}.csv")) -->
<!--   if(nrow(topAll) > 0) { -->
<!--     write.csv(topGO(goana(de = topAll$ENTREZID,  -->
<!--                           universe = treat.CD4$genes$ENTREZID,  -->
<!--                           species = "Hs"), number = 50), -->
<!--               file = here(glue("{dirName}/CD4-GO-{colnames(cont.CD4)[i]}.csv"))) -->
<!--   } -->
<!--   print(knitr::kable(top[,-c(1:3)], digits = 3)) -->
<!--   cat("\n") -->
<!--   if(nrow(top) > 1){ -->
<!--     exp <- normCounts(y.pb, log = TRUE) -->
<!--     keepCells <- y.pb$samples$cell_label == "CD4" -->
<!--     exp <- exp[rownames(exp) %in% top$SYMBOL, keepCells] -->
<!--     exp <- exp[, order(y.pb$samples$Group[keepCells])] -->
<!--     pheatmap::pheatmap(exp - rowMeans(exp), -->
<!--                        color = colorRampPalette(viridis_pal(option = "magma")(10))(100), -->
<!--                        annotation_col = data.frame(Group = y.pb$samples$Group[keepCells], -->
<!--                                                    Severity = toupper(substring(y.pb$samples$Severity[keepCells],1,1)), -->
<!--                                                    Treatment = toupper(substring(y.pb$samples$Treatment[keepCells],1,1)), -->
<!--                                                    row.names = rownames(y.pb$samples)[keepCells]), -->
<!--                        labels_col = y.pb$samples$donor[keepCells], -->
<!--                        scale = "none", -->
<!--                        cluster_cols = FALSE, -->
<!--                        cellwidth = 20, -->
<!--                        cellheight = 20, -->
<!--                        breaks = seq(-3, 3, length.out = 101)) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.CD4$coefficients[,1], treat.CD4$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "CD4 T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.CD4[,1] !=0 | dt.CD4[,2] != 0 -->
<!-- text(treat.CD4$coefficients[sig.genes,1], -->
<!--      treat.CD4$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.CD4)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.CD4$coefficients[,3], treat.CD4$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="CD4 T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.CD4[,1] !=0 | dt.CD4[,2] != 0 -->
<!-- text(treat.CD4$coefficients[sig.genes,1], -->
<!--      treat.CD4$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.CD4)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- ## B-cells -->
<!-- ### Overall DEG summary -->
<!-- ```{r} -->
<!-- cont.B <- makeContrasts(MUvMT = B.MU-B.MT, -->
<!--                            SUvST = B.SU-B.ST, -->
<!--                            All.MvS = 0.5*(B.MU + B.MT) - 0.5*(B.SU + B.ST), -->
<!--                            UT.MvS = B.MU-B.SU, -->
<!--                            All.UvT = 0.5*(B.MU + B.SU) - 0.5*(B.MT + B.ST), -->
<!--                            levels = design) -->
<!-- fit.B <- contrasts.fit(fit, contrasts = cont.B) -->
<!-- fit.B <- eBayes(fit.B, robust = TRUE) -->
<!-- summary(decideTests(fit.B)) -->
<!-- ``` -->
<!-- ### TREAT DEG summary -->
<!-- This tests for DEGs relative to a logFC threshold of 0.5 -->
<!-- ```{r} -->
<!-- treat.B <- treat(fit.B, lfc = 0.5) -->
<!-- dt.B <- decideTests(treat.B) -->
<!-- summary(dt.B) -->
<!-- ``` -->
<!-- ### Mean-Difference plot of DEGs -->
<!-- ```{r, fig.asp=0.7} -->
<!-- par(mfrow=c(2,3)) -->
<!-- for(i in 1:ncol(cont.B)){ -->
<!--   plotMD(treat.B, coef = i, status = dt.B[,i], hl.col = c("blue","red")) -->
<!--   abline(h = 0, col = "grey") -->
<!-- } -->
<!-- ``` -->
<!-- ### Barplot of DEGs -->
<!-- ```{r,fig.asp=1} -->
<!-- par(mfrow=c(1,1)) -->
<!-- par(mar=c(7,4,2,2)) -->
<!-- barplot(summary(dt.B)[-2,], -->
<!--         beside = TRUE, -->
<!--         legend = TRUE, -->
<!--         col = c("blue","red"), -->
<!--         ylab = "Number of significant genes", -->
<!--         las = 2, -->
<!--         xlab = "") -->
<!-- title("B-cells") -->
<!-- ``` -->
<!-- ### Top DEGs -->
<!-- ```{r, results='asis', fig.asp = 1} -->
<!-- t <- vector("list", ncol(cont.B)) -->
<!-- names(t) <- colnames(cont.B) -->
<!-- for(i in 1:ncol(cont.B)){ -->
<!--   cat("\n", colnames(cont.B)[i],"\n") -->
<!--   top <- topTreat(treat.B, coef = i, n = 20, p.value = 0.05) -->
<!--   topAll <- topTreat(treat.B, coef = i, n = Inf, p.value = 0.05) -->
<!--   write.csv(topAll, file = glue("{dirName}/B-DEGs-{colnames(cont.B)[i]}.csv")) -->
<!--   if(nrow(topAll) > 0) { -->
<!--     write.csv(topGO(goana(de = topAll$ENTREZID,  -->
<!--                           universe = treat.B$genes$ENTREZID,  -->
<!--                           species = "Hs"), number = 50), -->
<!--               file = here(glue("{dirName}/B-GO-{colnames(cont.B)[i]}.csv"))) -->
<!--   } -->
<!--   print(knitr::kable(top[,-c(1:3)], digits = 3)) -->
<!--   cat("\n") -->
<!--   if(nrow(top) > 1){ -->
<!--     exp <- normCounts(y.pb, log = TRUE) -->
<!--     keepCells <- y.pb$samples$cell_label == "B" -->
<!--     exp <- exp[rownames(exp) %in% top$SYMBOL, keepCells] -->
<!--     exp <- exp[, order(y.pb$samples$Group[keepCells])] -->
<!--     pheatmap::pheatmap(exp - rowMeans(exp), -->
<!--                        color = colorRampPalette(viridis_pal(option = "magma")(10))(100), -->
<!--                        annotation_col = data.frame(Group = y.pb$samples$Group[keepCells], -->
<!--                                                    Severity = toupper(substring(y.pb$samples$Severity[keepCells],1,1)), -->
<!--                                                    Treatment = toupper(substring(y.pb$samples$Treatment[keepCells],1,1)), -->
<!--                                                    row.names = rownames(y.pb$samples)[keepCells]), -->
<!--                        labels_col = y.pb$samples$donor[keepCells], -->
<!--                        scale = "none", -->
<!--                        cluster_cols = FALSE, -->
<!--                        cellwidth = 20, -->
<!--                        cellheight = 20, -->
<!--                        breaks = seq(-3, 3, length.out = 101)) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.B$coefficients[,1], treat.B$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "B-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.B[,1] !=0 | dt.B[,2] != 0 -->
<!-- text(treat.B$coefficients[sig.genes,1], -->
<!--      treat.B$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.B)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.B$coefficients[,3], treat.B$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="B-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.B[,1] !=0 | dt.B[,2] != 0 -->
<!-- text(treat.B$coefficients[sig.genes,1], -->
<!--      treat.B$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.B)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- ## NK Cells -->
<!-- ### Overall DEG summary -->
<!-- ```{r} -->
<!-- cont.NKT <- makeContrasts(MUvMT = NKT.MU-NKT.MT, -->
<!--                            SUvST = NKT.SU-NKT.ST, -->
<!--                            All.MvS = 0.5*(NKT.MU + NKT.MT) - 0.5*(NKT.SU + NKT.ST), -->
<!--                            UT.MvS = NKT.MU-NKT.SU, -->
<!--                            All.UvT = 0.5*(NKT.MU + NKT.SU) - 0.5*(NKT.MT + NKT.ST), -->
<!--                            levels = design) -->
<!-- fit.NKT <- contrasts.fit(fit, contrasts = cont.NKT) -->
<!-- fit.NKT <- eBayes(fit.NKT, robust = TRUE) -->
<!-- summary(decideTests(fit.NKT)) -->
<!-- ``` -->
<!-- ### TREAT DEG summary -->
<!-- This tests for DEGs relative to a logFC threshold of 0.5 -->
<!-- ```{r} -->
<!-- treat.NKT <- treat(fit.NKT, lfc = 0.5) -->
<!-- dt.NKT <- decideTests(treat.NKT) -->
<!-- summary(dt.NKT) -->
<!-- ``` -->
<!-- ### Mean-Difference plot of DEGs -->
<!-- ```{r, fig.asp=0.7} -->
<!-- par(mfrow=c(2,3)) -->
<!-- for(i in 1:ncol(cont.NKT)){ -->
<!--   plotMD(treat.NKT, coef = i, status = dt.NKT[,i], hl.col = c("blue","red")) -->
<!--   abline(h = 0, col = "grey") -->
<!-- } -->
<!-- ``` -->
<!-- ### Barplot of DEGs -->
<!-- ```{r,fig.asp=1} -->
<!-- par(mfrow=c(1,1)) -->
<!-- par(mar=c(7,4,2,2)) -->
<!-- barplot(summary(dt.NKT)[-2,], -->
<!--         beside = TRUE, -->
<!--         legend = TRUE, -->
<!--         col = c("blue","red"), -->
<!--         ylab = "Number of significant genes", -->
<!--         las = 2, -->
<!--         xlab = "") -->
<!-- title("NK T-cells") -->
<!-- ``` -->
<!-- ### Top DEGs -->
<!-- ```{r, results='asis', fig.asp = 1} -->
<!-- t <- vector("list", ncol(cont.NKT)) -->
<!-- names(t) <- colnames(cont.NKT) -->
<!-- for(i in 1:ncol(cont.NKT)){ -->
<!--   cat("\n", colnames(cont.NKT)[i],"\n") -->
<!--   top <- topTreat(treat.NKT, coef = i, n = 20, p.value = 0.05) -->
<!--   topAll <- topTreat(treat.NKT, coef = i, n = Inf, p.value = 0.05) -->
<!--   write.csv(topAll, file = glue("{dirName}/NKT-DEGs-{colnames(cont.NKT)[i]}.csv")) -->
<!--   if(nrow(topAll) > 0) { -->
<!--     write.csv(topGO(goana(de = topAll$ENTREZID,  -->
<!--                           universe = treat.NKT$genes$ENTREZID,  -->
<!--                           species = "Hs"), number = 50), -->
<!--               file = here(glue("{dirName}/NKT-GO-{colnames(cont.NKT)[i]}.csv"))) -->
<!--   } -->
<!--   print(knitr::kable(top[,-c(1:3)], digits = 3)) -->
<!--   cat("\n") -->
<!--   if(nrow(top) > 1){ -->
<!--     exp <- normCounts(y.pb, log = TRUE) -->
<!--     keepCells <- y.pb$samples$cell_label == "NKT" -->
<!--     exp <- exp[rownames(exp) %in% top$SYMBOL, keepCells] -->
<!--     exp <- exp[, order(y.pb$samples$Group[keepCells])] -->
<!--     pheatmap::pheatmap(exp - rowMeans(exp), -->
<!--                        color = colorRampPalette(viridis_pal(option = "magma")(10))(100), -->
<!--                        annotation_col = data.frame(Group = y.pb$samples$Group[keepCells], -->
<!--                                                    Severity = toupper(substring(y.pb$samples$Severity[keepCells],1,1)), -->
<!--                                                    Treatment = toupper(substring(y.pb$samples$Treatment[keepCells],1,1)), -->
<!--                                                    row.names = rownames(y.pb$samples)[keepCells]), -->
<!--                        labels_col = y.pb$samples$donor[keepCells], -->
<!--                        scale = "none", -->
<!--                        cluster_cols = FALSE, -->
<!--                        cellwidth = 20, -->
<!--                        cellheight = 20, -->
<!--                        breaks = seq(-3, 3, length.out = 101)) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ### Comparison of DEGs between contrasts -->
<!-- #### Treatment -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.NKT$coefficients[,1], treat.NKT$coefficients[,2], -->
<!--      xlab = "logFC: Mild Untreated vs Mild Treated",  -->
<!--      ylab = "logFC: Severe Untreated vs Severe Treated", -->
<!--      main = "NK T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.NKT[,1] !=0 | dt.NKT[,2] != 0 -->
<!-- text(treat.NKT$coefficients[sig.genes,1], -->
<!--      treat.NKT$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.NKT)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- #### Severity  -->
<!-- ```{r, fig.asp = 1} -->
<!-- par(mar=c(5,5,3,2)) -->
<!-- plot(treat.NKT$coefficients[,3], treat.NKT$coefficients[,4], -->
<!--      xlab="logFC: All Mild versus Severe",  -->
<!--      ylab="logFC: Untreated Mild versus Severe", -->
<!--      main="NK T-cells in Cystic Fibrosis", -->
<!--      col=rgb(0,0,1,alpha=0.25),pch=16,cex=0.8) -->
<!-- abline(h=0,v=0,col=colours()[c(175)]) -->
<!-- sig.genes <-  dt.NKT[,1] !=0 | dt.NKT[,2] != 0 -->
<!-- text(treat.NKT$coefficients[sig.genes,1], -->
<!--      treat.NKT$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.NKT)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<!-- =0 | dt.PrNKT[,2] != 0 -->
<!-- text(treat.PrNKT$coefficients[sig.genes,1], -->
<!--      treat.PrNKT$coefficients[sig.genes,2], -->
<!--      labels = rownames(dt.NKT)[sig.genes], col = 2, cex = 0.6) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.1.0/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.1.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices datasets  utils    
[8] methods   base     

other attached packages:
 [1] tidyHeatmap_1.7.0                                  
 [2] missMethyl_1.28.0                                  
 [3] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [4] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
 [5] minfi_1.40.0                                       
 [6] bumphunter_1.36.0                                  
 [7] locfit_1.5-9.4                                     
 [8] iterators_1.0.13                                   
 [9] foreach_1.5.1                                      
[10] Biostrings_2.62.0                                  
[11] XVector_0.34.0                                     
[12] TxDb.Hsapiens.UCSC.hg38.knownGene_3.14.0           
[13] speckle_0.0.3                                      
[14] scales_1.1.1                                       
[15] janitor_2.1.0                                      
[16] BiocParallel_1.28.3                                
[17] glmGamPoi_1.6.0                                    
[18] org.Hs.eg.db_3.14.0                                
[19] EGSEA_1.22.0                                       
[20] pathview_1.34.0                                    
[21] topGO_2.46.0                                       
[22] SparseM_1.81                                       
[23] GO.db_3.14.0                                       
[24] graph_1.72.0                                       
[25] gage_2.44.0                                        
[26] ensembldb_2.18.2                                   
[27] AnnotationFilter_1.18.0                            
[28] GenomicFeatures_1.46.3                             
[29] AnnotationDbi_1.56.2                               
[30] AnnotationHub_3.2.0                                
[31] BiocFileCache_2.2.0                                
[32] dbplyr_2.1.1                                       
[33] patchwork_1.1.1                                    
[34] edgeR_3.36.0                                       
[35] limma_3.50.0                                       
[36] Matrix_1.4-0                                       
[37] scater_1.22.0                                      
[38] scran_1.22.1                                       
[39] scuttle_1.4.0                                      
[40] DropletUtils_1.14.1                                
[41] SingleCellExperiment_1.16.0                        
[42] SummarizedExperiment_1.24.0                        
[43] Biobase_2.54.0                                     
[44] GenomicRanges_1.46.1                               
[45] GenomeInfoDb_1.30.1                                
[46] IRanges_2.28.0                                     
[47] S4Vectors_0.32.3                                   
[48] BiocGenerics_0.40.0                                
[49] MatrixGenerics_1.6.0                               
[50] matrixStats_0.61.0                                 
[51] glue_1.6.0                                         
[52] here_1.0.1                                         
[53] forcats_0.5.1                                      
[54] stringr_1.4.0                                      
[55] dplyr_1.0.7                                        
[56] purrr_0.3.4                                        
[57] readr_2.1.1                                        
[58] tidyr_1.1.4                                        
[59] tibble_3.1.6                                       
[60] ggplot2_3.3.5                                      
[61] tidyverse_1.3.1                                    
[62] SeuratObject_4.0.4                                 
[63] Seurat_4.0.6                                       
[64] workflowr_1.7.0                                    

loaded via a namespace (and not attached):
  [1] EGSEAdata_1.22.0              PADOG_1.36.0                 
  [3] pbapply_1.5-0                 lattice_0.20-45              
  [5] GSVA_1.42.0                   haven_2.4.3                  
  [7] vctrs_0.3.8                   mgcv_1.8-38                  
  [9] beanplot_1.3.1                blob_1.2.2                   
 [11] survival_3.2-13               spatstat.data_2.1-2          
 [13] later_1.3.0                   DBI_1.1.2                    
 [15] R.utils_2.11.0                rappdirs_0.3.3               
 [17] uwot_0.1.11                   dqrng_0.3.0                  
 [19] zlibbioc_1.40.0               GlobalOptions_0.1.2          
 [21] htmlwidgets_1.5.4             mvtnorm_1.1-3                
 [23] future_1.23.0                 leiden_0.3.9                 
 [25] irlba_2.3.5                   illuminaio_0.36.0            
 [27] Rcpp_1.0.7                    KernSmooth_2.23-20           
 [29] DT_0.20                       promises_1.2.0.1             
 [31] DelayedArray_0.20.0           fs_1.5.2                     
 [33] mnormt_2.0.2                  digest_0.6.29                
 [35] png_0.1-7                     bluster_1.4.0                
 [37] nor1mix_1.3-0                 sctransform_0.3.3            
 [39] cowplot_1.1.1                 pkgconfig_2.0.3              
 [41] DelayedMatrixStats_1.16.0     ggbeeswarm_0.6.0             
 [43] reticulate_1.22               circlize_0.4.13              
 [45] beeswarm_0.4.0                GetoptLong_1.0.5             
 [47] xfun_0.29                     bslib_0.3.1                  
 [49] zoo_1.8-9                     tidyselect_1.1.1             
 [51] reshape2_1.4.4                ica_1.0-2                    
 [53] viridisLite_0.4.0             rtracklayer_1.54.0           
 [55] rlang_0.4.12                  jquerylib_0.1.4              
 [57] metap_1.7                     RColorBrewer_1.1-2           
 [59] modelr_0.1.8                  labeling_0.4.2               
 [61] mutoss_0.1-12                 httpuv_1.6.5                 
 [63] preprocessCore_1.56.0         BiocNeighbors_1.12.0         
 [65] TH.data_1.1-0                 annotate_1.72.0              
 [67] jsonlite_1.7.2                tmvnsim_1.0-2                
 [69] bit_4.0.4                     mime_0.12                    
 [71] gridExtra_2.3                 gplots_3.1.1                 
 [73] Rsamtools_2.10.0              stringi_1.7.6                
 [75] processx_3.5.2                spatstat.sparse_2.1-0        
 [77] scattermore_0.7               rbibutils_2.2.7              
 [79] quadprog_1.5-8                bitops_1.0-7                 
 [81] cli_3.1.0                     Rdpack_2.1.3                 
 [83] rhdf5filters_1.6.0            RSQLite_2.2.9                
 [85] data.table_1.14.2             KEGGgraph_1.54.0             
 [87] rstudioapi_0.13               org.Mm.eg.db_3.14.0          
 [89] git2r_0.29.0                  GenomicAlignments_1.30.0     
 [91] nlme_3.1-153                  listenv_0.8.0                
 [93] miniUI_0.1.1.1                R.oo_1.24.0                  
 [95] sessioninfo_1.2.2             readxl_1.3.1                 
 [97] lifecycle_1.0.1               munsell_0.5.0                
 [99] cellranger_1.1.0              R.methodsS3_1.8.1            
[101] hwriter_1.3.2                 caTools_1.18.2               
[103] codetools_0.2-18              vipor_0.4.5                  
[105] lmtest_0.9-39                 xtable_1.8-4                 
[107] ROCR_1.0-11                   BiocManager_1.30.16          
[109] abind_1.4-5                   farver_2.1.0                 
[111] parallelly_1.30.0             RANN_2.6.1                   
[113] askpass_1.1                   hgu133a.db_3.13.0            
[115] BiocIO_1.4.0                  GEOquery_2.62.2              
[117] RcppAnnoy_0.0.19              goftest_1.2-3                
[119] cluster_2.1.2                 dendextend_1.15.2            
[121] future.apply_1.8.1            ellipsis_0.3.2               
[123] prettyunits_1.1.1             lubridate_1.8.0              
[125] ggridges_0.5.3                reprex_2.0.1                 
[127] mclust_5.4.9                  prismatic_1.1.0              
[129] igraph_1.2.11                 multtest_2.50.0              
[131] TFisher_0.2.0                 paletteer_1.4.0              
[133] spatstat.utils_2.3-0          htmltools_0.5.2              
[135] org.Rn.eg.db_3.14.0           yaml_2.2.1                   
[137] utf8_1.2.2                    plotly_4.10.0                
[139] interactiveDisplayBase_1.32.0 XML_3.99-0.8                 
[141] withr_2.4.3                   fitdistrplus_1.1-6           
[143] bit64_4.0.5                   BiasedUrn_1.07               
[145] rngtools_1.5.2                doRNG_1.8.2                  
[147] multcomp_1.4-18               ProtGenerics_1.26.0          
[149] GSA_1.03.1                    spatstat.core_2.3-2          
[151] rsvd_1.0.5                    ScaledMatrix_1.2.0           
[153] memoise_2.0.1                 evaluate_0.14                
[155] globaltest_5.48.0             tzdb_0.2.0                   
[157] callr_3.7.0                   ps_1.6.0                     
[159] curl_4.3.2                    fansi_1.0.0                  
[161] highr_0.9                     GSEABase_1.56.0              
[163] tensor_1.5                    renv_0.15.0-14               
[165] cachem_1.0.6                  deldir_1.0-6                 
[167] metapod_1.2.0                 rjson_0.2.21                 
[169] getPass_0.2-2                 ggrepel_0.9.1                
[171] clue_0.3-60                   hgu133plus2.db_3.13.0        
[173] rprojroot_2.0.2               tools_4.1.0                  
[175] sass_0.4.0                    sandwich_3.0-1               
[177] magrittr_2.0.1                RCurl_1.98-1.6               
[179] xml2_1.3.3                    httr_1.4.2                   
[181] assertthat_0.2.1              rmarkdown_2.11               
[183] globals_0.14.0                R6_2.5.1                     
[185] Rhdf5lib_1.16.0               progress_1.2.2               
[187] genefilter_1.76.0             KEGGREST_1.34.0              
[189] shape_1.4.6                   gtools_3.9.2                 
[191] statmod_1.4.36                beachmat_2.10.0              
[193] rematch2_2.1.2                BiocVersion_3.14.0           
[195] HDF5Array_1.22.1              BiocSingular_1.10.0          
[197] rhdf5_2.38.0                  splines_4.1.0                
[199] snakecase_0.11.0              colorspace_2.0-2             
[201] generics_0.1.1                pillar_1.6.4                 
[203] sn_2.0.1                      Rgraphviz_2.38.0             
[205] KEGGdzPathwaysGEO_1.32.0      GenomeInfoDbData_1.2.7       
[207] plyr_1.8.6                    gtable_0.3.0                 
[209] rvest_1.0.2                   restfulr_0.0.13              
[211] ComplexHeatmap_2.10.0         knitr_1.37                   
[213] biomaRt_2.50.1                fastmap_1.1.0                
[215] doParallel_1.0.16             broom_0.7.11                 
[217] openssl_1.4.6                 filelock_1.0.2               
[219] backports_1.4.1               plotrix_3.8-2                
[221] R2HTML_2.3.2                  base64_2.0                   
[223] vroom_1.5.7                   hms_1.1.1                    
[225] scrime_1.3.5                  Rtsne_0.15                   
[227] HTMLUtils_0.1.7               shiny_1.7.1                  
[229] polyclip_1.10-0               grid_4.1.0                   
[231] numDeriv_2016.8-1.1           mathjaxr_1.4-0               
[233] siggenes_1.68.0               lazyeval_0.2.2               
[235] whisker_0.4                   crayon_1.4.2                 
[237] MASS_7.3-53.1                 safe_3.34.0                  
[239] sparseMatrixStats_1.6.0       viridis_0.6.2                
[241] reshape_0.8.8                 rpart_4.1-15                 
[243] compiler_4.1.0                spatstat.geom_2.3-1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
