<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2022-06-15" />

<title>Analysis of scRNA-seq &amp; CITE-seq Data Combined</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-cf-cite-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of scRNA-seq &amp; CITE-seq Data Combined</h1>
<h3 class="subtitle">Differential Gene Expression Analysis</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">June 15, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks">
Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-15
</p>
<p>
<strong>Checks:</strong>
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
7
<span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
0
</p>
<p>
<strong>Knit directory:</strong>
<code>paed-cf-cite-seq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the
reproducibility checks that were applied when the results were created.
The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you
know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Environment:</strong> empty
</a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global
environment can affect the analysis in your R Markdown file in unknown ways.
For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210524code">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Seed:</strong> <code>set.seed(20210524)</code>
</a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210524)</code> was run prior to running the code in the R Markdown file.
Setting a seed ensures that any results that rely on randomness, e.g.
subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Session information:</strong> recorded
</a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is
critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Cache:</strong> none
</a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident
that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>File paths:</strong> relative
</a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project
makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree07fd1a8c4d8d7e4a4e9aea61492b3101632675ebtargetblank07fd1a8a">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Repository version:</strong> <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/07fd1a8c4d8d7e4a4e9aea61492b3101632675eb" target="_blank">07fd1a8</a>
</a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree07fd1a8c4d8d7e4a4e9aea61492b3101632675ebtargetblank07fd1a8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and
connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/07fd1a8c4d8d7e4a4e9aea61492b3101632675eb" target="_blank">07fd1a8</a>.
See the <em>Past versions</em> tab to see a history of the changes made to the
R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the
analysis have been committed to Git prior to generating the results (you can
use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only
checks the R Markdown file, but you know if there are other scripts or data
files that it depends on. Below is the status of the Git repository when the
results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/obsolete/
    Ignored:    code/obsolete/
    Ignored:    data/190930_A00152_0150_BHTYCMDSXX/
    Ignored:    data/CellRanger/
    Ignored:    data/GSE127465_RAW/
    Ignored:    data/SCEs/02_ZILIONIS.sct_normalised.SEU.rds
    Ignored:    data/SCEs/03_C133_Neeland.demultiplexed.SCE.rds
    Ignored:    data/SCEs/03_C133_Neeland.emptyDrops.SCE.rds
    Ignored:    data/SCEs/03_C133_Neeland.nuclear_fraction_calls.rds
    Ignored:    data/SCEs/03_C133_Neeland.preprocessed.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.CellRanger_v6.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.emptyDrops.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.nuclear_fraction_calls.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.preprocessed.SCE.rds
    Ignored:    data/SCEs/03_COMBO.clustered.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_azimuth.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_azimuth_v2.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.integrated.SEU.rds
    Ignored:    data/SCEs/03_COMBO.zilionis_mapped.SEU.rds
    Ignored:    data/SCEs/04_C133_Neeland.adt_dsb_normalised.rds
    Ignored:    data/SCEs/04_C133_Neeland.adt_integrated.rds
    Ignored:    data/SCEs/04_C133_Neeland.all_integrated.SEU.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.CellRanger_v6.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.emptyDrops.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.preprocessed.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.transfer_adt.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_clustered.SEU_bk.rds
    Ignored:    data/SCEs/04_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_integrated.SEU_bk.rds
    Ignored:    data/SCEs/04_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_adt_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_lung_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.zilionis_mapped.SEU.rds
    Ignored:    data/SCEs/05_CF_BAL_Pilot.transfer_adt.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_adt_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_lung_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/C133_Neeland.CellRanger.SCE.rds
    Ignored:    data/SCEs/obsolete/
    Ignored:    data/emptyDrops/
    Ignored:    data/obsolete/
    Ignored:    data/sample_sheets/obsolete/
    Ignored:    output/marker-analysis/obsolete/
    Ignored:    output/obsolete/
    Ignored:    rename_captures.R
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    wflow_background.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   renv.lock
    Modified:   renv/.gitignore
    Modified:   renv/activate.R
    Modified:   renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in
this status report because it is ok for generated content to have uncommitted
changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made
to the R Markdown (<code>analysis/05_COMBO.expression_analysis.Rmd</code>) and HTML (<code>docs/05_COMBO.expression_analysis.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table below to
view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/07fd1a8c4d8d7e4a4e9aea61492b3101632675eb/analysis/05_COMBO.expression_analysis.Rmd" target="_blank">07fd1a8</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-15
</td>
<td>
wflow_publish(c("analysis/index.Rmd", "analysis/03_C133_Neeland.preprocess.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/7b44e3893f55d9e2efa666c6c062135536c5bb79/docs/05_COMBO.expression_analysis.html" target="_blank">7b44e38</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/05_COMBO.expression_analysis.html" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/13b35cfc025c7b71a8d61a0d54ef3899cbd6b3f0/analysis/05_COMBO.expression_analysis.Rmd" target="_blank">13b35cf</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
<td>
wflow_publish("analysis/05_COMBO.expression_analysis.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages(library(BiocStyle))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(here))
suppressPackageStartupMessages(library(glue))
suppressPackageStartupMessages(library(DropletUtils))
suppressPackageStartupMessages(library(scran))
suppressPackageStartupMessages(library(scater))
suppressPackageStartupMessages(library(scuttle))
suppressPackageStartupMessages(library(Seurat))
suppressPackageStartupMessages(library(patchwork))
suppressPackageStartupMessages(library(clustree))
suppressPackageStartupMessages(library(glmGamPoi))
suppressPackageStartupMessages(library(BiocParallel))
suppressPackageStartupMessages(library(paletteer))
suppressPackageStartupMessages(library(glmGamPoi))
suppressPackageStartupMessages(library(tidyHeatmap))
suppressPackageStartupMessages(library(edgeR))
suppressPackageStartupMessages(library(limma))
suppressPackageStartupMessages(library(org.Hs.eg.db))
suppressPackageStartupMessages(library(EGSEA))
suppressPackageStartupMessages(library(TxDb.Hsapiens.UCSC.hg38.knownGene))
suppressPackageStartupMessages(library(missMethyl))
source(here(&quot;code/utility.R&quot;))
source(here(&quot;code/helper_functions.R&quot;))

set.seed(42)
options(scipen=0)
options(future.globals.maxSize = 6500 * 1024^2)</code></pre>
<div id="load-data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Load Data</h2>
<pre class="r"><code>seu1 &lt;- readRDS(here(&quot;data/SCEs/06_COMBO.clean_macrophages_diet.SEU.rds&quot;))
seu2 &lt;- readRDS(here(&quot;data/SCEs/06_COMBO.clean_tcells_diet.SEU.rds&quot;))
seu3 &lt;- readRDS(here(&quot;data/SCEs/06_COMBO.clean_others_diet.SEU.rds&quot;))
seu &lt;- merge(seu1, y = c(seu2, seu3))</code></pre>
<pre><code>            used   (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells  21742581 1161.2   39776324  2124.3   21959947  1172.8
Vcells 700440959 5344.0 1662770105 12686.0 1661625613 12677.2</code></pre>
</div>
<div id="load-clinical-information" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Load clinical information</h2>
<p>Import clinical characteristics and patient information and associate with <code>genetic_donor</code> IDs.</p>
<pre class="r"><code>info &lt;- read.csv(file = here(&quot;data/sample_sheets/Sample_information.csv&quot;))
tab &lt;- table(seu$HTO, seu$donor)

apply(tab, 2, function(x){
  names(which(x == max(x)))
}) %&gt;% data.frame %&gt;%
  dplyr::rename(&quot;HTO&quot; = &quot;.&quot;) %&gt;%
  rownames_to_column(var = &quot;donor&quot;) %&gt;%
  inner_join(info, by = c(&quot;HTO&quot; = &quot;Sample&quot;)) %&gt;%
  mutate(Batch = factor(Batch)) -&gt; info

info %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="left">HTO</th>
<th align="left">Participant</th>
<th align="left">Sex</th>
<th align="right">Age</th>
<th align="left">Disease</th>
<th align="left">Batch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="left">B1_CF</td>
<td align="left">M</td>
<td align="right">2.99</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="left">C1_CF</td>
<td align="left">M</td>
<td align="right">2.99</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="left">A1_Ctrl</td>
<td align="left">M</td>
<td align="right">3.00</td>
<td align="left">Ctrl</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="left">D1_CF</td>
<td align="left">M</td>
<td align="right">3.03</td>
<td align="left">CF</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">donor_A</td>
<td align="left">Human_HTO_8</td>
<td align="left">L2_CF</td>
<td align="left">M</td>
<td align="right">5.95</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_B</td>
<td align="left">Human_HTO_1</td>
<td align="left">E2_CF</td>
<td align="left">F</td>
<td align="right">5.99</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_C</td>
<td align="left">Human_HTO_4</td>
<td align="left">H2_CF</td>
<td align="left">F</td>
<td align="right">5.89</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_D</td>
<td align="left">Human_HTO_6</td>
<td align="left">J2_CF</td>
<td align="left">M</td>
<td align="right">5.05</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_E</td>
<td align="left">Human_HTO_3</td>
<td align="left">G2_CF</td>
<td align="left">F</td>
<td align="right">4.91</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_F</td>
<td align="left">Human_HTO_5</td>
<td align="left">I2_CF</td>
<td align="left">F</td>
<td align="right">5.93</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">donor_G</td>
<td align="left">Human_HTO_2</td>
<td align="left">F2_CF</td>
<td align="left">F</td>
<td align="right">6.02</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">donor_H</td>
<td align="left">Human_HTO_7</td>
<td align="left">K2_CF</td>
<td align="left">M</td>
<td align="right">4.92</td>
<td align="left">CF</td>
<td align="left">2</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="differences-in-gene-expression" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Differences in gene expression</h1>
<div id="test-for-dge-within-cell-types" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Test for DGE within cell types</h2>
<pre class="r"><code>seu@meta.data %&gt;%
  left_join(info, by = c(&quot;donor&quot; = &quot;donor&quot;)) -&gt; mdata

# Create single cell experiment object
sce &lt;- SingleCellExperiment(assays = list(counts = seu[[&quot;RNA&quot;]]@counts), 
                            colData = mdata)</code></pre>
<div id="add-gene-annotation" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Add gene annotation</h3>
<pre class="r"><code>columns(org.Hs.eg.db)</code></pre>
<pre><code> [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
 [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;    
[11] &quot;GENETYPE&quot;     &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;         
[16] &quot;OMIM&quot;         &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;        
[21] &quot;PMID&quot;         &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      
[26] &quot;UNIPROT&quot;     </code></pre>
<pre class="r"><code>ann &lt;- AnnotationDbi:::select(org.Hs.eg.db,
                              keys = rownames(sce),
                              columns=c(&quot;SYMBOL&quot;,
                                        &quot;ENTREZID&quot;,
                                        &quot;ENSEMBL&quot;,
                                        &quot;GENENAME&quot;,
                                        &quot;CHR&quot;),
                              keytype = &quot;SYMBOL&quot;)
m &lt;- match(rownames(sce), ann$SYMBOL)
ann &lt;- ann[m,]
all(ann$SYMBOL == rownames(sce))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>rowData(sce) &lt;- DataFrame(ann)</code></pre>
<p>Heatmap showing the abundance of cells with each combination of donor (row)
and cell type label (column). The color scale represents the log2-count for each
combination.</p>
<pre class="r"><code>table(sce$donor, sce$Broad) %&gt;% data.frame %&gt;%
  dplyr::rename(donor = Var1, `Cell Label` = Var2) %&gt;%
  inner_join(info) %&gt;%
  mutate(`Log2 No. Cells` = log2(Freq + 1),
         Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.row = Participant, 
          .column = `Cell Label`,
          .value = `Log2 No. Cells`,
          scale = &quot;none&quot;) %&gt;%
  add_tile(Batch) %&gt;%
  add_tile(Sex) %&gt;%
  add_tile(Age) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="create-pseudobulk-samples" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Create pseudobulk samples</h1>
<p>Using ‘label’ and ‘sample’ as our two factors; each column of the output
corresponds to one unique combination of these two factors.</p>
<pre class="r"><code>summed &lt;- aggregateAcrossCells(sce, 
                               id = colData(sce)[, c(&quot;Broad&quot;, &quot;donor&quot;)])
summed</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 16001 169 
metadata(0):
assays(1): counts
rownames(16001): FAM87B LINC00115 ... C21orf2 FAM207A
rowData names(5): SYMBOL ENTREZID ENSEMBL GENENAME CHR
colnames: NULL
colData names(67): orig.ident nCount_RNA ... donor ncells
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):</code></pre>
<pre class="r"><code>y.pb &lt;- DGEList(counts(summed), 
                samples = colData(summed))
y.pb</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
          Sample1 Sample2 Sample3 Sample4 Sample5 Sample6 Sample7 Sample8
FAM87B          0       0       0       0       0       0       0       0
LINC00115      24       2       5       1       0       0       0       2
FAM41C         36       1      12       2       3       4       0       0
SAMD11          0       1       0       1       0       0       0       0
NOC2L         303      54     130      32      28      48       2       6
          Sample9 Sample10 Sample11 Sample12 Sample13 Sample14 Sample15
FAM87B          0        0        0        0        0        0        0
LINC00115       0        1        0        1       12       10        5
FAM41C          2        0        0        0        9        2        3
SAMD11          0        0        0        0        1        0        0
NOC2L           4        2        2        4      101       63       72
          Sample16 Sample17 Sample18 Sample19 Sample20 Sample21 Sample22
FAM87B           0        0        0        0        0        0        0
LINC00115        1        4        1        1        0        0        0
FAM41C           2        2        1        1        1        1        0
SAMD11           0        0        0        0        0        0        0
NOC2L           59       39        4       18        6        6        9
          Sample23 Sample24 Sample25 Sample26 Sample27 Sample28 Sample29
FAM87B           0        0        0        0        0        0        0
LINC00115        5        1        0        0        6        4        0
FAM41C           3        1        6        1        4        4        0
SAMD11           0        0        0        0        0        0        0
NOC2L           51        4       56        8      105       44       16
          Sample30 Sample31 Sample32 Sample33 Sample34 Sample35 Sample36
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        1        0        0        0        0
FAM41C           0        2        0        0        0        1        0
SAMD11           0        0        1        0        0        0        0
NOC2L            2       11        5        2        3       17        1
          Sample37 Sample38 Sample39 Sample40 Sample41 Sample42 Sample43
FAM87B           1        1        0        0        0        0        0
LINC00115        6       14       35       12        3        0        0
FAM41C           4        3       13        1        0        0        3
SAMD11           0        2        1        0        0        1        0
NOC2L           32       93      187       94       26       11       10
          Sample44 Sample45 Sample46 Sample47 Sample48 Sample49 Sample50
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        2        1        2        0
FAM41C           2        2        0        1        0        0        0
SAMD11           0        0        0        1        0        0        0
NOC2L           10       18        5       18       13        1        2
          Sample51 Sample52 Sample53 Sample54 Sample55 Sample56 Sample57
FAM87B           0        0        0        0        0        0        0
LINC00115        1        3        3        0        1        0        0
FAM41C          21       12       12        0        3        7        0
SAMD11           1        0        1        0        0        0        0
NOC2L           55       45       77        2       20       31        1
          Sample58 Sample59 Sample60 Sample61 Sample62 Sample63 Sample64
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        3        0        0        0
FAM41C           0        1        1        0        1        0        0
SAMD11           0        0        0        1        0        0        0
NOC2L            0        1        3       23       20        2        5
          Sample65 Sample66 Sample67 Sample68 Sample69 Sample70 Sample71
FAM87B           0        0        0        0        0        0        0
LINC00115        0        0        0        0        1        0        1
FAM41C           0        0        0        0        1        0        0
SAMD11           0        0        0        0        0        0        0
NOC2L            3        0        1        0        6        0       18
          Sample72 Sample73 Sample74 Sample75 Sample76 Sample77 Sample78
FAM87B           0        0        0        0        0        0        0
LINC00115        2        0        3        0        0        0        1
FAM41C           0        0        1        0        0        0        0
SAMD11           0        0        0        0        0        0        0
NOC2L           13       12       59        1        1        1        3
          Sample79 Sample80 Sample81 Sample82 Sample83 Sample84 Sample85
FAM87B           0        0        0        0       12       17       22
LINC00115        0        0        1        0       77      122      129
FAM41C           0        0        0        1      236      222      293
SAMD11           0        0        0        0        1        2        2
NOC2L            1        0        9        2     1359     2998     4874
          Sample86 Sample87 Sample88 Sample89 Sample90 Sample91 Sample92
FAM87B          18        0        3        2        2        7        1
LINC00115      222        6        5        6       18       36       24
FAM41C         269        8       29       63       36      104       64
SAMD11           5        0        1        2        1        2        5
NOC2L         5540      202      350      637      658     1116     1118
          Sample93 Sample94 Sample95 Sample96 Sample97 Sample98 Sample99
FAM87B           3        2        0        0        0        0        0
LINC00115       18       25        0        0        0        0        0
FAM41C          48       66        2        0        0        1        0
SAMD11           2        3        0        0        0        0        0
NOC2L          611      888        2        1        1        3        0
          Sample100 Sample101 Sample102 Sample103 Sample104 Sample105 Sample106
FAM87B            0         0         0         1         0         0         0
LINC00115         0         0         3        11         5         0         0
FAM41C            0         0         1        10         1         1         0
SAMD11            0         0         0         0         0         0         0
NOC2L             0         4        23        64        42         2         1
          Sample107 Sample108 Sample109 Sample110 Sample111 Sample112 Sample113
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         2         1         0         0         0
FAM41C            0         0         0         0         1         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             3         6         7         4         5         7        11
          Sample114 Sample115 Sample116 Sample117 Sample118 Sample119 Sample120
FAM87B            0         0         0         0         0         0         0
LINC00115         0         4         1         0         1         0         0
FAM41C            0         3         0         1         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L            13        29         5         3         1         1         0
          Sample121 Sample122 Sample123 Sample124 Sample125 Sample126 Sample127
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         0         0         1         1         3
FAM41C            0         0         0         0         1         0         2
SAMD11            0         0         0         0         0         1         1
NOC2L             1         1         2         1        23        12        13
          Sample128 Sample129 Sample130 Sample131 Sample132 Sample133 Sample134
FAM87B            0         0         0         0         0         0         0
LINC00115         0         1         0         0         0         0         0
FAM41C            2         0         0         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             6         2         1         4         0         1         0
          Sample135 Sample136 Sample137 Sample138 Sample139 Sample140 Sample141
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         1         0         0         0         0
FAM41C            0         0         0         1         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L             5         0         7        26        15         6         2
          Sample142 Sample143 Sample144 Sample145 Sample146 Sample147 Sample148
FAM87B            0         0         0         0         0         0         0
LINC00115         0         0         0         0         0         0         1
FAM41C            0         0         0         0         0         0         3
SAMD11            0         0         0         0         0         0         0
NOC2L             2         2         0         0         1         3        26
          Sample149 Sample150 Sample151 Sample152 Sample153 Sample154 Sample155
FAM87B            2         2         0         0         0         0         0
LINC00115         5         9         9         1         0         1         1
FAM41C           14        16         9         2         1         3         2
SAMD11            0         0         0         0         0         0         0
NOC2L           199       217       226         8         8        35        35
          Sample156 Sample157 Sample158 Sample159 Sample160 Sample161 Sample162
FAM87B            0         1         0         1         0         0         0
LINC00115         6         2         0         4         1         2         0
FAM41C            9         4         2        11         0         0         0
SAMD11            2         0         0         0         0         0         0
NOC2L            81        75        30        84         7         5        10
          Sample163 Sample164 Sample165 Sample166 Sample167 Sample168 Sample169
FAM87B            0         0         0         0         0         0         0
LINC00115         1         0         0         0         0         0         0
FAM41C            1         1         0         0         0         0         0
SAMD11            0         0         0         0         0         0         0
NOC2L            18         0         0         0         2         4         0
15996 more rows ...

$samples
        group lib.size norm.factors orig.ident nCount_RNA nFeature_RNA Barcode
Sample1     1  3098600            1        A-1         NA           NA    &lt;NA&gt;
Sample2     1   390265            1        A-2         NA           NA    &lt;NA&gt;
Sample3     1  1734153            1        A-3         NA           NA    &lt;NA&gt;
Sample4     1   367065            1        A-4         NA           NA    &lt;NA&gt;
Sample5     1   301523            1       &lt;NA&gt;         NA           NA    &lt;NA&gt;
        capture nuclear_fraction cell_status       HTO.x   donor experiment sum
Sample1       A               NA        cell           A       A          1  NA
Sample2       B               NA        cell           B       B          1  NA
Sample3       C               NA        cell           C       C          1  NA
Sample4       D               NA        cell           D       D          1  NA
Sample5    &lt;NA&gt;               NA        cell Human_HTO_8 donor_A          2  NA
        detected subsets_Mito_sum subsets_Mito_detected subsets_Mito_percent
Sample1       NA               NA                    NA                   NA
Sample2       NA               NA                    NA                   NA
Sample3       NA               NA                    NA                   NA
Sample4       NA               NA                    NA                   NA
Sample5       NA               NA                    NA                   NA
        subsets_Ribo_sum subsets_Ribo_detected subsets_Ribo_percent total
Sample1               NA                    NA                   NA    NA
Sample2               NA                    NA                   NA    NA
Sample3               NA                    NA                   NA    NA
Sample4               NA                    NA                   NA    NA
Sample5               NA                    NA                   NA    NA
        zero_percent predicted.annotation.l1 predicted.annotation.l1.score
Sample1           NA                    &lt;NA&gt;                            NA
Sample2           NA                    &lt;NA&gt;                            NA
Sample3           NA                    &lt;NA&gt;                            NA
Sample4           NA                    &lt;NA&gt;                            NA
Sample5           NA                    &lt;NA&gt;                            NA
        mapping.score nCount_SCT nFeature_SCT integrated_snn_res.0.1
Sample1            NA         NA           NA                   &lt;NA&gt;
Sample2            NA         NA           NA                   &lt;NA&gt;
Sample3            NA         NA           NA                   &lt;NA&gt;
Sample4            NA         NA           NA                   &lt;NA&gt;
Sample5            NA         NA           NA                   &lt;NA&gt;
        integrated_snn_res.0.2 integrated_snn_res.0.3 integrated_snn_res.0.4
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample4                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
        integrated_snn_res.0.5 integrated_snn_res.0.6 integrated_snn_res.0.7
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample4                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                   &lt;NA&gt;
        integrated_snn_res.0.8 integrated_snn_res.0.9 integrated_snn_res.1
Sample1                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample2                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample3                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample4                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
Sample5                   &lt;NA&gt;                   &lt;NA&gt;                 &lt;NA&gt;
        seurat_clusters predicted.celltype.score predicted.celltype
Sample1            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample2            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample3            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample4            &lt;NA&gt;                       NA               &lt;NA&gt;
Sample5            &lt;NA&gt;                       NA               &lt;NA&gt;
        predicted.ann_level_1 predicted.ann_level_2 predicted.ann_level_3
Sample1                  &lt;NA&gt;                  &lt;NA&gt;                  &lt;NA&gt;
Sample2                  &lt;NA&gt;                  &lt;NA&gt;                  &lt;NA&gt;
Sample3                Immune              Lymphoid        B cell lineage
Sample4                Immune                  &lt;NA&gt;                  &lt;NA&gt;
Sample5                  &lt;NA&gt;                  &lt;NA&gt;                  &lt;NA&gt;
        predicted.ann_level_4 predicted.ann_level_5 predicted.ann_finest_level
Sample1                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample2                  &lt;NA&gt;                  &lt;NA&gt;                       &lt;NA&gt;
Sample3                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample4                  &lt;NA&gt;                  None                       &lt;NA&gt;
Sample5                  &lt;NA&gt;                  None                       &lt;NA&gt;
        predicted.ann_level_1.score predicted.ann_level_2.score
Sample1                          NA                          NA
Sample2                          NA                          NA
Sample3                          NA                          NA
Sample4                          NA                          NA
Sample5                          NA                          NA
        predicted.ann_level_3.score predicted.ann_level_4.score
Sample1                          NA                          NA
Sample2                          NA                          NA
Sample3                          NA                          NA
Sample4                          NA                          NA
Sample5                          NA                          NA
        predicted.ann_level_5.score predicted.ann_finest_level.score nCount_ADT
Sample1                          NA                               NA          0
Sample2                          NA                               NA          0
Sample3                          NA                               NA          0
Sample4                          NA                               NA          0
Sample5                          NA                               NA         NA
        nFeature_ADT Annotation   Broad nCount_ADT.dsb nFeature_ADT.dsb
Sample1            0       &lt;NA&gt; B cells              0                0
Sample2            0       &lt;NA&gt; B cells              0                0
Sample3            0       &lt;NA&gt; B cells              0                0
Sample4            0       &lt;NA&gt; B cells              0                0
Sample5           NA       &lt;NA&gt; B cells             NA               NA
        nCount_ADT.raw nFeature_ADT.raw       HTO.y Participant Sex  Age
Sample1              0                0           A       B1_CF   M 2.99
Sample2              0                0           B       C1_CF   M 2.99
Sample3              0                0           C     A1_Ctrl   M 3.00
Sample4              0                0           D       D1_CF   M 3.03
Sample5             NA               NA Human_HTO_8       L2_CF   M 5.95
        Disease Batch Broad.1 donor.1 ncells
Sample1      CF     1 B cells       A    594
Sample2      CF     1 B cells       B     99
Sample3    Ctrl     1 B cells       C    364
Sample4      CF     1 B cells       D     87
Sample5      CF     2 B cells donor_A    131
164 more rows ...</code></pre>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(cpm(y.pb[, y.pb$samples$Broad == &quot;Macrophages&quot;], log=TRUE), 
                 gene.selection = &quot;common&quot;,
                 plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(rownames_to_column(y.pb$samples, var = &quot;sample&quot;)) %&gt;%
    mutate(Batch = as.factor(Batch))-&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            shape = Batch,
                            color = Disease))+
    geom_point(size = 3) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Disease&quot;,
         shape = &quot;Batch&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))  +
    scale_colour_brewer(palette = &quot;Set1&quot;) 
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="differential-expression-analysis-within-cell-types" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Differential expression analysis within cell types</h1>
<p>This analysis was adapted from <a href="https://bphipson.github.io/Human_Development_snRNAseq/">this</a>
analysis of single nuclei RNAseq data from human heart samples.</p>
<pre class="r"><code>y.pb$samples %&gt;%
  mutate(cell_label = fct_recode(Broad,
                                 Macr = &quot;Macrophages&quot;,
                                 PMacr = &quot;Proliferating macrophages&quot;,
                                 B = &quot;B cells&quot;,
                                 CD4 = &quot;CD4 T&quot;,
                                 CD8 = &quot;CD8 T&quot;,
                                 GDT = &quot;Gamma delta T&quot;,
                                 Mast = &quot;Mast cells&quot;,
                                 NK = &quot;NK cells&quot;,
                                 NKT = &quot;NK T&quot;,
                                 IL = &quot;Innate lymphocyte&quot;,
                                 Neu = &quot;Neutrophils&quot;,
                                 Mono = &quot;Monocytes&quot;,
                                 Epi = &quot;Epithelial&quot;,
                                 PNKT = &quot;Proliferating NK/T&quot;,
                                 Dend = &quot;Dendritic&quot;,
                                 Endo = &quot;Endothelial&quot;
                                 )) -&gt; y.pb$samples

newgrp &lt;- paste(y.pb$sample$cell_label, y.pb$samples$Disease, sep=&quot;.&quot;)
newgrp &lt;- factor(newgrp)

design &lt;- model.matrix(~ 0 + newgrp)
colnames(design) &lt;- levels(newgrp)

keep &lt;- filterByExpr(y.pb, design = design)
y &lt;- y.pb[keep,]
summary(keep)</code></pre>
<pre><code>   Mode   FALSE    TRUE 
logical    4572   11429 </code></pre>
<pre class="r"><code>v &lt;- voom(y, design, plot = TRUE, 
          normalize.method = &quot;cyclicloess&quot;)</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fit &lt;- lmFit(v, design)

y &lt;- calcNormFactors(y) # add normalisation factors for downstream visualization</code></pre>
<pre class="r"><code>y$samples %&gt;% dplyr::select(Broad,
                            Disease,
                            ncells) %&gt;%
  group_by(Broad, Disease) %&gt;%
  summarise(agg = sum(ncells),
            ind = n()) %&gt;%
  dplyr::rename(cell = Broad,
                grp = Disease) -&gt; dat

ggplot(dat, aes(x = cell,  y = ind, fill = grp)) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)) +
  labs(y = &quot;No. individuals&quot;, x = &quot;Cell type&quot;, fill = &quot;Disease&quot;) +
  geom_text(size = 3, position = position_stack(vjust = 0.5),
            aes(label = agg))</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-14-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="overall-deg-summary" class="section level3" number="4.0.1">
<h3><span class="header-section-number">4.0.1</span> Overall DEG summary</h3>
<pre class="r"><code>cont &lt;- makeContrasts(Macr.CvCF = Macr.Ctrl - Macr.CF,
                      B.CvCF = B.Ctrl - B.CF,
                      CD4.CvCF = CD4.Ctrl - CD4.CF,
                      CD8.CvCF = CD8.Ctrl - CD8.CF,
                      Den.CvCF = Dend.Ctrl - Dend.CF,
                      Epi.CvCF = Epi.Ctrl - Epi.CF,
                      GDT.CvCF = GDT.Ctrl - GDT.CF,
                      Mast.CvCF = Mast.Ctrl - Mast.CF,
                      Mono.CvCF = Mono.Ctrl - Mono.CF,
                      Neu.CvCF = Neu.Ctrl - Neu.CF,
                      NK.CvCF = NK.Ctrl - NK.CF,
                      NKT.CvCF = NKT.Ctrl - NKT.CF,
                      PNKT.CvCF = PNKT.Ctrl - PNKT.CF,
                      IL.CvCF = IL.Ctrl - IL.CF,
                      PMacr.CvCF = PMacr.Ctrl - PMacr.CF,
                      levels = design)

cont.fit &lt;- contrasts.fit(fit, contrasts = cont)
fit2 &lt;- eBayes(cont.fit, robust = TRUE)

summary(decideTests(fit2))</code></pre>
<pre><code>       Macr.CvCF B.CvCF CD4.CvCF CD8.CvCF Den.CvCF Epi.CvCF GDT.CvCF Mast.CvCF
Down         202     25        1        5        0      308       70        49
NotSig     11087  11224    11426    11376    11340    11081    11359     11374
Up           140    180        2       48       89       40        0         6
       Mono.CvCF Neu.CvCF NK.CvCF NKT.CvCF PNKT.CvCF IL.CvCF PMacr.CvCF
Down           1       12      20       41       151      16          1
NotSig     11358    11410   11409    11380     11271   11411      11406
Up            70        7       0        8         7       2         22</code></pre>
</div>
<div id="treat-deg-summary" class="section level3" number="4.0.2">
<h3><span class="header-section-number">4.0.2</span> TREAT DEG summary</h3>
<p>This tests for DEGs relative to a logFC threshold of 0.5</p>
<pre class="r"><code>fit.treat &lt;- treat(fit2, lfc = 0.5)
dt &lt;- decideTests(fit.treat)
summary(dt)</code></pre>
<pre><code>       Macr.CvCF B.CvCF CD4.CvCF CD8.CvCF Den.CvCF Epi.CvCF GDT.CvCF Mast.CvCF
Down          80     12        0        0        0      215       18         3
NotSig     11290  11336    11428    11406    11399    11192    11411     11426
Up            59     81        1       23       30       22        0         0
       Mono.CvCF Neu.CvCF NK.CvCF NKT.CvCF PNKT.CvCF IL.CvCF PMacr.CvCF
Down           1        4       4        7        53       5          0
NotSig     11386    11423   11425    11420     11375   11422      11424
Up            42        2       0        2         1       2          5</code></pre>
<pre class="r"><code>pal &lt;- paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;, 2)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;)

summary(dt) %&gt;% 
  data.frame %&gt;%
  dplyr::rename(Direction = Var1, Cell = Var2) %&gt;%
  dplyr::filter(Direction != &quot;NotSig&quot;) %&gt;%
  mutate(Cell = gsub(&quot;.CvCF&quot;, &quot;&quot;, Cell)) %&gt;%
  ggplot(aes(x = Cell, y = Freq, fill = Direction)) +
  geom_bar(position = &quot;dodge&quot;, stat = &quot;identity&quot;) +
  geom_text(aes(label = Freq), stat = &quot;identity&quot;,
            position = position_dodge(width = .9),
            vjust = -0.5, colour = &quot;black&quot;, size = 2) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90,
                                   vjust = 0.5,
                                   hjust = 1)) +
  labs(y = &quot;No. DGE (FDR &lt; 0.05)&quot;) +
  scale_fill_manual(values = pal)</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="macrophages" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Macrophages</h2>
<div id="mean-difference-plot-of-degs" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,1],
                  status = as.factor(as.vector(dt[,1]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 1, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,1] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;Macr.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 20) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) -&gt; f5a
f5a</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-19-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 1, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<colgroup>
<col width="16%" />
<col width="15%" />
<col width="13%" />
<col width="15%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">UTS2</td>
<td align="right">5.862007</td>
<td align="right">2.713501</td>
<td align="right">13.951714</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">CEACAM6</td>
<td align="right">3.631857</td>
<td align="right">2.445749</td>
<td align="right">11.079756</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">UGT2A1</td>
<td align="right">2.821002</td>
<td align="right">2.313806</td>
<td align="right">10.291653</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">TSPAN8</td>
<td align="right">3.788274</td>
<td align="right">2.492056</td>
<td align="right">8.562761</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">TEKT4</td>
<td align="right">3.358874</td>
<td align="right">2.761809</td>
<td align="right">8.517565</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">GCNT4</td>
<td align="right">3.198563</td>
<td align="right">2.420036</td>
<td align="right">8.326925</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">IGHA2</td>
<td align="right">4.405958</td>
<td align="right">2.642874</td>
<td align="right">8.126118</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000001</td>
</tr>
<tr class="even">
<td align="left">SLC5A8</td>
<td align="right">2.757411</td>
<td align="right">2.325958</td>
<td align="right">7.934654</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000003</td>
</tr>
<tr class="odd">
<td align="left">IGHG4</td>
<td align="right">5.409163</td>
<td align="right">2.609476</td>
<td align="right">7.792670</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000007</td>
</tr>
<tr class="even">
<td align="left">CYP2F1</td>
<td align="right">3.452080</td>
<td align="right">2.502012</td>
<td align="right">7.662353</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000012</td>
</tr>
<tr class="odd">
<td align="left">FRMPD2</td>
<td align="right">3.576618</td>
<td align="right">2.624740</td>
<td align="right">7.529787</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000024</td>
</tr>
<tr class="even">
<td align="left">RBMS3</td>
<td align="right">-5.986931</td>
<td align="right">2.709507</td>
<td align="right">-6.881678</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000000736</td>
</tr>
<tr class="odd">
<td align="left">GABRP</td>
<td align="right">2.330100</td>
<td align="right">2.425643</td>
<td align="right">6.772929</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000001208</td>
</tr>
<tr class="even">
<td align="left">CXCL6</td>
<td align="right">3.000939</td>
<td align="right">2.457529</td>
<td align="right">6.671729</td>
<td align="right">0.0000000002</td>
<td align="right">0.0000001910</td>
</tr>
<tr class="odd">
<td align="left">KDR</td>
<td align="right">-3.031158</td>
<td align="right">2.254323</td>
<td align="right">-6.562908</td>
<td align="right">0.0000000004</td>
<td align="right">0.0000003147</td>
</tr>
<tr class="even">
<td align="left">SNCG</td>
<td align="right">-3.318880</td>
<td align="right">2.207711</td>
<td align="right">-6.463583</td>
<td align="right">0.0000000007</td>
<td align="right">0.0000004934</td>
</tr>
<tr class="odd">
<td align="left">SPRR3</td>
<td align="right">2.891999</td>
<td align="right">2.335707</td>
<td align="right">6.102621</td>
<td align="right">0.0000000043</td>
<td align="right">0.0000029082</td>
</tr>
<tr class="even">
<td align="left">CCL15</td>
<td align="right">-4.477277</td>
<td align="right">2.930575</td>
<td align="right">-5.958375</td>
<td align="right">0.0000000089</td>
<td align="right">0.0000056274</td>
</tr>
<tr class="odd">
<td align="left">PTP4A1</td>
<td align="right">7.380823</td>
<td align="right">3.986594</td>
<td align="right">5.798239</td>
<td align="right">0.0000000197</td>
<td align="right">0.0000118424</td>
</tr>
<tr class="even">
<td align="left">TSPOAP1</td>
<td align="right">3.327775</td>
<td align="right">3.884451</td>
<td align="right">5.714154</td>
<td align="right">0.0000000292</td>
<td align="right">0.0000166646</td>
</tr>
<tr class="odd">
<td align="left">PTPRG</td>
<td align="right">-2.569488</td>
<td align="right">2.252961</td>
<td align="right">-5.649415</td>
<td align="right">0.0000000398</td>
<td align="right">0.0000216586</td>
</tr>
<tr class="even">
<td align="left">ZNF667</td>
<td align="right">2.661115</td>
<td align="right">3.125318</td>
<td align="right">5.555272</td>
<td align="right">0.0000000623</td>
<td align="right">0.0000323714</td>
</tr>
<tr class="odd">
<td align="left">AGAP6</td>
<td align="right">4.948099</td>
<td align="right">3.075654</td>
<td align="right">5.482151</td>
<td align="right">0.0000000882</td>
<td align="right">0.0000438196</td>
</tr>
<tr class="even">
<td align="left">SPARCL1</td>
<td align="right">-2.885917</td>
<td align="right">2.322192</td>
<td align="right">-5.420035</td>
<td align="right">0.0000001177</td>
<td align="right">0.0000560599</td>
</tr>
<tr class="odd">
<td align="left">EHD2</td>
<td align="right">-2.615356</td>
<td align="right">2.212725</td>
<td align="right">-5.340006</td>
<td align="right">0.0000001708</td>
<td align="right">0.0000780634</td>
</tr>
<tr class="even">
<td align="left">LYPD2</td>
<td align="right">3.338787</td>
<td align="right">2.505069</td>
<td align="right">5.280452</td>
<td align="right">0.0000002247</td>
<td align="right">0.0000987842</td>
</tr>
<tr class="odd">
<td align="left">EFNB3</td>
<td align="right">-2.482948</td>
<td align="right">2.262176</td>
<td align="right">-5.218009</td>
<td align="right">0.0000002990</td>
<td align="right">0.0001260893</td>
</tr>
<tr class="even">
<td align="left">SCNN1B</td>
<td align="right">2.261281</td>
<td align="right">2.349556</td>
<td align="right">5.210892</td>
<td align="right">0.0000003089</td>
<td align="right">0.0001260893</td>
</tr>
<tr class="odd">
<td align="left">IGHM</td>
<td align="right">4.191090</td>
<td align="right">4.141473</td>
<td align="right">5.187602</td>
<td align="right">0.0000003438</td>
<td align="right">0.0001354886</td>
</tr>
<tr class="even">
<td align="left">LAMA3</td>
<td align="right">-3.553021</td>
<td align="right">2.455797</td>
<td align="right">-5.175146</td>
<td align="right">0.0000003635</td>
<td align="right">0.0001384811</td>
</tr>
<tr class="odd">
<td align="left">BICDL2</td>
<td align="right">2.178937</td>
<td align="right">2.517024</td>
<td align="right">5.150600</td>
<td align="right">0.0000004062</td>
<td align="right">0.0001468721</td>
</tr>
<tr class="even">
<td align="left">CCDC148</td>
<td align="right">-3.381283</td>
<td align="right">2.458126</td>
<td align="right">-5.147886</td>
<td align="right">0.0000004112</td>
<td align="right">0.0001468721</td>
</tr>
<tr class="odd">
<td align="left">GALNT15</td>
<td align="right">-3.253900</td>
<td align="right">2.189739</td>
<td align="right">-5.133314</td>
<td align="right">0.0000004392</td>
<td align="right">0.0001521045</td>
</tr>
<tr class="even">
<td align="left">HOXA9</td>
<td align="right">-4.606467</td>
<td align="right">2.526269</td>
<td align="right">-5.055521</td>
<td align="right">0.0000006245</td>
<td align="right">0.0002099263</td>
</tr>
<tr class="odd">
<td align="left">GBP3</td>
<td align="right">-2.641806</td>
<td align="right">5.733257</td>
<td align="right">-4.993655</td>
<td align="right">0.0000008203</td>
<td align="right">0.0002678552</td>
</tr>
<tr class="even">
<td align="left">FAM110D</td>
<td align="right">-2.564753</td>
<td align="right">2.234322</td>
<td align="right">-4.956733</td>
<td align="right">0.0000009658</td>
<td align="right">0.0003066160</td>
</tr>
<tr class="odd">
<td align="left">PPP1R14C</td>
<td align="right">2.066667</td>
<td align="right">2.887982</td>
<td align="right">4.931255</td>
<td align="right">0.0000010805</td>
<td align="right">0.0003337643</td>
</tr>
<tr class="even">
<td align="left">FBN1</td>
<td align="right">-3.011915</td>
<td align="right">3.184503</td>
<td align="right">-4.886648</td>
<td align="right">0.0000013141</td>
<td align="right">0.0003952243</td>
</tr>
<tr class="odd">
<td align="left">PDGFD</td>
<td align="right">-3.335840</td>
<td align="right">3.807812</td>
<td align="right">-4.873228</td>
<td align="right">0.0000013937</td>
<td align="right">0.0004084118</td>
</tr>
<tr class="even">
<td align="left">NGFR</td>
<td align="right">-3.382599</td>
<td align="right">2.282537</td>
<td align="right">-4.857996</td>
<td align="right">0.0000014894</td>
<td align="right">0.0004220437</td>
</tr>
<tr class="odd">
<td align="left">COBL</td>
<td align="right">-3.516822</td>
<td align="right">2.510669</td>
<td align="right">-4.854268</td>
<td align="right">0.0000015140</td>
<td align="right">0.0004220437</td>
</tr>
<tr class="even">
<td align="left">COL4A1</td>
<td align="right">-2.667119</td>
<td align="right">2.147238</td>
<td align="right">-4.723250</td>
<td align="right">0.0000026634</td>
<td align="right">0.0007247552</td>
</tr>
<tr class="odd">
<td align="left">NUPR2</td>
<td align="right">-2.253806</td>
<td align="right">2.239889</td>
<td align="right">-4.675721</td>
<td align="right">0.0000032612</td>
<td align="right">0.0008349096</td>
</tr>
<tr class="even">
<td align="left">IL33</td>
<td align="right">-2.892216</td>
<td align="right">2.341083</td>
<td align="right">-4.674602</td>
<td align="right">0.0000032770</td>
<td align="right">0.0008349096</td>
</tr>
<tr class="odd">
<td align="left">MUC5B</td>
<td align="right">2.926775</td>
<td align="right">2.565709</td>
<td align="right">4.673862</td>
<td align="right">0.0000032873</td>
<td align="right">0.0008349096</td>
</tr>
<tr class="even">
<td align="left">CACNA2D1</td>
<td align="right">-2.644489</td>
<td align="right">2.177479</td>
<td align="right">-4.656808</td>
<td align="right">0.0000035336</td>
<td align="right">0.0008779431</td>
</tr>
<tr class="odd">
<td align="left">CLDN5</td>
<td align="right">-2.363169</td>
<td align="right">2.234311</td>
<td align="right">-4.624240</td>
<td align="right">0.0000040547</td>
<td align="right">0.0009859931</td>
</tr>
<tr class="even">
<td align="left">MCF2L-AS1</td>
<td align="right">-2.738597</td>
<td align="right">2.338666</td>
<td align="right">-4.616447</td>
<td align="right">0.0000041903</td>
<td align="right">0.0009977228</td>
</tr>
<tr class="odd">
<td align="left">TMPRSS11B</td>
<td align="right">1.864160</td>
<td align="right">2.215410</td>
<td align="right">4.603853</td>
<td align="right">0.0000044180</td>
<td align="right">0.0010304735</td>
</tr>
<tr class="even">
<td align="left">CXCL17</td>
<td align="right">2.441603</td>
<td align="right">2.895101</td>
<td align="right">4.509433</td>
<td align="right">0.0000065519</td>
<td align="right">0.0014976255</td>
</tr>
<tr class="odd">
<td align="left">CCL7</td>
<td align="right">-2.727805</td>
<td align="right">2.502773</td>
<td align="right">-4.438855</td>
<td align="right">0.0000087648</td>
<td align="right">0.0019641682</td>
</tr>
<tr class="even">
<td align="left">C16orf89</td>
<td align="right">2.166467</td>
<td align="right">2.357485</td>
<td align="right">4.408514</td>
<td align="right">0.0000099219</td>
<td align="right">0.0021807213</td>
</tr>
<tr class="odd">
<td align="left">NR2F1</td>
<td align="right">-2.173938</td>
<td align="right">2.257981</td>
<td align="right">-4.323870</td>
<td align="right">0.0000139831</td>
<td align="right">0.0030153328</td>
</tr>
<tr class="even">
<td align="left">NEU4</td>
<td align="right">2.636091</td>
<td align="right">2.847328</td>
<td align="right">4.316010</td>
<td align="right">0.0000144337</td>
<td align="right">0.0030548621</td>
</tr>
<tr class="odd">
<td align="left">ADGRB2</td>
<td align="right">-2.515205</td>
<td align="right">2.288961</td>
<td align="right">-4.300795</td>
<td align="right">0.0000153426</td>
<td align="right">0.0031881837</td>
</tr>
<tr class="even">
<td align="left">HOXA10</td>
<td align="right">-2.451632</td>
<td align="right">2.628175</td>
<td align="right">-4.278149</td>
<td align="right">0.0000167986</td>
<td align="right">0.0034160552</td>
</tr>
<tr class="odd">
<td align="left">NNMT</td>
<td align="right">-3.084712</td>
<td align="right">2.619793</td>
<td align="right">-4.274764</td>
<td align="right">0.0000170369</td>
<td align="right">0.0034160552</td>
</tr>
<tr class="even">
<td align="left">FCRLA</td>
<td align="right">2.984239</td>
<td align="right">2.619683</td>
<td align="right">4.267959</td>
<td align="right">0.0000175034</td>
<td align="right">0.0034490701</td>
</tr>
<tr class="odd">
<td align="left">MFAP2</td>
<td align="right">-2.227250</td>
<td align="right">2.176890</td>
<td align="right">-4.220319</td>
<td align="right">0.0000211432</td>
<td align="right">0.0040510390</td>
</tr>
<tr class="even">
<td align="left">SERPINA6</td>
<td align="right">-2.041625</td>
<td align="right">2.183037</td>
<td align="right">-4.218840</td>
<td align="right">0.0000212672</td>
<td align="right">0.0040510390</td>
</tr>
<tr class="odd">
<td align="left">CYP2A13</td>
<td align="right">-2.266084</td>
<td align="right">2.180351</td>
<td align="right">-4.212257</td>
<td align="right">0.0000218285</td>
<td align="right">0.0040669820</td>
</tr>
<tr class="even">
<td align="left">NUAK1</td>
<td align="right">-2.179022</td>
<td align="right">2.233585</td>
<td align="right">-4.209558</td>
<td align="right">0.0000220625</td>
<td align="right">0.0040669820</td>
</tr>
<tr class="odd">
<td align="left">CLEC9A</td>
<td align="right">2.634166</td>
<td align="right">2.765039</td>
<td align="right">4.184551</td>
<td align="right">0.0000243527</td>
<td align="right">0.0044178833</td>
</tr>
<tr class="even">
<td align="left">SLC18A2</td>
<td align="right">-3.695858</td>
<td align="right">2.462275</td>
<td align="right">-4.154343</td>
<td align="right">0.0000275152</td>
<td align="right">0.0047576963</td>
</tr>
<tr class="odd">
<td align="left">EGR1</td>
<td align="right">3.204173</td>
<td align="right">5.493262</td>
<td align="right">4.151228</td>
<td align="right">0.0000277846</td>
<td align="right">0.0047576963</td>
</tr>
<tr class="even">
<td align="left">CNGA3</td>
<td align="right">-2.621621</td>
<td align="right">2.171762</td>
<td align="right">-4.148535</td>
<td align="right">0.0000280495</td>
<td align="right">0.0047576963</td>
</tr>
<tr class="odd">
<td align="left">PRRX2</td>
<td align="right">-2.875204</td>
<td align="right">2.474116</td>
<td align="right">-4.146292</td>
<td align="right">0.0000283046</td>
<td align="right">0.0047576963</td>
</tr>
<tr class="even">
<td align="left">EFHB</td>
<td align="right">-2.385610</td>
<td align="right">2.914609</td>
<td align="right">-4.146169</td>
<td align="right">0.0000283072</td>
<td align="right">0.0047576963</td>
</tr>
<tr class="odd">
<td align="left">SCT</td>
<td align="right">-3.900694</td>
<td align="right">2.602384</td>
<td align="right">-4.142087</td>
<td align="right">0.0000289168</td>
<td align="right">0.0047897151</td>
</tr>
<tr class="even">
<td align="left">IGHD</td>
<td align="right">3.055782</td>
<td align="right">2.774144</td>
<td align="right">4.128571</td>
<td align="right">0.0000303445</td>
<td align="right">0.0049543832</td>
</tr>
<tr class="odd">
<td align="left">NTS</td>
<td align="right">-2.895366</td>
<td align="right">2.234923</td>
<td align="right">-4.065333</td>
<td align="right">0.0000387681</td>
<td align="right">0.0062405667</td>
</tr>
<tr class="even">
<td align="left">ADH1C</td>
<td align="right">2.405662</td>
<td align="right">2.513190</td>
<td align="right">4.040319</td>
<td align="right">0.0000426632</td>
<td align="right">0.0067626591</td>
</tr>
<tr class="odd">
<td align="left">ITGB1-DT</td>
<td align="right">-3.560199</td>
<td align="right">2.513560</td>
<td align="right">-4.038024</td>
<td align="right">0.0000431949</td>
<td align="right">0.0067626591</td>
</tr>
<tr class="even">
<td align="left">ST8SIA6</td>
<td align="right">-3.140236</td>
<td align="right">2.341614</td>
<td align="right">-4.021663</td>
<td align="right">0.0000458881</td>
<td align="right">0.0070872256</td>
</tr>
<tr class="odd">
<td align="left">FCRL4</td>
<td align="right">3.101568</td>
<td align="right">2.575871</td>
<td align="right">3.998124</td>
<td align="right">0.0000502018</td>
<td align="right">0.0076283244</td>
</tr>
<tr class="even">
<td align="left">CTTNBP2</td>
<td align="right">1.835282</td>
<td align="right">2.543146</td>
<td align="right">3.994236</td>
<td align="right">0.0000508812</td>
<td align="right">0.0076283244</td>
</tr>
<tr class="odd">
<td align="left">GXYLT2</td>
<td align="right">-2.228172</td>
<td align="right">2.380615</td>
<td align="right">-3.989502</td>
<td align="right">0.0000518083</td>
<td align="right">0.0076283244</td>
</tr>
<tr class="even">
<td align="left">C21orf62</td>
<td align="right">-5.605389</td>
<td align="right">2.466816</td>
<td align="right">-3.998859</td>
<td align="right">0.0000520614</td>
<td align="right">0.0076283244</td>
</tr>
<tr class="odd">
<td align="left">CREB3L3</td>
<td align="right">2.273824</td>
<td align="right">2.379804</td>
<td align="right">3.975370</td>
<td align="right">0.0000546679</td>
<td align="right">0.0079088537</td>
</tr>
<tr class="even">
<td align="left">FAM84A</td>
<td align="right">-2.010766</td>
<td align="right">2.326654</td>
<td align="right">-3.914547</td>
<td align="right">0.0000687767</td>
<td align="right">0.0097637883</td>
</tr>
<tr class="odd">
<td align="left">MUCL1</td>
<td align="right">-5.153326</td>
<td align="right">2.477942</td>
<td align="right">-3.921647</td>
<td align="right">0.0000691983</td>
<td align="right">0.0097637883</td>
</tr>
<tr class="even">
<td align="left">DSP</td>
<td align="right">2.226220</td>
<td align="right">3.267715</td>
<td align="right">3.893399</td>
<td align="right">0.0000744517</td>
<td align="right">0.0103769358</td>
</tr>
<tr class="odd">
<td align="left">UPK1B</td>
<td align="right">1.936639</td>
<td align="right">2.564542</td>
<td align="right">3.889551</td>
<td align="right">0.0000755273</td>
<td align="right">0.0104000180</td>
</tr>
<tr class="even">
<td align="left">CFAP57</td>
<td align="right">-2.656793</td>
<td align="right">2.672766</td>
<td align="right">-3.878843</td>
<td align="right">0.0000786428</td>
<td align="right">0.0107001027</td>
</tr>
<tr class="odd">
<td align="left">SERPINB3</td>
<td align="right">2.678247</td>
<td align="right">2.983725</td>
<td align="right">3.869099</td>
<td align="right">0.0000815554</td>
<td align="right">0.0109658404</td>
</tr>
<tr class="even">
<td align="left">SPATA9</td>
<td align="right">-2.500811</td>
<td align="right">2.363668</td>
<td align="right">-3.839654</td>
<td align="right">0.0000909565</td>
<td align="right">0.0120876933</td>
</tr>
<tr class="odd">
<td align="left">MUC5AC</td>
<td align="right">2.046414</td>
<td align="right">2.544260</td>
<td align="right">3.824700</td>
<td align="right">0.0000961049</td>
<td align="right">0.0126054838</td>
</tr>
<tr class="even">
<td align="left">CYR61</td>
<td align="right">-2.025816</td>
<td align="right">2.256598</td>
<td align="right">-3.822024</td>
<td align="right">0.0000970586</td>
<td align="right">0.0126054838</td>
</tr>
<tr class="odd">
<td align="left">SOX18</td>
<td align="right">-2.215386</td>
<td align="right">2.131568</td>
<td align="right">-3.804506</td>
<td align="right">0.0001035354</td>
<td align="right">0.0132955784</td>
</tr>
<tr class="even">
<td align="left">UCKL1-AS1</td>
<td align="right">1.980620</td>
<td align="right">3.454806</td>
<td align="right">3.800398</td>
<td align="right">0.0001051060</td>
<td align="right">0.0133208728</td>
</tr>
<tr class="odd">
<td align="left">LCN2</td>
<td align="right">3.073328</td>
<td align="right">3.567471</td>
<td align="right">3.798521</td>
<td align="right">0.0001060635</td>
<td align="right">0.0133208728</td>
</tr>
<tr class="even">
<td align="left">MYCT1</td>
<td align="right">-2.486898</td>
<td align="right">2.402289</td>
<td align="right">-3.791333</td>
<td align="right">0.0001086945</td>
<td align="right">0.0135029256</td>
</tr>
<tr class="odd">
<td align="left">SPP1</td>
<td align="right">-4.652893</td>
<td align="right">3.785170</td>
<td align="right">-3.779180</td>
<td align="right">0.0001166927</td>
<td align="right">0.0143406507</td>
</tr>
<tr class="even">
<td align="left">TM4SF18</td>
<td align="right">-2.052556</td>
<td align="right">2.131772</td>
<td align="right">-3.767910</td>
<td align="right">0.0001183966</td>
<td align="right">0.0143952621</td>
</tr>
<tr class="odd">
<td align="left">CALB2</td>
<td align="right">-2.049530</td>
<td align="right">2.146066</td>
<td align="right">-3.758354</td>
<td align="right">0.0001225989</td>
<td align="right">0.0147492921</td>
</tr>
<tr class="even">
<td align="left">ZSWIM8-AS1</td>
<td align="right">-4.959136</td>
<td align="right">2.438503</td>
<td align="right">-3.758472</td>
<td align="right">0.0001270083</td>
<td align="right">0.0148749638</td>
</tr>
<tr class="odd">
<td align="left">WFDC2</td>
<td align="right">2.863752</td>
<td align="right">3.944572</td>
<td align="right">3.748685</td>
<td align="right">0.0001271669</td>
<td align="right">0.0148749638</td>
</tr>
<tr class="even">
<td align="left">SLCO2A1</td>
<td align="right">-2.046132</td>
<td align="right">2.161991</td>
<td align="right">-3.747490</td>
<td align="right">0.0001275480</td>
<td align="right">0.0148749638</td>
</tr>
<tr class="odd">
<td align="left">S100A12</td>
<td align="right">-2.780639</td>
<td align="right">2.837807</td>
<td align="right">-3.743318</td>
<td align="right">0.0001296344</td>
<td align="right">0.0149655737</td>
</tr>
<tr class="even">
<td align="left">CHD1L</td>
<td align="right">2.661906</td>
<td align="right">4.146493</td>
<td align="right">3.717864</td>
<td align="right">0.0001421293</td>
<td align="right">0.0162439543</td>
</tr>
<tr class="odd">
<td align="left">NFATC2</td>
<td align="right">2.215347</td>
<td align="right">4.957781</td>
<td align="right">3.693688</td>
<td align="right">0.0001549875</td>
<td align="right">0.0173750852</td>
</tr>
<tr class="even">
<td align="left">NXPH4</td>
<td align="right">2.079670</td>
<td align="right">2.504485</td>
<td align="right">3.693529</td>
<td align="right">0.0001550668</td>
<td align="right">0.0173750852</td>
</tr>
<tr class="odd">
<td align="left">ELF3-AS1</td>
<td align="right">-4.120005</td>
<td align="right">2.853089</td>
<td align="right">-3.685970</td>
<td align="right">0.0001622255</td>
<td align="right">0.0179071144</td>
</tr>
<tr class="even">
<td align="left">RAB6B</td>
<td align="right">-2.522288</td>
<td align="right">2.309746</td>
<td align="right">-3.679871</td>
<td align="right">0.0001629486</td>
<td align="right">0.0179071144</td>
</tr>
<tr class="odd">
<td align="left">CNTD1</td>
<td align="right">-2.259104</td>
<td align="right">3.038356</td>
<td align="right">-3.665197</td>
<td align="right">0.0001716928</td>
<td align="right">0.0186883489</td>
</tr>
<tr class="even">
<td align="left">ACKR1</td>
<td align="right">-2.316434</td>
<td align="right">2.158453</td>
<td align="right">-3.657016</td>
<td align="right">0.0001768036</td>
<td align="right">0.0190630996</td>
</tr>
<tr class="odd">
<td align="left">IRX5</td>
<td align="right">1.846038</td>
<td align="right">2.515168</td>
<td align="right">3.654155</td>
<td align="right">0.0001785861</td>
<td align="right">0.0190753339</td>
</tr>
<tr class="even">
<td align="left">CCL14</td>
<td align="right">-3.153287</td>
<td align="right">2.326173</td>
<td align="right">-3.641413</td>
<td align="right">0.0001876434</td>
<td align="right">0.0198571856</td>
</tr>
<tr class="odd">
<td align="left">PRMT8</td>
<td align="right">-1.787325</td>
<td align="right">2.315710</td>
<td align="right">-3.635255</td>
<td align="right">0.0001910391</td>
<td align="right">0.0200310675</td>
</tr>
<tr class="even">
<td align="left">TMEM246</td>
<td align="right">-2.014305</td>
<td align="right">2.310262</td>
<td align="right">-3.626412</td>
<td align="right">0.0001971494</td>
<td align="right">0.0204838275</td>
</tr>
<tr class="odd">
<td align="left">G0S2</td>
<td align="right">3.122591</td>
<td align="right">4.477964</td>
<td align="right">3.617647</td>
<td align="right">0.0002041827</td>
<td align="right">0.0210234622</td>
</tr>
<tr class="even">
<td align="left">S100A14</td>
<td align="right">1.715372</td>
<td align="right">2.566960</td>
<td align="right">3.599455</td>
<td align="right">0.0002169077</td>
<td align="right">0.0221342721</td>
</tr>
<tr class="odd">
<td align="left">ASIC1</td>
<td align="right">-2.259971</td>
<td align="right">2.441399</td>
<td align="right">-3.589598</td>
<td align="right">0.0002246330</td>
<td align="right">0.0227197381</td>
</tr>
<tr class="even">
<td align="left">ADGRL4</td>
<td align="right">-2.204105</td>
<td align="right">2.106213</td>
<td align="right">-3.585455</td>
<td align="right">0.0002279275</td>
<td align="right">0.0228507281</td>
</tr>
<tr class="odd">
<td align="left">CHTF8</td>
<td align="right">3.064582</td>
<td align="right">4.151596</td>
<td align="right">3.568458</td>
<td align="right">0.0002428918</td>
<td align="right">0.0241392251</td>
</tr>
<tr class="even">
<td align="left">MS4A1</td>
<td align="right">2.820123</td>
<td align="right">3.686587</td>
<td align="right">3.548454</td>
<td align="right">0.0002600769</td>
<td align="right">0.0256243055</td>
</tr>
<tr class="odd">
<td align="left">MYRIP</td>
<td align="right">-2.368332</td>
<td align="right">2.188446</td>
<td align="right">-3.532564</td>
<td align="right">0.0002744378</td>
<td align="right">0.0268081207</td>
</tr>
<tr class="even">
<td align="left">SRXN1</td>
<td align="right">-4.515838</td>
<td align="right">3.322142</td>
<td align="right">-3.515615</td>
<td align="right">0.0003016451</td>
<td align="right">0.0291446207</td>
</tr>
<tr class="odd">
<td align="left">NR5A2</td>
<td align="right">-2.531241</td>
<td align="right">2.296196</td>
<td align="right">-3.502690</td>
<td align="right">0.0003046270</td>
<td align="right">0.0291446207</td>
</tr>
<tr class="even">
<td align="left">C16orf71</td>
<td align="right">-1.996979</td>
<td align="right">2.391356</td>
<td align="right">-3.501128</td>
<td align="right">0.0003060070</td>
<td align="right">0.0291446207</td>
</tr>
<tr class="odd">
<td align="left">BPIFB1</td>
<td align="right">2.615884</td>
<td align="right">3.226672</td>
<td align="right">3.473512</td>
<td align="right">0.0003370941</td>
<td align="right">0.0317454979</td>
</tr>
<tr class="even">
<td align="left">GPR15</td>
<td align="right">-2.638611</td>
<td align="right">2.188404</td>
<td align="right">-3.472021</td>
<td align="right">0.0003388705</td>
<td align="right">0.0317454979</td>
</tr>
<tr class="odd">
<td align="left">TMEM56</td>
<td align="right">-2.020514</td>
<td align="right">2.281380</td>
<td align="right">-3.462065</td>
<td align="right">0.0003501632</td>
<td align="right">0.0322784525</td>
</tr>
<tr class="even">
<td align="left">OSM</td>
<td align="right">2.448479</td>
<td align="right">4.425266</td>
<td align="right">3.462232</td>
<td align="right">0.0003502081</td>
<td align="right">0.0322784525</td>
</tr>
<tr class="odd">
<td align="left">DOC2B</td>
<td align="right">-1.858369</td>
<td align="right">2.312310</td>
<td align="right">-3.445541</td>
<td align="right">0.0003705626</td>
<td align="right">0.0338812821</td>
</tr>
<tr class="even">
<td align="left">CLMP</td>
<td align="right">-1.970626</td>
<td align="right">2.300935</td>
<td align="right">-3.438174</td>
<td align="right">0.0003800344</td>
<td align="right">0.0344715301</td>
</tr>
<tr class="odd">
<td align="left">ENHO</td>
<td align="right">2.271082</td>
<td align="right">3.124004</td>
<td align="right">3.431242</td>
<td align="right">0.0003892686</td>
<td align="right">0.0350311133</td>
</tr>
<tr class="even">
<td align="left">KDF1</td>
<td align="right">-2.449588</td>
<td align="right">2.369429</td>
<td align="right">-3.427288</td>
<td align="right">0.0003947511</td>
<td align="right">0.0352469559</td>
</tr>
<tr class="odd">
<td align="left">KLF12</td>
<td align="right">2.078521</td>
<td align="right">4.475907</td>
<td align="right">3.405449</td>
<td align="right">0.0004248898</td>
<td align="right">0.0376439225</td>
</tr>
<tr class="even">
<td align="left">BAALC</td>
<td align="right">-2.928380</td>
<td align="right">2.741552</td>
<td align="right">-3.377277</td>
<td align="right">0.0004695292</td>
<td align="right">0.0412788370</td>
</tr>
<tr class="odd">
<td align="left">CXCL12</td>
<td align="right">-2.233036</td>
<td align="right">2.148625</td>
<td align="right">-3.361503</td>
<td align="right">0.0004930372</td>
<td align="right">0.0430146758</td>
</tr>
<tr class="even">
<td align="left">C11orf52</td>
<td align="right">-3.946309</td>
<td align="right">2.494017</td>
<td align="right">-3.362762</td>
<td align="right">0.0005039327</td>
<td align="right">0.0436321709</td>
</tr>
<tr class="odd">
<td align="left">TSPAN18</td>
<td align="right">-2.799707</td>
<td align="right">2.411065</td>
<td align="right">-3.339633</td>
<td align="right">0.0005323348</td>
<td align="right">0.0457447721</td>
</tr>
<tr class="even">
<td align="left">KRT19</td>
<td align="right">2.118870</td>
<td align="right">3.241186</td>
<td align="right">3.330737</td>
<td align="right">0.0005464980</td>
<td align="right">0.0466113822</td>
</tr>
<tr class="odd">
<td align="left">CDRT4</td>
<td align="right">-2.567343</td>
<td align="right">2.791182</td>
<td align="right">-3.322615</td>
<td align="right">0.0005624595</td>
<td align="right">0.0476174010</td>
</tr>
<tr class="even">
<td align="left">MUC13</td>
<td align="right">1.677931</td>
<td align="right">2.549000</td>
<td align="right">3.309124</td>
<td align="right">0.0005871644</td>
<td align="right">0.0487844705</td>
</tr>
<tr class="odd">
<td align="left">MSMB</td>
<td align="right">2.382626</td>
<td align="right">2.879050</td>
<td align="right">3.309354</td>
<td align="right">0.0005873030</td>
<td align="right">0.0487844705</td>
</tr>
<tr class="even">
<td align="left">CCL3L1</td>
<td align="right">3.993840</td>
<td align="right">4.201053</td>
<td align="right">3.317266</td>
<td align="right">0.0005890504</td>
<td align="right">0.0487844705</td>
</tr>
<tr class="odd">
<td align="left">NDUFA7</td>
<td align="right">-2.812082</td>
<td align="right">6.382546</td>
<td align="right">-3.300935</td>
<td align="right">0.0006059515</td>
<td align="right">0.0498231670</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/DEGs-{colnames(cont)[1]}.csv&quot;))</code></pre>
<pre class="r"><code>## calculate gene lengths
ex &lt;- exonsBy(TxDb.Hsapiens.UCSC.hg38.knownGene, &quot;gene&quot;)
## reduce to longest exon extents
ex &lt;- reduce(ex)
## compute exon lengths, by gene
exlen &lt;- relist(width(unlist(ex)), ex)
## add them up
exlens &lt;- sapply(exlen, sum)

## create full annotation
ann &lt;- rowData(summed) %&gt;% 
  data.frame %&gt;%
  right_join(data.frame(SYMBOL = rownames(y))) %&gt;%
  right_join(exlens %&gt;% 
               data.frame %&gt;% 
               rownames_to_column(var = &quot;ENTREZID&quot;) %&gt;% 
               dplyr::rename(Length = &quot;.&quot;))

gsAnnots &lt;- buildIdx(entrezIDs = ann$ENTREZID, 
                     species = &quot;human&quot;,
                     msigdb.gsets = c(&quot;c2&quot;,&quot;c5&quot;))</code></pre>
<pre><code>[1] &quot;Loading MSigDB Gene Sets ... &quot;
[1] &quot;Loaded gene sets for the collection c2 ...&quot;
[1] &quot;Indexed the collection c2 ...&quot;
[1] &quot;Created annotation for the collection c2 ...&quot;
[1] &quot;Loaded gene sets for the collection c5 ...&quot;
[1] &quot;Indexed the collection c5 ...&quot;
[1] &quot;Created annotation for the collection c5 ...&quot;
[1] &quot;Building KEGG pathways annotation object ... &quot;</code></pre>
<pre class="r"><code>react &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
                universe = ann$ENTREZID, 
                collection = gsAnnots$c2@original[grep(&quot;REACTOME&quot;, 
                                                       names(gsAnnots$c2@original))],
                gene.length = ann$Length)</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">6</td>
<td align="right">0.0000002</td>
<td align="right">0.0001498</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">4</td>
<td align="right">0.0000020</td>
<td align="right">0.0006848</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">7</td>
<td align="right">0.0000224</td>
<td align="right">0.0041716</td>
</tr>
<tr class="even">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">7</td>
<td align="right">0.0000248</td>
<td align="right">0.0041716</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">4</td>
<td align="right">0.0001221</td>
<td align="right">0.0164528</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">7</td>
<td align="right">0.0005435</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">8</td>
<td align="right">0.0006161</td>
<td align="right">0.0560222</td>
</tr>
<tr class="even">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">5</td>
<td align="right">0.0006650</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">0.0026122</td>
<td align="right">0.1956237</td>
</tr>
<tr class="even">
<td align="left">REACTOME_METABOLISM_OF_PROTEINS</td>
<td align="right">427</td>
<td align="right">7</td>
<td align="right">0.0045843</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0051455</td>
<td align="right">0.2996069</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE1_FUNCTIONALIZATION_OF_COMPOUNDS</td>
<td align="right">69</td>
<td align="right">3</td>
<td align="right">0.0053342</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0158286</td>
<td align="right">0.8206532</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NUCLEAR_RECEPTOR_TRANSCRIPTION_PATHWAY</td>
<td align="right">47</td>
<td align="right">2</td>
<td align="right">0.0198569</td>
<td align="right">0.9559694</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CYTOCHROME_P450_ARRANGED_BY_SUBSTRATE_TYPE</td>
<td align="right">50</td>
<td align="right">2</td>
<td align="right">0.0273852</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0427644</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_REGULATED_PROTEOLYSIS_OF_P75NTR</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0455794</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0467892</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0474190</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_ETHANOL_OXIDATION</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0475130</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="6%" />
<col width="3%" />
<col width="10%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_POSITIVE_REGULATION_OF_RESPONSE_TO_EXTERNAL_STIMULUS</td>
<td align="right">296</td>
<td align="right">13</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000047</td>
</tr>
<tr class="even">
<td align="left">GO_EXTRACELLULAR_SPACE</td>
<td align="right">1343</td>
<td align="right">25</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000047</td>
</tr>
<tr class="odd">
<td align="left">GO_CHEMOKINE_RECEPTOR_BINDING</td>
<td align="right">57</td>
<td align="right">7</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000074</td>
</tr>
<tr class="even">
<td align="left">GO_O_GLYCAN_PROCESSING</td>
<td align="right">57</td>
<td align="right">7</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000133</td>
</tr>
<tr class="odd">
<td align="left">GO_PROTEIN_O_LINKED_GLYCOSYLATION</td>
<td align="right">100</td>
<td align="right">8</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000214</td>
</tr>
<tr class="even">
<td align="left">GO_CHEMOKINE_MEDIATED_SIGNALING_PATHWAY</td>
<td align="right">72</td>
<td align="right">7</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000331</td>
</tr>
<tr class="odd">
<td align="left">GO_CHEMOKINE_ACTIVITY</td>
<td align="right">48</td>
<td align="right">6</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000331</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_INFLAMMATORY_RESPONSE</td>
<td align="right">113</td>
<td align="right">8</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000331</td>
</tr>
<tr class="odd">
<td align="left">GO_RECEPTOR_BINDING</td>
<td align="right">1452</td>
<td align="right">23</td>
<td align="right">1.00e-07</td>
<td align="right">0.0000969</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOKINE_ACTIVITY</td>
<td align="right">219</td>
<td align="right">9</td>
<td align="right">5.00e-07</td>
<td align="right">0.0003355</td>
</tr>
<tr class="odd">
<td align="left">GO_MONOCYTE_CHEMOTAXIS</td>
<td align="right">41</td>
<td align="right">5</td>
<td align="right">7.00e-07</td>
<td align="right">0.0003943</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_RESPONSE_TO_WOUNDING</td>
<td align="right">162</td>
<td align="right">8</td>
<td align="right">8.00e-07</td>
<td align="right">0.0004057</td>
</tr>
<tr class="odd">
<td align="left">GO_CYTOKINE_RECEPTOR_BINDING</td>
<td align="right">271</td>
<td align="right">9</td>
<td align="right">3.50e-06</td>
<td align="right">0.0016412</td>
</tr>
<tr class="even">
<td align="left">GO_CELL_CHEMOTAXIS</td>
<td align="right">162</td>
<td align="right">7</td>
<td align="right">8.40e-06</td>
<td align="right">0.0036797</td>
</tr>
<tr class="odd">
<td align="left">GO_CCR_CHEMOKINE_RECEPTOR_BINDING</td>
<td align="right">35</td>
<td align="right">4</td>
<td align="right">1.16e-05</td>
<td align="right">0.0047640</td>
</tr>
<tr class="even">
<td align="left">GO_LOCOMOTION</td>
<td align="right">1113</td>
<td align="right">17</td>
<td align="right">1.29e-05</td>
<td align="right">0.0049847</td>
</tr>
<tr class="odd">
<td align="left">GO_GRANULOCYTE_MIGRATION</td>
<td align="right">75</td>
<td align="right">5</td>
<td align="right">1.61e-05</td>
<td align="right">0.0058335</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_CHEMOTAXIS</td>
<td align="right">120</td>
<td align="right">6</td>
<td align="right">1.87e-05</td>
<td align="right">0.0063890</td>
</tr>
<tr class="odd">
<td align="left">GO_IMMUNE_SYSTEM_PROCESS</td>
<td align="right">1949</td>
<td align="right">23</td>
<td align="right">2.16e-05</td>
<td align="right">0.0070077</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_RESPONSE_TO_EXTERNAL_STIMULUS</td>
<td align="right">926</td>
<td align="right">15</td>
<td align="right">2.33e-05</td>
<td align="right">0.0070216</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-{colnames(cont)[1]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-{colnames(cont)[1]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;Macrophages&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;)) %&gt;%
  mutate(`Z-score` = `Log2 CPM`)


dat %&gt;%
  mutate(Batch = as.factor(Batch)) %&gt;%
  as_tibble %&gt;%
  heatmap(.row = Gene,
          .column = Participant,
          .value = `Z-score`,
          .scale = &quot;row&quot;,
          show_column_names = TRUE,
          column_title_side = &quot;top&quot;,
          row_names_gp = grid::gpar(fontsize = 7),
          column_names_gp = grid::gpar(fontsize = 10),
          palette_value = circlize::colorRamp2(seq(-5, 5, length.out = 256),
                                               viridis::magma(256)),
          heatmap_legend_param = list(direction = &quot;horizontal&quot;)) %&gt;%
  add_tile(Disease, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9),
           annotation_legend_param = list(direction = &quot;horizontal&quot;)) %&gt;%
  add_tile(Sex, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9),
           annotation_legend_param = list(direction = &quot;horizontal&quot;)) %&gt;%
  add_tile(Age, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9),
           annotation_legend_param = list(direction = &quot;horizontal&quot;)) %&gt;%
  add_tile(Batch, size = unit(0.25, &quot;cm&quot;), 
           annotation_name_gp = grid::gpar(fontsize = 9),
           annotation_legend_param = list(direction = &quot;horizontal&quot;)) %&gt;%
  as_ComplexHeatmap() %&gt;%
  ComplexHeatmap::draw(heatmap_legend_side = &quot;bottom&quot;,  
                       annotation_legend_side = &quot;bottom&quot;) -&gt; f5b</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, nrow(top))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="b-cells" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> B cells</h2>
<div id="mean-difference-plot-of-degs-1" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,2],
                  status = as.factor(as.vector(dt[,2]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-25-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 2, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,2] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;B.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/7b44e3893f55d9e2efa666c6c062135536c5bb79/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">7b44e38</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs-1" class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 2, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ALPL</td>
<td align="right">4.319432</td>
<td align="right">2.397972</td>
<td align="right">9.059561</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">IGHG4</td>
<td align="right">6.259901</td>
<td align="right">2.609476</td>
<td align="right">8.539303</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">PKHD1L1</td>
<td align="right">4.144839</td>
<td align="right">2.279969</td>
<td align="right">7.897186</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000011</td>
</tr>
<tr class="even">
<td align="left">IGHD</td>
<td align="right">4.288344</td>
<td align="right">2.774144</td>
<td align="right">7.180298</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000443</td>
</tr>
<tr class="odd">
<td align="left">LARGE1</td>
<td align="right">3.845261</td>
<td align="right">2.793621</td>
<td align="right">6.981215</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000001036</td>
</tr>
<tr class="even">
<td align="left">ACY3</td>
<td align="right">4.403560</td>
<td align="right">2.602140</td>
<td align="right">6.902362</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000001239</td>
</tr>
<tr class="odd">
<td align="left">IGHM</td>
<td align="right">4.306975</td>
<td align="right">4.141473</td>
<td align="right">6.884739</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000001239</td>
</tr>
<tr class="even">
<td align="left">UTS2</td>
<td align="right">4.555232</td>
<td align="right">2.713501</td>
<td align="right">6.579470</td>
<td align="right">0.0000000004</td>
<td align="right">0.0000005413</td>
</tr>
<tr class="odd">
<td align="left">FCRL4</td>
<td align="right">4.199270</td>
<td align="right">2.575871</td>
<td align="right">6.538634</td>
<td align="right">0.0000000005</td>
<td align="right">0.0000005949</td>
</tr>
<tr class="even">
<td align="left">TNFRSF17</td>
<td align="right">3.643815</td>
<td align="right">2.542688</td>
<td align="right">6.517248</td>
<td align="right">0.0000000005</td>
<td align="right">0.0000005982</td>
</tr>
<tr class="odd">
<td align="left">HIST1H2BJ</td>
<td align="right">3.645157</td>
<td align="right">2.611150</td>
<td align="right">6.240792</td>
<td align="right">0.0000000022</td>
<td align="right">0.0000022419</td>
</tr>
<tr class="even">
<td align="left">JCHAIN</td>
<td align="right">4.189326</td>
<td align="right">3.569987</td>
<td align="right">5.983326</td>
<td align="right">0.0000000078</td>
<td align="right">0.0000074603</td>
</tr>
<tr class="odd">
<td align="left">SCNN1B</td>
<td align="right">3.198775</td>
<td align="right">2.349556</td>
<td align="right">5.854025</td>
<td align="right">0.0000000148</td>
<td align="right">0.0000130038</td>
</tr>
<tr class="even">
<td align="left">TTC24</td>
<td align="right">4.212988</td>
<td align="right">2.610741</td>
<td align="right">5.588380</td>
<td align="right">0.0000000533</td>
<td align="right">0.0000410990</td>
</tr>
<tr class="odd">
<td align="left">UGT8</td>
<td align="right">3.074373</td>
<td align="right">2.494762</td>
<td align="right">5.585671</td>
<td align="right">0.0000000539</td>
<td align="right">0.0000410990</td>
</tr>
<tr class="even">
<td align="left">FCRLA</td>
<td align="right">3.374339</td>
<td align="right">2.619683</td>
<td align="right">5.511477</td>
<td align="right">0.0000000767</td>
<td align="right">0.0000547529</td>
</tr>
<tr class="odd">
<td align="left">C12orf42</td>
<td align="right">3.873799</td>
<td align="right">2.467282</td>
<td align="right">5.439485</td>
<td align="right">0.0000001075</td>
<td align="right">0.0000722865</td>
</tr>
<tr class="even">
<td align="left">RHEX</td>
<td align="right">4.061839</td>
<td align="right">2.612493</td>
<td align="right">5.389413</td>
<td align="right">0.0000001358</td>
<td align="right">0.0000862558</td>
</tr>
<tr class="odd">
<td align="left">LINC02362</td>
<td align="right">3.515397</td>
<td align="right">2.786115</td>
<td align="right">5.284798</td>
<td align="right">0.0000002203</td>
<td align="right">0.0001325147</td>
</tr>
<tr class="even">
<td align="left">CRY1</td>
<td align="right">-4.572031</td>
<td align="right">4.419139</td>
<td align="right">-5.261744</td>
<td align="right">0.0000002453</td>
<td align="right">0.0001401671</td>
</tr>
<tr class="odd">
<td align="left">SDC1</td>
<td align="right">3.426729</td>
<td align="right">2.593031</td>
<td align="right">5.246120</td>
<td align="right">0.0000002630</td>
<td align="right">0.0001429518</td>
</tr>
<tr class="even">
<td align="left">IL17A</td>
<td align="right">4.365010</td>
<td align="right">2.516590</td>
<td align="right">5.236519</td>
<td align="right">0.0000002752</td>
<td align="right">0.0001429518</td>
</tr>
<tr class="odd">
<td align="left">GATD3A</td>
<td align="right">3.627865</td>
<td align="right">2.748692</td>
<td align="right">5.054742</td>
<td align="right">0.0000006252</td>
<td align="right">0.0003106722</td>
</tr>
<tr class="even">
<td align="left">FBLN2</td>
<td align="right">3.630360</td>
<td align="right">2.474426</td>
<td align="right">4.981966</td>
<td align="right">0.0000008643</td>
<td align="right">0.0004115691</td>
</tr>
<tr class="odd">
<td align="left">IGLC7</td>
<td align="right">3.598307</td>
<td align="right">2.227017</td>
<td align="right">4.948552</td>
<td align="right">0.0000010017</td>
<td align="right">0.0004579527</td>
</tr>
<tr class="even">
<td align="left">NXPH4</td>
<td align="right">3.423765</td>
<td align="right">2.504485</td>
<td align="right">4.930432</td>
<td align="right">0.0000010848</td>
<td align="right">0.0004749074</td>
</tr>
<tr class="odd">
<td align="left">RASSF6</td>
<td align="right">3.163523</td>
<td align="right">2.643794</td>
<td align="right">4.922728</td>
<td align="right">0.0000011219</td>
<td align="right">0.0004749074</td>
</tr>
<tr class="even">
<td align="left">GRAMD1C</td>
<td align="right">3.015066</td>
<td align="right">2.664814</td>
<td align="right">4.790640</td>
<td align="right">0.0000019941</td>
<td align="right">0.0008139576</td>
</tr>
<tr class="odd">
<td align="left">TUBB2B</td>
<td align="right">4.285814</td>
<td align="right">2.550330</td>
<td align="right">4.767140</td>
<td align="right">0.0000022127</td>
<td align="right">0.0008720323</td>
</tr>
<tr class="even">
<td align="left">TNFSF11</td>
<td align="right">3.088457</td>
<td align="right">2.612865</td>
<td align="right">4.753345</td>
<td align="right">0.0000023415</td>
<td align="right">0.0008920303</td>
</tr>
<tr class="odd">
<td align="left">TNFRSF13B</td>
<td align="right">2.604596</td>
<td align="right">2.875782</td>
<td align="right">4.691186</td>
<td align="right">0.0000030537</td>
<td align="right">0.0011258308</td>
</tr>
<tr class="even">
<td align="left">IGHG1</td>
<td align="right">3.376495</td>
<td align="right">2.767682</td>
<td align="right">4.675543</td>
<td align="right">0.0000032652</td>
<td align="right">0.0011661893</td>
</tr>
<tr class="odd">
<td align="left">MARC2</td>
<td align="right">2.675986</td>
<td align="right">2.432783</td>
<td align="right">4.629830</td>
<td align="right">0.0000039604</td>
<td align="right">0.0013716205</td>
</tr>
<tr class="even">
<td align="left">CCDC189</td>
<td align="right">3.010559</td>
<td align="right">3.042438</td>
<td align="right">4.588312</td>
<td align="right">0.0000047167</td>
<td align="right">0.0015855101</td>
</tr>
<tr class="odd">
<td align="left">HRK</td>
<td align="right">2.915655</td>
<td align="right">2.632944</td>
<td align="right">4.577134</td>
<td align="right">0.0000049426</td>
<td align="right">0.0016139617</td>
</tr>
<tr class="even">
<td align="left">HIST1H1A</td>
<td align="right">3.034156</td>
<td align="right">2.406247</td>
<td align="right">4.543329</td>
<td align="right">0.0000056925</td>
<td align="right">0.0017759913</td>
</tr>
<tr class="odd">
<td align="left">CADM1</td>
<td align="right">3.707028</td>
<td align="right">3.105903</td>
<td align="right">4.541249</td>
<td align="right">0.0000057496</td>
<td align="right">0.0017759913</td>
</tr>
<tr class="even">
<td align="left">HRASLS2</td>
<td align="right">3.063559</td>
<td align="right">2.694374</td>
<td align="right">4.461128</td>
<td align="right">0.0000080005</td>
<td align="right">0.0024062645</td>
</tr>
<tr class="odd">
<td align="left">SPC25</td>
<td align="right">3.675657</td>
<td align="right">3.028454</td>
<td align="right">4.440624</td>
<td align="right">0.0000087165</td>
<td align="right">0.0025158641</td>
</tr>
<tr class="even">
<td align="left">CR2</td>
<td align="right">3.523614</td>
<td align="right">2.455004</td>
<td align="right">4.438028</td>
<td align="right">0.0000088052</td>
<td align="right">0.0025158641</td>
</tr>
<tr class="odd">
<td align="left">CSF3R</td>
<td align="right">-4.941139</td>
<td align="right">4.391477</td>
<td align="right">-4.423411</td>
<td align="right">0.0000094542</td>
<td align="right">0.0026354032</td>
</tr>
<tr class="even">
<td align="left">VPREB3</td>
<td align="right">2.760700</td>
<td align="right">2.733157</td>
<td align="right">4.355818</td>
<td align="right">0.0000122930</td>
<td align="right">0.0033451487</td>
</tr>
<tr class="odd">
<td align="left">SOX5</td>
<td align="right">2.618195</td>
<td align="right">2.646066</td>
<td align="right">4.284695</td>
<td align="right">0.0000163655</td>
<td align="right">0.0043398150</td>
</tr>
<tr class="even">
<td align="left">IGHA2</td>
<td align="right">2.658944</td>
<td align="right">2.642874</td>
<td align="right">4.279527</td>
<td align="right">0.0000167077</td>
<td align="right">0.0043398150</td>
</tr>
<tr class="odd">
<td align="left">MCM10</td>
<td align="right">3.438047</td>
<td align="right">2.998359</td>
<td align="right">4.230641</td>
<td align="right">0.0000203315</td>
<td align="right">0.0050890764</td>
</tr>
<tr class="even">
<td align="left">TCL1A</td>
<td align="right">4.061658</td>
<td align="right">2.786837</td>
<td align="right">4.229798</td>
<td align="right">0.0000204828</td>
<td align="right">0.0050890764</td>
</tr>
<tr class="odd">
<td align="left">C16orf45</td>
<td align="right">2.977165</td>
<td align="right">2.709143</td>
<td align="right">4.215432</td>
<td align="right">0.0000215667</td>
<td align="right">0.0051379662</td>
</tr>
<tr class="even">
<td align="left">GAS2L1</td>
<td align="right">-4.384247</td>
<td align="right">4.370777</td>
<td align="right">-4.217544</td>
<td align="right">0.0000215786</td>
<td align="right">0.0051379662</td>
</tr>
<tr class="odd">
<td align="left">KIF19</td>
<td align="right">2.601965</td>
<td align="right">2.681817</td>
<td align="right">4.174967</td>
<td align="right">0.0000252876</td>
<td align="right">0.0058982088</td>
</tr>
<tr class="even">
<td align="left">ZNF608</td>
<td align="right">2.876240</td>
<td align="right">2.771798</td>
<td align="right">4.165145</td>
<td align="right">0.0000262892</td>
<td align="right">0.0060091927</td>
</tr>
<tr class="odd">
<td align="left">DEPDC1B</td>
<td align="right">3.119022</td>
<td align="right">3.100315</td>
<td align="right">4.145860</td>
<td align="right">0.0000283674</td>
<td align="right">0.0063570691</td>
</tr>
<tr class="even">
<td align="left">MYL9</td>
<td align="right">3.151153</td>
<td align="right">3.365629</td>
<td align="right">4.046588</td>
<td align="right">0.0000417020</td>
<td align="right">0.0091628578</td>
</tr>
<tr class="odd">
<td align="left">PTPRG</td>
<td align="right">2.066790</td>
<td align="right">2.252961</td>
<td align="right">4.037045</td>
<td align="right">0.0000431998</td>
<td align="right">0.0091628578</td>
</tr>
<tr class="even">
<td align="left">CD19</td>
<td align="right">2.242201</td>
<td align="right">3.194841</td>
<td align="right">4.036489</td>
<td align="right">0.0000432929</td>
<td align="right">0.0091628578</td>
</tr>
<tr class="odd">
<td align="left">KIF14</td>
<td align="right">3.180335</td>
<td align="right">3.264090</td>
<td align="right">4.027870</td>
<td align="right">0.0000448163</td>
<td align="right">0.0092907555</td>
</tr>
<tr class="even">
<td align="left">JAM3</td>
<td align="right">3.110815</td>
<td align="right">2.469564</td>
<td align="right">4.022489</td>
<td align="right">0.0000457380</td>
<td align="right">0.0092907555</td>
</tr>
<tr class="odd">
<td align="left">MZB1</td>
<td align="right">2.748803</td>
<td align="right">3.777388</td>
<td align="right">4.018857</td>
<td align="right">0.0000463359</td>
<td align="right">0.0092907555</td>
</tr>
<tr class="even">
<td align="left">CDK5R1</td>
<td align="right">2.983503</td>
<td align="right">3.121548</td>
<td align="right">4.011701</td>
<td align="right">0.0000476461</td>
<td align="right">0.0093887455</td>
</tr>
<tr class="odd">
<td align="left">CCDC170</td>
<td align="right">-4.769245</td>
<td align="right">4.275079</td>
<td align="right">-4.010480</td>
<td align="right">0.0000488199</td>
<td align="right">0.0094569960</td>
</tr>
<tr class="even">
<td align="left">OSTN-AS1</td>
<td align="right">2.765725</td>
<td align="right">2.773341</td>
<td align="right">3.970603</td>
<td align="right">0.0000556911</td>
<td align="right">0.0106082305</td>
</tr>
<tr class="odd">
<td align="left">FXYD7</td>
<td align="right">3.646285</td>
<td align="right">2.814158</td>
<td align="right">3.967081</td>
<td align="right">0.0000566972</td>
<td align="right">0.0106228284</td>
</tr>
<tr class="even">
<td align="left">FAM129C</td>
<td align="right">3.105676</td>
<td align="right">2.600085</td>
<td align="right">3.914987</td>
<td align="right">0.0000687816</td>
<td align="right">0.0125571553</td>
</tr>
<tr class="odd">
<td align="left">HIST1H3G</td>
<td align="right">2.977066</td>
<td align="right">2.321503</td>
<td align="right">3.911928</td>
<td align="right">0.0000695398</td>
<td align="right">0.0125571553</td>
</tr>
<tr class="even">
<td align="left">RLN1</td>
<td align="right">3.014732</td>
<td align="right">2.638882</td>
<td align="right">3.909005</td>
<td align="right">0.0000703174</td>
<td align="right">0.0125571553</td>
</tr>
<tr class="odd">
<td align="left">TUSC1</td>
<td align="right">-3.708185</td>
<td align="right">3.686750</td>
<td align="right">-3.885251</td>
<td align="right">0.0000772690</td>
<td align="right">0.0134311773</td>
</tr>
<tr class="even">
<td align="left">MAML3</td>
<td align="right">-4.069002</td>
<td align="right">4.062416</td>
<td align="right">-3.885537</td>
<td align="right">0.0000775621</td>
<td align="right">0.0134311773</td>
</tr>
<tr class="odd">
<td align="left">ACPP</td>
<td align="right">2.979294</td>
<td align="right">2.640771</td>
<td align="right">3.831679</td>
<td align="right">0.0000938004</td>
<td align="right">0.0160006619</td>
</tr>
<tr class="even">
<td align="left">FOXM1</td>
<td align="right">3.087004</td>
<td align="right">3.354528</td>
<td align="right">3.820933</td>
<td align="right">0.0000976562</td>
<td align="right">0.0162813768</td>
</tr>
<tr class="odd">
<td align="left">CDC6</td>
<td align="right">2.884238</td>
<td align="right">3.122812</td>
<td align="right">3.818913</td>
<td align="right">0.0000982951</td>
<td align="right">0.0162813768</td>
</tr>
<tr class="even">
<td align="left">AICDA</td>
<td align="right">3.105370</td>
<td align="right">2.888371</td>
<td align="right">3.804366</td>
<td align="right">0.0001038241</td>
<td align="right">0.0169515139</td>
</tr>
<tr class="odd">
<td align="left">FCER2</td>
<td align="right">3.008546</td>
<td align="right">2.783726</td>
<td align="right">3.798425</td>
<td align="right">0.0001060636</td>
<td align="right">0.0170732555</td>
</tr>
<tr class="even">
<td align="left">MYBL2</td>
<td align="right">3.104687</td>
<td align="right">3.371776</td>
<td align="right">3.718649</td>
<td align="right">0.0001420321</td>
<td align="right">0.0225456203</td>
</tr>
<tr class="odd">
<td align="left">CAV1</td>
<td align="right">2.959222</td>
<td align="right">2.548271</td>
<td align="right">3.693758</td>
<td align="right">0.0001552641</td>
<td align="right">0.0243083957</td>
</tr>
<tr class="even">
<td align="left">CENPA</td>
<td align="right">3.021664</td>
<td align="right">3.516133</td>
<td align="right">3.683464</td>
<td align="right">0.0001611965</td>
<td align="right">0.0248961452</td>
</tr>
<tr class="odd">
<td align="left">SMIM24</td>
<td align="right">2.664149</td>
<td align="right">2.338751</td>
<td align="right">3.652661</td>
<td align="right">0.0001797109</td>
<td align="right">0.0273855469</td>
</tr>
<tr class="even">
<td align="left">MVB12B</td>
<td align="right">-3.765686</td>
<td align="right">4.316426</td>
<td align="right">-3.643987</td>
<td align="right">0.0001874631</td>
<td align="right">0.0281909904</td>
</tr>
<tr class="odd">
<td align="left">ASB2</td>
<td align="right">2.864194</td>
<td align="right">3.691630</td>
<td align="right">3.626165</td>
<td align="right">0.0001977052</td>
<td align="right">0.0293451009</td>
</tr>
<tr class="even">
<td align="left">KCNK12</td>
<td align="right">2.964810</td>
<td align="right">2.801422</td>
<td align="right">3.620216</td>
<td align="right">0.0002020654</td>
<td align="right">0.0296077657</td>
</tr>
<tr class="odd">
<td align="left">IGLC2</td>
<td align="right">3.352676</td>
<td align="right">3.434153</td>
<td align="right">3.581216</td>
<td align="right">0.0002329575</td>
<td align="right">0.0332850856</td>
</tr>
<tr class="even">
<td align="left">NEK2</td>
<td align="right">2.951447</td>
<td align="right">3.049205</td>
<td align="right">3.579416</td>
<td align="right">0.0002334630</td>
<td align="right">0.0332850856</td>
</tr>
<tr class="odd">
<td align="left">SHISA8</td>
<td align="right">2.257761</td>
<td align="right">2.456828</td>
<td align="right">3.574091</td>
<td align="right">0.0002372459</td>
<td align="right">0.0332850856</td>
</tr>
<tr class="even">
<td align="left">LINC01781</td>
<td align="right">2.279553</td>
<td align="right">2.245882</td>
<td align="right">3.572227</td>
<td align="right">0.0002388115</td>
<td align="right">0.0332850856</td>
</tr>
<tr class="odd">
<td align="left">CYTH3</td>
<td align="right">-3.744854</td>
<td align="right">3.895357</td>
<td align="right">-3.546429</td>
<td align="right">0.0002651619</td>
<td align="right">0.0365124702</td>
</tr>
<tr class="even">
<td align="left">WWC1</td>
<td align="right">2.165672</td>
<td align="right">2.527773</td>
<td align="right">3.536257</td>
<td align="right">0.0002708493</td>
<td align="right">0.0366754403</td>
</tr>
<tr class="odd">
<td align="left">CC2D2A</td>
<td align="right">-3.724604</td>
<td align="right">3.785765</td>
<td align="right">-3.538288</td>
<td align="right">0.0002727634</td>
<td align="right">0.0366754403</td>
</tr>
<tr class="even">
<td align="left">SNX22</td>
<td align="right">2.435666</td>
<td align="right">3.012458</td>
<td align="right">3.505211</td>
<td align="right">0.0003018828</td>
<td align="right">0.0401188218</td>
</tr>
<tr class="odd">
<td align="left">IGF2BP2</td>
<td align="right">-3.932521</td>
<td align="right">3.901662</td>
<td align="right">-3.502346</td>
<td align="right">0.0003109043</td>
<td align="right">0.0408428183</td>
</tr>
<tr class="even">
<td align="left">DOCK1</td>
<td align="right">-3.649242</td>
<td align="right">3.890820</td>
<td align="right">-3.491827</td>
<td align="right">0.0003204798</td>
<td align="right">0.0412371083</td>
</tr>
<tr class="odd">
<td align="left">TLR10</td>
<td align="right">2.263578</td>
<td align="right">3.318400</td>
<td align="right">3.487261</td>
<td align="right">0.0003211219</td>
<td align="right">0.0412371083</td>
</tr>
<tr class="even">
<td align="left">ZBED2</td>
<td align="right">2.792755</td>
<td align="right">2.617383</td>
<td align="right">3.463775</td>
<td align="right">0.0003489779</td>
<td align="right">0.0440104560</td>
</tr>
<tr class="odd">
<td align="left">RAPH1</td>
<td align="right">-3.563778</td>
<td align="right">3.817997</td>
<td align="right">-3.465633</td>
<td align="right">0.0003504201</td>
<td align="right">0.0440104560</td>
</tr>
<tr class="even">
<td align="left">POLQ</td>
<td align="right">2.477416</td>
<td align="right">3.328333</td>
<td align="right">3.450923</td>
<td align="right">0.0003641151</td>
<td align="right">0.0452333880</td>
</tr>
<tr class="odd">
<td align="left">C17orf51</td>
<td align="right">2.554487</td>
<td align="right">2.521448</td>
<td align="right">3.432758</td>
<td align="right">0.0003876118</td>
<td align="right">0.0476345716</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/DEGs-{colnames(cont)[2]}.csv&quot;))</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">6</td>
<td align="right">0.0000002</td>
<td align="right">0.0001498</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">4</td>
<td align="right">0.0000020</td>
<td align="right">0.0006848</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">7</td>
<td align="right">0.0000224</td>
<td align="right">0.0041716</td>
</tr>
<tr class="even">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">7</td>
<td align="right">0.0000248</td>
<td align="right">0.0041716</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">4</td>
<td align="right">0.0001221</td>
<td align="right">0.0164528</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">7</td>
<td align="right">0.0005435</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">8</td>
<td align="right">0.0006161</td>
<td align="right">0.0560222</td>
</tr>
<tr class="even">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">5</td>
<td align="right">0.0006650</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">0.0026122</td>
<td align="right">0.1956237</td>
</tr>
<tr class="even">
<td align="left">REACTOME_METABOLISM_OF_PROTEINS</td>
<td align="right">427</td>
<td align="right">7</td>
<td align="right">0.0045843</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0051455</td>
<td align="right">0.2996069</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE1_FUNCTIONALIZATION_OF_COMPOUNDS</td>
<td align="right">69</td>
<td align="right">3</td>
<td align="right">0.0053342</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0158286</td>
<td align="right">0.8206532</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NUCLEAR_RECEPTOR_TRANSCRIPTION_PATHWAY</td>
<td align="right">47</td>
<td align="right">2</td>
<td align="right">0.0198569</td>
<td align="right">0.9559694</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CYTOCHROME_P450_ARRANGED_BY_SUBSTRATE_TYPE</td>
<td align="right">50</td>
<td align="right">2</td>
<td align="right">0.0273852</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0427644</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_REGULATED_PROTEOLYSIS_OF_P75NTR</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0455794</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0467892</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0474190</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_ETHANOL_OXIDATION</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0475130</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="69%" />
<col width="5%" />
<col width="3%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_ADAPTIVE_IMMUNE_RESPONSE</td>
<td align="right">259</td>
<td align="right">8</td>
<td align="right">0.0000006</td>
<td align="right">0.0025702</td>
</tr>
<tr class="even">
<td align="left">GO_MICROTUBULE_CYTOSKELETON_ORGANIZATION</td>
<td align="right">348</td>
<td align="right">9</td>
<td align="right">0.0000008</td>
<td align="right">0.0025702</td>
</tr>
<tr class="odd">
<td align="left">GO_MICROTUBULE_BASED_PROCESS</td>
<td align="right">521</td>
<td align="right">10</td>
<td align="right">0.0000028</td>
<td align="right">0.0049279</td>
</tr>
<tr class="even">
<td align="left">GO_IMMUNE_RESPONSE</td>
<td align="right">1066</td>
<td align="right">13</td>
<td align="right">0.0000044</td>
<td align="right">0.0049279</td>
</tr>
<tr class="odd">
<td align="left">GO_IMMUNE_SYSTEM_PROCESS</td>
<td align="right">1949</td>
<td align="right">18</td>
<td align="right">0.0000046</td>
<td align="right">0.0049279</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_IMMUNE_SYSTEM_PROCESS</td>
<td align="right">1371</td>
<td align="right">15</td>
<td align="right">0.0000048</td>
<td align="right">0.0049279</td>
</tr>
<tr class="odd">
<td align="left">GO_CELL_MOTILITY</td>
<td align="right">834</td>
<td align="right">11</td>
<td align="right">0.0000216</td>
<td align="right">0.0166727</td>
</tr>
<tr class="even">
<td align="left">GO_LOCALIZATION_OF_CELL</td>
<td align="right">834</td>
<td align="right">11</td>
<td align="right">0.0000216</td>
<td align="right">0.0166727</td>
</tr>
<tr class="odd">
<td align="left">GO_CYTOSKELETON_ORGANIZATION</td>
<td align="right">837</td>
<td align="right">11</td>
<td align="right">0.0000280</td>
<td align="right">0.0192069</td>
</tr>
<tr class="even">
<td align="left">GO_EXTERNAL_SIDE_OF_PLASMA_MEMBRANE</td>
<td align="right">219</td>
<td align="right">6</td>
<td align="right">0.0000383</td>
<td align="right">0.0236374</td>
</tr>
<tr class="odd">
<td align="left">GO_MOVEMENT_OF_CELL_OR_SUBCELLULAR_COMPONENT</td>
<td align="right">1273</td>
<td align="right">13</td>
<td align="right">0.0000565</td>
<td align="right">0.0316848</td>
</tr>
<tr class="even">
<td align="left">GO_LOCOMOTION</td>
<td align="right">1113</td>
<td align="right">12</td>
<td align="right">0.0000641</td>
<td align="right">0.0329474</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MICROTUBULE_BASED_PROCESS</td>
<td align="right">243</td>
<td align="right">6</td>
<td align="right">0.0000809</td>
<td align="right">0.0383523</td>
</tr>
<tr class="even">
<td align="left">GO_POSITIVE_REGULATION_OF_IMMUNE_SYSTEM_PROCESS</td>
<td align="right">835</td>
<td align="right">10</td>
<td align="right">0.0001180</td>
<td align="right">0.0502951</td>
</tr>
<tr class="odd">
<td align="left">GO_ESTABLISHMENT_OF_CELL_POLARITY</td>
<td align="right">88</td>
<td align="right">4</td>
<td align="right">0.0001224</td>
<td align="right">0.0502951</td>
</tr>
<tr class="even">
<td align="left">GO_SIDE_OF_MEMBRANE</td>
<td align="right">409</td>
<td align="right">7</td>
<td align="right">0.0001742</td>
<td align="right">0.0647558</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_MICROTUBULE_POLYMERIZATION_OR_DEPOLYMERIZATION</td>
<td align="right">178</td>
<td align="right">5</td>
<td align="right">0.0001785</td>
<td align="right">0.0647558</td>
</tr>
<tr class="even">
<td align="left">GO_MICROTUBULE_POLYMERIZATION_OR_DEPOLYMERIZATION</td>
<td align="right">41</td>
<td align="right">3</td>
<td align="right">0.0002502</td>
<td align="right">0.0857108</td>
</tr>
<tr class="odd">
<td align="left">GO_IMMUNOGLOBULIN_PRODUCTION</td>
<td align="right">46</td>
<td align="right">3</td>
<td align="right">0.0002899</td>
<td align="right">0.0940813</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_TRANSPORT</td>
<td align="right">1802</td>
<td align="right">14</td>
<td align="right">0.0005130</td>
<td align="right">0.1472695</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-{colnames(cont)[2]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-{colnames(cont)[2]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;B cells&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;)) %&gt;%
  mutate(`Z-score` = `Log2 CPM`)

p &lt;- vector(&quot;list&quot;, nrow(top))

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-29-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="epithelial" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Epithelial</h2>
<div id="mean-difference-plot-of-degs-2" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Mean-Difference plot of DEGs</h3>
<pre class="r"><code>dat &lt;- data.frame(x = fit.treat$Amean, 
                  y = fit.treat$coefficients[,6],
                  status = as.factor(as.vector(dt[,6]))) %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
  arrange(desc(status))

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;)
names(pal) &lt;- c(&quot;Up&quot;, &quot;Down&quot;, &quot;NotSig&quot;)
ggplot(dat, aes(x = x, y = y, colour = status)) +
  geom_point() +
  ggrepel::geom_text_repel(data = dat[dat$status != &quot;NotSig&quot;,], 
                  aes(x = x, y = y, label = gene), 
                  max.overlaps = 20, size = 2, colour = &quot;black&quot;) +
  scale_colour_manual(values = pal) +
  labs(x = &quot;Average-log-expression&quot;, 
       y = &quot;Log-fold-change&quot;,
       colour = &quot;FDR &lt; 0.05&quot;) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/7b44e3893f55d9e2efa666c6c062135536c5bb79/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-30-1.png" target="_blank">7b44e38</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-30-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>topTreat(fit.treat, coef = 6, num = Inf) %&gt;% 
  mutate(sig = ifelse(adj.P.Val &lt;= 0.05, &quot;&lt;= 0.05&quot;, &quot;&gt; 0.05&quot;)) %&gt;%
  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
  left_join(dt[,6] %&gt;% 
              data.frame %&gt;%
              rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
              dplyr::rename(status = &quot;Epi.CvCF&quot;)) %&gt;%
    mutate(status = ifelse(status == 1, &quot;Up&quot;,
                         ifelse(status == -1, &quot;Down&quot;,
                                &quot;NotSig&quot;))) %&gt;%
  mutate(status = as.factor(status)) %&gt;%
  mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) -&gt; dat

ggplot(dat, aes(x = logFC, y = -log10(P.Value), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = subset(dat, adj.P.Val &lt; 0.05), 
            aes(x = logFC, y = -log10(P.Value), 
                label = SYMBOL), 
            size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
  labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
       y = expression(~-log[10]~&quot;(P-value)&quot;),
       colour = &quot;FDR &lt; 0.05&quot;) +
  theme_classic() +
  scale_colour_manual(values = pal) </code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/7b44e3893f55d9e2efa666c6c062135536c5bb79/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-31-1.png" target="_blank">7b44e38</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-31-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="top-degs-2" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Top DEGs</h3>
<pre class="r"><code>dirName &lt;- here(&quot;output/DEGs&quot;)
if(!dir.exists(dirName)) dir.create(dirName)

top &lt;- topTreat(fit.treat, coef = 6, n = Inf, p.value = 0.05)
knitr::kable(top, digits = 10)</code></pre>
<table>
<colgroup>
<col width="17%" />
<col width="14%" />
<col width="13%" />
<col width="14%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">AveExpr</th>
<th align="right">t</th>
<th align="right">P.Value</th>
<th align="right">adj.P.Val</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SPRR3</td>
<td align="right">6.788332</td>
<td align="right">2.335707</td>
<td align="right">13.227307</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">TMPRSS11B</td>
<td align="right">5.694174</td>
<td align="right">2.215410</td>
<td align="right">12.745828</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="odd">
<td align="left">SPRR1B</td>
<td align="right">3.967066</td>
<td align="right">2.160902</td>
<td align="right">8.580770</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000000</td>
</tr>
<tr class="even">
<td align="left">STC1</td>
<td align="right">-4.638106</td>
<td align="right">2.176213</td>
<td align="right">-7.872950</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000010</td>
</tr>
<tr class="odd">
<td align="left">CYGB</td>
<td align="right">-6.248283</td>
<td align="right">2.823966</td>
<td align="right">-7.419269</td>
<td align="right">0.0000000000</td>
<td align="right">0.0000000096</td>
</tr>
<tr class="even">
<td align="left">CALB2</td>
<td align="right">-4.150674</td>
<td align="right">2.146066</td>
<td align="right">-6.940595</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000001073</td>
</tr>
<tr class="odd">
<td align="left">UGT2A1</td>
<td align="right">3.029301</td>
<td align="right">2.313806</td>
<td align="right">6.768798</td>
<td align="right">0.0000000001</td>
<td align="right">0.0000002292</td>
</tr>
<tr class="even">
<td align="left">ECM1</td>
<td align="right">4.371072</td>
<td align="right">2.980127</td>
<td align="right">6.279004</td>
<td align="right">0.0000000018</td>
<td align="right">0.0000025398</td>
</tr>
<tr class="odd">
<td align="left">CLMP</td>
<td align="right">-4.217637</td>
<td align="right">2.300935</td>
<td align="right">-6.075626</td>
<td align="right">0.0000000050</td>
<td align="right">0.0000062875</td>
</tr>
<tr class="even">
<td align="left">EMCN</td>
<td align="right">-3.687657</td>
<td align="right">2.137346</td>
<td align="right">-6.041915</td>
<td align="right">0.0000000059</td>
<td align="right">0.0000064886</td>
</tr>
<tr class="odd">
<td align="left">KRT78</td>
<td align="right">4.309530</td>
<td align="right">2.248139</td>
<td align="right">6.029024</td>
<td align="right">0.0000000062</td>
<td align="right">0.0000064886</td>
</tr>
<tr class="even">
<td align="left">C1QTNF1</td>
<td align="right">-3.935318</td>
<td align="right">2.207371</td>
<td align="right">-5.995434</td>
<td align="right">0.0000000074</td>
<td align="right">0.0000070255</td>
</tr>
<tr class="odd">
<td align="left">COL4A1</td>
<td align="right">-3.931574</td>
<td align="right">2.147238</td>
<td align="right">-5.847482</td>
<td align="right">0.0000000153</td>
<td align="right">0.0000134278</td>
</tr>
<tr class="even">
<td align="left">AARD</td>
<td align="right">-3.887435</td>
<td align="right">2.220522</td>
<td align="right">-5.701805</td>
<td align="right">0.0000000310</td>
<td align="right">0.0000252689</td>
</tr>
<tr class="odd">
<td align="left">SLC5A8</td>
<td align="right">2.837601</td>
<td align="right">2.325958</td>
<td align="right">5.564721</td>
<td align="right">0.0000000596</td>
<td align="right">0.0000434763</td>
</tr>
<tr class="even">
<td align="left">MSMB</td>
<td align="right">4.293452</td>
<td align="right">2.879050</td>
<td align="right">5.553367</td>
<td align="right">0.0000000629</td>
<td align="right">0.0000434763</td>
</tr>
<tr class="odd">
<td align="left">TMEM233</td>
<td align="right">-3.757644</td>
<td align="right">2.255817</td>
<td align="right">-5.547472</td>
<td align="right">0.0000000647</td>
<td align="right">0.0000434763</td>
</tr>
<tr class="even">
<td align="left">SOX18</td>
<td align="right">-3.689213</td>
<td align="right">2.131568</td>
<td align="right">-5.525063</td>
<td align="right">0.0000000719</td>
<td align="right">0.0000456503</td>
</tr>
<tr class="odd">
<td align="left">FAM110D</td>
<td align="right">-3.788130</td>
<td align="right">2.234322</td>
<td align="right">-5.397877</td>
<td align="right">0.0000001306</td>
<td align="right">0.0000785392</td>
</tr>
<tr class="even">
<td align="left">PCDHGA11</td>
<td align="right">-3.608989</td>
<td align="right">2.153725</td>
<td align="right">-5.206359</td>
<td align="right">0.0000003154</td>
<td align="right">0.0001802519</td>
</tr>
<tr class="odd">
<td align="left">CLEC14A</td>
<td align="right">-3.727114</td>
<td align="right">2.304745</td>
<td align="right">-5.169584</td>
<td align="right">0.0000003728</td>
<td align="right">0.0002029127</td>
</tr>
<tr class="even">
<td align="left">C7</td>
<td align="right">-3.455607</td>
<td align="right">2.123291</td>
<td align="right">-5.153043</td>
<td align="right">0.0000004018</td>
<td align="right">0.0002087148</td>
</tr>
<tr class="odd">
<td align="left">MMRN1</td>
<td align="right">-3.967753</td>
<td align="right">2.235068</td>
<td align="right">-5.108276</td>
<td align="right">0.0000004920</td>
<td align="right">0.0002444657</td>
</tr>
<tr class="even">
<td align="left">CABP4</td>
<td align="right">-4.147749</td>
<td align="right">3.205449</td>
<td align="right">-5.025104</td>
<td align="right">0.0000007142</td>
<td align="right">0.0003401298</td>
</tr>
<tr class="odd">
<td align="left">GREM1</td>
<td align="right">-3.746093</td>
<td align="right">2.507863</td>
<td align="right">-4.939299</td>
<td align="right">0.0000010436</td>
<td align="right">0.0004771112</td>
</tr>
<tr class="even">
<td align="left">SMAD9</td>
<td align="right">-3.859904</td>
<td align="right">2.230209</td>
<td align="right">-4.864750</td>
<td align="right">0.0000014473</td>
<td align="right">0.0006361928</td>
</tr>
<tr class="odd">
<td align="left">EDN1</td>
<td align="right">-4.201084</td>
<td align="right">2.605500</td>
<td align="right">-4.842709</td>
<td align="right">0.0000015948</td>
<td align="right">0.0006750548</td>
</tr>
<tr class="even">
<td align="left">ADGRL4</td>
<td align="right">-3.460036</td>
<td align="right">2.106213</td>
<td align="right">-4.829112</td>
<td align="right">0.0000016889</td>
<td align="right">0.0006893560</td>
</tr>
<tr class="odd">
<td align="left">RAMP3</td>
<td align="right">-3.457493</td>
<td align="right">2.116022</td>
<td align="right">-4.804081</td>
<td align="right">0.0000018823</td>
<td align="right">0.0007418115</td>
</tr>
<tr class="even">
<td align="left">KDM4A-AS1</td>
<td align="right">-4.802205</td>
<td align="right">3.131668</td>
<td align="right">-4.789162</td>
<td align="right">0.0000020183</td>
<td align="right">0.0007689236</td>
</tr>
<tr class="odd">
<td align="left">SPDYE5</td>
<td align="right">-3.720759</td>
<td align="right">2.405777</td>
<td align="right">-4.766658</td>
<td align="right">0.0000022131</td>
<td align="right">0.0008107397</td>
</tr>
<tr class="even">
<td align="left">RASAL2-AS1</td>
<td align="right">-3.637597</td>
<td align="right">2.726616</td>
<td align="right">-4.760713</td>
<td align="right">0.0000022700</td>
<td align="right">0.0008107397</td>
</tr>
<tr class="odd">
<td align="left">MEOX1</td>
<td align="right">-3.583909</td>
<td align="right">2.260391</td>
<td align="right">-4.749500</td>
<td align="right">0.0000023818</td>
<td align="right">0.0008248881</td>
</tr>
<tr class="even">
<td align="left">RAD54L</td>
<td align="right">-3.881837</td>
<td align="right">2.839259</td>
<td align="right">-4.724694</td>
<td align="right">0.0000026508</td>
<td align="right">0.0008910740</td>
</tr>
<tr class="odd">
<td align="left">ACKR1</td>
<td align="right">-3.446421</td>
<td align="right">2.158453</td>
<td align="right">-4.678389</td>
<td align="right">0.0000032263</td>
<td align="right">0.0010535260</td>
</tr>
<tr class="even">
<td align="left">GAREM2</td>
<td align="right">-3.846967</td>
<td align="right">2.539076</td>
<td align="right">-4.603323</td>
<td align="right">0.0000044359</td>
<td align="right">0.0014082624</td>
</tr>
<tr class="odd">
<td align="left">PCDH12</td>
<td align="right">-3.571990</td>
<td align="right">2.310748</td>
<td align="right">-4.596239</td>
<td align="right">0.0000045661</td>
<td align="right">0.0014104407</td>
</tr>
<tr class="even">
<td align="left">PI3</td>
<td align="right">2.933058</td>
<td align="right">2.360329</td>
<td align="right">4.585842</td>
<td align="right">0.0000047656</td>
<td align="right">0.0014333123</td>
</tr>
<tr class="odd">
<td align="left">CD300LG</td>
<td align="right">-3.435213</td>
<td align="right">2.200966</td>
<td align="right">-4.536948</td>
<td align="right">0.0000058491</td>
<td align="right">0.0017140941</td>
</tr>
<tr class="even">
<td align="left">FLRT2</td>
<td align="right">-3.401293</td>
<td align="right">2.343805</td>
<td align="right">-4.530467</td>
<td align="right">0.0000060086</td>
<td align="right">0.0017168038</td>
</tr>
<tr class="odd">
<td align="left">COL15A1</td>
<td align="right">-3.781979</td>
<td align="right">2.815673</td>
<td align="right">-4.508399</td>
<td align="right">0.0000065928</td>
<td align="right">0.0018140598</td>
</tr>
<tr class="even">
<td align="left">CXCL12</td>
<td align="right">-3.449000</td>
<td align="right">2.148625</td>
<td align="right">-4.498694</td>
<td align="right">0.0000068561</td>
<td align="right">0.0018140598</td>
</tr>
<tr class="odd">
<td align="left">CAV1</td>
<td align="right">-4.662133</td>
<td align="right">2.548271</td>
<td align="right">-4.498947</td>
<td align="right">0.0000068979</td>
<td align="right">0.0018140598</td>
</tr>
<tr class="even">
<td align="left">IL12RB1</td>
<td align="right">-4.293888</td>
<td align="right">4.967612</td>
<td align="right">-4.495195</td>
<td align="right">0.0000069839</td>
<td align="right">0.0018140598</td>
</tr>
<tr class="odd">
<td align="left">ARHGEF15</td>
<td align="right">-3.445842</td>
<td align="right">2.160656</td>
<td align="right">-4.459368</td>
<td align="right">0.0000080642</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="even">
<td align="left">SYN2</td>
<td align="right">-3.847229</td>
<td align="right">2.452106</td>
<td align="right">-4.457451</td>
<td align="right">0.0000081400</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="odd">
<td align="left">CLSPN</td>
<td align="right">-4.881017</td>
<td align="right">4.199885</td>
<td align="right">-4.458084</td>
<td align="right">0.0000081894</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="even">
<td align="left">RDH16</td>
<td align="right">-3.881772</td>
<td align="right">2.333804</td>
<td align="right">-4.455131</td>
<td align="right">0.0000082196</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="odd">
<td align="left">CDK15</td>
<td align="right">-3.451736</td>
<td align="right">2.138190</td>
<td align="right">-4.447731</td>
<td align="right">0.0000084596</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="even">
<td align="left">GINS1</td>
<td align="right">-4.035590</td>
<td align="right">3.271133</td>
<td align="right">-4.447145</td>
<td align="right">0.0000085014</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="odd">
<td align="left">NGFR</td>
<td align="right">-3.731279</td>
<td align="right">2.282537</td>
<td align="right">-4.446486</td>
<td align="right">0.0000085111</td>
<td align="right">0.0019073195</td>
</tr>
<tr class="even">
<td align="left">CNGA3</td>
<td align="right">-3.384123</td>
<td align="right">2.171762</td>
<td align="right">-4.424510</td>
<td align="right">0.0000093031</td>
<td align="right">0.0020447080</td>
</tr>
<tr class="odd">
<td align="left">CRHBP</td>
<td align="right">-3.396759</td>
<td align="right">2.367373</td>
<td align="right">-4.419092</td>
<td align="right">0.0000095119</td>
<td align="right">0.0020511611</td>
</tr>
<tr class="even">
<td align="left">ITGA2B</td>
<td align="right">-3.596221</td>
<td align="right">2.203083</td>
<td align="right">-4.411426</td>
<td align="right">0.0000098208</td>
<td align="right">0.0020785569</td>
</tr>
<tr class="odd">
<td align="left">HIST1H4F</td>
<td align="right">-3.629960</td>
<td align="right">2.192315</td>
<td align="right">-4.400076</td>
<td align="right">0.0000102879</td>
<td align="right">0.0021378333</td>
</tr>
<tr class="even">
<td align="left">HOXA10</td>
<td align="right">-3.785462</td>
<td align="right">2.628175</td>
<td align="right">-4.383895</td>
<td align="right">0.0000109969</td>
<td align="right">0.0022443467</td>
</tr>
<tr class="odd">
<td align="left">MUC5AC</td>
<td align="right">3.052561</td>
<td align="right">2.544260</td>
<td align="right">4.376789</td>
<td align="right">0.0000112941</td>
<td align="right">0.0022645665</td>
</tr>
<tr class="even">
<td align="left">ADAM11</td>
<td align="right">-3.431419</td>
<td align="right">2.215238</td>
<td align="right">-4.365039</td>
<td align="right">0.0000118554</td>
<td align="right">0.0023050362</td>
</tr>
<tr class="odd">
<td align="left">ST8SIA6</td>
<td align="right">-3.938905</td>
<td align="right">2.341614</td>
<td align="right">-4.364710</td>
<td align="right">0.0000118993</td>
<td align="right">0.0023050362</td>
</tr>
<tr class="even">
<td align="left">LINC01781</td>
<td align="right">-3.658784</td>
<td align="right">2.245882</td>
<td align="right">-4.352319</td>
<td align="right">0.0000124934</td>
<td align="right">0.0023797867</td>
</tr>
<tr class="odd">
<td align="left">GALNT15</td>
<td align="right">-3.438194</td>
<td align="right">2.189739</td>
<td align="right">-4.341087</td>
<td align="right">0.0000130629</td>
<td align="right">0.0024288527</td>
</tr>
<tr class="even">
<td align="left">MAMSTR</td>
<td align="right">-3.960752</td>
<td align="right">2.761319</td>
<td align="right">-4.339596</td>
<td align="right">0.0000131760</td>
<td align="right">0.0024288527</td>
</tr>
<tr class="odd">
<td align="left">DEPDC1B</td>
<td align="right">-4.307687</td>
<td align="right">3.100315</td>
<td align="right">-4.334183</td>
<td align="right">0.0000135060</td>
<td align="right">0.0024501514</td>
</tr>
<tr class="even">
<td align="left">AATK</td>
<td align="right">-4.397457</td>
<td align="right">2.925527</td>
<td align="right">-4.313865</td>
<td align="right">0.0000146747</td>
<td align="right">0.0026205796</td>
</tr>
<tr class="odd">
<td align="left">PREX2</td>
<td align="right">-4.310684</td>
<td align="right">2.274274</td>
<td align="right">-4.277693</td>
<td align="right">0.0000169577</td>
<td align="right">0.0029816906</td>
</tr>
<tr class="even">
<td align="left">SLC47A1</td>
<td align="right">-3.808938</td>
<td align="right">3.873415</td>
<td align="right">-4.267731</td>
<td align="right">0.0000175732</td>
<td align="right">0.0030430983</td>
</tr>
<tr class="odd">
<td align="left">HRCT1</td>
<td align="right">-3.746575</td>
<td align="right">2.229948</td>
<td align="right">-4.263770</td>
<td align="right">0.0000178466</td>
<td align="right">0.0030443177</td>
</tr>
<tr class="even">
<td align="left">LINC02361</td>
<td align="right">-3.945348</td>
<td align="right">3.405743</td>
<td align="right">-4.247794</td>
<td align="right">0.0000190477</td>
<td align="right">0.0032014061</td>
</tr>
<tr class="odd">
<td align="left">CYP2F1</td>
<td align="right">2.846555</td>
<td align="right">2.502012</td>
<td align="right">4.240434</td>
<td align="right">0.0000195276</td>
<td align="right">0.0032345040</td>
</tr>
<tr class="even">
<td align="left">DNAJB5</td>
<td align="right">-3.848417</td>
<td align="right">3.153785</td>
<td align="right">-4.207334</td>
<td align="right">0.0000223530</td>
<td align="right">0.0036496009</td>
</tr>
<tr class="odd">
<td align="left">TSPAN18</td>
<td align="right">-3.869895</td>
<td align="right">2.411065</td>
<td align="right">-4.194942</td>
<td align="right">0.0000234800</td>
<td align="right">0.0037796132</td>
</tr>
<tr class="even">
<td align="left">ODAPH</td>
<td align="right">-3.588439</td>
<td align="right">2.233733</td>
<td align="right">-4.189976</td>
<td align="right">0.0000238994</td>
<td align="right">0.0037937054</td>
</tr>
<tr class="odd">
<td align="left">FSCN1</td>
<td align="right">-4.592688</td>
<td align="right">3.536904</td>
<td align="right">-4.181716</td>
<td align="right">0.0000249418</td>
<td align="right">0.0039049268</td>
</tr>
<tr class="even">
<td align="left">ZNF367</td>
<td align="right">-4.166974</td>
<td align="right">3.370324</td>
<td align="right">-4.171792</td>
<td align="right">0.0000257983</td>
<td align="right">0.0039844412</td>
</tr>
<tr class="odd">
<td align="left">SMIM24</td>
<td align="right">-3.565350</td>
<td align="right">2.338751</td>
<td align="right">-4.161582</td>
<td align="right">0.0000267221</td>
<td align="right">0.0040720963</td>
</tr>
<tr class="even">
<td align="left">IL21</td>
<td align="right">-3.434141</td>
<td align="right">2.205003</td>
<td align="right">-4.121802</td>
<td align="right">0.0000312009</td>
<td align="right">0.0046779427</td>
</tr>
<tr class="odd">
<td align="left">SCARB1</td>
<td align="right">-4.215130</td>
<td align="right">4.069015</td>
<td align="right">-4.120946</td>
<td align="right">0.0000315165</td>
<td align="right">0.0046779427</td>
</tr>
<tr class="even">
<td align="left">C11orf96</td>
<td align="right">-4.388398</td>
<td align="right">2.588931</td>
<td align="right">-4.081947</td>
<td align="right">0.0000367792</td>
<td align="right">0.0053891011</td>
</tr>
<tr class="odd">
<td align="left">COL13A1</td>
<td align="right">-3.583701</td>
<td align="right">2.261269</td>
<td align="right">-4.051926</td>
<td align="right">0.0000409499</td>
<td align="right">0.0059242549</td>
</tr>
<tr class="even">
<td align="left">TNXB</td>
<td align="right">-3.968113</td>
<td align="right">2.365336</td>
<td align="right">-4.044600</td>
<td align="right">0.0000422731</td>
<td align="right">0.0060110493</td>
</tr>
<tr class="odd">
<td align="left">RUNDC3A-AS1</td>
<td align="right">-3.566611</td>
<td align="right">2.333432</td>
<td align="right">-4.041629</td>
<td align="right">0.0000426017</td>
<td align="right">0.0060110493</td>
</tr>
<tr class="even">
<td align="left">MS4A2</td>
<td align="right">-3.625682</td>
<td align="right">2.509230</td>
<td align="right">-4.030846</td>
<td align="right">0.0000444258</td>
<td align="right">0.0061919793</td>
</tr>
<tr class="odd">
<td align="left">HIST1H2AD</td>
<td align="right">-3.443127</td>
<td align="right">2.170992</td>
<td align="right">-4.022974</td>
<td align="right">0.0000457287</td>
<td align="right">0.0062967917</td>
</tr>
<tr class="even">
<td align="left">MGP</td>
<td align="right">-3.107587</td>
<td align="right">2.429693</td>
<td align="right">-4.011484</td>
<td align="right">0.0000477052</td>
<td align="right">0.0064907437</td>
</tr>
<tr class="odd">
<td align="left">ENPP1</td>
<td align="right">-4.321531</td>
<td align="right">2.843752</td>
<td align="right">-3.980859</td>
<td align="right">0.0000542591</td>
<td align="right">0.0072215115</td>
</tr>
<tr class="even">
<td align="left">FAM153C</td>
<td align="right">-4.878368</td>
<td align="right">2.638307</td>
<td align="right">-3.983209</td>
<td align="right">0.0000543398</td>
<td align="right">0.0072215115</td>
</tr>
<tr class="odd">
<td align="left">SHE</td>
<td align="right">-3.410600</td>
<td align="right">2.299672</td>
<td align="right">-3.970206</td>
<td align="right">0.0000559224</td>
<td align="right">0.0073464036</td>
</tr>
<tr class="even">
<td align="left">ROBO4</td>
<td align="right">-3.422763</td>
<td align="right">2.249783</td>
<td align="right">-3.936650</td>
<td align="right">0.0000635099</td>
<td align="right">0.0082483516</td>
</tr>
<tr class="odd">
<td align="left">CEACAM21</td>
<td align="right">-4.362450</td>
<td align="right">4.669784</td>
<td align="right">-3.933843</td>
<td align="right">0.0000649462</td>
<td align="right">0.0083401122</td>
</tr>
<tr class="even">
<td align="left">CHRNA6</td>
<td align="right">-3.397408</td>
<td align="right">2.363886</td>
<td align="right">-3.923129</td>
<td align="right">0.0000668217</td>
<td align="right">0.0084856140</td>
</tr>
<tr class="odd">
<td align="left">LDLRAD3</td>
<td align="right">-3.871241</td>
<td align="right">3.414730</td>
<td align="right">-3.910838</td>
<td align="right">0.0000703216</td>
<td align="right">0.0087555014</td>
</tr>
<tr class="even">
<td align="left">TPSD1</td>
<td align="right">-3.597757</td>
<td align="right">2.197147</td>
<td align="right">-3.905326</td>
<td align="right">0.0000715756</td>
<td align="right">0.0087555014</td>
</tr>
<tr class="odd">
<td align="left">MND1</td>
<td align="right">-4.585170</td>
<td align="right">3.348023</td>
<td align="right">-3.907916</td>
<td align="right">0.0000719559</td>
<td align="right">0.0087555014</td>
</tr>
<tr class="even">
<td align="left">SCGB3A2</td>
<td align="right">-3.653242</td>
<td align="right">2.268346</td>
<td align="right">-3.903861</td>
<td align="right">0.0000720113</td>
<td align="right">0.0087555014</td>
</tr>
<tr class="odd">
<td align="left">BMP6</td>
<td align="right">-3.561418</td>
<td align="right">2.348615</td>
<td align="right">-3.899621</td>
<td align="right">0.0000731010</td>
<td align="right">0.0087647657</td>
</tr>
<tr class="even">
<td align="left">RBP5</td>
<td align="right">-3.425338</td>
<td align="right">2.239152</td>
<td align="right">-3.894695</td>
<td align="right">0.0000743764</td>
<td align="right">0.0087647657</td>
</tr>
<tr class="odd">
<td align="left">F2RL3</td>
<td align="right">-3.418203</td>
<td align="right">2.268544</td>
<td align="right">-3.894638</td>
<td align="right">0.0000743882</td>
<td align="right">0.0087647657</td>
</tr>
<tr class="even">
<td align="left">HIST1H3F</td>
<td align="right">-3.631802</td>
<td align="right">2.200883</td>
<td align="right">-3.873605</td>
<td align="right">0.0000806449</td>
<td align="right">0.0092599902</td>
</tr>
<tr class="odd">
<td align="left">LINC00243</td>
<td align="right">-3.942871</td>
<td align="right">2.594630</td>
<td align="right">-3.874120</td>
<td align="right">0.0000808043</td>
<td align="right">0.0092599902</td>
</tr>
<tr class="even">
<td align="left">SEMA3G</td>
<td align="right">-3.417506</td>
<td align="right">2.271409</td>
<td align="right">-3.871811</td>
<td align="right">0.0000810219</td>
<td align="right">0.0092599902</td>
</tr>
<tr class="odd">
<td align="left">OTUD7A</td>
<td align="right">-3.636303</td>
<td align="right">2.734260</td>
<td align="right">-3.849351</td>
<td align="right">0.0000882891</td>
<td align="right">0.0099906595</td>
</tr>
<tr class="even">
<td align="left">CCNE1</td>
<td align="right">-3.869029</td>
<td align="right">3.327947</td>
<td align="right">-3.833026</td>
<td align="right">0.0000940988</td>
<td align="right">0.0105436806</td>
</tr>
<tr class="odd">
<td align="left">SIGLEC8</td>
<td align="right">-3.444464</td>
<td align="right">2.165902</td>
<td align="right">-3.826778</td>
<td align="right">0.0000958215</td>
<td align="right">0.0106324629</td>
</tr>
<tr class="even">
<td align="left">RNF165</td>
<td align="right">-3.731399</td>
<td align="right">2.579006</td>
<td align="right">-3.814675</td>
<td align="right">0.0001005389</td>
<td align="right">0.0110486455</td>
</tr>
<tr class="odd">
<td align="left">HOXB7</td>
<td align="right">-3.936474</td>
<td align="right">3.398998</td>
<td align="right">-3.794024</td>
<td align="right">0.0001088449</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="even">
<td align="left">GPR15</td>
<td align="right">-3.438546</td>
<td align="right">2.188404</td>
<td align="right">-3.790833</td>
<td align="right">0.0001094186</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="odd">
<td align="left">GCSAML</td>
<td align="right">-3.403968</td>
<td align="right">2.331022</td>
<td align="right">-3.783661</td>
<td align="right">0.0001123053</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="even">
<td align="left">ZNF385D</td>
<td align="right">-4.122761</td>
<td align="right">2.422526</td>
<td align="right">-3.785466</td>
<td align="right">0.0001127056</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="odd">
<td align="left">ADAMTSL2</td>
<td align="right">-3.382429</td>
<td align="right">2.444270</td>
<td align="right">-3.780134</td>
<td align="right">0.0001137467</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="even">
<td align="left">LINC02207</td>
<td align="right">-3.378364</td>
<td align="right">2.466286</td>
<td align="right">-3.779571</td>
<td align="right">0.0001139777</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="odd">
<td align="left">PPP1R14A</td>
<td align="right">-4.284556</td>
<td align="right">3.416767</td>
<td align="right">-3.782080</td>
<td align="right">0.0001144875</td>
<td align="right">0.0117880867</td>
</tr>
<tr class="even">
<td align="left">HIST1H2AH</td>
<td align="right">-3.856943</td>
<td align="right">2.458824</td>
<td align="right">-3.776608</td>
<td align="right">0.0001159185</td>
<td align="right">0.0118288579</td>
</tr>
<tr class="odd">
<td align="left">HIST1H2BL</td>
<td align="right">-3.446301</td>
<td align="right">2.158908</td>
<td align="right">-3.765134</td>
<td align="right">0.0001202585</td>
<td align="right">0.0121631375</td>
</tr>
<tr class="even">
<td align="left">TRBV3-1</td>
<td align="right">-3.434238</td>
<td align="right">2.204638</td>
<td align="right">-3.758762</td>
<td align="right">0.0001230820</td>
<td align="right">0.0123395072</td>
</tr>
<tr class="odd">
<td align="left">SNHG22</td>
<td align="right">-3.335338</td>
<td align="right">2.754604</td>
<td align="right">-3.756088</td>
<td align="right">0.0001241692</td>
<td align="right">0.0123402610</td>
</tr>
<tr class="even">
<td align="left">KCNA2</td>
<td align="right">-3.383515</td>
<td align="right">2.438480</td>
<td align="right">-3.747645</td>
<td align="right">0.0001281191</td>
<td align="right">0.0126230409</td>
</tr>
<tr class="odd">
<td align="left">SPTB</td>
<td align="right">-3.425135</td>
<td align="right">2.239991</td>
<td align="right">-3.739329</td>
<td align="right">0.0001321243</td>
<td align="right">0.0128378464</td>
</tr>
<tr class="even">
<td align="left">PDE2A</td>
<td align="right">-3.723316</td>
<td align="right">2.545712</td>
<td align="right">-3.738674</td>
<td align="right">0.0001329473</td>
<td align="right">0.0128378464</td>
</tr>
<tr class="odd">
<td align="left">DLGAP3</td>
<td align="right">-3.632197</td>
<td align="right">2.759290</td>
<td align="right">-3.735277</td>
<td align="right">0.0001344317</td>
<td align="right">0.0128378464</td>
</tr>
<tr class="even">
<td align="left">ELOVL6</td>
<td align="right">-4.258504</td>
<td align="right">3.114868</td>
<td align="right">-3.735828</td>
<td align="right">0.0001356581</td>
<td align="right">0.0128378464</td>
</tr>
<tr class="odd">
<td align="left">SPARCL1</td>
<td align="right">-2.575145</td>
<td align="right">2.322192</td>
<td align="right">-3.730138</td>
<td align="right">0.0001359156</td>
<td align="right">0.0128378464</td>
</tr>
<tr class="even">
<td align="left">KRT1</td>
<td align="right">-3.438745</td>
<td align="right">2.187645</td>
<td align="right">-3.726681</td>
<td align="right">0.0001383737</td>
<td align="right">0.0129628891</td>
</tr>
<tr class="odd">
<td align="left">FAM111B</td>
<td align="right">-4.278732</td>
<td align="right">3.575654</td>
<td align="right">-3.723946</td>
<td align="right">0.0001417530</td>
<td align="right">0.0131698821</td>
</tr>
<tr class="even">
<td align="left">KRT17</td>
<td align="right">2.795374</td>
<td align="right">2.415781</td>
<td align="right">3.716519</td>
<td align="right">0.0001428879</td>
<td align="right">0.0131698821</td>
</tr>
<tr class="odd">
<td align="left">NTM</td>
<td align="right">-3.425606</td>
<td align="right">2.238047</td>
<td align="right">-3.710158</td>
<td align="right">0.0001469111</td>
<td align="right">0.0134323783</td>
</tr>
<tr class="even">
<td align="left">SCNN1B</td>
<td align="right">2.437361</td>
<td align="right">2.349556</td>
<td align="right">3.706327</td>
<td align="right">0.0001481139</td>
<td align="right">0.0134348731</td>
</tr>
<tr class="odd">
<td align="left">A1BG-AS1</td>
<td align="right">-3.727817</td>
<td align="right">3.911770</td>
<td align="right">-3.703780</td>
<td align="right">0.0001509688</td>
<td align="right">0.0135090667</td>
</tr>
<tr class="even">
<td align="left">HIST2H2AB</td>
<td align="right">-3.758429</td>
<td align="right">2.360336</td>
<td align="right">-3.703318</td>
<td align="right">0.0001512959</td>
<td align="right">0.0135090667</td>
</tr>
<tr class="odd">
<td align="left">CACNA2D4</td>
<td align="right">-3.848260</td>
<td align="right">4.103024</td>
<td align="right">-3.699417</td>
<td align="right">0.0001536855</td>
<td align="right">0.0136160571</td>
</tr>
<tr class="even">
<td align="left">ERVFRD-1</td>
<td align="right">-3.411797</td>
<td align="right">2.294797</td>
<td align="right">-3.682797</td>
<td align="right">0.0001621699</td>
<td align="right">0.0141911544</td>
</tr>
<tr class="odd">
<td align="left">PTPRB</td>
<td align="right">-3.576703</td>
<td align="right">2.283668</td>
<td align="right">-3.682556</td>
<td align="right">0.0001626600</td>
<td align="right">0.0141911544</td>
</tr>
<tr class="even">
<td align="left">MLXIPL</td>
<td align="right">-3.555057</td>
<td align="right">2.386265</td>
<td align="right">-3.676781</td>
<td align="right">0.0001660356</td>
<td align="right">0.0143623165</td>
</tr>
<tr class="odd">
<td align="left">MAL</td>
<td align="right">4.097729</td>
<td align="right">3.544483</td>
<td align="right">3.677616</td>
<td align="right">0.0001671352</td>
<td align="right">0.0143623165</td>
</tr>
<tr class="even">
<td align="left">MCM10</td>
<td align="right">-4.098110</td>
<td align="right">2.998359</td>
<td align="right">-3.675024</td>
<td align="right">0.0001687127</td>
<td align="right">0.0143896792</td>
</tr>
<tr class="odd">
<td align="left">GATA2</td>
<td align="right">-3.548613</td>
<td align="right">2.416722</td>
<td align="right">-3.668165</td>
<td align="right">0.0001712573</td>
<td align="right">0.0144069077</td>
</tr>
<tr class="even">
<td align="left">MIR9-3HG</td>
<td align="right">-3.657762</td>
<td align="right">2.457287</td>
<td align="right">-3.666218</td>
<td align="right">0.0001727434</td>
<td align="right">0.0144069077</td>
</tr>
<tr class="odd">
<td align="left">HIST1H3A</td>
<td align="right">-3.690327</td>
<td align="right">2.533430</td>
<td align="right">-3.666084</td>
<td align="right">0.0001729168</td>
<td align="right">0.0144069077</td>
</tr>
<tr class="even">
<td align="left">SERPINB10</td>
<td align="right">-3.275051</td>
<td align="right">2.295762</td>
<td align="right">-3.662883</td>
<td align="right">0.0001739569</td>
<td align="right">0.0144069077</td>
</tr>
<tr class="odd">
<td align="left">PDE1B</td>
<td align="right">-3.511094</td>
<td align="right">3.992173</td>
<td align="right">-3.638450</td>
<td align="right">0.0001904402</td>
<td align="right">0.0155844557</td>
</tr>
<tr class="even">
<td align="left">TNFSF11</td>
<td align="right">-3.723836</td>
<td align="right">2.612865</td>
<td align="right">-3.635935</td>
<td align="right">0.0001928155</td>
<td align="right">0.0155844557</td>
</tr>
<tr class="odd">
<td align="left">HIST1H2AK</td>
<td align="right">-3.705617</td>
<td align="right">2.497142</td>
<td align="right">-3.634483</td>
<td align="right">0.0001937601</td>
<td align="right">0.0155844557</td>
</tr>
<tr class="even">
<td align="left">JAM2</td>
<td align="right">-3.551157</td>
<td align="right">2.396931</td>
<td align="right">-3.632812</td>
<td align="right">0.0001944370</td>
<td align="right">0.0155844557</td>
</tr>
<tr class="odd">
<td align="left">PRG4</td>
<td align="right">-3.440394</td>
<td align="right">2.181381</td>
<td align="right">-3.631572</td>
<td align="right">0.0001949932</td>
<td align="right">0.0155844557</td>
</tr>
<tr class="even">
<td align="left">SIGLEC7</td>
<td align="right">-3.779825</td>
<td align="right">3.990483</td>
<td align="right">-3.619679</td>
<td align="right">0.0002045767</td>
<td align="right">0.0162154808</td>
</tr>
<tr class="odd">
<td align="left">DNASE1L3</td>
<td align="right">-3.867084</td>
<td align="right">3.200887</td>
<td align="right">-3.618584</td>
<td align="right">0.0002057262</td>
<td align="right">0.0162154808</td>
</tr>
<tr class="even">
<td align="left">ZNF366</td>
<td align="right">-3.600733</td>
<td align="right">3.045439</td>
<td align="right">-3.601436</td>
<td align="right">0.0002176403</td>
<td align="right">0.0169266832</td>
</tr>
<tr class="odd">
<td align="left">NKD1</td>
<td align="right">-3.763226</td>
<td align="right">2.911158</td>
<td align="right">-3.602157</td>
<td align="right">0.0002177113</td>
<td align="right">0.0169266832</td>
</tr>
<tr class="even">
<td align="left">ELN</td>
<td align="right">-2.635520</td>
<td align="right">2.209570</td>
<td align="right">-3.584898</td>
<td align="right">0.0002285849</td>
<td align="right">0.0175991382</td>
</tr>
<tr class="odd">
<td align="left">CDC45</td>
<td align="right">-3.963538</td>
<td align="right">3.041343</td>
<td align="right">-3.588585</td>
<td align="right">0.0002294402</td>
<td align="right">0.0175991382</td>
</tr>
<tr class="even">
<td align="left">CDHR1</td>
<td align="right">-3.426327</td>
<td align="right">2.235068</td>
<td align="right">-3.581638</td>
<td align="right">0.0002328495</td>
<td align="right">0.0176876393</td>
</tr>
<tr class="odd">
<td align="left">GATD3B</td>
<td align="right">-4.907991</td>
<td align="right">3.025364</td>
<td align="right">-3.590798</td>
<td align="right">0.0002336892</td>
<td align="right">0.0176876393</td>
</tr>
<tr class="even">
<td align="left">GNRH1</td>
<td align="right">-3.603027</td>
<td align="right">2.941175</td>
<td align="right">-3.570766</td>
<td align="right">0.0002426692</td>
<td align="right">0.0182464851</td>
</tr>
<tr class="odd">
<td align="left">AGRP</td>
<td align="right">-3.721460</td>
<td align="right">3.918619</td>
<td align="right">-3.567792</td>
<td align="right">0.0002457726</td>
<td align="right">0.0183590532</td>
</tr>
<tr class="even">
<td align="left">ANKDD1A</td>
<td align="right">-4.074248</td>
<td align="right">3.712936</td>
<td align="right">-3.550006</td>
<td align="right">0.0002638012</td>
<td align="right">0.0194806212</td>
</tr>
<tr class="odd">
<td align="left">TRBV14</td>
<td align="right">-3.432535</td>
<td align="right">2.211044</td>
<td align="right">-3.545867</td>
<td align="right">0.0002641960</td>
<td align="right">0.0194806212</td>
</tr>
<tr class="even">
<td align="left">CYYR1</td>
<td align="right">-3.555012</td>
<td align="right">2.469857</td>
<td align="right">-3.538741</td>
<td align="right">0.0002714513</td>
<td align="right">0.0198872896</td>
</tr>
<tr class="odd">
<td align="left">RTN4RL2</td>
<td align="right">-3.548286</td>
<td align="right">2.504082</td>
<td align="right">-3.532477</td>
<td align="right">0.0002774575</td>
<td align="right">0.0201588878</td>
</tr>
<tr class="even">
<td align="left">HAND2-AS1</td>
<td align="right">-3.399740</td>
<td align="right">2.351602</td>
<td align="right">-3.530512</td>
<td align="right">0.0002786862</td>
<td align="right">0.0201588878</td>
</tr>
<tr class="odd">
<td align="left">NTS</td>
<td align="right">-3.085865</td>
<td align="right">2.234923</td>
<td align="right">-3.516101</td>
<td align="right">0.0002918781</td>
<td align="right">0.0209803433</td>
</tr>
<tr class="even">
<td align="left">HIST1H2AL</td>
<td align="right">-3.966698</td>
<td align="right">2.600144</td>
<td align="right">-3.511839</td>
<td align="right">0.0003009792</td>
<td align="right">0.0214993198</td>
</tr>
<tr class="odd">
<td align="left">JPH4</td>
<td align="right">-3.492134</td>
<td align="right">3.216389</td>
<td align="right">-3.504094</td>
<td align="right">0.0003061312</td>
<td align="right">0.0217200105</td>
</tr>
<tr class="even">
<td align="left">TRAV13-2</td>
<td align="right">-3.387471</td>
<td align="right">2.417258</td>
<td align="right">-3.501962</td>
<td align="right">0.0003078696</td>
<td align="right">0.0217200105</td>
</tr>
<tr class="odd">
<td align="left">ST6GALNAC3</td>
<td align="right">-3.349126</td>
<td align="right">2.652197</td>
<td align="right">-3.498189</td>
<td align="right">0.0003117532</td>
<td align="right">0.0218590647</td>
</tr>
<tr class="even">
<td align="left">PRR19</td>
<td align="right">-3.575303</td>
<td align="right">2.646519</td>
<td align="right">-3.496879</td>
<td align="right">0.0003144180</td>
<td align="right">0.0219114837</td>
</tr>
<tr class="odd">
<td align="left">CREB3L3</td>
<td align="right">-3.394446</td>
<td align="right">2.379804</td>
<td align="right">-3.483787</td>
<td align="right">0.0003280041</td>
<td align="right">0.0227197513</td>
</tr>
<tr class="even">
<td align="left">DLC1</td>
<td align="right">-3.406469</td>
<td align="right">3.329316</td>
<td align="right">-3.471149</td>
<td align="right">0.0003427734</td>
<td align="right">0.0235997443</td>
</tr>
<tr class="odd">
<td align="left">MAPK12</td>
<td align="right">-3.909657</td>
<td align="right">2.724970</td>
<td align="right">-3.467040</td>
<td align="right">0.0003514815</td>
<td align="right">0.0240072471</td>
</tr>
<tr class="even">
<td align="left">S1PR5</td>
<td align="right">-3.618556</td>
<td align="right">2.234865</td>
<td align="right">-3.463930</td>
<td align="right">0.0003528933</td>
<td align="right">0.0240072471</td>
</tr>
<tr class="odd">
<td align="left">FCGBP</td>
<td align="right">3.302150</td>
<td align="right">3.362444</td>
<td align="right">3.451576</td>
<td align="right">0.0003661292</td>
<td align="right">0.0247602997</td>
</tr>
<tr class="even">
<td align="left">BTNL9</td>
<td align="right">-3.139993</td>
<td align="right">2.474799</td>
<td align="right">-3.438497</td>
<td align="right">0.0003820616</td>
<td align="right">0.0256857771</td>
</tr>
<tr class="odd">
<td align="left">ZBTB20-AS5</td>
<td align="right">-3.969969</td>
<td align="right">2.501968</td>
<td align="right">-3.441266</td>
<td align="right">0.0003849933</td>
<td align="right">0.0257315108</td>
</tr>
<tr class="even">
<td align="left">S100A14</td>
<td align="right">2.233740</td>
<td align="right">2.566960</td>
<td align="right">3.431894</td>
<td align="right">0.0003883758</td>
<td align="right">0.0257370294</td>
</tr>
<tr class="odd">
<td align="left">KIR2DL1</td>
<td align="right">-3.429234</td>
<td align="right">2.223448</td>
<td align="right">-3.434197</td>
<td align="right">0.0003895797</td>
<td align="right">0.0257370294</td>
</tr>
<tr class="even">
<td align="left">DCHS1</td>
<td align="right">-3.493574</td>
<td align="right">2.711362</td>
<td align="right">-3.421589</td>
<td align="right">0.0004073494</td>
<td align="right">0.0267562991</td>
</tr>
<tr class="odd">
<td align="left">LINC00299</td>
<td align="right">-4.101281</td>
<td align="right">2.841865</td>
<td align="right">-3.422839</td>
<td align="right">0.0004118833</td>
<td align="right">0.0268995088</td>
</tr>
<tr class="even">
<td align="left">LRRC32</td>
<td align="right">-3.410879</td>
<td align="right">2.298535</td>
<td align="right">-3.415530</td>
<td align="right">0.0004152262</td>
<td align="right">0.0269637532</td>
</tr>
<tr class="odd">
<td align="left">CEACAM6</td>
<td align="right">2.042031</td>
<td align="right">2.445749</td>
<td align="right">3.407180</td>
<td align="right">0.0004223875</td>
<td align="right">0.0272738255</td>
</tr>
<tr class="even">
<td align="left">UPK1B</td>
<td align="right">2.375657</td>
<td align="right">2.564542</td>
<td align="right">3.386655</td>
<td align="right">0.0004530961</td>
<td align="right">0.0290923336</td>
</tr>
<tr class="odd">
<td align="left">HAS1</td>
<td align="right">-3.426417</td>
<td align="right">2.234694</td>
<td align="right">-3.383726</td>
<td align="right">0.0004630352</td>
<td align="right">0.0295644109</td>
</tr>
<tr class="even">
<td align="left">MEFV</td>
<td align="right">-3.532760</td>
<td align="right">3.413281</td>
<td align="right">-3.380549</td>
<td align="right">0.0004691468</td>
<td align="right">0.0296965561</td>
</tr>
<tr class="odd">
<td align="left">PALD1</td>
<td align="right">-3.499259</td>
<td align="right">3.672849</td>
<td align="right">-3.379609</td>
<td align="right">0.0004703016</td>
<td align="right">0.0296965561</td>
</tr>
<tr class="even">
<td align="left">TSPAN8</td>
<td align="right">2.461516</td>
<td align="right">2.492056</td>
<td align="right">3.370572</td>
<td align="right">0.0004785523</td>
<td align="right">0.0300515058</td>
</tr>
<tr class="odd">
<td align="left">HOXB6</td>
<td align="right">-3.803065</td>
<td align="right">3.507831</td>
<td align="right">-3.369468</td>
<td align="right">0.0004905151</td>
<td align="right">0.0306344080</td>
</tr>
<tr class="even">
<td align="left">SNCG</td>
<td align="right">-2.341894</td>
<td align="right">2.207711</td>
<td align="right">-3.357569</td>
<td align="right">0.0004997489</td>
<td align="right">0.0310357519</td>
</tr>
<tr class="odd">
<td align="left">RUBCNL</td>
<td align="right">-3.356717</td>
<td align="right">3.811752</td>
<td align="right">-3.359306</td>
<td align="right">0.0005023724</td>
<td align="right">0.0310357519</td>
</tr>
<tr class="even">
<td align="left">TMEM255B</td>
<td align="right">-3.606099</td>
<td align="right">3.632980</td>
<td align="right">-3.353941</td>
<td align="right">0.0005144657</td>
<td align="right">0.0316119790</td>
</tr>
<tr class="odd">
<td align="left">NFE2</td>
<td align="right">-3.626578</td>
<td align="right">3.479323</td>
<td align="right">-3.349460</td>
<td align="right">0.0005226124</td>
<td align="right">0.0317756816</td>
</tr>
<tr class="even">
<td align="left">MYBL1</td>
<td align="right">-3.946295</td>
<td align="right">4.353361</td>
<td align="right">-3.352039</td>
<td align="right">0.0005226904</td>
<td align="right">0.0317756816</td>
</tr>
<tr class="odd">
<td align="left">CLEC4F</td>
<td align="right">-3.485441</td>
<td align="right">2.777726</td>
<td align="right">-3.345634</td>
<td align="right">0.0005276295</td>
<td align="right">0.0318646444</td>
</tr>
<tr class="even">
<td align="left">ADCY4</td>
<td align="right">-3.451932</td>
<td align="right">3.142176</td>
<td align="right">-3.344234</td>
<td align="right">0.0005297298</td>
<td align="right">0.0318646444</td>
</tr>
<tr class="odd">
<td align="left">C1orf61</td>
<td align="right">-3.591653</td>
<td align="right">2.405603</td>
<td align="right">-3.341119</td>
<td align="right">0.0005371175</td>
<td align="right">0.0321362004</td>
</tr>
<tr class="even">
<td align="left">CD300LB</td>
<td align="right">-3.514285</td>
<td align="right">3.024992</td>
<td align="right">-3.339051</td>
<td align="right">0.0005398679</td>
<td align="right">0.0321362004</td>
</tr>
<tr class="odd">
<td align="left">LINC00426</td>
<td align="right">-3.647023</td>
<td align="right">3.453340</td>
<td align="right">-3.333240</td>
<td align="right">0.0005524067</td>
<td align="right">0.0327122074</td>
</tr>
<tr class="even">
<td align="left">NNMT</td>
<td align="right">-3.958854</td>
<td align="right">2.619793</td>
<td align="right">-3.331459</td>
<td align="right">0.0005607713</td>
<td align="right">0.0330363647</td>
</tr>
<tr class="odd">
<td align="left">MYBL2</td>
<td align="right">-4.138462</td>
<td align="right">3.371776</td>
<td align="right">-3.331091</td>
<td align="right">0.0005647638</td>
<td align="right">0.0331009528</td>
</tr>
<tr class="even">
<td align="left">IFNL1</td>
<td align="right">-3.373214</td>
<td align="right">2.496936</td>
<td align="right">-3.321027</td>
<td align="right">0.0005718475</td>
<td align="right">0.0333451272</td>
</tr>
<tr class="odd">
<td align="left">LINC02019</td>
<td align="right">-3.510167</td>
<td align="right">3.250600</td>
<td align="right">-3.319739</td>
<td align="right">0.0005761489</td>
<td align="right">0.0334063829</td>
</tr>
<tr class="even">
<td align="left">PDE10A</td>
<td align="right">-2.095157</td>
<td align="right">2.173568</td>
<td align="right">-3.313093</td>
<td align="right">0.0005795970</td>
<td align="right">0.0334063829</td>
</tr>
<tr class="odd">
<td align="left">ZNF521</td>
<td align="right">-3.623781</td>
<td align="right">2.402447</td>
<td align="right">-3.317766</td>
<td align="right">0.0005816668</td>
<td align="right">0.0334063829</td>
</tr>
<tr class="even">
<td align="left">CADPS</td>
<td align="right">-3.415807</td>
<td align="right">2.278392</td>
<td align="right">-3.307856</td>
<td align="right">0.0005982812</td>
<td align="right">0.0341887796</td>
</tr>
<tr class="odd">
<td align="left">IL2</td>
<td align="right">-3.368624</td>
<td align="right">2.524972</td>
<td align="right">-3.300071</td>
<td align="right">0.0006134375</td>
<td align="right">0.0348804856</td>
</tr>
<tr class="even">
<td align="left">GIMAP8</td>
<td align="right">-3.659040</td>
<td align="right">4.120120</td>
<td align="right">-3.293549</td>
<td align="right">0.0006316163</td>
<td align="right">0.0357363494</td>
</tr>
<tr class="odd">
<td align="left">UTS2</td>
<td align="right">-3.340562</td>
<td align="right">2.713501</td>
<td align="right">-3.288983</td>
<td align="right">0.0006362070</td>
<td align="right">0.0358187679</td>
</tr>
<tr class="even">
<td align="left">PVALB</td>
<td align="right">-3.438194</td>
<td align="right">3.133083</td>
<td align="right">-3.280409</td>
<td align="right">0.0006561790</td>
<td align="right">0.0367621078</td>
</tr>
<tr class="odd">
<td align="left">CXCL5</td>
<td align="right">-4.553403</td>
<td align="right">3.417964</td>
<td align="right">-3.285015</td>
<td align="right">0.0006705053</td>
<td align="right">0.0373814873</td>
</tr>
<tr class="even">
<td align="left">ADGRG3</td>
<td align="right">-3.616115</td>
<td align="right">2.426386</td>
<td align="right">-3.273119</td>
<td align="right">0.0006755039</td>
<td align="right">0.0374629315</td>
</tr>
<tr class="odd">
<td align="left">RHBDL3</td>
<td align="right">-2.614819</td>
<td align="right">2.375263</td>
<td align="right">-3.264802</td>
<td align="right">0.0006817225</td>
<td align="right">0.0374629315</td>
</tr>
<tr class="even">
<td align="left">PIEZO2</td>
<td align="right">-2.729491</td>
<td align="right">2.328162</td>
<td align="right">-3.264239</td>
<td align="right">0.0006837436</td>
<td align="right">0.0374629315</td>
</tr>
<tr class="odd">
<td align="left">PLXNA4</td>
<td align="right">-3.385932</td>
<td align="right">2.425513</td>
<td align="right">-3.267092</td>
<td align="right">0.0006850777</td>
<td align="right">0.0374629315</td>
</tr>
<tr class="even">
<td align="left">A1BG</td>
<td align="right">-3.661520</td>
<td align="right">5.629545</td>
<td align="right">-3.266546</td>
<td align="right">0.0006913891</td>
<td align="right">0.0376280300</td>
</tr>
<tr class="odd">
<td align="left">PRSS27</td>
<td align="right">2.770486</td>
<td align="right">2.925203</td>
<td align="right">3.258676</td>
<td align="right">0.0006967349</td>
<td align="right">0.0377392564</td>
</tr>
<tr class="even">
<td align="left">SLC18A2</td>
<td align="right">-3.379081</td>
<td align="right">2.462275</td>
<td align="right">-3.255318</td>
<td align="right">0.0007122874</td>
<td align="right">0.0383152109</td>
</tr>
<tr class="odd">
<td align="left">IL21R-AS1</td>
<td align="right">-3.565626</td>
<td align="right">2.459198</td>
<td align="right">-3.256034</td>
<td align="right">0.0007140730</td>
<td align="right">0.0383152109</td>
</tr>
<tr class="even">
<td align="left">CCM2L</td>
<td align="right">-3.540994</td>
<td align="right">2.452538</td>
<td align="right">-3.252761</td>
<td align="right">0.0007213928</td>
<td align="right">0.0385270971</td>
</tr>
<tr class="odd">
<td align="left">SUCNR1</td>
<td align="right">-3.829894</td>
<td align="right">4.123454</td>
<td align="right">-3.250128</td>
<td align="right">0.0007341227</td>
<td align="right">0.0389025682</td>
</tr>
<tr class="even">
<td align="left">OLFM1</td>
<td align="right">-3.756577</td>
<td align="right">2.575758</td>
<td align="right">-3.247882</td>
<td align="right">0.0007378939</td>
<td align="right">0.0389025682</td>
</tr>
<tr class="odd">
<td align="left">SEMA6A</td>
<td align="right">-2.676515</td>
<td align="right">2.451847</td>
<td align="right">-3.238141</td>
<td align="right">0.0007446749</td>
<td align="right">0.0389025682</td>
</tr>
<tr class="even">
<td align="left">LINC00996</td>
<td align="right">-3.738398</td>
<td align="right">4.099541</td>
<td align="right">-3.244725</td>
<td align="right">0.0007452598</td>
<td align="right">0.0389025682</td>
</tr>
<tr class="odd">
<td align="left">NID1</td>
<td align="right">-3.447254</td>
<td align="right">3.068001</td>
<td align="right">-3.242114</td>
<td align="right">0.0007454425</td>
<td align="right">0.0389025682</td>
</tr>
<tr class="even">
<td align="left">CYSLTR2</td>
<td align="right">-3.494900</td>
<td align="right">3.570216</td>
<td align="right">-3.239509</td>
<td align="right">0.0007528611</td>
<td align="right">0.0391111331</td>
</tr>
<tr class="odd">
<td align="left">SLC24A3</td>
<td align="right">-2.428559</td>
<td align="right">2.225369</td>
<td align="right">-3.231260</td>
<td align="right">0.0007601262</td>
<td align="right">0.0393098774</td>
</tr>
<tr class="even">
<td align="left">LINC01480</td>
<td align="right">-3.609508</td>
<td align="right">3.341788</td>
<td align="right">-3.227660</td>
<td align="right">0.0007855530</td>
<td align="right">0.0403126833</td>
</tr>
<tr class="odd">
<td align="left">GTSF1</td>
<td align="right">-3.548499</td>
<td align="right">4.002530</td>
<td align="right">-3.226726</td>
<td align="right">0.0007865717</td>
<td align="right">0.0403126833</td>
</tr>
<tr class="even">
<td align="left">P2RX1</td>
<td align="right">-3.434184</td>
<td align="right">3.920735</td>
<td align="right">-3.222732</td>
<td align="right">0.0007944813</td>
<td align="right">0.0405362785</td>
</tr>
<tr class="odd">
<td align="left">CD5</td>
<td align="right">-3.616686</td>
<td align="right">4.162880</td>
<td align="right">-3.218127</td>
<td align="right">0.0008108823</td>
<td align="right">0.0411892194</td>
</tr>
<tr class="even">
<td align="left">PMCH</td>
<td align="right">-3.418119</td>
<td align="right">2.268888</td>
<td align="right">-3.212153</td>
<td align="right">0.0008223056</td>
<td align="right">0.0415846506</td>
</tr>
<tr class="odd">
<td align="left">TRDV1</td>
<td align="right">-3.396433</td>
<td align="right">2.369123</td>
<td align="right">-3.209035</td>
<td align="right">0.0008302994</td>
<td align="right">0.0418039270</td>
</tr>
<tr class="even">
<td align="left">ADH1C</td>
<td align="right">2.557753</td>
<td align="right">2.513190</td>
<td align="right">3.199618</td>
<td align="right">0.0008435804</td>
<td align="right">0.0422863163</td>
</tr>
<tr class="odd">
<td align="left">PRICKLE1</td>
<td align="right">-3.920465</td>
<td align="right">2.745075</td>
<td align="right">-3.204098</td>
<td align="right">0.0008580129</td>
<td align="right">0.0428219604</td>
</tr>
<tr class="even">
<td align="left">PSMA8</td>
<td align="right">-3.358787</td>
<td align="right">2.585598</td>
<td align="right">-3.188381</td>
<td align="right">0.0008876011</td>
<td align="right">0.0441060542</td>
</tr>
<tr class="odd">
<td align="left">SCT</td>
<td align="right">-3.356265</td>
<td align="right">2.602384</td>
<td align="right">-3.183964</td>
<td align="right">0.0009004539</td>
<td align="right">0.0445510308</td>
</tr>
<tr class="even">
<td align="left">KIR2DL3</td>
<td align="right">-3.382241</td>
<td align="right">2.445274</td>
<td align="right">-3.179154</td>
<td align="right">0.0009153687</td>
<td align="right">0.0450937428</td>
</tr>
<tr class="odd">
<td align="left">FUT7</td>
<td align="right">-3.454371</td>
<td align="right">3.902375</td>
<td align="right">-3.177358</td>
<td align="right">0.0009226442</td>
<td align="right">0.0452570854</td>
</tr>
<tr class="even">
<td align="left">ILDR2</td>
<td align="right">-3.365178</td>
<td align="right">2.546048</td>
<td align="right">-3.168907</td>
<td align="right">0.0009460447</td>
<td align="right">0.0462066021</td>
</tr>
<tr class="odd">
<td align="left">SIGLEC14</td>
<td align="right">-3.430643</td>
<td align="right">4.069498</td>
<td align="right">-3.167501</td>
<td align="right">0.0009521633</td>
<td align="right">0.0463075505</td>
</tr>
<tr class="even">
<td align="left">PLAT</td>
<td align="right">2.909000</td>
<td align="right">3.052799</td>
<td align="right">3.162225</td>
<td align="right">0.0009567766</td>
<td align="right">0.0463347448</td>
</tr>
<tr class="odd">
<td align="left">ITGB1-DT</td>
<td align="right">-3.993028</td>
<td align="right">2.513560</td>
<td align="right">-3.153847</td>
<td align="right">0.0010151613</td>
<td align="right">0.0489547595</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(top, file = glue(&quot;{dirName}/DEGs-{colnames(cont)[6]}.csv&quot;))</code></pre>
<pre class="r"><code>go &lt;- gsaseq(ann$ENTREZID[ann$SYMBOL %in% rownames(top)], 
             universe = ann$ENTREZID, 
             collection = gsAnnots$c5@original,
             gene.length = ann$Length)

topGSA(react) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="67%" />
<col width="4%" />
<col width="3%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">REACTOME_O_LINKED_GLYCOSYLATION_OF_MUCINS</td>
<td align="right">55</td>
<td align="right">6</td>
<td align="right">0.0000002</td>
<td align="right">0.0001498</td>
</tr>
<tr class="even">
<td align="left">REACTOME_TERMINATION_OF_O_GLYCAN_BIOSYNTHESIS</td>
<td align="right">21</td>
<td align="right">4</td>
<td align="right">0.0000020</td>
<td align="right">0.0006848</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_PEPTIDE_LIGAND_BINDING_RECEPTORS</td>
<td align="right">184</td>
<td align="right">7</td>
<td align="right">0.0000224</td>
<td align="right">0.0041716</td>
</tr>
<tr class="even">
<td align="left">REACTOME_POST_TRANSLATIONAL_PROTEIN_MODIFICATION</td>
<td align="right">183</td>
<td align="right">7</td>
<td align="right">0.0000248</td>
<td align="right">0.0041716</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CHEMOKINE_RECEPTORS_BIND_CHEMOKINES</td>
<td align="right">55</td>
<td align="right">4</td>
<td align="right">0.0001221</td>
<td align="right">0.0164528</td>
</tr>
<tr class="even">
<td align="left">REACTOME_CLASS_A1_RHODOPSIN_LIKE_RECEPTORS</td>
<td align="right">298</td>
<td align="right">7</td>
<td align="right">0.0005435</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_GPCR_LIGAND_BINDING</td>
<td align="right">399</td>
<td align="right">8</td>
<td align="right">0.0006161</td>
<td align="right">0.0560222</td>
</tr>
<tr class="even">
<td align="left">REACTOME_BIOLOGICAL_OXIDATIONS</td>
<td align="right">137</td>
<td align="right">5</td>
<td align="right">0.0006650</td>
<td align="right">0.0560222</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_XENOBIOTICS</td>
<td align="right">15</td>
<td align="right">2</td>
<td align="right">0.0026122</td>
<td align="right">0.1956237</td>
</tr>
<tr class="even">
<td align="left">REACTOME_METABOLISM_OF_PROTEINS</td>
<td align="right">427</td>
<td align="right">7</td>
<td align="right">0.0045843</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_INTEGRIN_CELL_SURFACE_INTERACTIONS</td>
<td align="right">79</td>
<td align="right">3</td>
<td align="right">0.0051455</td>
<td align="right">0.2996069</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE1_FUNCTIONALIZATION_OF_COMPOUNDS</td>
<td align="right">69</td>
<td align="right">3</td>
<td align="right">0.0053342</td>
<td align="right">0.2996069</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_SIGNALING_BY_PDGF</td>
<td align="right">118</td>
<td align="right">3</td>
<td align="right">0.0158286</td>
<td align="right">0.8206532</td>
</tr>
<tr class="even">
<td align="left">REACTOME_NUCLEAR_RECEPTOR_TRANSCRIPTION_PATHWAY</td>
<td align="right">47</td>
<td align="right">2</td>
<td align="right">0.0198569</td>
<td align="right">0.9559694</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_CYTOCHROME_P450_ARRANGED_BY_SUBSTRATE_TYPE</td>
<td align="right">50</td>
<td align="right">2</td>
<td align="right">0.0273852</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_VEGF_LIGAND_RECEPTOR_INTERACTIONS</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0427644</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_REGULATED_PROTEOLYSIS_OF_P75NTR</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0455794</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_PHASE_II_CONJUGATION</td>
<td align="right">69</td>
<td align="right">2</td>
<td align="right">0.0467892</td>
<td align="right">1.0000000</td>
</tr>
<tr class="odd">
<td align="left">REACTOME_NOREPINEPHRINE_NEUROTRANSMITTER_RELEASE_CYCLE</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0474190</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">REACTOME_ETHANOL_OXIDATION</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">0.0475130</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>topGSA(go) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="68%" />
<col width="5%" />
<col width="3%" />
<col width="10%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">N</th>
<th align="right">DE</th>
<th align="right">P.DE</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GO_INTRINSIC_COMPONENT_OF_PLASMA_MEMBRANE</td>
<td align="right">1644</td>
<td align="right">45</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">GO_EXTRACELLULAR_SPACE</td>
<td align="right">1343</td>
<td align="right">33</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000150</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_CELLULAR_COMPONENT_MOVEMENT</td>
<td align="right">771</td>
<td align="right">23</td>
<td align="right">1.00e-07</td>
<td align="right">0.0002820</td>
</tr>
<tr class="even">
<td align="left">GO_RECEPTOR_BINDING</td>
<td align="right">1452</td>
<td align="right">32</td>
<td align="right">2.00e-07</td>
<td align="right">0.0003197</td>
</tr>
<tr class="odd">
<td align="left">GO_IMMUNE_SYSTEM_PROCESS</td>
<td align="right">1949</td>
<td align="right">38</td>
<td align="right">4.00e-07</td>
<td align="right">0.0005336</td>
</tr>
<tr class="even">
<td align="left">GO_REGULATION_OF_RESPONSE_TO_EXTERNAL_STIMULUS</td>
<td align="right">926</td>
<td align="right">24</td>
<td align="right">7.00e-07</td>
<td align="right">0.0006789</td>
</tr>
<tr class="odd">
<td align="left">GO_HORMONE_ACTIVITY</td>
<td align="right">118</td>
<td align="right">8</td>
<td align="right">8.00e-07</td>
<td align="right">0.0006789</td>
</tr>
<tr class="even">
<td align="left">GO_CYTOKINE_RECEPTOR_BINDING</td>
<td align="right">271</td>
<td align="right">12</td>
<td align="right">1.00e-06</td>
<td align="right">0.0006789</td>
</tr>
<tr class="odd">
<td align="left">GO_DIVALENT_INORGANIC_CATION_HOMEOSTASIS</td>
<td align="right">343</td>
<td align="right">14</td>
<td align="right">1.10e-06</td>
<td align="right">0.0006789</td>
</tr>
<tr class="even">
<td align="left">GO_CHEMICAL_HOMEOSTASIS</td>
<td align="right">871</td>
<td align="right">23</td>
<td align="right">1.10e-06</td>
<td align="right">0.0006789</td>
</tr>
<tr class="odd">
<td align="left">GO_ION_HOMEOSTASIS</td>
<td align="right">575</td>
<td align="right">18</td>
<td align="right">1.50e-06</td>
<td align="right">0.0008543</td>
</tr>
<tr class="even">
<td align="left">GO_NEURON_PART</td>
<td align="right">1264</td>
<td align="right">28</td>
<td align="right">3.20e-06</td>
<td align="right">0.0016335</td>
</tr>
<tr class="odd">
<td align="left">GO_EXTRACELLULAR_MATRIX</td>
<td align="right">426</td>
<td align="right">15</td>
<td align="right">3.50e-06</td>
<td align="right">0.0016721</td>
</tr>
<tr class="even">
<td align="left">GO_IMMUNE_RESPONSE</td>
<td align="right">1066</td>
<td align="right">24</td>
<td align="right">4.00e-06</td>
<td align="right">0.0017773</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_BODY_FLUID_LEVELS</td>
<td align="right">505</td>
<td align="right">16</td>
<td align="right">4.90e-06</td>
<td align="right">0.0019992</td>
</tr>
<tr class="even">
<td align="left">GO_LOCOMOTION</td>
<td align="right">1113</td>
<td align="right">25</td>
<td align="right">6.60e-06</td>
<td align="right">0.0025587</td>
</tr>
<tr class="odd">
<td align="left">GO_NEGATIVE_REGULATION_OF_MULTICELLULAR_ORGANISMAL_PROCESS</td>
<td align="right">983</td>
<td align="right">23</td>
<td align="right">7.50e-06</td>
<td align="right">0.0027094</td>
</tr>
<tr class="even">
<td align="left">GO_TISSUE_DEVELOPMENT</td>
<td align="right">1514</td>
<td align="right">30</td>
<td align="right">8.50e-06</td>
<td align="right">0.0028689</td>
</tr>
<tr class="odd">
<td align="left">GO_REGULATION_OF_RESPONSE_TO_WOUNDING</td>
<td align="right">413</td>
<td align="right">14</td>
<td align="right">8.80e-06</td>
<td align="right">0.0028689</td>
</tr>
<tr class="even">
<td align="left">GO_ORGAN_MORPHOGENESIS</td>
<td align="right">841</td>
<td align="right">21</td>
<td align="right">1.03e-05</td>
<td align="right">0.0031901</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write_csv(topGSA(react, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/REACTOME-{colnames(cont)[6]}.csv&quot;))
write_csv(topGSA(go, number = 100) %&gt;% data.frame, 
          file = glue(&quot;{dirName}/GO-{colnames(cont)[6]}.csv&quot;))</code></pre>
<pre class="r"><code>exp &lt;- cpm(y, log = TRUE)
exp &lt;- exp[rownames(exp) %in% rownames(top), 
           y$samples$Broad == &quot;Epithelial&quot;]

dat &lt;- exp %&gt;% 
  data.frame %&gt;% 
  rownames_to_column(var = &quot;Gene&quot;) %&gt;% 
  pivot_longer(-Gene, 
               names_to = &quot;ID&quot;, 
               values_to = &quot;Log2 CPM&quot;) %&gt;%
  left_join(rownames_to_column(y$samples, var = &quot;ID&quot;)) %&gt;%
  mutate(`Z-score` = `Log2 CPM`)

p &lt;- vector(&quot;list&quot;, 50)

for(i in 1:length(p)){
p[[i]] &lt;- ggplot(data = subset(dat, dat$Gene == rownames(top)[i]), 
       aes(x = Disease, y = `Log2 CPM`)) +
  geom_jitter(width = 0.25) +
  labs(x = &quot;Disease&quot;, y = &quot;Log2 CPM&quot;) +
  ggtitle(rownames(top)[i]) +
  theme(plot.title = element_text(size = 8),
        plot.subtitle = element_text(size = 7),
        axis.title = element_text(size = 8),
        axis.text.x = element_text(size = 7))
}

wrap_plots(p, guides = &quot;collect&quot;, ncol = 3) &amp; 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-34-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="panel-figures" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Panel figures</h1>
<pre class="r"><code>layout = &quot;AABB
          AABB
          ##BB
          ##BB
          ##BB&quot;

(f5a + theme(legend.position = &quot;bottom&quot;,
             legend.direction = &quot;horizontal&quot;)) +
  wrap_elements(grid::grid.grabExpr(ComplexHeatmap::draw(f5b))) + 
  plot_layout(design = layout) +
  plot_annotation(tag_levels = &quot;A&quot;) &amp;
  theme(plot.tag = element_text(size = 14, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/7b44e3893f55d9e2efa666c6c062135536c5bb79/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-35-1.png" target="_blank">7b44e38</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/f61b44185208faa116eda47b7d533082cdb2132b/docs/figure/05_COMBO.expression_analysis.Rmd/unnamed-chunk-35-1.png" target="_blank">f61b441</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-info" class="section level1 appendix" number="6">
<h1><span class="header-section-number">6</span> Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.0 (2021-05-18)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-06-15
 pandoc   2.17.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package                                       * version    date (UTC) lib source
 P abind                                           1.4-5      2016-07-21 [?] CRAN (R 4.1.0)
 P annotate                                        1.72.0     2021-10-26 [?] Bioconductor
 P AnnotationDbi                                 * 1.56.2     2021-11-09 [?] Bioconductor
 P askpass                                         1.1        2019-01-13 [?] CRAN (R 4.0.2)
 P assertthat                                      0.2.1      2019-03-21 [?] CRAN (R 4.1.0)
 P backports                                       1.4.1      2021-12-13 [?] CRAN (R 4.1.0)
 P base64                                          2.0        2016-05-10 [?] CRAN (R 4.1.0)
 P beachmat                                        2.10.0     2021-10-26 [?] Bioconductor
 P beanplot                                        1.3.1      2022-04-09 [?] CRAN (R 4.1.0)
 P beeswarm                                        0.4.0      2021-06-01 [?] CRAN (R 4.1.0)
 P BiasedUrn                                       1.07       2015-12-28 [?] CRAN (R 4.1.0)
 P Biobase                                       * 2.54.0     2021-10-26 [?] Bioconductor
 P BiocFileCache                                   2.2.0      2021-10-26 [?] Bioconductor
 P BiocGenerics                                  * 0.40.0     2021-10-26 [?] Bioconductor
 P BiocIO                                          1.4.0      2021-10-26 [?] Bioconductor
 P BiocManager                                     1.30.16    2021-06-15 [?] CRAN (R 4.1.0)
 P BiocNeighbors                                   1.12.0     2021-10-26 [?] Bioconductor
 P BiocParallel                                  * 1.28.3     2021-12-09 [?] Bioconductor
 P BiocSingular                                    1.10.0     2021-10-26 [?] Bioconductor
 P BiocStyle                                     * 2.22.0     2021-10-26 [?] Bioconductor
 P biomaRt                                         2.50.1     2021-11-21 [?] Bioconductor
 P Biostrings                                    * 2.62.0     2021-10-26 [?] Bioconductor
 P bit                                             4.0.4      2020-08-04 [?] CRAN (R 4.1.0)
 P bit64                                           4.0.5      2020-08-30 [?] CRAN (R 4.0.2)
 P bitops                                          1.0-7      2021-04-24 [?] CRAN (R 4.0.2)
 P blob                                            1.2.2      2021-07-23 [?] CRAN (R 4.1.0)
 P bluster                                         1.4.0      2021-10-26 [?] Bioconductor
 P bookdown                                        0.24       2021-09-02 [?] CRAN (R 4.1.0)
 P broom                                           0.7.11     2022-01-03 [?] CRAN (R 4.1.0)
 P bslib                                           0.3.1      2021-10-06 [?] CRAN (R 4.1.0)
 P bumphunter                                    * 1.36.0     2021-10-27 [?] Bioconductor
 P cachem                                          1.0.6      2021-08-19 [?] CRAN (R 4.1.0)
 P callr                                           3.7.0      2021-04-20 [?] CRAN (R 4.1.0)
 P caTools                                         1.18.2     2021-03-28 [?] CRAN (R 4.1.0)
 P cellranger                                      1.1.0      2016-07-27 [?] CRAN (R 4.1.0)
 P circlize                                        0.4.13     2021-06-09 [?] CRAN (R 4.1.0)
 P cli                                             3.1.0      2021-10-27 [?] CRAN (R 4.1.0)
 P clue                                            0.3-60     2021-10-11 [?] CRAN (R 4.1.0)
 P cluster                                         2.1.2      2021-04-17 [?] CRAN (R 4.1.0)
 P clustree                                      * 0.4.4      2021-11-08 [?] CRAN (R 4.1.0)
 P codetools                                       0.2-18     2020-11-04 [?] CRAN (R 4.1.0)
 P colorspace                                      2.0-2      2021-06-24 [?] CRAN (R 4.0.2)
 P ComplexHeatmap                                  2.10.0     2021-10-26 [?] Bioconductor
 P cowplot                                         1.1.1      2020-12-30 [?] CRAN (R 4.0.2)
 P crayon                                          1.4.2      2021-10-29 [?] CRAN (R 4.1.0)
 P curl                                            4.3.2      2021-06-23 [?] CRAN (R 4.1.0)
 P data.table                                      1.14.2     2021-09-27 [?] CRAN (R 4.1.0)
 P DBI                                             1.1.2      2021-12-20 [?] CRAN (R 4.1.0)
 P dbplyr                                          2.1.1      2021-04-06 [?] CRAN (R 4.1.0)
 P DelayedArray                                    0.20.0     2021-10-26 [?] Bioconductor
 P DelayedMatrixStats                              1.16.0     2021-10-26 [?] Bioconductor
 P deldir                                          1.0-6      2021-10-23 [?] CRAN (R 4.1.0)
 P dendextend                                      1.15.2     2021-10-28 [?] CRAN (R 4.1.0)
 P digest                                          0.6.29     2021-12-01 [?] CRAN (R 4.1.0)
 P doParallel                                      1.0.16     2020-10-16 [?] CRAN (R 4.0.2)
 P doRNG                                           1.8.2      2020-01-27 [?] CRAN (R 4.1.0)
 P dplyr                                         * 1.0.7      2021-06-18 [?] CRAN (R 4.1.0)
 P dqrng                                           0.3.0      2021-05-01 [?] CRAN (R 4.1.0)
 P DropletUtils                                  * 1.14.1     2021-11-08 [?] Bioconductor
 P DT                                              0.20       2021-11-15 [?] CRAN (R 4.1.0)
 P edgeR                                         * 3.36.0     2021-10-26 [?] Bioconductor
 P EGSEA                                         * 1.22.0     2021-10-26 [?] Bioconductor
 P EGSEAdata                                       1.22.0     2021-10-30 [?] Bioconductor
 P ellipsis                                        0.3.2      2021-04-29 [?] CRAN (R 4.0.2)
 P evaluate                                        0.14       2019-05-28 [?] CRAN (R 4.0.2)
 P fansi                                           1.0.0      2022-01-10 [?] CRAN (R 4.1.0)
 P farver                                          2.1.0      2021-02-28 [?] CRAN (R 4.0.2)
 P fastmap                                         1.1.0      2021-01-25 [?] CRAN (R 4.1.0)
 P filelock                                        1.0.2      2018-10-05 [?] CRAN (R 4.1.0)
 P fitdistrplus                                    1.1-6      2021-09-28 [?] CRAN (R 4.1.0)
 P forcats                                       * 0.5.1      2021-01-27 [?] CRAN (R 4.1.0)
 P foreach                                       * 1.5.1      2020-10-15 [?] CRAN (R 4.0.2)
 P fs                                              1.5.2      2021-12-08 [?] CRAN (R 4.1.0)
 P future                                          1.23.0     2021-10-31 [?] CRAN (R 4.1.0)
 P future.apply                                    1.8.1      2021-08-10 [?] CRAN (R 4.1.0)
 P gage                                          * 2.44.0     2021-10-26 [?] Bioconductor
 P genefilter                                      1.76.0     2021-10-26 [?] Bioconductor
 P generics                                        0.1.1      2021-10-25 [?] CRAN (R 4.1.0)
   GenomeInfoDb                                  * 1.30.1     2022-01-30 [1] Bioconductor
 P GenomeInfoDbData                                1.2.7      2021-12-21 [?] Bioconductor
 P GenomicAlignments                               1.30.0     2021-10-26 [?] Bioconductor
 P GenomicFeatures                               * 1.46.3     2021-12-30 [?] Bioconductor
 P GenomicRanges                                 * 1.46.1     2021-11-18 [?] Bioconductor
 P GEOquery                                        2.62.2     2022-01-11 [?] Bioconductor
 P GetoptLong                                      1.0.5      2020-12-15 [?] CRAN (R 4.0.2)
 P getPass                                         0.2-2      2017-07-21 [?] CRAN (R 4.0.2)
 P ggbeeswarm                                      0.6.0      2017-08-07 [?] CRAN (R 4.1.0)
 P ggforce                                         0.3.3      2021-03-05 [?] CRAN (R 4.1.0)
 P ggplot2                                       * 3.3.5      2021-06-25 [?] CRAN (R 4.0.2)
 P ggraph                                        * 2.0.5      2021-02-23 [?] CRAN (R 4.1.0)
 P ggrepel                                         0.9.1      2021-01-15 [?] CRAN (R 4.1.0)
 P ggridges                                        0.5.3      2021-01-08 [?] CRAN (R 4.1.0)
 P git2r                                           0.29.0     2021-11-22 [?] CRAN (R 4.1.0)
 P glmGamPoi                                     * 1.6.0      2021-10-26 [?] Bioconductor
 P GlobalOptions                                   0.1.2      2020-06-10 [?] CRAN (R 4.1.0)
 P globals                                         0.14.0     2020-11-22 [?] CRAN (R 4.0.2)
 P globaltest                                      5.48.0     2021-10-26 [?] Bioconductor
 P glue                                          * 1.6.0      2021-12-17 [?] CRAN (R 4.1.0)
 P GO.db                                         * 3.14.0     2021-12-21 [?] Bioconductor
 P goftest                                         1.2-3      2021-10-07 [?] CRAN (R 4.1.0)
 P gplots                                          3.1.1      2020-11-28 [?] CRAN (R 4.0.2)
 P graph                                         * 1.72.0     2021-10-26 [?] Bioconductor
 P graphlayouts                                    0.8.0      2022-01-03 [?] CRAN (R 4.1.0)
 P gridExtra                                       2.3        2017-09-09 [?] CRAN (R 4.1.0)
 P GSA                                             1.03.1     2019-01-31 [?] CRAN (R 4.1.0)
 P GSEABase                                        1.56.0     2021-10-26 [?] Bioconductor
 P GSVA                                            1.42.0     2021-10-26 [?] Bioconductor
 P gtable                                          0.3.0      2019-03-25 [?] CRAN (R 4.1.0)
 P gtools                                          3.9.2      2021-06-06 [?] CRAN (R 4.1.0)
 P haven                                           2.4.3      2021-08-04 [?] CRAN (R 4.1.0)
 P HDF5Array                                       1.22.1     2021-11-14 [?] Bioconductor
 P here                                          * 1.0.1      2020-12-13 [?] CRAN (R 4.0.2)
 P hgu133a.db                                      3.13.0     2022-01-24 [?] Bioconductor
 P hgu133plus2.db                                  3.13.0     2022-01-24 [?] Bioconductor
 P highr                                           0.9        2021-04-16 [?] CRAN (R 4.1.0)
 P hms                                             1.1.1      2021-09-26 [?] CRAN (R 4.1.0)
 P htmltools                                       0.5.2      2021-08-25 [?] CRAN (R 4.1.0)
 P HTMLUtils                                       0.1.7      2015-01-17 [?] CRAN (R 4.1.0)
 P htmlwidgets                                     1.5.4      2021-09-08 [?] CRAN (R 4.1.0)
 P httpuv                                          1.6.5      2022-01-05 [?] CRAN (R 4.1.0)
 P httr                                            1.4.2      2020-07-20 [?] CRAN (R 4.1.0)
 P hwriter                                         1.3.2      2014-09-10 [?] CRAN (R 4.1.0)
 P ica                                             1.0-2      2018-05-24 [?] CRAN (R 4.1.0)
 P igraph                                          1.2.11     2022-01-04 [?] CRAN (R 4.1.0)
 P IlluminaHumanMethylation450kanno.ilmn12.hg19  * 0.6.0      2022-05-10 [?] Bioconductor
 P IlluminaHumanMethylationEPICanno.ilm10b4.hg19 * 0.6.0      2022-05-10 [?] Bioconductor
 P illuminaio                                      0.36.0     2021-10-26 [?] Bioconductor
 P IRanges                                       * 2.28.0     2021-10-26 [?] Bioconductor
 P irlba                                           2.3.5      2021-12-06 [?] CRAN (R 4.1.0)
 P iterators                                     * 1.0.13     2020-10-15 [?] CRAN (R 4.0.2)
 P jquerylib                                       0.1.4      2021-04-26 [?] CRAN (R 4.1.0)
 P jsonlite                                        1.7.2      2020-12-09 [?] CRAN (R 4.0.2)
 P KEGGdzPathwaysGEO                               1.32.0     2021-10-30 [?] Bioconductor
 P KEGGgraph                                       1.54.0     2021-10-26 [?] Bioconductor
 P KEGGREST                                        1.34.0     2021-10-26 [?] Bioconductor
 P KernSmooth                                      2.23-20    2021-05-03 [?] CRAN (R 4.1.0)
 P knitr                                           1.37       2021-12-16 [?] CRAN (R 4.1.0)
 P labeling                                        0.4.2      2020-10-20 [?] CRAN (R 4.0.2)
 P later                                           1.3.0      2021-08-18 [?] CRAN (R 4.1.0)
 P lattice                                         0.20-45    2021-09-22 [?] CRAN (R 4.1.0)
 P lazyeval                                        0.2.2      2019-03-15 [?] CRAN (R 4.1.0)
 P leiden                                          0.3.9      2021-07-27 [?] CRAN (R 4.1.0)
 P lifecycle                                       1.0.1      2021-09-24 [?] CRAN (R 4.1.0)
 P limma                                         * 3.50.0     2021-10-26 [?] Bioconductor
 P listenv                                         0.8.0      2019-12-05 [?] CRAN (R 4.1.0)
 P lmtest                                          0.9-39     2021-11-07 [?] CRAN (R 4.1.0)
 P locfit                                        * 1.5-9.4    2020-03-25 [?] CRAN (R 4.1.0)
 P lubridate                                       1.8.0      2021-10-07 [?] CRAN (R 4.1.0)
 P magrittr                                        2.0.1      2020-11-17 [?] CRAN (R 4.0.2)
 P MASS                                            7.3-53.1   2021-02-12 [?] CRAN (R 4.0.2)
 P mathjaxr                                        1.4-0      2021-03-01 [?] CRAN (R 4.1.0)
 P Matrix                                          1.4-0      2021-12-08 [?] CRAN (R 4.1.0)
 P MatrixGenerics                                * 1.6.0      2021-10-26 [?] Bioconductor
 P matrixStats                                   * 0.61.0     2021-09-17 [?] CRAN (R 4.1.0)
 P mclust                                          5.4.9      2021-12-17 [?] CRAN (R 4.1.0)
 P memoise                                         2.0.1      2021-11-26 [?] CRAN (R 4.1.0)
 P metap                                           1.7        2021-12-16 [?] CRAN (R 4.1.0)
 P metapod                                         1.2.0      2021-10-26 [?] Bioconductor
 P mgcv                                            1.8-38     2021-10-06 [?] CRAN (R 4.1.0)
 P mime                                            0.12       2021-09-28 [?] CRAN (R 4.1.0)
 P minfi                                         * 1.40.0     2021-10-26 [?] Bioconductor
 P miniUI                                          0.1.1.1    2018-05-18 [?] CRAN (R 4.1.0)
 P missMethyl                                    * 1.28.0     2021-10-26 [?] Bioconductor
 P mnormt                                          2.0.2      2020-09-01 [?] CRAN (R 4.0.2)
 P modelr                                          0.1.8      2020-05-19 [?] CRAN (R 4.0.2)
 P multcomp                                        1.4-18     2022-01-04 [?] CRAN (R 4.1.0)
 P multtest                                        2.50.0     2021-10-26 [?] Bioconductor
 P munsell                                         0.5.0      2018-06-12 [?] CRAN (R 4.1.0)
 P mutoss                                          0.1-12     2017-12-04 [?] CRAN (R 4.1.0)
 P mvtnorm                                         1.1-3      2021-10-08 [?] CRAN (R 4.1.0)
 P nlme                                            3.1-153    2021-09-07 [?] CRAN (R 4.1.0)
 P nor1mix                                         1.3-0      2019-06-13 [?] CRAN (R 4.1.0)
 P numDeriv                                        2016.8-1.1 2019-06-06 [?] CRAN (R 4.1.0)
 P openssl                                         1.4.6      2021-12-19 [?] CRAN (R 4.1.0)
 P org.Hs.eg.db                                  * 3.14.0     2021-12-21 [?] Bioconductor
 P org.Mm.eg.db                                    3.14.0     2022-01-24 [?] Bioconductor
 P org.Rn.eg.db                                    3.14.0     2022-01-24 [?] Bioconductor
 P PADOG                                           1.36.0     2021-10-26 [?] Bioconductor
 P paletteer                                     * 1.4.0      2021-07-20 [?] CRAN (R 4.1.0)
 P parallelly                                      1.30.0     2021-12-17 [?] CRAN (R 4.1.0)
 P patchwork                                     * 1.1.1      2020-12-17 [?] CRAN (R 4.0.2)
 P pathview                                      * 1.34.0     2021-10-26 [?] Bioconductor
 P pbapply                                         1.5-0      2021-09-16 [?] CRAN (R 4.1.0)
 P pillar                                          1.6.4      2021-10-18 [?] CRAN (R 4.1.0)
 P pkgconfig                                       2.0.3      2019-09-22 [?] CRAN (R 4.1.0)
 P plotly                                          4.10.0     2021-10-09 [?] CRAN (R 4.1.0)
 P plotrix                                         3.8-2      2021-09-08 [?] CRAN (R 4.1.0)
 P plyr                                            1.8.6      2020-03-03 [?] CRAN (R 4.0.2)
 P png                                             0.1-7      2013-12-03 [?] CRAN (R 4.1.0)
 P polyclip                                        1.10-0     2019-03-14 [?] CRAN (R 4.1.0)
 P preprocessCore                                  1.56.0     2021-10-26 [?] Bioconductor
 P prettyunits                                     1.1.1      2020-01-24 [?] CRAN (R 4.0.2)
 P prismatic                                       1.1.0      2021-10-17 [?] CRAN (R 4.1.0)
 P processx                                        3.5.2      2021-04-30 [?] CRAN (R 4.1.0)
 P progress                                        1.2.2      2019-05-16 [?] CRAN (R 4.1.0)
 P promises                                        1.2.0.1    2021-02-11 [?] CRAN (R 4.0.2)
 P ps                                              1.6.0      2021-02-28 [?] CRAN (R 4.1.0)
 P purrr                                         * 0.3.4      2020-04-17 [?] CRAN (R 4.0.2)
 P quadprog                                        1.5-8      2019-11-20 [?] CRAN (R 4.1.0)
 P R.methodsS3                                     1.8.1      2020-08-26 [?] CRAN (R 4.0.2)
 P R.oo                                            1.24.0     2020-08-26 [?] CRAN (R 4.0.2)
 P R.utils                                         2.11.0     2021-09-26 [?] CRAN (R 4.1.0)
 P R2HTML                                          2.3.2      2016-06-23 [?] CRAN (R 4.1.0)
 P R6                                              2.5.1      2021-08-19 [?] CRAN (R 4.1.0)
 P RANN                                            2.6.1      2019-01-08 [?] CRAN (R 4.1.0)
 P rappdirs                                        0.3.3      2021-01-31 [?] CRAN (R 4.0.2)
 P rbibutils                                       2.2.7      2021-12-07 [?] CRAN (R 4.1.0)
 P RColorBrewer                                    1.1-2      2014-12-07 [?] CRAN (R 4.0.2)
 P Rcpp                                            1.0.7      2021-07-07 [?] CRAN (R 4.1.0)
 P RcppAnnoy                                       0.0.19     2021-07-30 [?] CRAN (R 4.1.0)
   RCurl                                           1.98-1.6   2022-02-08 [1] CRAN (R 4.1.0)
 P Rdpack                                          2.1.3      2021-12-08 [?] CRAN (R 4.1.0)
 P readr                                         * 2.1.1      2021-11-30 [?] CRAN (R 4.1.0)
 P readxl                                          1.3.1      2019-03-13 [?] CRAN (R 4.1.0)
 P rematch2                                        2.1.2      2020-05-01 [?] CRAN (R 4.1.0)
 P renv                                            0.15.0-14  2022-01-10 [?] Github (rstudio/renv@a3b90eb)
 P reprex                                          2.0.1      2021-08-05 [?] CRAN (R 4.1.0)
 P reshape                                         0.8.8      2018-10-23 [?] CRAN (R 4.0.2)
 P reshape2                                        1.4.4      2020-04-09 [?] CRAN (R 4.1.0)
 P restfulr                                        0.0.13     2017-08-06 [?] CRAN (R 4.1.0)
 P reticulate                                      1.22       2021-09-17 [?] CRAN (R 4.1.0)
 P Rgraphviz                                       2.38.0     2021-10-26 [?] Bioconductor
 P rhdf5                                           2.38.0     2021-10-26 [?] Bioconductor
 P rhdf5filters                                    1.6.0      2021-10-26 [?] Bioconductor
 P Rhdf5lib                                        1.16.0     2021-10-26 [?] Bioconductor
 P rjson                                           0.2.21     2022-01-09 [?] CRAN (R 4.1.0)
 P rlang                                           0.4.12     2021-10-18 [?] CRAN (R 4.1.0)
 P rmarkdown                                       2.11       2021-09-14 [?] CRAN (R 4.1.0)
 P rngtools                                        1.5.2      2021-09-20 [?] CRAN (R 4.1.0)
 P ROCR                                            1.0-11     2020-05-02 [?] CRAN (R 4.1.0)
 P rpart                                           4.1-15     2019-04-12 [?] CRAN (R 4.1.0)
 P rprojroot                                       2.0.2      2020-11-15 [?] CRAN (R 4.0.2)
 P Rsamtools                                       2.10.0     2021-10-26 [?] Bioconductor
 P RSQLite                                         2.2.9      2021-12-06 [?] CRAN (R 4.1.0)
 P rstudioapi                                      0.13       2020-11-12 [?] CRAN (R 4.0.2)
 P rsvd                                            1.0.5      2021-04-16 [?] CRAN (R 4.1.0)
 P rtracklayer                                     1.54.0     2021-10-26 [?] Bioconductor
 P Rtsne                                           0.15       2018-11-10 [?] CRAN (R 4.1.0)
 P rvest                                           1.0.2      2021-10-16 [?] CRAN (R 4.1.0)
 P S4Vectors                                     * 0.32.3     2021-11-21 [?] Bioconductor
 P safe                                            3.34.0     2021-10-26 [?] Bioconductor
 P sandwich                                        3.0-1      2021-05-18 [?] CRAN (R 4.1.0)
 P sass                                            0.4.0      2021-05-12 [?] CRAN (R 4.1.0)
 P ScaledMatrix                                    1.2.0      2021-10-26 [?] Bioconductor
 P scales                                          1.1.1      2020-05-11 [?] CRAN (R 4.0.2)
 P scater                                        * 1.22.0     2021-10-26 [?] Bioconductor
 P scattermore                                     0.7        2020-11-24 [?] CRAN (R 4.1.0)
 P scran                                         * 1.22.1     2021-11-14 [?] Bioconductor
 P scrime                                          1.3.5      2018-12-01 [?] CRAN (R 4.1.0)
 P sctransform                                     0.3.3      2022-01-13 [?] CRAN (R 4.1.0)
 P scuttle                                       * 1.4.0      2021-10-26 [?] Bioconductor
 P sessioninfo                                     1.2.2      2021-12-06 [?] CRAN (R 4.1.0)
 P Seurat                                        * 4.0.6      2021-12-16 [?] CRAN (R 4.1.0)
 P SeuratObject                                  * 4.0.4      2021-11-23 [?] CRAN (R 4.1.0)
 P shape                                           1.4.6      2021-05-19 [?] CRAN (R 4.1.0)
 P shiny                                           1.7.1      2021-10-02 [?] CRAN (R 4.1.0)
 P siggenes                                        1.68.0     2021-10-26 [?] Bioconductor
 P SingleCellExperiment                          * 1.16.0     2021-10-26 [?] Bioconductor
 P sn                                              2.0.1      2021-11-26 [?] CRAN (R 4.1.0)
 P SparseM                                       * 1.81       2021-02-18 [?] CRAN (R 4.1.0)
 P sparseMatrixStats                               1.6.0      2021-10-26 [?] Bioconductor
 P spatstat.core                                   2.3-2      2021-11-26 [?] CRAN (R 4.1.0)
 P spatstat.data                                   2.1-2      2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.geom                                   2.3-1      2021-12-10 [?] CRAN (R 4.1.0)
 P spatstat.sparse                                 2.1-0      2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.utils                                  2.3-0      2021-12-12 [?] CRAN (R 4.1.0)
 P statmod                                         1.4.36     2021-05-10 [?] CRAN (R 4.1.0)
 P stringi                                         1.7.6      2021-11-29 [?] CRAN (R 4.1.0)
 P stringr                                       * 1.4.0      2019-02-10 [?] CRAN (R 4.0.2)
 P SummarizedExperiment                          * 1.24.0     2021-10-26 [?] Bioconductor
 P survival                                        3.2-13     2021-08-24 [?] CRAN (R 4.1.0)
 P tensor                                          1.5        2012-05-05 [?] CRAN (R 4.1.0)
 P TFisher                                         0.2.0      2018-03-21 [?] CRAN (R 4.1.0)
 P TH.data                                         1.1-0      2021-09-27 [?] CRAN (R 4.1.0)
 P tibble                                        * 3.1.6      2021-11-07 [?] CRAN (R 4.1.0)
 P tidygraph                                       1.2.0      2020-05-12 [?] CRAN (R 4.0.2)
 P tidyHeatmap                                   * 1.7.0      2022-05-13 [?] Github (stemangiola/tidyHeatmap@241aec2)
 P tidyr                                         * 1.1.4      2021-09-27 [?] CRAN (R 4.1.0)
 P tidyselect                                      1.1.1      2021-04-30 [?] CRAN (R 4.1.0)
 P tidyverse                                     * 1.3.1      2021-04-15 [?] CRAN (R 4.1.0)
 P tmvnsim                                         1.0-2      2016-12-15 [?] CRAN (R 4.1.0)
 P topGO                                         * 2.46.0     2021-10-26 [?] Bioconductor
 P tweenr                                          1.0.2      2021-03-23 [?] CRAN (R 4.1.0)
 P TxDb.Hsapiens.UCSC.hg38.knownGene             * 3.14.0     2022-05-10 [?] Bioconductor
 P tzdb                                            0.2.0      2021-10-27 [?] CRAN (R 4.1.0)
 P utf8                                            1.2.2      2021-07-24 [?] CRAN (R 4.1.0)
 P uwot                                            0.1.11     2021-12-02 [?] CRAN (R 4.1.0)
 P vctrs                                           0.3.8      2021-04-29 [?] CRAN (R 4.0.2)
 P vipor                                           0.4.5      2017-03-22 [?] CRAN (R 4.1.0)
 P viridis                                         0.6.2      2021-10-13 [?] CRAN (R 4.1.0)
 P viridisLite                                     0.4.0      2021-04-13 [?] CRAN (R 4.0.2)
 P vroom                                           1.5.7      2021-11-30 [?] CRAN (R 4.1.0)
 P whisker                                         0.4        2019-08-28 [?] CRAN (R 4.0.2)
 P withr                                           2.4.3      2021-11-30 [?] CRAN (R 4.1.0)
 P workflowr                                     * 1.7.0      2021-12-21 [?] CRAN (R 4.1.0)
 P xfun                                            0.29       2021-12-14 [?] CRAN (R 4.1.0)
 P XML                                             3.99-0.8   2021-09-17 [?] CRAN (R 4.1.0)
 P xml2                                            1.3.3      2021-11-30 [?] CRAN (R 4.1.0)
 P xtable                                          1.8-4      2019-04-21 [?] CRAN (R 4.1.0)
 P XVector                                       * 0.34.0     2021-10-26 [?] Bioconductor
 P yaml                                            2.2.1      2020-02-01 [?] CRAN (R 4.0.2)
 P zlibbioc                                        1.40.0     2021-10-26 [?] Bioconductor
 P zoo                                             1.8-9      2021-03-09 [?] CRAN (R 4.1.0)

 [1] /oshlack_lab/jovana.maksimovic/projects/MCRI/melanie.neeland/paed-cf-cite-seq/renv/library/R-4.1/x86_64-pc-linux-gnu
 [2] /config/binaries/R/4.1.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
<div id="refs">

</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.1.0/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.1.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices datasets  utils    
[8] methods   base     

other attached packages:
 [1] missMethyl_1.28.0                                  
 [2] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [3] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
 [4] minfi_1.40.0                                       
 [5] bumphunter_1.36.0                                  
 [6] locfit_1.5-9.4                                     
 [7] iterators_1.0.13                                   
 [8] foreach_1.5.1                                      
 [9] Biostrings_2.62.0                                  
[10] XVector_0.34.0                                     
[11] TxDb.Hsapiens.UCSC.hg38.knownGene_3.14.0           
[12] GenomicFeatures_1.46.3                             
[13] EGSEA_1.22.0                                       
[14] pathview_1.34.0                                    
[15] topGO_2.46.0                                       
[16] SparseM_1.81                                       
[17] GO.db_3.14.0                                       
[18] graph_1.72.0                                       
[19] gage_2.44.0                                        
[20] org.Hs.eg.db_3.14.0                                
[21] AnnotationDbi_1.56.2                               
[22] edgeR_3.36.0                                       
[23] limma_3.50.0                                       
[24] tidyHeatmap_1.7.0                                  
[25] paletteer_1.4.0                                    
[26] BiocParallel_1.28.3                                
[27] glmGamPoi_1.6.0                                    
[28] clustree_0.4.4                                     
[29] ggraph_2.0.5                                       
[30] patchwork_1.1.1                                    
[31] SeuratObject_4.0.4                                 
[32] Seurat_4.0.6                                       
[33] scater_1.22.0                                      
[34] scran_1.22.1                                       
[35] scuttle_1.4.0                                      
[36] DropletUtils_1.14.1                                
[37] SingleCellExperiment_1.16.0                        
[38] SummarizedExperiment_1.24.0                        
[39] Biobase_2.54.0                                     
[40] GenomicRanges_1.46.1                               
[41] GenomeInfoDb_1.30.1                                
[42] IRanges_2.28.0                                     
[43] S4Vectors_0.32.3                                   
[44] BiocGenerics_0.40.0                                
[45] MatrixGenerics_1.6.0                               
[46] matrixStats_0.61.0                                 
[47] glue_1.6.0                                         
[48] here_1.0.1                                         
[49] forcats_0.5.1                                      
[50] stringr_1.4.0                                      
[51] dplyr_1.0.7                                        
[52] purrr_0.3.4                                        
[53] readr_2.1.1                                        
[54] tidyr_1.1.4                                        
[55] tibble_3.1.6                                       
[56] ggplot2_3.3.5                                      
[57] tidyverse_1.3.1                                    
[58] BiocStyle_2.22.0                                   
[59] workflowr_1.7.0                                    

loaded via a namespace (and not attached):
  [1] EGSEAdata_1.22.0          PADOG_1.36.0             
  [3] graphlayouts_0.8.0        pbapply_1.5-0            
  [5] lattice_0.20-45           GSVA_1.42.0              
  [7] haven_2.4.3               vctrs_0.3.8              
  [9] mgcv_1.8-38               beanplot_1.3.1           
 [11] blob_1.2.2                survival_3.2-13          
 [13] spatstat.data_2.1-2       later_1.3.0              
 [15] DBI_1.1.2                 R.utils_2.11.0           
 [17] rappdirs_0.3.3            uwot_0.1.11              
 [19] dqrng_0.3.0               zlibbioc_1.40.0          
 [21] htmlwidgets_1.5.4         mvtnorm_1.1-3            
 [23] GlobalOptions_0.1.2       future_1.23.0            
 [25] leiden_0.3.9              irlba_2.3.5              
 [27] illuminaio_0.36.0         tidygraph_1.2.0          
 [29] Rcpp_1.0.7                KernSmooth_2.23-20       
 [31] DT_0.20                   promises_1.2.0.1         
 [33] DelayedArray_0.20.0       fs_1.5.2                 
 [35] mnormt_2.0.2              digest_0.6.29            
 [37] png_0.1-7                 bluster_1.4.0            
 [39] bookdown_0.24             nor1mix_1.3-0            
 [41] sctransform_0.3.3         cowplot_1.1.1            
 [43] pkgconfig_2.0.3           DelayedMatrixStats_1.16.0
 [45] ggbeeswarm_0.6.0          reticulate_1.22          
 [47] circlize_0.4.13           beeswarm_0.4.0           
 [49] GetoptLong_1.0.5          xfun_0.29                
 [51] bslib_0.3.1               zoo_1.8-9                
 [53] tidyselect_1.1.1          reshape2_1.4.4           
 [55] ica_1.0-2                 viridisLite_0.4.0        
 [57] rtracklayer_1.54.0        rlang_0.4.12             
 [59] jquerylib_0.1.4           metap_1.7                
 [61] RColorBrewer_1.1-2        modelr_0.1.8             
 [63] labeling_0.4.2            mutoss_0.1-12            
 [65] httpuv_1.6.5              preprocessCore_1.56.0    
 [67] BiocNeighbors_1.12.0      TH.data_1.1-0            
 [69] annotate_1.72.0           jsonlite_1.7.2           
 [71] tmvnsim_1.0-2             bit_4.0.4                
 [73] mime_0.12                 gridExtra_2.3            
 [75] gplots_3.1.1              Rsamtools_2.10.0         
 [77] stringi_1.7.6             processx_3.5.2           
 [79] spatstat.sparse_2.1-0     scattermore_0.7          
 [81] rbibutils_2.2.7           quadprog_1.5-8           
 [83] bitops_1.0-7              cli_3.1.0                
 [85] Rdpack_2.1.3              rhdf5filters_1.6.0       
 [87] RSQLite_2.2.9             data.table_1.14.2        
 [89] KEGGgraph_1.54.0          rstudioapi_0.13          
 [91] org.Mm.eg.db_3.14.0       git2r_0.29.0             
 [93] GenomicAlignments_1.30.0  nlme_3.1-153             
 [95] listenv_0.8.0             miniUI_0.1.1.1           
 [97] R.oo_1.24.0               dbplyr_2.1.1             
 [99] sessioninfo_1.2.2         readxl_1.3.1             
[101] lifecycle_1.0.1           munsell_0.5.0            
[103] cellranger_1.1.0          R.methodsS3_1.8.1        
[105] hwriter_1.3.2             caTools_1.18.2           
[107] codetools_0.2-18          vipor_0.4.5              
[109] lmtest_0.9-39             xtable_1.8-4             
[111] ROCR_1.0-11               BiocManager_1.30.16      
[113] abind_1.4-5               farver_2.1.0             
[115] parallelly_1.30.0         RANN_2.6.1               
[117] askpass_1.1               hgu133a.db_3.13.0        
[119] BiocIO_1.4.0              GEOquery_2.62.2          
[121] RcppAnnoy_0.0.19          goftest_1.2-3            
[123] cluster_2.1.2             future.apply_1.8.1       
[125] dendextend_1.15.2         Matrix_1.4-0             
[127] ellipsis_0.3.2            prettyunits_1.1.1        
[129] lubridate_1.8.0           ggridges_0.5.3           
[131] reprex_2.0.1              mclust_5.4.9             
[133] prismatic_1.1.0           igraph_1.2.11            
[135] multtest_2.50.0           TFisher_0.2.0            
[137] spatstat.utils_2.3-0      htmltools_0.5.2          
[139] BiocFileCache_2.2.0       org.Rn.eg.db_3.14.0      
[141] yaml_2.2.1                utf8_1.2.2               
[143] plotly_4.10.0             XML_3.99-0.8             
[145] withr_2.4.3               fitdistrplus_1.1-6       
[147] bit64_4.0.5               BiasedUrn_1.07           
[149] rngtools_1.5.2            doRNG_1.8.2              
[151] multcomp_1.4-18           GSA_1.03.1               
[153] spatstat.core_2.3-2       rsvd_1.0.5               
[155] ScaledMatrix_1.2.0        memoise_2.0.1            
[157] evaluate_0.14             globaltest_5.48.0        
[159] tzdb_0.2.0                callr_3.7.0              
[161] ps_1.6.0                  curl_4.3.2               
[163] fansi_1.0.0               highr_0.9                
[165] GSEABase_1.56.0           tensor_1.5               
[167] renv_0.15.0-14            cachem_1.0.6             
[169] deldir_1.0-6              metapod_1.2.0            
[171] rjson_0.2.21              getPass_0.2-2            
[173] ggrepel_0.9.1             hgu133plus2.db_3.13.0    
[175] clue_0.3-60               rprojroot_2.0.2          
[177] tools_4.1.0               sass_0.4.0               
[179] sandwich_3.0-1            magrittr_2.0.1           
[181] RCurl_1.98-1.6            xml2_1.3.3               
[183] httr_1.4.2                assertthat_0.2.1         
[185] rmarkdown_2.11            globals_0.14.0           
[187] R6_2.5.1                  Rhdf5lib_1.16.0          
[189] genefilter_1.76.0         progress_1.2.2           
[191] KEGGREST_1.34.0           gtools_3.9.2             
[193] shape_1.4.6               statmod_1.4.36           
[195] beachmat_2.10.0           rematch2_2.1.2           
[197] HDF5Array_1.22.1          BiocSingular_1.10.0      
[199] rhdf5_2.38.0              splines_4.1.0            
[201] colorspace_2.0-2          generics_0.1.1           
[203] pillar_1.6.4              sn_2.0.1                 
[205] Rgraphviz_2.38.0          tweenr_1.0.2             
[207] KEGGdzPathwaysGEO_1.32.0  GenomeInfoDbData_1.2.7   
[209] plyr_1.8.6                gtable_0.3.0             
[211] rvest_1.0.2               restfulr_0.0.13          
[213] knitr_1.37                ComplexHeatmap_2.10.0    
[215] biomaRt_2.50.1            fastmap_1.1.0            
[217] doParallel_1.0.16         broom_0.7.11             
[219] openssl_1.4.6             scales_1.1.1             
[221] filelock_1.0.2            backports_1.4.1          
[223] plotrix_3.8-2             R2HTML_2.3.2             
[225] base64_2.0                vroom_1.5.7              
[227] hms_1.1.1                 ggforce_0.3.3            
[229] scrime_1.3.5              Rtsne_0.15               
[231] HTMLUtils_0.1.7           shiny_1.7.1              
[233] polyclip_1.10-0           grid_4.1.0               
[235] numDeriv_2016.8-1.1       mathjaxr_1.4-0           
[237] siggenes_1.68.0           lazyeval_0.2.2           
[239] whisker_0.4               crayon_1.4.2             
[241] MASS_7.3-53.1             safe_3.34.0              
[243] sparseMatrixStats_1.6.0   viridis_0.6.2            
[245] reshape_0.8.8             rpart_4.1-15             
[247] compiler_4.1.0            spatstat.geom_2.3-1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
