<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2022-06-16" />

<title>Analysis of scRNA-seq &amp; CITE-seq Data Combined</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-cf-cite-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of scRNA-seq &amp; CITE-seq Data Combined</h1>
<h3 class="subtitle">Normalise and Transfer ADT data</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">June 16, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks">
Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-16
</p>
<p>
<strong>Checks:</strong>
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
7
<span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
0
</p>
<p>
<strong>Knit directory:</strong>
<code>paed-cf-cite-seq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the
reproducibility checks that were applied when the results were created.
The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you
know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Environment:</strong> empty
</a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global
environment can affect the analysis in your R Markdown file in unknown ways.
For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210524code">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Seed:</strong> <code>set.seed(20210524)</code>
</a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210524code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210524)</code> was run prior to running the code in the R Markdown file.
Setting a seed ensures that any results that rely on randomness, e.g.
subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Session information:</strong> recorded
</a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is
critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Cache:</strong> none
</a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident
that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>File paths:</strong> relative
</a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project
makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree8255c241a80231919c022a91ab9f226505b1e73ctargetblank8255c24a">
<span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
<strong>Repository version:</strong> <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/8255c241a80231919c022a91ab9f226505b1e73c" target="_blank">8255c24</a>
</a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefsshgitatlassianpetermacorgau7999ospaedcfciteseqtree8255c241a80231919c022a91ab9f226505b1e73ctargetblank8255c24a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and
connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/tree/8255c241a80231919c022a91ab9f226505b1e73c" target="_blank">8255c24</a>.
See the <em>Past versions</em> tab to see a history of the changes made to the
R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the
analysis have been committed to Git prior to generating the results (you can
use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only
checks the R Markdown file, but you know if there are other scripts or data
files that it depends on. Below is the status of the Git repository when the
results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/obsolete/
    Ignored:    code/obsolete/
    Ignored:    data/190930_A00152_0150_BHTYCMDSXX/
    Ignored:    data/CellRanger/
    Ignored:    data/GSE127465_RAW/
    Ignored:    data/SCEs/02_ZILIONIS.sct_normalised.SEU.rds
    Ignored:    data/SCEs/03_C133_Neeland.demultiplexed.SCE.rds
    Ignored:    data/SCEs/03_C133_Neeland.emptyDrops.SCE.rds
    Ignored:    data/SCEs/03_C133_Neeland.nuclear_fraction_calls.rds
    Ignored:    data/SCEs/03_C133_Neeland.preprocessed.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.CellRanger_v6.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.emptyDrops.SCE.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.nuclear_fraction_calls.rds
    Ignored:    data/SCEs/03_CF_BAL_Pilot.preprocessed.SCE.rds
    Ignored:    data/SCEs/03_COMBO.clustered.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_azimuth.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_azimuth_v2.SEU.rds
    Ignored:    data/SCEs/03_COMBO.clustered_diet.SEU.rds
    Ignored:    data/SCEs/03_COMBO.integrated.SEU.rds
    Ignored:    data/SCEs/03_COMBO.zilionis_mapped.SEU.rds
    Ignored:    data/SCEs/04_C133_Neeland.adt_dsb_normalised.rds
    Ignored:    data/SCEs/04_C133_Neeland.adt_integrated.rds
    Ignored:    data/SCEs/04_C133_Neeland.all_integrated.SEU.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.CellRanger_v6.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.emptyDrops.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.preprocessed.SCE.rds
    Ignored:    data/SCEs/04_CF_BAL_Pilot.transfer_adt.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_clustered.SEU_bk.rds
    Ignored:    data/SCEs/04_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_integrated.SEU_bk.rds
    Ignored:    data/SCEs/04_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_adt_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_lung_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.clustered_diet.SEU.rds
    Ignored:    data/SCEs/04_COMBO.integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/04_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/04_COMBO.zilionis_mapped.SEU.rds
    Ignored:    data/SCEs/05_CF_BAL_Pilot.transfer_adt.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_adt_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_lung_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_others_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.clustered_annotated_tcells_diet.SEU.rds
    Ignored:    data/SCEs/05_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/05_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/05_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_macrophages_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_others_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.clean_tcells_diet.SEU.rds
    Ignored:    data/SCEs/06_COMBO.macrophages_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.macrophages_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.others_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.others_integrated.SEU.rds
    Ignored:    data/SCEs/06_COMBO.tcells_clustered.SEU.rds
    Ignored:    data/SCEs/06_COMBO.tcells_integrated.SEU.rds
    Ignored:    data/SCEs/C133_Neeland.CellRanger.SCE.rds
    Ignored:    data/SCEs/obsolete/
    Ignored:    data/emptyDrops/
    Ignored:    data/obsolete/
    Ignored:    data/sample_sheets/obsolete/
    Ignored:    output/marker-analysis/obsolete/
    Ignored:    output/obsolete/
    Ignored:    rename_captures.R
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    wflow_background.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   .renvignore
    Deleted:    analysis/03_C133_Neeland.demultiplex.Rmd
    Deleted:    analysis/03_C133_Neeland.preprocess.Rmd
    Deleted:    analysis/03_COMBO.clustering_annotation.Rmd
    Deleted:    analysis/04_CF_BAL_Pilot.emptyDrops.Rmd
    Deleted:    analysis/04_CF_BAL_Pilot.preprocess.Rmd
    Deleted:    analysis/04_COMBO.transfer_proteins.Rmd
    Deleted:    analysis/05_COMBO.cluster_macrophages.Rmd
    Deleted:    analysis/05_COMBO.cluster_others.Rmd
    Deleted:    analysis/05_COMBO.cluster_tcells.Rmd
    Deleted:    analysis/05_COMBO.expression_analysis.Rmd
    Deleted:    analysis/05_COMBO.postprocess_all.Rmd
    Deleted:    analysis/05_COMBO.postprocess_macrophages.Rmd
    Deleted:    analysis/05_COMBO.postprocess_others.Rmd
    Deleted:    analysis/05_COMBO.postprocess_tcells.Rmd
    Modified:   renv/.gitignore
    Modified:   renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in
this status report because it is ok for generated content to have uncommitted
changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made
to the R Markdown (<code>analysis/07_COMBO.transfer_proteins.Rmd</code>) and HTML (<code>docs/07_COMBO.transfer_proteins.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table below to
view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="ssh://git@atlassian.petermac.org.au:7999/os/paed-cf-cite-seq/blob/8255c241a80231919c022a91ab9f226505b1e73c/analysis/07_COMBO.transfer_proteins.Rmd" target="_blank">8255c24</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2022-06-16
</td>
<td>
wflow_publish(c(paste0("analysis/", list.files(path = here::here("analysis"),
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Load libraries</h1>
<pre><code>loaded dsb package version 1.0.1 please cite DOI: 10.1101/2020.02.24.963603</code></pre>
</div>
<div id="load-data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Load data</h1>
<div id="load-c133_neeland-raw-and-preprocessed-cite-seq-data" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Load C133_Neeland raw and preprocessed CITE-seq data</h2>
<pre class="r"><code>sceRaw &lt;- readRDS(here(&quot;data/SCEs/C133_Neeland.CellRanger.SCE.rds&quot;))
is_hto &lt;- rownames(altExp(sceRaw, &quot;Antibody Capture&quot;)) %in%
  paste0(&quot;Human_HTO_&quot;, 1:8)
altExp(sceRaw, &quot;HTO&quot;) &lt;- altExp(sceRaw, &quot;Antibody Capture&quot;)[is_hto, ]
altExp(sceRaw, &quot;ADT&quot;) &lt;- altExp(sceRaw, &quot;Antibody Capture&quot;)[!is_hto, ]
altExp(sceRaw, &quot;Antibody Capture&quot;) &lt;- NULL

# Load C133_Neeland ADT data 
scePrep &lt;- readRDS(here(&quot;data&quot;, &quot;SCEs&quot;, 
                           &quot;03_C133_Neeland.preprocessed.SCE.rds&quot;))</code></pre>
<div id="plot-raw-background-and-cell-adt-expression" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Plot raw background and cell ADT expression</h3>
<pre class="r"><code>md &lt;- data.frame(
  rna.size = log10(Matrix::colSums(counts(sceRaw))), 
  prot.size = log10(Matrix::colSums(counts(altExp(sceRaw, &quot;ADT&quot;)))))
md &lt;- md[md$rna.size &gt; 0 &amp; md$prot.size &gt; 0, ]
md$type &lt;- ifelse(rownames(md) %in% colnames(scePrep), &quot;cell&quot;, &quot;background&quot;)

ggplot(md[md$type == &quot;cell&quot;,], aes(x = prot.size,
                                   y = rna.size)) +
  geom_hex() + 
  ggtitle(&quot;cells&quot;) +
  xlim(c(0, 5)) +
  ylim(c(0,5)) +
  scale_fill_viridis_c(option = &quot;magma&quot;) -&gt; p1

ggplot(md[md$type != &quot;cell&quot;,], aes(x = prot.size,
                                   y = rna.size)) +
  geom_hex() + 
  xlim(c(0, 5)) +
  ylim(c(0,5)) +
  ggtitle(&quot;background&quot;) + 
  scale_fill_viridis_c(option = &quot;magma&quot;) -&gt; p2

(p1 | p2) &amp; theme(legend.position = &quot;bottom&quot;,
                  legend.text = element_text(size = 6))</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dsb-normalise-adt-data" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> DSB normalise ADT data</h2>
<pre class="r"><code># remove low end ADT expression droplets
md &lt;- md[md$prot.size &gt; 1.5 &amp; md$rna.size &gt; 1.5, ]
background.adt.mtx &lt;- counts(altExp(sceRaw, &quot;ADT&quot;))[, colnames(sceRaw) %in% rownames(md[md$type == &quot;background&quot;,])]
keep &lt;- grepl(&quot;^A0&quot;, rownames(background.adt.mtx))
background.adt.mtx &lt;- background.adt.mtx[keep, ] 
cell.adt.mtx &lt;- counts(altExp(scePrep, &quot;ADT&quot;))
cell.adt.mtx &lt;- cell.adt.mtx[keep, ]

read_csv(file = here(&quot;data/sample_sheets/TotalSeq-A_Universal_Cocktail_v1.0.csv&quot;)) %&gt;%
  dplyr::filter(grepl(&quot;[Ii]sotype&quot;, name)) %&gt;%
  pull(id) -&gt; isotype.controls 

# normalize and denoise with dsb with 
out &lt;- here(&quot;data/SCEs/04_C133_Neeland.adt_dsb_normalised.rds&quot;)

if(!file.exists(out)){
  cells.dsb.norm &lt;- DSBNormalizeProtein(cell_protein_matrix = cell.adt.mtx, 
                                        empty_drop_matrix = background.adt.mtx, 
                                        denoise.counts = TRUE, 
                                        use.isotype.control = TRUE, 
                                        isotype.control.name.vec = isotype.controls)
  saveRDS(cells.dsb.norm, file = out)
} else {
  cells.dsb.norm &lt;- readRDS(out)
}</code></pre>
</div>
<div id="load-processed-rna-data" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Load processed RNA data</h2>
<p>Load integrated, clustered data that has been mapped to Zilionis reference.</p>
<pre class="r"><code>out &lt;- here(&quot;data/SCEs/04_COMBO.zilionis_mapped.SEU.rds&quot;)
seuInt &lt;- readRDS(out)

## Add Azimuth HCLA v1.0 labels
seuInt &lt;- AddAzimuthResults(seuInt,
                            filename = here(&quot;data/SCEs/03_COMBO.clustered_azimuth.SEU.rds&quot;))
seuInt$predicted.annotation.l1 &lt;- fct_drop(seuInt$predicted.annotation.l1)

## Add Azimuth HCLA v2.0 labels
seuInt &lt;- AddAzimuthResults(seuInt,
                            filename = here(&quot;data/SCEs/03_COMBO.clustered_azimuth_v2.SEU.rds&quot;))
seuInt$predicted.ann_level_1 &lt;- fct_drop(seuInt$predicted.ann_level_1)
seuInt$predicted.ann_level_2 &lt;- fct_drop(seuInt$predicted.ann_level_2)
seuInt$predicted.ann_level_3 &lt;- fct_drop(seuInt$predicted.ann_level_3)
seuInt$predicted.ann_level_4 &lt;- fct_drop(seuInt$predicted.ann_level_4)
seuInt$predicted.ann_finest_level &lt;- fct_drop(seuInt$predicted.ann_finest_level)</code></pre>
</div>
<div id="add-adt-data-to-seurat-object" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Add ADT data to <code>Seurat</code> object</h2>
<pre class="r"><code># Extract C133_Neeland RNA data from Seurat object
DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
seuAdt &lt;- DietSeurat(seuInt[, seuInt$experiment == 2], assays = &quot;RNA&quot;) 
# Create cell ID that matched SCE object
colnames(cells.dsb.norm) &lt;- paste0(&quot;B-&quot;, colnames(cells.dsb.norm))
colnames(cell.adt.mtx) &lt;- paste0(&quot;B-&quot;, colnames(cell.adt.mtx))

# Check that all cells in Seurat object are also in SCE object
all(colnames(seuAdt) %in% colnames(cells.dsb.norm))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># Match up and subset Seurat and SCE objects 
m &lt;- match(colnames(seuAdt), colnames(cells.dsb.norm))
cells.dsb.norm &lt;- cells.dsb.norm[, m]
cell.adt.mtx &lt;- cell.adt.mtx[, m]

# Check that cell IDs match
all(colnames(seuAdt) == colnames(cells.dsb.norm))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># Add ADT data to Seurat object
# Create a new assay to store ADT information
adt.raw &lt;- CreateAssayObject(counts = cell.adt.mtx)
adt.dsb &lt;- CreateAssayObject(counts = cells.dsb.norm)

# add this assay to the previously created Seurat object
seuAdt[[&quot;ADT.raw&quot;]] &lt;- adt.raw
seuAdt[[&quot;ADT.dsb&quot;]] &lt;- adt.dsb

# Validate that the object now contains multiple assays
seuAdt</code></pre>
<pre><code>An object of class Seurat 
19442 features across 18474 samples within 3 assays 
Active assay: RNA (19120 features, 0 variable features)
 2 other assays present: ADT.raw, ADT.dsb</code></pre>
<pre><code>             used    (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells   12516199   668.5   38124784  2036.1   47655980  2545.2
Vcells 1699493140 12966.2 2887796642 22032.2 2842229724 21684.5</code></pre>
</div>
<div id="integrate-adt-data" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Integrate ADT data</h2>
<pre class="r"><code>out &lt;- here(&quot;data/SCEs/04_C133_Neeland.adt_integrated.rds&quot;)

if(!file.exists(out)){
  DefaultAssay(seuAdt) &lt;- &quot;ADT.raw&quot;
  seuAdt &lt;- intDat(seuAdt, type = &quot;ADT&quot;, int.assay.name = &quot;int.adt.raw&quot;)
  
  DefaultAssay(seuAdt) &lt;- &quot;ADT.dsb&quot;
  seuAdt &lt;- intDat(seuAdt, type = &quot;ADT&quot;, adt.norm = &quot;DSB&quot;, 
                   int.assay.name = &quot;int.adt.dsb&quot;)
  saveRDS(seuAdt, file = out)
  
} else {
  seuAdt &lt;- readRDS(out)
}</code></pre>
</div>
<div id="cluster-adt" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Cluster ADT</h2>
<div id="raw-with-clr-normalisation" class="section level3" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Raw with CLR normalisation</h3>
<pre class="r"><code># define proteins to use in clustering (non-isptype controls)
prots &lt;- rownames(cells.dsb.norm)[!rownames(cells.dsb.norm) %in% isotype.controls]

DefaultAssay(seuAdt) &lt;- &quot;int.adt.raw&quot;
VariableFeatures(seuAdt) &lt;- prots
seuAdt &lt;- ScaleData(seuAdt) %&gt;%
  RunPCA(verbose = FALSE, dims = 1:30, reduction.name = &quot;int.adt.raw.pca&quot;)

seuAdt &lt;- FindNeighbors(object = seuAdt, dims = 1:30, assay = &#39;int.adt.raw&#39;,  
                        k.param = 30, verbose = FALSE,
                        reduction = &quot;int.adt.raw.pca&quot;)
seuAdt &lt;- FindClusters(object = seuAdt, resolution = 1, 
                       algorithm = 3, verbose = FALSE)
seuAdt &lt;- RunUMAP(seuAdt, verbose = FALSE, dims = 1:30, 
                  reduction = &quot;int.adt.raw.pca&quot;, 
                  reduction.name = &quot;int.adt.raw.umap&quot;)

DimPlot(seuAdt, group.by = &quot;seurat_clusters&quot;, label = TRUE,
        label.size = 2.5, reduction = &quot;int.adt.raw.umap&quot;) + NoLegend() -&gt; p1
DimPlot(seuAdt, group.by = &quot;predicted.ann_level_3&quot;, label = TRUE,
        label.size = 2.5, reduction = &quot;int.adt.raw.umap&quot;) + NoLegend() -&gt; p2
DimPlot(seuAdt, group.by = &quot;donor&quot;, reduction = &quot;int.adt.raw.umap&quot;) -&gt; p3

# make results dataframe 
d &lt;- cbind(seuAdt@meta.data, 
           as.data.frame(t(seuAdt@assays$ADT.raw@data)))
# calculate the median protein expression separately for each cluster 
raw.adt.plot &lt;- d %&gt;% 
  dplyr::group_by(seurat_clusters) %&gt;% 
  dplyr::summarize_at(.vars = prots, .funs = median) %&gt;% 
  tibble::remove_rownames() %&gt;% 
  tibble::column_to_rownames(&quot;seurat_clusters&quot;) </code></pre>
</div>
<div id="dsb-normalised" class="section level3" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> DSB normalised</h3>
<pre class="r"><code>DefaultAssay(seuAdt) &lt;- &quot;int.adt.dsb&quot;
VariableFeatures(seuAdt) &lt;- prots
seuAdt &lt;- ScaleData(seuAdt) %&gt;%
  RunPCA(verbose = FALSE, dims = 1:30, reduction.name = &quot;int.adt.dsb.pca&quot;)

seuAdt &lt;- FindNeighbors(object = seuAdt, dims = 1:30, assay = &#39;int.adt.dsb&#39;,  
                        k.param = 30, verbose = FALSE,
                        reduction = &quot;int.adt.dsb.pca&quot;)
seuAdt &lt;- FindClusters(object = seuAdt, resolution = 1, 
                       algorithm = 3, verbose = FALSE)
seuAdt &lt;- RunUMAP(seuAdt, verbose = FALSE, dims = 1:30, 
                  reduction = &quot;int.adt.dsb.pca&quot;, 
                  reduction.name = &quot;int.adt.dsb.umap&quot;)

DimPlot(seuAdt, group.by = &quot;seurat_clusters&quot;, label = TRUE,
        label.size = 2.5, reduction = &quot;int.adt.dsb.umap&quot;) + NoLegend() -&gt; p4
DimPlot(seuAdt, group.by = &quot;predicted.ann_level_3&quot;, label = TRUE,
        label.size = 2.5, reduction = &quot;int.adt.dsb.umap&quot;) + NoLegend() -&gt; p5
DimPlot(seuAdt, group.by = &quot;donor&quot;, reduction = &quot;int.adt.dsb.umap&quot;) -&gt; p6

d &lt;- cbind(seuAdt@meta.data, 
           as.data.frame(t(seuAdt@assays$ADT.dsb@data))) %&gt;% 
  dplyr::group_by(seurat_clusters) %&gt;% 
  dplyr::summarize_at(.vars = prots, .funs = median) %&gt;% 
  tibble::remove_rownames() %&gt;% 
  tibble::column_to_rownames(&quot;seurat_clusters&quot;) -&gt; dsb.adt.plot</code></pre>
<pre class="r"><code>((p1 + scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;) | 
    p4 + scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;)) / 
   ((p2 + scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;) | 
       p5 + scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;)) + 
      plot_layout(guides = &quot;collect&quot;)) / 
   ((p3 | p6) + 
      plot_layout(guides = &quot;collect&quot;))) &amp; 
  theme(legend.position = &quot;bottom&quot;,
        legend.text = element_text(size = 8))</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>read_csv(file = here(&quot;data/sample_sheets/TotalSeq-A_Universal_Cocktail_v1.0.csv&quot;)) -&gt; dat
dat &lt;- dat[dat$id %in% prots,]
all(colnames(raw.adt.plot) == dat$id)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>pheatmap::pheatmap(t(raw.adt.plot), 
                   color = viridis::viridis(25, option = &quot;B&quot;), 
                   fontsize_row = 8, border_color = NA,
                   labels_row = gsub(&quot;anti-human&quot;, &quot;&quot;, dat$name))</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pheatmap::pheatmap(t(dsb.adt.plot), 
                   color = viridis::viridis(25, option = &quot;B&quot;), 
                   fontsize_row = 8, border_color = NA,
                   labels_row = gsub(&quot;anti-human&quot;, &quot;&quot;, dat$name))</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualise-rna" class="section level3" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> Visualise RNA</h3>
<pre class="r"><code>DefaultAssay(seuAdt) &lt;- &quot;RNA&quot;
seuAdt &lt;- NormalizeData(seuAdt) %&gt;% 
  FindVariableFeatures() %&gt;% 
  ScaleData() %&gt;% 
  RunPCA(verbose = FALSE, dims = 1:30, reduction.name = &quot;rna.pca&quot;) %&gt;%
  RunUMAP(verbose = FALSE, dims = 1:30, reduction = &quot;rna.pca&quot;, 
          reduction.name = &quot;rna.umap&quot;)</code></pre>
<pre class="r"><code>DimPlot(seuAdt, group.by = &quot;donor&quot;, combine = FALSE, reduction = &quot;rna.umap&quot;)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuAdt, group.by = &quot;predicted.ann_level_3&quot;, 
        combine = FALSE, reduction = &quot;rna.umap&quot;)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="integrate-rna-data" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Integrate RNA data</h2>
<pre class="r"><code>DefaultAssay(seuAdt) &lt;- &quot;RNA&quot;
out &lt;- here(&quot;data/SCEs/04_C133_Neeland.all_integrated.SEU.rds&quot;)

if(!file.exists(out)) {
  seuAdt &lt;- intDat(seuAdt, type = &quot;RNA&quot;)
  seuAdt &lt;- RunPCA(seuAdt, verbose = FALSE, dims = 1:30, 
                 reduction.name = &quot;rna.pca&quot;) %&gt;%
  RunUMAP(verbose = FALSE, dims = 1:30, reduction = &quot;rna.pca&quot;, 
          reduction.name = &quot;rna.umap&quot;)
  saveRDS(seuAdt, file = out)
  
} else {
  seuAdt &lt;- readRDS(out)
  
}

DefaultAssay(seuAdt) &lt;- &quot;integrated&quot;
DimPlot(seuAdt, group.by = &quot;donor&quot;, combine = FALSE, reduction = &quot;rna.umap&quot;)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="map-cf_bal_pilot-to-c133_neeland-and-transfer-protein-data" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Map CF_BAL_Pilot to C133_Neeland and transfer protein data</h2>
<pre class="r"><code>out &lt;- here(&quot;data/SCEs/05_CF_BAL_Pilot.transfer_adt.SEU.rds&quot;)

DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
seuPilot &lt;- DietSeurat(seuInt[, seuInt$experiment == 1], assays = &quot;RNA&quot;)

if(!file.exists(out)) {
  seuSct &lt;- SCTransform(seuPilot, method = &quot;glmGamPoi&quot;)
  anchors &lt;- FindTransferAnchors(reference = seuAdt, query = seuSct,
                                 dims = 1:30, reference.reduction = &quot;rna.pca&quot;,
                                 normalization.method = &quot;SCT&quot;)
  adt.raw &lt;- TransferData(anchorset = anchors, 
                         refdata = GetAssayData(seuAdt[[&quot;ADT.raw&quot;]]), dims = 1:30)
  adt.dsb &lt;- TransferData(anchorset = anchors, 
                         refdata = GetAssayData(seuAdt[[&quot;ADT.dsb&quot;]]), dims = 1:30)
  seuPilot[[&quot;ADT.raw&quot;]] &lt;- adt.raw
  seuPilot[[&quot;ADT.dsb&quot;]] &lt;- adt.dsb
  saveRDS(seuPilot, file = out)
  
} else {
  seuPilot &lt;- readRDS(out)
  
}</code></pre>
</div>
<div id="visualise-transferred-adt-data" class="section level2" number="2.9">
<h2><span class="header-section-number">2.9</span> Visualise transferred ADT data</h2>
<div id="by-donor-and-hlca-v2.0-level-4-annotations" class="section level3" number="2.9.1">
<h3><span class="header-section-number">2.9.1</span> By donor and HLCA v2.0 level 4 annotations</h3>
<pre class="r"><code>DefaultAssay(seuPilot) &lt;- &quot;ADT.raw&quot;
VariableFeatures(seuPilot) &lt;- prots
seuPilot &lt;- ScaleData(seuPilot) %&gt;% 
  RunPCA(verbose = FALSE, dims = 1:30, reduction.name = &quot;adt.pca&quot;) %&gt;%
  RunUMAP(verbose = FALSE, dims = 1:30, reduction = &quot;adt.pca&quot;, 
          reduction.name = &quot;adt.umap&quot;)

DimPlot(seuPilot, group.by = &quot;donor&quot;, repel = TRUE,
        reduction = &quot;adt.umap&quot;, label = TRUE, label.size = 2.5) </code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuPilot, group.by = &quot;predicted.ann_level_4&quot;, repel = TRUE,
        reduction = &quot;adt.umap&quot;, label = TRUE, label.size = 2.5) &amp; NoLegend()</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" />
### By various marker genes</p>
<pre class="r"><code>markers &lt;- read_csv(
  here(&quot;data/sample_sheets/TotalSeq_A_Human_Universal_Cocktail_Proteins_of_interest_29.09.21.csv&quot;))</code></pre>
<pre class="r"><code>DefaultAssay(seuPilot) &lt;- &quot;RNA&quot;
Idents(seuPilot) &lt;- &quot;predicted.ann_level_4&quot;
options(ggrepel.max.overlaps = Inf) 

p &lt;- vector(&quot;list&quot;, nrow(markers))
for(i in 1:nrow(markers)){
  if(markers$`Gene Name`[i] %in% rownames(seuPilot[[&quot;RNA&quot;]]) &amp;
     markers$DNA_ID[i] %in% rownames(seuPilot[[&quot;ADT.raw&quot;]])){
    
    DefaultAssay(seuPilot) &lt;- &quot;ADT.raw&quot;
    p1 &lt;- FeaturePlot(seuPilot, features = markers$DNA_ID[i], label = TRUE,
                      repel = TRUE, label.size = 2.5,
                      reduction = &#39;adt.umap&#39;, keep.scale = &quot;all&quot;, 
                      cols = c(&quot;lightgrey&quot;,&quot;darkgreen&quot;)) +
      ggtitle(markers$`Gene Name`[i], subtitle = &quot;ADT.raw&quot;) +
      theme(title = element_text(size = 8),
            axis.text = element_text(size = 8)) + 
      NoLegend()
    
    DefaultAssay(seuPilot) &lt;- &quot;ADT.dsb&quot;
    p2 &lt;- FeaturePlot(seuPilot, features = markers$DNA_ID[i], label = TRUE,
                      repel = TRUE, label.size = 2.5,
                      reduction = &#39;adt.umap&#39;, keep.scale = &quot;all&quot;, 
                      cols = c(&quot;lightgrey&quot;,&quot;darkblue&quot;)) +
      ggtitle(markers$`Gene Name`[i], subtitle = &quot;ADT.dsb&quot;) +
      theme(title = element_text(size = 8),
            axis.text = element_text(size = 8)) + 
      NoLegend()
    
    DefaultAssay(seuPilot) &lt;- &quot;RNA&quot;
    p3 &lt;- FeaturePlot(seuPilot, features = markers$`Gene Name`[i], label = TRUE,
                      repel = TRUE, label.size = 2.5,
                      reduction = &#39;adt.umap&#39;, keep.scale = &quot;all&quot;, 
                      cols = c(&quot;lightgrey&quot;,&quot;purple&quot;)) +
      ggtitle(markers$`Gene Name`[i], subtitle = &quot;RNA&quot;) +
      theme(title = element_text(size = 8),
            axis.text = element_text(size = 8)) + 
      NoLegend()
    
    p[[i]] &lt;- (p1 | p2 | p3)
  }
}

p[!sapply(p, is.null)]</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-5.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-6.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-7.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-8.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[9]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-9.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[10]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-10.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[11]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-11.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[12]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-12.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[13]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-13.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[14]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-14.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[15]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-15.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[16]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-16.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[17]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-17.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[18]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-18.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[19]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-19.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[20]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-20.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[21]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-21.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[22]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-22.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[23]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-23.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[24]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-24.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[25]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-25.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[26]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-26.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[27]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-27.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[28]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-28.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[29]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-29.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[30]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-30.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[31]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-31.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[32]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-32.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[33]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-33.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[34]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-34.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[35]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-35.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[36]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-36.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[37]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-37.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[38]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-38.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[39]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-39.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[40]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-40.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[41]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-41.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[42]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-42.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[43]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-43.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[44]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-44.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[45]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-45.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[46]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-46.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[47]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-47.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[48]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-48.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[49]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-49.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[50]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-50.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[51]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-51.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[52]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-52.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[53]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-53.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[54]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-54.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[55]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-55.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[56]]</code></pre>
<p><img src="figure/07_COMBO.transfer_proteins.Rmd/unnamed-chunk-21-56.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="merge-c133_neeland-and-cf_bal_pilot-data-with-adts" class="section level2" number="2.10">
<h2><span class="header-section-number">2.10</span> Merge C133_Neeland and CF_BAL_Pilot data with ADTs</h2>
<pre class="r"><code>DefaultAssay(seuPilot) &lt;- &quot;RNA&quot;
DefaultAssay(seuAdt) &lt;- &quot;RNA&quot;
seuMerge &lt;- merge(DietSeurat(seuPilot, assays = c(&quot;RNA&quot;, &quot;ADT.dsb&quot;, &quot;ADT.raw&quot;)),
                  y = DietSeurat(seuAdt, assays = c(&quot;RNA&quot;,&quot;ADT.dsb&quot;, &quot;ADT.raw&quot;)))
seuMerge</code></pre>
<pre><code>An object of class Seurat 
19442 features across 45590 samples within 3 assays 
Active assay: RNA (19120 features, 0 variable features)
 2 other assays present: ADT.raw, ADT.dsb</code></pre>
</div>
</div>
<div id="save-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Save data</h1>
<pre class="r"><code>out &lt;- here(glue(&quot;data/SCEs/05_COMBO.clustered_annotated_adt_diet.SEU.rds&quot;))

if(!file.exists(out)) {
  saveRDS(seuMerge, file = out)
} </code></pre>
</div>
<div id="session-info" class="section level1 appendix" number="4">
<h1><span class="header-section-number">4</span> Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.1.0 (2021-05-18)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_AU.UTF-8
 ctype    en_AU.UTF-8
 tz       Australia/Melbourne
 date     2022-06-16
 pandoc   2.17.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package              * version   date (UTC) lib source
 P abind                  1.4-5     2016-07-21 [?] CRAN (R 4.1.0)
 P AnnotationDbi          1.56.2    2021-11-09 [?] Bioconductor
 P AnnotationFilter       1.18.0    2021-10-26 [?] Bioconductor
 P assertthat             0.2.1     2019-03-21 [?] CRAN (R 4.1.0)
 P backports              1.4.1     2021-12-13 [?] CRAN (R 4.1.0)
 P beachmat               2.10.0    2021-10-26 [?] Bioconductor
 P beeswarm               0.4.0     2021-06-01 [?] CRAN (R 4.1.0)
 P Biobase              * 2.54.0    2021-10-26 [?] Bioconductor
 P BiocFileCache          2.2.0     2021-10-26 [?] Bioconductor
 P BiocGenerics         * 0.40.0    2021-10-26 [?] Bioconductor
 P BiocIO                 1.4.0     2021-10-26 [?] Bioconductor
 P BiocManager            1.30.16   2021-06-15 [?] CRAN (R 4.1.0)
 P BiocNeighbors          1.12.0    2021-10-26 [?] Bioconductor
 P BiocParallel         * 1.28.3    2021-12-09 [?] Bioconductor
 P BiocSingular           1.10.0    2021-10-26 [?] Bioconductor
 P BiocStyle            * 2.22.0    2021-10-26 [?] Bioconductor
 P biomaRt                2.50.1    2021-11-21 [?] Bioconductor
 P Biostrings             2.62.0    2021-10-26 [?] Bioconductor
 P bit                    4.0.4     2020-08-04 [?] CRAN (R 4.1.0)
 P bit64                  4.0.5     2020-08-30 [?] CRAN (R 4.0.2)
 P bitops                 1.0-7     2021-04-24 [?] CRAN (R 4.0.2)
 P blob                   1.2.2     2021-07-23 [?] CRAN (R 4.1.0)
 P bluster                1.4.0     2021-10-26 [?] Bioconductor
 P bookdown               0.24      2021-09-02 [?] CRAN (R 4.1.0)
 P broom                  0.7.11    2022-01-03 [?] CRAN (R 4.1.0)
 P bslib                  0.3.1     2021-10-06 [?] CRAN (R 4.1.0)
 P cachem                 1.0.6     2021-08-19 [?] CRAN (R 4.1.0)
 P callr                  3.7.0     2021-04-20 [?] CRAN (R 4.1.0)
 P cellranger             1.1.0     2016-07-27 [?] CRAN (R 4.1.0)
 P cli                    3.1.0     2021-10-27 [?] CRAN (R 4.1.0)
 P cluster                2.1.2     2021-04-17 [?] CRAN (R 4.1.0)
 P clustree             * 0.4.4     2021-11-08 [?] CRAN (R 4.1.0)
 P codetools              0.2-18    2020-11-04 [?] CRAN (R 4.1.0)
 P colorspace             2.0-2     2021-06-24 [?] CRAN (R 4.0.2)
 P cowplot                1.1.1     2020-12-30 [?] CRAN (R 4.0.2)
 P crayon                 1.4.2     2021-10-29 [?] CRAN (R 4.1.0)
 P curl                   4.3.2     2021-06-23 [?] CRAN (R 4.1.0)
 P data.table             1.14.2    2021-09-27 [?] CRAN (R 4.1.0)
 P DBI                    1.1.2     2021-12-20 [?] CRAN (R 4.1.0)
 P dbplyr                 2.1.1     2021-04-06 [?] CRAN (R 4.1.0)
 P DelayedArray           0.20.0    2021-10-26 [?] Bioconductor
 P DelayedMatrixStats     1.16.0    2021-10-26 [?] Bioconductor
 P deldir                 1.0-6     2021-10-23 [?] CRAN (R 4.1.0)
 P digest                 0.6.29    2021-12-01 [?] CRAN (R 4.1.0)
 P dplyr                * 1.0.7     2021-06-18 [?] CRAN (R 4.1.0)
 P dqrng                  0.3.0     2021-05-01 [?] CRAN (R 4.1.0)
 P DropletUtils         * 1.14.1    2021-11-08 [?] Bioconductor
 P dsb                  * 1.0.1     2022-03-14 [?] CRAN (R 4.1.0)
 P edgeR                  3.36.0    2021-10-26 [?] Bioconductor
 P ellipsis               0.3.2     2021-04-29 [?] CRAN (R 4.0.2)
 P ensembldb              2.18.2    2021-11-08 [?] Bioconductor
 P evaluate               0.14      2019-05-28 [?] CRAN (R 4.0.2)
 P fansi                  1.0.0     2022-01-10 [?] CRAN (R 4.1.0)
 P farver                 2.1.0     2021-02-28 [?] CRAN (R 4.0.2)
 P fastmap                1.1.0     2021-01-25 [?] CRAN (R 4.1.0)
 P filelock               1.0.2     2018-10-05 [?] CRAN (R 4.1.0)
 P fitdistrplus           1.1-6     2021-09-28 [?] CRAN (R 4.1.0)
 P forcats              * 0.5.1     2021-01-27 [?] CRAN (R 4.1.0)
 P fs                     1.5.2     2021-12-08 [?] CRAN (R 4.1.0)
 P future                 1.23.0    2021-10-31 [?] CRAN (R 4.1.0)
 P future.apply           1.8.1     2021-08-10 [?] CRAN (R 4.1.0)
 P generics               0.1.1     2021-10-25 [?] CRAN (R 4.1.0)
   GenomeInfoDb         * 1.30.1    2022-01-30 [1] Bioconductor
 P GenomeInfoDbData       1.2.7     2021-12-21 [?] Bioconductor
 P GenomicAlignments      1.30.0    2021-10-26 [?] Bioconductor
 P GenomicFeatures        1.46.3    2021-12-30 [?] Bioconductor
 P GenomicRanges        * 1.46.1    2021-11-18 [?] Bioconductor
 P getPass                0.2-2     2017-07-21 [?] CRAN (R 4.0.2)
 P ggbeeswarm             0.6.0     2017-08-07 [?] CRAN (R 4.1.0)
 P ggforce                0.3.3     2021-03-05 [?] CRAN (R 4.1.0)
 P ggplot2              * 3.3.5     2021-06-25 [?] CRAN (R 4.0.2)
 P ggraph               * 2.0.5     2021-02-23 [?] CRAN (R 4.1.0)
 P ggrepel                0.9.1     2021-01-15 [?] CRAN (R 4.1.0)
 P ggridges               0.5.3     2021-01-08 [?] CRAN (R 4.1.0)
 P git2r                  0.29.0    2021-11-22 [?] CRAN (R 4.1.0)
 P glmGamPoi            * 1.6.0     2021-10-26 [?] Bioconductor
 P globals                0.14.0    2020-11-22 [?] CRAN (R 4.0.2)
 P glue                 * 1.6.0     2021-12-17 [?] CRAN (R 4.1.0)
 P goftest                1.2-3     2021-10-07 [?] CRAN (R 4.1.0)
 P graphlayouts           0.8.0     2022-01-03 [?] CRAN (R 4.1.0)
 P gridExtra              2.3       2017-09-09 [?] CRAN (R 4.1.0)
 P gtable                 0.3.0     2019-03-25 [?] CRAN (R 4.1.0)
 P haven                  2.4.3     2021-08-04 [?] CRAN (R 4.1.0)
 P HDF5Array              1.22.1    2021-11-14 [?] Bioconductor
 P here                 * 1.0.1     2020-12-13 [?] CRAN (R 4.0.2)
   hexbin                 1.28.2    2021-01-08 [1] CRAN (R 4.1.0)
 P highr                  0.9       2021-04-16 [?] CRAN (R 4.1.0)
 P hms                    1.1.1     2021-09-26 [?] CRAN (R 4.1.0)
 P htmltools              0.5.2     2021-08-25 [?] CRAN (R 4.1.0)
 P htmlwidgets            1.5.4     2021-09-08 [?] CRAN (R 4.1.0)
 P httpuv                 1.6.5     2022-01-05 [?] CRAN (R 4.1.0)
 P httr                   1.4.2     2020-07-20 [?] CRAN (R 4.1.0)
 P ica                    1.0-2     2018-05-24 [?] CRAN (R 4.1.0)
 P igraph                 1.2.11    2022-01-04 [?] CRAN (R 4.1.0)
 P IRanges              * 2.28.0    2021-10-26 [?] Bioconductor
 P irlba                  2.3.5     2021-12-06 [?] CRAN (R 4.1.0)
 P jquerylib              0.1.4     2021-04-26 [?] CRAN (R 4.1.0)
 P jsonlite               1.7.2     2020-12-09 [?] CRAN (R 4.0.2)
 P KEGGREST               1.34.0    2021-10-26 [?] Bioconductor
 P KernSmooth             2.23-20   2021-05-03 [?] CRAN (R 4.1.0)
 P knitr                  1.37      2021-12-16 [?] CRAN (R 4.1.0)
 P labeling               0.4.2     2020-10-20 [?] CRAN (R 4.0.2)
 P later                  1.3.0     2021-08-18 [?] CRAN (R 4.1.0)
 P lattice                0.20-45   2021-09-22 [?] CRAN (R 4.1.0)
 P lazyeval               0.2.2     2019-03-15 [?] CRAN (R 4.1.0)
 P leiden                 0.3.9     2021-07-27 [?] CRAN (R 4.1.0)
 P lifecycle              1.0.1     2021-09-24 [?] CRAN (R 4.1.0)
 P limma                  3.50.0    2021-10-26 [?] Bioconductor
 P listenv                0.8.0     2019-12-05 [?] CRAN (R 4.1.0)
 P lmtest                 0.9-39    2021-11-07 [?] CRAN (R 4.1.0)
 P locfit                 1.5-9.4   2020-03-25 [?] CRAN (R 4.1.0)
 P lubridate              1.8.0     2021-10-07 [?] CRAN (R 4.1.0)
 P magrittr               2.0.1     2020-11-17 [?] CRAN (R 4.0.2)
 P MASS                   7.3-53.1  2021-02-12 [?] CRAN (R 4.0.2)
 P Matrix                 1.4-0     2021-12-08 [?] CRAN (R 4.1.0)
 P MatrixGenerics       * 1.6.0     2021-10-26 [?] Bioconductor
 P matrixStats          * 0.61.0    2021-09-17 [?] CRAN (R 4.1.0)
 P mclust                 5.4.9     2021-12-17 [?] CRAN (R 4.1.0)
 P memoise                2.0.1     2021-11-26 [?] CRAN (R 4.1.0)
 P metapod                1.2.0     2021-10-26 [?] Bioconductor
 P mgcv                   1.8-38    2021-10-06 [?] CRAN (R 4.1.0)
 P mime                   0.12      2021-09-28 [?] CRAN (R 4.1.0)
 P miniUI                 0.1.1.1   2018-05-18 [?] CRAN (R 4.1.0)
 P modelr                 0.1.8     2020-05-19 [?] CRAN (R 4.0.2)
 P munsell                0.5.0     2018-06-12 [?] CRAN (R 4.1.0)
 P nlme                   3.1-153   2021-09-07 [?] CRAN (R 4.1.0)
 P paletteer            * 1.4.0     2021-07-20 [?] CRAN (R 4.1.0)
 P parallelly             1.30.0    2021-12-17 [?] CRAN (R 4.1.0)
 P patchwork            * 1.1.1     2020-12-17 [?] CRAN (R 4.0.2)
 P pbapply                1.5-0     2021-09-16 [?] CRAN (R 4.1.0)
 P pheatmap               1.0.12    2019-01-04 [?] CRAN (R 4.1.0)
 P pillar                 1.6.4     2021-10-18 [?] CRAN (R 4.1.0)
 P pkgconfig              2.0.3     2019-09-22 [?] CRAN (R 4.1.0)
 P plotly                 4.10.0    2021-10-09 [?] CRAN (R 4.1.0)
 P plyr                   1.8.6     2020-03-03 [?] CRAN (R 4.0.2)
 P png                    0.1-7     2013-12-03 [?] CRAN (R 4.1.0)
 P polyclip               1.10-0    2019-03-14 [?] CRAN (R 4.1.0)
 P prettyunits            1.1.1     2020-01-24 [?] CRAN (R 4.0.2)
 P prismatic              1.1.0     2021-10-17 [?] CRAN (R 4.1.0)
 P processx               3.5.2     2021-04-30 [?] CRAN (R 4.1.0)
 P progress               1.2.2     2019-05-16 [?] CRAN (R 4.1.0)
 P promises               1.2.0.1   2021-02-11 [?] CRAN (R 4.0.2)
 P ProtGenerics           1.26.0    2021-10-26 [?] Bioconductor
 P ps                     1.6.0     2021-02-28 [?] CRAN (R 4.1.0)
 P purrr                * 0.3.4     2020-04-17 [?] CRAN (R 4.0.2)
 P R.methodsS3            1.8.1     2020-08-26 [?] CRAN (R 4.0.2)
 P R.oo                   1.24.0    2020-08-26 [?] CRAN (R 4.0.2)
 P R.utils                2.11.0    2021-09-26 [?] CRAN (R 4.1.0)
 P R6                     2.5.1     2021-08-19 [?] CRAN (R 4.1.0)
 P RANN                   2.6.1     2019-01-08 [?] CRAN (R 4.1.0)
 P rappdirs               0.3.3     2021-01-31 [?] CRAN (R 4.0.2)
 P RColorBrewer           1.1-2     2014-12-07 [?] CRAN (R 4.0.2)
 P Rcpp                   1.0.7     2021-07-07 [?] CRAN (R 4.1.0)
 P RcppAnnoy              0.0.19    2021-07-30 [?] CRAN (R 4.1.0)
   RCurl                  1.98-1.6  2022-02-08 [1] CRAN (R 4.1.0)
 P readr                * 2.1.1     2021-11-30 [?] CRAN (R 4.1.0)
 P readxl                 1.3.1     2019-03-13 [?] CRAN (R 4.1.0)
 P rematch2               2.1.2     2020-05-01 [?] CRAN (R 4.1.0)
 P renv                   0.15.0-14 2022-01-10 [?] Github (rstudio/renv@a3b90eb)
 P reprex                 2.0.1     2021-08-05 [?] CRAN (R 4.1.0)
 P reshape2               1.4.4     2020-04-09 [?] CRAN (R 4.1.0)
 P restfulr               0.0.13    2017-08-06 [?] CRAN (R 4.1.0)
 P reticulate             1.22      2021-09-17 [?] CRAN (R 4.1.0)
 P rhdf5                  2.38.0    2021-10-26 [?] Bioconductor
 P rhdf5filters           1.6.0     2021-10-26 [?] Bioconductor
 P Rhdf5lib               1.16.0    2021-10-26 [?] Bioconductor
 P rjson                  0.2.21    2022-01-09 [?] CRAN (R 4.1.0)
 P rlang                  0.4.12    2021-10-18 [?] CRAN (R 4.1.0)
 P rmarkdown              2.11      2021-09-14 [?] CRAN (R 4.1.0)
 P ROCR                   1.0-11    2020-05-02 [?] CRAN (R 4.1.0)
 P rpart                  4.1-15    2019-04-12 [?] CRAN (R 4.1.0)
 P rprojroot              2.0.2     2020-11-15 [?] CRAN (R 4.0.2)
 P Rsamtools              2.10.0    2021-10-26 [?] Bioconductor
 P RSpectra               0.16-0    2019-12-01 [?] CRAN (R 4.1.0)
 P RSQLite                2.2.9     2021-12-06 [?] CRAN (R 4.1.0)
 P rstudioapi             0.13      2020-11-12 [?] CRAN (R 4.0.2)
 P rsvd                   1.0.5     2021-04-16 [?] CRAN (R 4.1.0)
 P rtracklayer            1.54.0    2021-10-26 [?] Bioconductor
 P Rtsne                  0.15      2018-11-10 [?] CRAN (R 4.1.0)
 P rvest                  1.0.2     2021-10-16 [?] CRAN (R 4.1.0)
 P S4Vectors            * 0.32.3    2021-11-21 [?] Bioconductor
 P sass                   0.4.0     2021-05-12 [?] CRAN (R 4.1.0)
 P ScaledMatrix           1.2.0     2021-10-26 [?] Bioconductor
 P scales                 1.1.1     2020-05-11 [?] CRAN (R 4.0.2)
 P scater               * 1.22.0    2021-10-26 [?] Bioconductor
 P scattermore            0.7       2020-11-24 [?] CRAN (R 4.1.0)
 P scran                * 1.22.1    2021-11-14 [?] Bioconductor
 P sctransform            0.3.3     2022-01-13 [?] CRAN (R 4.1.0)
 P scuttle              * 1.4.0     2021-10-26 [?] Bioconductor
 P sessioninfo            1.2.2     2021-12-06 [?] CRAN (R 4.1.0)
 P Seurat               * 4.0.6     2021-12-16 [?] CRAN (R 4.1.0)
 P SeuratObject         * 4.0.4     2021-11-23 [?] CRAN (R 4.1.0)
 P shiny                  1.7.1     2021-10-02 [?] CRAN (R 4.1.0)
 P SingleCellExperiment * 1.16.0    2021-10-26 [?] Bioconductor
 P sparseMatrixStats      1.6.0     2021-10-26 [?] Bioconductor
 P spatstat.core          2.3-2     2021-11-26 [?] CRAN (R 4.1.0)
 P spatstat.data          2.1-2     2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.geom          2.3-1     2021-12-10 [?] CRAN (R 4.1.0)
 P spatstat.sparse        2.1-0     2021-12-17 [?] CRAN (R 4.1.0)
 P spatstat.utils         2.3-0     2021-12-12 [?] CRAN (R 4.1.0)
 P statmod                1.4.36    2021-05-10 [?] CRAN (R 4.1.0)
 P stringi                1.7.6     2021-11-29 [?] CRAN (R 4.1.0)
 P stringr              * 1.4.0     2019-02-10 [?] CRAN (R 4.0.2)
 P SummarizedExperiment * 1.24.0    2021-10-26 [?] Bioconductor
 P survival               3.2-13    2021-08-24 [?] CRAN (R 4.1.0)
 P tensor                 1.5       2012-05-05 [?] CRAN (R 4.1.0)
 P tibble               * 3.1.6     2021-11-07 [?] CRAN (R 4.1.0)
 P tidygraph              1.2.0     2020-05-12 [?] CRAN (R 4.0.2)
 P tidyr                * 1.1.4     2021-09-27 [?] CRAN (R 4.1.0)
 P tidyselect             1.1.1     2021-04-30 [?] CRAN (R 4.1.0)
 P tidyverse            * 1.3.1     2021-04-15 [?] CRAN (R 4.1.0)
 P tweenr                 1.0.2     2021-03-23 [?] CRAN (R 4.1.0)
 P tzdb                   0.2.0     2021-10-27 [?] CRAN (R 4.1.0)
 P utf8                   1.2.2     2021-07-24 [?] CRAN (R 4.1.0)
 P uwot                   0.1.11    2021-12-02 [?] CRAN (R 4.1.0)
 P vctrs                  0.3.8     2021-04-29 [?] CRAN (R 4.0.2)
 P vipor                  0.4.5     2017-03-22 [?] CRAN (R 4.1.0)
 P viridis                0.6.2     2021-10-13 [?] CRAN (R 4.1.0)
 P viridisLite            0.4.0     2021-04-13 [?] CRAN (R 4.0.2)
 P vroom                  1.5.7     2021-11-30 [?] CRAN (R 4.1.0)
 P whisker                0.4       2019-08-28 [?] CRAN (R 4.0.2)
 P withr                  2.4.3     2021-11-30 [?] CRAN (R 4.1.0)
 P workflowr            * 1.7.0     2021-12-21 [?] CRAN (R 4.1.0)
 P xfun                   0.29      2021-12-14 [?] CRAN (R 4.1.0)
 P XML                    3.99-0.8  2021-09-17 [?] CRAN (R 4.1.0)
 P xml2                   1.3.3     2021-11-30 [?] CRAN (R 4.1.0)
 P xtable                 1.8-4     2019-04-21 [?] CRAN (R 4.1.0)
 P XVector                0.34.0    2021-10-26 [?] Bioconductor
 P yaml                   2.2.1     2020-02-01 [?] CRAN (R 4.0.2)
 P zlibbioc               1.40.0    2021-10-26 [?] Bioconductor
 P zoo                    1.8-9     2021-03-09 [?] CRAN (R 4.1.0)

 [1] /oshlack_lab/jovana.maksimovic/projects/MCRI/melanie.neeland/paed-cf-cite-seq/renv/library/R-4.1/x86_64-pc-linux-gnu
 [2] /config/binaries/R/4.1.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</details>
<div id="refs">

</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.1.0/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.1.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] dsb_1.0.1                   paletteer_1.4.0            
 [3] BiocParallel_1.28.3         glmGamPoi_1.6.0            
 [5] clustree_0.4.4              ggraph_2.0.5               
 [7] patchwork_1.1.1             SeuratObject_4.0.4         
 [9] Seurat_4.0.6                scater_1.22.0              
[11] scran_1.22.1                scuttle_1.4.0              
[13] DropletUtils_1.14.1         SingleCellExperiment_1.16.0
[15] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[17] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[19] IRanges_2.28.0              S4Vectors_0.32.3           
[21] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[23] matrixStats_0.61.0          glue_1.6.0                 
[25] here_1.0.1                  forcats_0.5.1              
[27] stringr_1.4.0               dplyr_1.0.7                
[29] purrr_0.3.4                 readr_2.1.1                
[31] tidyr_1.1.4                 tibble_3.1.6               
[33] ggplot2_3.3.5               tidyverse_1.3.1            
[35] BiocStyle_2.22.0            workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            rtracklayer_1.54.0       
  [3] scattermore_0.7           R.methodsS3_1.8.1        
  [5] bit64_4.0.5               knitr_1.37               
  [7] irlba_2.3.5               DelayedArray_0.20.0      
  [9] R.utils_2.11.0            data.table_1.14.2        
 [11] rpart_4.1-15              AnnotationFilter_1.18.0  
 [13] KEGGREST_1.34.0           RCurl_1.98-1.6           
 [15] generics_0.1.1            GenomicFeatures_1.46.3   
 [17] ScaledMatrix_1.2.0        callr_3.7.0              
 [19] cowplot_1.1.1             RSQLite_2.2.9            
 [21] RANN_2.6.1                future_1.23.0            
 [23] bit_4.0.4                 tzdb_0.2.0               
 [25] spatstat.data_2.1-2       xml2_1.3.3               
 [27] lubridate_1.8.0           httpuv_1.6.5             
 [29] assertthat_0.2.1          viridis_0.6.2            
 [31] xfun_0.29                 hms_1.1.1                
 [33] jquerylib_0.1.4           evaluate_0.14            
 [35] promises_1.2.0.1          progress_1.2.2           
 [37] restfulr_0.0.13           fansi_1.0.0              
 [39] dbplyr_2.1.1              readxl_1.3.1             
 [41] igraph_1.2.11             DBI_1.1.2                
 [43] htmlwidgets_1.5.4         spatstat.geom_2.3-1      
 [45] ellipsis_0.3.2            RSpectra_0.16-0          
 [47] backports_1.4.1           bookdown_0.24            
 [49] prismatic_1.1.0           biomaRt_2.50.1           
 [51] deldir_1.0-6              sparseMatrixStats_1.6.0  
 [53] vctrs_0.3.8               ensembldb_2.18.2         
 [55] ROCR_1.0-11               abind_1.4-5              
 [57] cachem_1.0.6              withr_2.4.3              
 [59] ggforce_0.3.3             vroom_1.5.7              
 [61] sctransform_0.3.3         GenomicAlignments_1.30.0 
 [63] prettyunits_1.1.1         mclust_5.4.9             
 [65] goftest_1.2-3             cluster_2.1.2            
 [67] lazyeval_0.2.2            crayon_1.4.2             
 [69] labeling_0.4.2            edgeR_3.36.0             
 [71] pkgconfig_2.0.3           tweenr_1.0.2             
 [73] ProtGenerics_1.26.0       nlme_3.1-153             
 [75] vipor_0.4.5               rlang_0.4.12             
 [77] globals_0.14.0            lifecycle_1.0.1          
 [79] miniUI_0.1.1.1            filelock_1.0.2           
 [81] BiocFileCache_2.2.0       modelr_0.1.8             
 [83] rsvd_1.0.5                cellranger_1.1.0         
 [85] rprojroot_2.0.2           polyclip_1.10-0          
 [87] lmtest_0.9-39             Matrix_1.4-0             
 [89] Rhdf5lib_1.16.0           zoo_1.8-9                
 [91] reprex_2.0.1              beeswarm_0.4.0           
 [93] pheatmap_1.0.12           whisker_0.4              
 [95] ggridges_0.5.3            processx_3.5.2           
 [97] rjson_0.2.21              png_0.1-7                
 [99] viridisLite_0.4.0         bitops_1.0-7             
[101] getPass_0.2-2             R.oo_1.24.0              
[103] KernSmooth_2.23-20        rhdf5filters_1.6.0       
[105] Biostrings_2.62.0         blob_1.2.2               
[107] DelayedMatrixStats_1.16.0 parallelly_1.30.0        
[109] beachmat_2.10.0           scales_1.1.1             
[111] memoise_2.0.1             hexbin_1.28.2            
[113] magrittr_2.0.1            plyr_1.8.6               
[115] ica_1.0-2                 zlibbioc_1.40.0          
[117] compiler_4.1.0            BiocIO_1.4.0             
[119] dqrng_0.3.0               RColorBrewer_1.1-2       
[121] fitdistrplus_1.1-6        Rsamtools_2.10.0         
[123] cli_3.1.0                 XVector_0.34.0           
[125] listenv_0.8.0             pbapply_1.5-0            
[127] ps_1.6.0                  MASS_7.3-53.1            
[129] mgcv_1.8-38               tidyselect_1.1.1         
[131] stringi_1.7.6             highr_0.9                
[133] yaml_2.2.1                BiocSingular_1.10.0      
[135] locfit_1.5-9.4            ggrepel_0.9.1            
[137] grid_4.1.0                sass_0.4.0               
[139] tools_4.1.0               future.apply_1.8.1       
[141] parallel_4.1.0            rstudioapi_0.13          
[143] bluster_1.4.0             git2r_0.29.0             
[145] metapod_1.2.0             gridExtra_2.3            
[147] farver_2.1.0              Rtsne_0.15               
[149] digest_0.6.29             BiocManager_1.30.16      
[151] shiny_1.7.1               Rcpp_1.0.7               
[153] broom_0.7.11              later_1.3.0              
[155] RcppAnnoy_0.0.19          AnnotationDbi_1.56.2     
[157] httr_1.4.2                colorspace_2.0-2         
[159] XML_3.99-0.8              rvest_1.0.2              
[161] fs_1.5.2                  tensor_1.5               
[163] reticulate_1.22           splines_4.1.0            
[165] uwot_0.1.11               statmod_1.4.36           
[167] rematch2_2.1.2            spatstat.utils_2.3-0     
[169] graphlayouts_0.8.0        renv_0.15.0-14           
[171] sessioninfo_1.2.2         plotly_4.10.0            
[173] xtable_1.8-4              jsonlite_1.7.2           
[175] tidygraph_1.2.0           R6_2.5.1                 
[177] pillar_1.6.4              htmltools_0.5.2          
[179] mime_0.12                 fastmap_1.1.0            
[181] BiocNeighbors_1.12.0      codetools_0.2-18         
[183] utf8_1.2.2                lattice_0.20-45          
[185] bslib_0.3.1               spatstat.sparse_2.1-0    
[187] curl_4.3.2                ggbeeswarm_0.6.0         
[189] leiden_0.3.9              survival_3.2-13          
[191] limma_3.50.0              rmarkdown_2.11           
[193] munsell_0.5.0             rhdf5_2.38.0             
[195] GenomeInfoDbData_1.2.7    HDF5Array_1.22.1         
[197] haven_2.4.3               reshape2_1.4.4           
[199] gtable_0.3.0              spatstat.core_2.3-2      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
