---
title: "Multimodal single-cell analysis of bronchoalveolar lavage from children with cystic fibrosis reveals novel cell phenotypes"
description: |
author:
  - name: Jovana Maksimovic
date: "`r Sys.Date()`"
bibliography: ref.bib
site: workflowr::wflow_site
output:
  html_document:
    theme: cosmo
---

```{r, echo = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, number_sections = FALSE)
```


This site presents the code and results of the analyses described in the 
pre-print: [*“Multimodal single-cell analysis of bronchoalveolar lavage from children with cystic fibrosis reveals novel cell phenotypes”*](https://www.biorxiv.org/content/10.1101/2022.06.17.496207v1).

All the code and results of this analysis are available from GitHub at
https://github.com/Oshlack/paed-cf-cite-seq. To reproduce the complete analysis 
follow the instructions on the [getting started](https://oshlacklab.com/paed-cf-cite-seq/gettingStarted.html) page. 
The raw single cell RNA-seq and CITE-seq count data generated for this study 
can be downloaded as RDS files from [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.6651465.svg)](https://doi.org/10.5281/zenodo.6651465).

Follow the links below to view the different parts of the analysis. 


# Abstract {-}

## Rationale
Inflammation is a key driver of cystic fibrosis (CF) lung disease, which is not addressed by current standard care. Improved understanding of the mechanisms leading to aberrant inflammation may assist the development of effective anti-inflammatory therapy. Single-cell RNA sequencing allows profiling of cell composition and function at previously unprecedented resolution.  Here, we employ multiomic single-cell RNA sequencing (scRNA-seq) to generate the first immune cell atlas of the pediatric lower airway in preschool children with CF.

## Objectives
To use multimodal single-cell analysis to comprehensively define immune cell phenotypes, proportions and functional characteristics in preschool children with CF.

## Methods
We analyzed 42,658 cells from bronchoalveolar lavage of 11 preschool children with CF and a healthy control using  scRNA-seq and parallel assessment of 154 cell surface proteins. Validation of cell types identified by single-cell RNA seq was achieved by assessment of samples by  spectral flow cytometry. 

## Measurements and Main Results
Analysis of transcriptome expression and cell surface protein expression, combined with functional pathway analysis, revealed 41 immune and epithelial cell populations in BAL. Spectral flow cytometry analysis of over 256,000 cells from a subset of the same patients revealed high correlation in major cell type proportions across the two technologies. Macrophages consisted of 13 functionally distinct sub populations, including previously undescribed populations enriched for markers of vesicle production and regulatory/repair functions. Other novel cell populations included CD4 T cells expressing inflammatory IFNα/β and NFκB signalling genes. 

## Conclusions
Our work provides a comprehensive cellular analysis of the pediatric lower airway in preschool children with CF, reveals novel cell types and provides a reference for investigation of inflammation in early life CF.



# Authors {-}

Jovana Maksimovic^1,2,9\*^, Shivanthan Shanthikumar^2,3,4\*^, George Howitt^1,9^, Peter Hickey^5,6^, William Ho^5^, Casey Anttila^5^, Daniel V. Brown^5^, Anne Senabouth^7,8^, Dominick Kaczorowski^7,8^, Daniela Amann-Zalcenstein^5^, Joseph E. Powell^7,8^, Sarath C. Ranganathan^2,3,4^, Alicia Oshlack^1,9,10^, Melanie R. Neeland^2,3,11+^

^1^ Computational Biology Program, Peter MacCallum Cancer Centre, Parkville, VIC, Australia

^2^ Respiratory Diseases, Murdoch Children’s Research Institute, Parkville, VIC, Australia

^3^ Department of Paediatrics, University of Melbourne, Parkville, VIC, Australia

^4^ Respiratory and Sleep Medicine, Royal Children’s Hospital, Parkville, VIC, Australia 

^5^ Cellular Genomics Projects Team, Advanced Technology and Biology Division, WEHI, Parkville, VIC, Australia

^6^ Department of Medical Biology, University of Melbourne, Parkville, VIC, Australia 

^7^ Garvan-Weizmann Centre for Cellular Genomics, Garvan Institute of Medical Research, Sydney, NSW, Australia

^8^ UNSW Cellular Genomics Futures Institute, University of New South Wales, Sydney, NSW, Australia

^9^ Sir Peter MacCallum Department of Oncology, University of Melbourne, Parkville, VIC, Australia

^10^ School of Bioscience, University of Melbourne, Parkville, VIC, Australia

^11^ Molecular Immunity, Murdoch Children’s Research Institute, Parkville, VIC, Australia 

    
^+^ corresponding author

^\*^ contributed equally
 
 
# Analysis Overview {-}

1. scRNA-seq data quality control
    a) [Droplet selection](01_CF_BAL_Pilot.emptyDrops.html)
    b) [Preprocessing](02_CF_BAL_Pilot.preprocess.html)

2. CITE-seq data quality control and sample demultiplexing
    a) [Droplet selection](03_C133_Neeland.emptyDrops.html)
    b) [Demultiplexing](04_C133_Neeland.demultiplex.html)
    c) [Preprocessing](05_C133_Neeland.preprocess.html)      

3. Clustering, reference annotation and antibody derived tag (ADT) processing
    a) [Clustering and annotation](06_COMBO.clustering_annotation.html)
    b) [ADT processing](07_COMBO.transfer_proteins.html) 

4. Sub-clustering of cell type groups     
    a) [Sub-clustering of macrophages](08_COMBO.cluster_macrophages.html)
    b) [Sub-clustering of T/NK cells](09_COMBO.cluster_tcells.html)
    c) [Sub-clustering of "other" cells](10_COMBO.cluster_others.html)
    d) [Sub-clustering of macrophages (round 2)](08_COMBO.cluster_macrophages_round2.html)
    
5. Manual annotation 
    a) [Post-processing of macrophages](11_COMBO.postprocess_macrophages.html) 
    b) [Post-processing of T/NK cells](12_COMBO.postprocess_tcells.html)
    c) [Post-processing of "other" cells](13_COMBO.postprocess_others.html)
    d) [Post-processing of all cells combined](14_COMBO.postprocess_all.html) 


# Licenses {-}

The code in this analysis is covered by the [MIT license][mit] and the written
content on this website is covered by a [Creative Commons CC-BY][cc] license.


# Citations {-}



# Version Information  {-}

**R version**: `r R.version.string`

**Bioconductor version**: `r BiocManager::version()`


[mit]: https://choosealicense.com/licenses/mit/ "MIT License"
[cc]: https://choosealicense.com/licenses/cc-by-4.0/ "CC-BY License"
[pp]: https://www.biorxiv.org/content/10.1101/2020.08.24.265702v1 "Pre-print"
